diff --git i/test/test_utils.rb w/test/test_utils.rb
index 60e1153e..efc0d67b 100644
--- i/test/test_utils.rb
+++ w/test/test_utils.rb
@@ -1,7 +1,6 @@
 # encoding: UTF-8
 # frozen_string_literal: true
 
-raise 'Tests must be run with bundler, e.g. bundle exec rake test' unless defined?(Bundler)
 
 # Workaround an issue on JRuby where the stdlib directory can end up before the
 # minitest gem in the load path. This causes an old built-in version of minitest
