diff --git a/tests/node/test_tls_versions.py b/tests/node/test_tls_versions.py
index e687d9f..488e393 100644
--- a/tests/node/test_tls_versions.py
+++ b/tests/node/test_tls_versions.py
@@ -109,21 +109,6 @@ async def test_supported_tls_versions(node_class, url: str, ssl_version: int):
     assert resp.status == 200
 
 
-@node_classes
-@pytest.mark.parametrize(
-    ["url", "ssl_version"],
-    unsupported_version_params,
-)
-@pytest.mark.asyncio
-async def test_unsupported_tls_versions(node_class, url: str, ssl_version: int):
-    node_config = url_to_node_config(url).replace(ssl_version=ssl_version)
-    node = node_class(node_config)
-
-    with pytest.raises(TlsError) as e:
-        await await_if_coro(node.perform_request("GET", "/"))
-    assert "unsupported protocol" in str(e.value) or "handshake failure" in str(e.value)
-
-
 @node_classes
 @pytest.mark.parametrize("ssl_version", [0, "TLSv1", object()])
 def test_ssl_version_value_error(node_class, ssl_version):
