diff --git a/cpp/cmake_modules/Findlz4Alt.cmake b/cpp/cmake_modules/Findlz4Alt.cmake
index 77a22957f..ce35d81eb 100644
--- a/cpp/cmake_modules/Findlz4Alt.cmake
+++ b/cpp/cmake_modules/Findlz4Alt.cmake
@@ -33,6 +33,9 @@ if(lz4_FOUND)
   if(NOT TARGET LZ4::lz4 AND TARGET lz4::lz4)
     add_library(LZ4::lz4 ALIAS lz4::lz4)
   endif()
+  if(NOT TARGET LZ4::lz4 AND TARGET LZ4::lz4_shared)
+    add_library(LZ4::lz4 ALIAS LZ4::lz4_shared)
+  endif()
   return()
 endif()
 
