--- systemtap-4.0/stapbpf/Makefile.am.orig	2019-02-14 17:37:56.083512013 +0100
+++ systemtap-4.0/stapbpf/Makefile.am	2019-02-14 17:38:34.603641107 +0100
@@ -42,10 +42,5 @@
 # Why the "id -u" condition?  This way, an unprivileged user can run
 # make install, and have "sudo stap ...." or "sudo stapbpf ...." work later.
 install-exec-hook:
-	if [ `id -u` -eq 0 ]; then \
-		getent group stapusr >/dev/null || groupadd -g 156 -r stapusr 2>/dev/null || groupadd -r stapusr; \
-		getent group stapusr >/dev/null \
-		&& chgrp stapusr "$(DESTDIR)$(bindir)/stapbpf" \
-		&& chmod 04110 "$(DESTDIR)$(bindir)/stapbpf"; \
-	fi
+	echo "Skipping grpadd..."
 endif
--- systemtap-4.0/staprun/Makefile.am.orig	2019-02-14 17:38:06.216879308 +0100
+++ systemtap-4.0/staprun/Makefile.am	2019-02-14 17:38:54.897042437 +0100
@@ -77,8 +77,4 @@
 # Why the "id -u" condition?  This way, an unprivileged user can run
 # make install, and have "sudo stap ...." or "sudo staprun ...." work later.
 install-exec-hook:
-	if [ `id -u` -eq 0 ]; then \
-		getent group stapusr >/dev/null || groupadd -g 156 -r stapusr 2>/dev/null || groupadd -r stapusr; \
-		getent group stapusr >/dev/null && chgrp stapusr "$(DESTDIR)$(bindir)/staprun"; \
-		chmod 04110 "$(DESTDIR)$(bindir)/staprun"; \
-	fi
+	echo "Skipping grpadd..."
--- systemtap-4.0/aclocal.m4.orig	2019-02-14 17:43:30.424632028 +0100
+++ systemtap-4.0/aclocal.m4	2019-02-14 17:43:55.388048948 +0100
@@ -32,10 +32,10 @@
 # generated from the m4 files accompanying Automake X.Y.
 # (This private macro should not be called outside this file.)
 AC_DEFUN([AM_AUTOMAKE_VERSION],
-[am__api_version='1.15'
+[am__api_version='1.16'
 dnl Some users find AM_AUTOMAKE_VERSION and mistake it for a way to
 dnl require some minimum version.  Point them to the right macro.
-m4_if([$1], [1.15.1], [],
+m4_if([$1], [1.16.1], [],
       [AC_FATAL([Do not call $0, use AM_INIT_AUTOMAKE([$1]).])])dnl
 ])
 
