diff --git i/Autostart/Systemd/sysusers w/Autostart/Systemd/sysusers
index 9a7e53cf3..6fee5c255 100644
--- i/Autostart/Systemd/sysusers
+++ w/Autostart/Systemd/sysusers
@@ -20,7 +20,7 @@
 # keep these synchronized with brltty@.service #
 ################################################
 
-u brltty - "Braille Device Daemon" /var/lib/brltty
+u! brltty - "Braille Device Daemon" /var/lib/brltty
 
 # for reading screen content (/dev/vcs*)
 # for virtual console monitoring and control (/dev/tty<n>)
