diff --git a/spa/plugins/avb/avbtp/packets.h b/spa/plugins/avb/avbtp/packets.h
index 7047456bf..97731348c 100644
--- a/spa/plugins/avb/avbtp/packets.h
+++ b/spa/plugins/avb/avbtp/packets.h
@@ -186,7 +186,7 @@ struct spa_avbtp_packet_aaf {
 #define SPA_AVBTP_PACKET_AAF_SET_MR(p,v)		((p)->mr = (v))
 #define SPA_AVBTP_PACKET_AAF_SET_GV(p,v)		((p)->gv = (v))
 #define SPA_AVBTP_PACKET_AAF_SET_TV(p,v)		((p)->tv = (v))
-#define SPA_AVBTP_PACKET_AAF_SET_SEQ_NUM(p,v)		((p)->seq_num = (v))
+#define SPA_AVBTP_PACKET_AAF_SET_SEQ_NUM(p,v)		((p)->seq_number = (v))
 #define SPA_AVBTP_PACKET_AAF_SET_TU(p,v)		((p)->tu = (v))
 #define SPA_AVBTP_PACKET_AAF_SET_STREAM_ID(p,v)		((p)->stream_id = htobe64(v))
 #define SPA_AVBTP_PACKET_AAF_SET_TIMESTAMP(p,v)		((p)->timestamp = htonl(v))
@@ -204,7 +204,7 @@ struct spa_avbtp_packet_aaf {
 #define SPA_AVBTP_PACKET_AAF_GET_MR(p)			((p)->mr)
 #define SPA_AVBTP_PACKET_AAF_GET_GV(p)			((p)->gv)
 #define SPA_AVBTP_PACKET_AAF_GET_TV(p)			((p)->tv)
-#define SPA_AVBTP_PACKET_AAF_GET_SEQ_NUM(p)		((p)->seq_num)
+#define SPA_AVBTP_PACKET_AAF_GET_SEQ_NUM(p)		((p)->seq_number)
 #define SPA_AVBTP_PACKET_AAF_GET_TU(p)			((p)->tu)
 #define SPA_AVBTP_PACKET_AAF_GET_STREAM_ID(p)		be64toh((p)->stream_id)
 #define SPA_AVBTP_PACKET_AAF_GET_TIMESTAMP(p)		ntohl((p)->timestamp)
diff --git a/src/modules/module-avb/stream.c b/src/modules/module-avb/stream.c
index b86f0814c..cdfa68640 100644
--- a/src/modules/module-avb/stream.c
+++ b/src/modules/module-avb/stream.c
@@ -131,7 +131,7 @@ static int flush_write(struct stream *stream, uint64_t current_time)
 			index % stream->buffer_size,
 			&stream->iov[1], stream->payload_size);
 
-		p->seq_num = stream->pdu_seq++;
+		p->seq_number = stream->pdu_seq++;
 		p->tv = 1;
 		p->timestamp = ptime;
 		p->dbc = dbc;
