--- gupnp-av/meson.build.orig	2024-10-25 23:48:51.226065532 +0200
+++ gupnp-av/meson.build	2024-10-25 23:49:04.372724549 +0200
@@ -71,7 +71,6 @@
 
 common_cflags = cc.get_supported_arguments(test_cflags)
 
-add_project_arguments(cc.get_supported_arguments('-Werror=deprecated-declarations'), language: 'c')
 conf = configuration_data()
 conf.set_quoted('DATADIR', join_paths(get_option('prefix'), get_option('datadir'), 'gupnp-av'))
 conf.set_quoted('VERSION', meson.project_version())
