diff -aur elasticsearch-py-7.9.0/test_elasticsearch/test_serializer.py elasticsearch-py-7.9.0.old/test_elasticsearch/test_serializer.py
--- elasticsearch-py-7.9.0/test_elasticsearch/test_serializer.py	2020-08-18 18:39:49.000000000 +0200
+++ elasticsearch-py-7.9.0.old/test_elasticsearch/test_serializer.py	2024-04-06 18:28:26.153403837 +0200
@@ -85,7 +85,7 @@
             np.float32,
             np.float64,
         ):
-            self.assertRegexpMatches(
+            self.assertRegex(
                 ser.dumps({"d": np_type(1.2)}), r'^\{"d":1\.2[\d]*}$'
             )
 
