diff --git i/test_elasticsearch/test_async/test_connection.py w/test_elasticsearch/test_async/test_connection.py
index fd2ac62d..63c7a98d 100644
--- i/test_elasticsearch/test_async/test_connection.py
+++ w/test_elasticsearch/test_async/test_connection.py
@@ -19,7 +19,7 @@
 import ssl
 import gzip
 import io
-from mock import patch
+from unittest.mock import patch
 import warnings
 from platform import python_version
 import aiohttp
diff --git i/test_elasticsearch/test_async/test_transport.py w/test_elasticsearch/test_async/test_transport.py
index 293363ff..7d56e1f1 100644
--- i/test_elasticsearch/test_async/test_transport.py
+++ w/test_elasticsearch/test_async/test_transport.py
@@ -19,7 +19,7 @@
 from __future__ import unicode_literals
 import asyncio
 import json
-from mock import patch
+from unittest.mock import patch
 import pytest
 
 from elasticsearch import AsyncTransport
diff --git i/test_elasticsearch/test_connection.py w/test_elasticsearch/test_connection.py
index 711ab53c..5971c4a5 100644
--- i/test_elasticsearch/test_connection.py
+++ w/test_elasticsearch/test_connection.py
@@ -20,7 +20,7 @@ import re
 import ssl
 import gzip
 import io
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 import urllib3
 from urllib3._collections import HTTPHeaderDict
 import warnings
diff --git i/test_elasticsearch/test_helpers.py w/test_elasticsearch/test_helpers.py
index 8ea7c74c..1966d88f 100644
--- i/test_elasticsearch/test_helpers.py
+++ w/test_elasticsearch/test_helpers.py
@@ -16,7 +16,7 @@
 #  specific language governing permissions and limitations
 #  under the License.
 
-import mock
+from unittest import mock
 import time
 import threading
 import pytest
diff --git i/test_elasticsearch/test_server/test_helpers.py w/test_elasticsearch/test_server/test_helpers.py
index a9abc315..a4d56edc 100644
--- i/test_elasticsearch/test_server/test_helpers.py
+++ w/test_elasticsearch/test_server/test_helpers.py
@@ -15,7 +15,7 @@
 #  specific language governing permissions and limitations
 #  under the License.
 
-from mock import patch
+from unittest.mock import patch
 
 from elasticsearch import helpers, TransportError
 from elasticsearch.helpers import ScanError
diff --git i/test_elasticsearch/test_transport.py w/test_elasticsearch/test_transport.py
index 50e68c97..a4d97568 100644
--- i/test_elasticsearch/test_transport.py
+++ w/test_elasticsearch/test_transport.py
@@ -19,7 +19,7 @@
 from __future__ import unicode_literals
 import json
 import time
-from mock import patch
+from unittest.mock import patch
 
 from elasticsearch.transport import Transport, get_host_info
 from elasticsearch.connection import Connection
