diff --git a/cliff/tests/test_help.py b/cliff/tests/test_help.py
index 571211d..247b822 100644
--- a/cliff/tests/test_help.py
+++ b/cliff/tests/test_help.py
@@ -111,7 +111,7 @@ class TestHelp(base.TestBase):
             pass
         help_text = stdout.getvalue()
         basecommand = os.path.split(sys.argv[0])[1]
-        self.assertIn(f'usage: {basecommand} [--version]', help_text)
+        self.assertIn(f'usage: python -m stestr.subunit_runner.run [--version]', help_text)
         self.assertRegex(help_text, 'option(s|al arguments):\n  --version')
         expected = (
             '  one            Test command\n'
