--- clang-16.0.4.src/lib/Basic/Targets/PPC.h.orig	2023-05-24 14:28:01.176621490 +0200
+++ clang-16.0.4.src/lib/Basic/Targets/PPC.h	2023-05-24 14:29:38.414872784 +0200
@@ -425,14 +425,10 @@
       LongDoubleFormat = &llvm::APFloat::IEEEdouble();
     } else if ((Triple.getArch() == llvm::Triple::ppc64le)) {
       DataLayout = "e-m:e-i64:64-n32:64";
-      ABI = "elfv2";
     } else {
       DataLayout = "E-m:e-i64:64-n32:64";
-      if (Triple.isPPC64ELFv2ABI())
-        ABI = "elfv2";
-      else
-        ABI = "elfv1";
     }
+    ABI = "elfv2";
 
     if (Triple.isOSFreeBSD() || Triple.isOSOpenBSD() || Triple.isMusl()) {
       LongDoubleWidth = LongDoubleAlign = 64;
--- clang-16.0.4.src/lib/Driver/ToolChains/Clang.cpp.orig	2023-05-24 14:32:51.659489490 +0200
+++ clang-16.0.4.src/lib/Driver/ToolChains/Clang.cpp	2023-05-24 14:33:30.235618677 +0200
@@ -2000,13 +2000,7 @@
   const char *ABIName = nullptr;
   if (T.isOSBinFormatELF()) {
     switch (getToolChain().getArch()) {
-    case llvm::Triple::ppc64: {
-      if (T.isPPC64ELFv2ABI())
-        ABIName = "elfv2";
-      else
-        ABIName = "elfv1";
-      break;
-    }
+    case llvm::Triple::ppc64:
     case llvm::Triple::ppc64le:
       ABIName = "elfv2";
       break;
