Doesn't work on PowerPC.

--- clang-14.0.6.src/unittests/DirectoryWatcher/CMakeLists.txt.old	2022-06-22 16:46:24.000000000 +0000
+++ clang-14.0.6.src/unittests/DirectoryWatcher/CMakeLists.txt	2022-12-05 01:14:23.863133395 +0000
@@ -1,4 +1,4 @@
-if(APPLE OR CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME STREQUAL Windows)
+if(APPLE OR (CMAKE_SYSTEM_NAME MATCHES "Linux" AND NOT (CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc")) OR CMAKE_SYSTEM_NAME STREQUAL Windows)
 
   set(LLVM_LINK_COMPONENTS
     Support
