From c6e03f208c85288b81a780f26967b98ace976e60 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Sat, 27 Jan 2024 23:44:34 +0100
Subject: [PATCH] config: disable anonymous access by default

---
 vsftpd.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vsftpd.conf b/vsftpd.conf
index cc1c607..f613efe 100644
--- a/vsftpd.conf
+++ b/vsftpd.conf
@@ -9,7 +9,7 @@
 # capabilities.
 #
 # Allow anonymous FTP? (Beware - allowed by default if you comment this out).
-anonymous_enable=YES
+anonymous_enable=NO
 #
 # Uncomment this to allow local users to log in.
 #local_enable=YES
-- 
2.43.0

