From efe3fa360454f86800ed60eab403c00713cf8e92 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Sat, 27 Jan 2024 23:48:42 +0100
Subject: [PATCH] config: set PAM service name to vsftpd

---
 vsftpd.conf | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vsftpd.conf b/vsftpd.conf
index f613efe..ce9d10a 100644
--- a/vsftpd.conf
+++ b/vsftpd.conf
@@ -115,3 +115,7 @@ listen=YES
 # sockets, you must run two copies of vsftpd with two configuration files.
 # Make sure, that one of the listen options is commented !!
 #listen_ipv6=YES
+
+# Set own PAM service name to detect authentication settings specified
+# for vsftpd by the system package.
+pam_service_name=vsftpd
-- 
2.43.0

