From 4dd04b995fd51dbbeadd3d6ad0417f128924a932 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Sat, 27 Jan 2024 20:27:51 +0100
Subject: [PATCH] fix: build with openssl 1.1

---
 vsf_findlibs.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vsf_findlibs.sh b/vsf_findlibs.sh
index 4538685..6e65e2e 100755
--- a/vsf_findlibs.sh
+++ b/vsf_findlibs.sh
@@ -66,7 +66,7 @@ locate_library /usr/shlib/librt.so && echo "-lrt";
 locate_library /usr/lib/libsendfile.so && echo "-lsendfile";
 
 # OpenSSL
-if find_func SSL_library_init ssl.o; then
+if find_func SSL_CTX_new ssl.o; then
   echo "-lssl -lcrypto";
 elif find_func SSL_new ssl.o; then
   echo "-lssl -lcrypto";
-- 
2.43.0

