From ca9a5c7719f6c1a285ab80d0660e1b1fd9d0d8a3 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Sat, 27 Jan 2024 20:13:59 +0100
Subject: [PATCH] fix: ssl_tlsv documentation and config tunables

---
 README.ssl    | 4 ++--
 vsftpd.conf.5 | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/README.ssl b/README.ssl
index 2ec70cb..5327679 100644
--- a/README.ssl
+++ b/README.ssl
@@ -35,6 +35,6 @@ go to that directory and type e.g. "make vsftpd.pem". Then answer the
 questions you are asked. Alternatively, read the man page for "openssl".
 - Also be aware of the following SSL related parameters. Read the vsftpd.conf.5
 manual page to learn about them: allow_anon_ssl, force_local_logins_ssl,
-force_local_data_ssl, ssl_sslv2, ssl_sslv3, ssl_tlsv1, rsa_cert_file,
-dsa_cert_file, ssl_ciphers.
+force_local_data_ssl, ssl_sslv2, ssl_sslv3, ssl_tlsv1, ssl_tlsv11, ssl_tlsv12,
+ssl_tlsv13, rsa_cert_file, dsa_cert_file, ssl_ciphers.
 
diff --git a/vsftpd.conf.5 b/vsftpd.conf.5
index 8d469e9..56ab251 100644
--- a/vsftpd.conf.5
+++ b/vsftpd.conf.5
@@ -499,7 +499,7 @@ TLS v1.2+ connections are preferred.
 
 Default: NO
 .TP
-.B ssl_tlsv1_1
+.B ssl_tlsv11
 Only applies if
 .BR ssl_enable
 is activated. If enabled, this option will permit TLS v1.1 protocol connections.
@@ -507,7 +507,7 @@ TLS v1.2+ connections are preferred.
 
 Default: NO
 .TP
-.B ssl_tlsv1_2
+.B ssl_tlsv12
 Only applies if
 .BR ssl_enable
 is activated. If enabled, this option will permit TLS v1.2 protocol connections.
@@ -515,7 +515,7 @@ TLS v1.2+ connections are preferred.
 
 Default: YES
 .TP
-.B ssl_tlsv1_3
+.B ssl_tlsv13
 Only applies if
 .BR ssl_enable
 is activated. If enabled, this option will permit TLS v1.3 protocol connections.
-- 
2.43.0

