diff --git i/audisp/plugins/af_unix/af_unix.conf w/audisp/plugins/af_unix/af_unix.conf
index f07a5fc1..26ad3148 100644
--- i/audisp/plugins/af_unix/af_unix.conf
+++ w/audisp/plugins/af_unix/af_unix.conf
@@ -6,7 +6,7 @@
 # the output format, and optionally the queue depth.
 
 active = no
-path = /sbin/audisp-af_unix
+path = /usr/bin/audisp-af_unix
 type = always
 args = 0640 /run/audit/audispd_events string
 # To change the queue depth from the default of 512, append the value:
diff --git i/audisp/plugins/filter/filter.conf w/audisp/plugins/filter/filter.conf
index deb84df1..5070418e 100644
--- i/audisp/plugins/filter/filter.conf
+++ w/audisp/plugins/filter/filter.conf
@@ -10,7 +10,7 @@
 # See audisp-filter(8)
 
 active = no
-path = /sbin/audisp-filter
+path = /usr/bin/audisp-filter
 type = always
-args = allowlist /etc/audit/audisp-filter.conf /sbin/audisp-syslog LOG_USER LOG_INFO interpret
+args = allowlist /etc/audit/audisp-filter.conf /usr/bin/audisp-syslog LOG_USER LOG_INFO interpret
 format = string
diff --git i/audisp/plugins/ids/audisp-ids.conf w/audisp/plugins/ids/audisp-ids.conf
index 1e2046c4..0e890b95 100644
--- i/audisp/plugins/ids/audisp-ids.conf
+++ w/audisp/plugins/ids/audisp-ids.conf
@@ -2,7 +2,7 @@
 # audisp-ids plugin.
 
 active = no
-path = /usr/sbin/audisp-ids
+path = /usr/bin/audisp-ids
 type = always 
 args = 1
 format = string
diff --git i/audisp/plugins/remote/au-remote.conf w/audisp/plugins/remote/au-remote.conf
index d9a6db8b..e10097f3 100644
--- i/audisp/plugins/remote/au-remote.conf
+++ w/audisp/plugins/remote/au-remote.conf
@@ -4,7 +4,7 @@
 # a remote machine (Central Logger).
 
 active = no
-path = /sbin/audisp-remote
+path = /usr/bin/audisp-remote
 type = always
 #args =
 format = string
diff --git i/audisp/plugins/statsd/au-statsd.conf w/audisp/plugins/statsd/au-statsd.conf
index 84316e07..b9d16ad7 100644
--- i/audisp/plugins/statsd/au-statsd.conf
+++ w/audisp/plugins/statsd/au-statsd.conf
@@ -2,7 +2,7 @@
 # It picks out metrics and writes them to statsd.
 
 active = no
-path = /sbin/audisp-statsd
+path = /usr/bin/audisp-statsd
 type = always 
 # args =
 format = string
diff --git i/audisp/plugins/syslog/syslog.conf w/audisp/plugins/syslog/syslog.conf
index ad29c5bf..0d3e00ac 100644
--- i/audisp/plugins/syslog/syslog.conf
+++ w/audisp/plugins/syslog/syslog.conf
@@ -7,7 +7,7 @@
 # LOG_AUTHPRIV, LOG_DAEMON, LOG_SYSLOG, and LOG_USER.
 
 active = no
-path = /sbin/audisp-syslog
+path = /usr/bin/audisp-syslog
 type = always 
 args = LOG_INFO
 format = string
diff --git i/audisp/plugins/zos-remote/audispd-zos-remote.conf w/audisp/plugins/zos-remote/audispd-zos-remote.conf
index 555d9e8a..680aaa8b 100644
--- i/audisp/plugins/zos-remote/audispd-zos-remote.conf
+++ w/audisp/plugins/zos-remote/audispd-zos-remote.conf
@@ -7,7 +7,7 @@
 # See audispd-zos-remote(8) 
 
 active = no
-path = /sbin/audispd-zos-remote
+path = /usr/bin/audispd-zos-remote
 type = always 
 args = /etc/audit/zos-remote.conf
 format = string
