diff --git a/cmake_modules/ThirdpartyToolchain.cmake b/cmake_modules/ThirdpartyToolchain.cmake
index c03afa8f2..0c29abec2 100644
--- a/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cmake_modules/ThirdpartyToolchain.cmake
@@ -376,8 +376,6 @@ else ()
       GIT_REPOSITORY "https://github.com/madler/zlib.git"
       GIT_TAG 5a82f71ed1dfc0bec044d9702463dbdf84ea3b71
       FIND_PACKAGE_ARGS
-      NAMES ZLIB
-      CONFIG
       )
     fetchcontent_makeavailable(ZLIB)

