--- /home/antonio/Software/test/texlive/texmf-dist/web2c/texmfcnf.lua	2023-05-27 18:02:06.978367099 +0200
+++ ./texmfcnf.lua	2023-05-29 10:26:45.444599722 +0200
@@ -1,6 +1,12 @@
 -- todo: come up with an auto-texlive identification (texmf-dist)
 
-local hiddentexlivepath = ".texlive2023"
+local hiddentexlivepath = ".texlive"
+
+-- ConTeXt needs a properly expanded TEXMFLOCAL, so here is a
+-- bit of lua code to make that happen
+
+local texmflocal = resolvers.prefixes.selfautoparent();
+texmflocal = string.gsub(texmflocal, "20%d%d$", "texmf-local");
 
 return {
 
@@ -52,7 +58,7 @@
 
             TEXMFVAR        = "home:" .. hiddentexlivepath .. "/texmf-var",
             TEXMFCONFIG     = "home:" .. hiddentexlivepath .. "/texmf-config",
-            TEXMFSYSVAR     = "selfautoparent:texmf-var",
+            TEXMFSYSVAR     = "/var/lib/texmf",
             TEXMFCACHE      = "$TEXMFSYSVAR;$TEXMFVAR",
 
             -- I don't like this texmf under home and texmf-home would make more sense. One never knows
@@ -62,7 +68,7 @@
             -- By using prefixes we don't get expanded paths in the cache __path__ entry. This makes the
             -- tex root relocatable.
 
-            TEXMFOS         = "selfautodir:",
+            TEXMFOS         = "selfautodir:share",
 
             -- standalone:
 
@@ -73,12 +79,12 @@
 
             -- texlive:
 
-            TEXMFDIST       = "selfautoparent:texmf-dist",
-            TEXMFSYSCONFIG  = "selfautoparent:texmf-config",
+            TEXMFDIST       = "selfautodir:share/texmf-dist",
+            TEXMFSYSCONFIG  = "/etc/texmf",
 
             -- The texmf-local path is only used for (maybe) some additional configuration file.
 
-            TEXMFLOCAL      = "selfautoparent:texmf-local",
+            TEXMFLOCAL      = texmflocal,
             TEXMFFONTS      = "selfautoparent:texmf-fonts",
             TEXMFPROJECT    = "selfautoparent:texmf-project",
 
@@ -93,7 +99,6 @@
          -- TEXMF           = "{$TEXMFHOME,!!$TEXMFPROJECT,!!$TEXMFFONTS,!!$TEXMFLOCAL,!!$TEXMFMODULES,!!$TEXMFCONTEXT,!!$TEXMFSYSTEM,!!$TEXMFMAIN}",
 
             -- texlive:
-
             TEXMF           = "{$TEXMFCONFIG,$TEXMFHOME,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFPROJECT,!!$TEXMFFONTS,!!$TEXMFLOCAL,!!$TEXMFDIST}",
 
             TEXFONTMAPS     = ".;$TEXMF/fonts/data//;$TEXMF/fonts/map/{pdftex,dvips}//",
