diff --git a/h5py/__init__.py b/h5py/__init__.py
index 20d11225..e3e5eb54 100644
--- a/h5py/__init__.py
+++ b/h5py/__init__.py
@@ -32,14 +32,6 @@ except ImportError:
 
 from . import version
 
-if version.hdf5_version_tuple != version.hdf5_built_version_tuple:
-    _warn(("h5py is running against HDF5 {0} when it was built against {1}, "
-           "this may cause problems").format(
-            '{0}.{1}.{2}'.format(*version.hdf5_version_tuple),
-            '{0}.{1}.{2}'.format(*version.hdf5_built_version_tuple)
-    ))
-
-
 _errors.silence_errors()
 
 from ._conv import register_converters as _register_converters, \
