diff --unified --recursive --text --new-file lilv-0.26.2.orig/bindings/test/meson.build lilv-0.26.2/bindings/test/meson.build
--- lilv-0.26.2.orig/bindings/test/meson.build	2025-12-30 17:19:02.807287590 +0100
+++ lilv-0.26.2/bindings/test/meson.build	2025-12-30 17:23:02.511927952 +0100
@@ -10,7 +10,10 @@
     py,
     args: ['-m', 'unittest', 'discover', meson.current_source_dir()],
     depends: [liblilv],
-    env: ['PYTHONPATH=' + meson.current_source_dir() / '..' / 'python'],
+    env: [
+      'PYTHONPATH=' + meson.current_source_dir() / '..' / 'python',
+      'LILV_BUILD_DIR=' + meson.project_build_root(),
+    ],
     suite: 'bindings',
   )
 endif
diff --unified --recursive --text --new-file lilv-0.26.2.orig/bindings/test/test_api.py lilv-0.26.2/bindings/test/test_api.py
--- lilv-0.26.2.orig/bindings/test/test_api.py	2025-12-30 17:19:02.807552574 +0100
+++ lilv-0.26.2/bindings/test/test_api.py	2025-12-30 17:34:27.848847447 +0100
@@ -6,19 +6,11 @@
 import os
 import sys
 import unittest
+from pathlib import Path
 
-path = os.path.abspath("bindings/bindings_test_plugin.lv2/")
-
-if sys.version_info[0] == 2:
-    import urllib
-    import urlparse
-
-    location = urlparse.urljoin("file:", urllib.pathname2url(path) + "/")
-else:
-    from urllib.parse import urljoin
-    from urllib.request import pathname2url
-
-    location = urljoin("file:", pathname2url(path) + "/")
+build_dir = os.environ.get("LILV_BUILD_DIR", os.getcwd())
+path = os.path.abspath(os.path.join(build_dir, "bindings/bindings_test_plugin.lv2"))
+location = Path(path).as_uri() + "/"
 
 
 class NodeTests(unittest.TestCase):
