diff --git a/src/plugin.c b/src/plugin.c
index 06f505b3..6c5a3963 100644
--- a/src/plugin.c
+++ b/src/plugin.c
@@ -957,7 +957,7 @@ lilv_plugin_get_uis(const LilvPlugin* plugin)
         lilv_world_get_unique(plugin->world, ui, plugin->world->uris.ui_binary);
     }
 
-    if (sord_node_get_type(ui) != SORD_URI ||
+    if (!ui || !type || !binary || sord_node_get_type(ui) != SORD_URI ||
         sord_node_get_type(type) != SORD_URI ||
         sord_node_get_type(binary) != SORD_URI) {
       LILV_ERRORF("Corrupt UI <%s>\n", sord_node_get_string(ui));
