diff -wbBur tigervnc/CMakeLists.txt tigervnc.my/CMakeLists.txt
--- tigervnc/CMakeLists.txt	2026-01-24 02:47:57.694863453 +0300
+++ tigervnc.my/CMakeLists.txt	2026-01-24 02:44:39.591532429 +0300
@@ -267,12 +267,12 @@
   set(FLTK_SKIP_OPENGL TRUE)
   set(FLTK_SKIP_FORMS TRUE)
   if(BUILD_VIEWER STREQUAL "AUTO")
-    find_package(FLTK)
+    find_package(FLTK1.3)
   else()
-    find_package(FLTK REQUIRED)
+    find_package(FLTK1.3 REQUIRED)
   endif()
 
-  if(NOT FLTK_FOUND)
+  if(NOT FLTK1.3_FOUND)
     message(WARNING "FLTK NOT found. TigerVNC viewer disabled.")
     set(BUILD_VIEWER 0)
   endif()
