--- Python-3.7.4/Makefile.pre.in.orig	2019-10-07 10:29:16.018090689 +0200
+++ Python-3.7.4/Makefile.pre.in	2019-10-07 10:29:25.917059075 +0200
@@ -1103,7 +1103,7 @@
 		test_multibytecodec test_urllib2_localnet test_itertools \
 		test_multiprocessing_fork test_multiprocessing_spawn \
 		test_multiprocessing_forkserver \
-		test_mailbox test_socket test_poll \
+		test_mailbox test_poll \
 		test_select test_zipfile test_concurrent_futures
 quicktest:	@DEF_MAKE_RULE@ platform
 		$(TESTRUNNER) $(QUICKTESTOPTS)
--- Python-3.7.4/PCbuild/lib.pyproj.orig	2019-10-07 12:10:57.600657428 +0200
+++ Python-3.7.4/PCbuild/lib.pyproj	2019-10-07 12:11:09.889359825 +0200
@@ -1272,7 +1272,6 @@
     <Compile Include="test\test_smtplib.py" />
     <Compile Include="test\test_smtpnet.py" />
     <Compile Include="test\test_sndhdr.py" />
-    <Compile Include="test\test_socket.py" />
     <Compile Include="test\test_socketserver.py" />
     <Compile Include="test\test_sort.py" />
     <Compile Include="test\test_source_encoding.py" />
@@ -1816,4 +1815,4 @@
     <Folder Include="xml\sax" />
   </ItemGroup>
   <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\Python Tools\Microsoft.PythonTools.targets" />
-</Project>
\ No newline at end of file
+</Project>
