diff --git a/tests/test.c b/tests/test.c
index 8a4a6806..24a007b2 100644
--- a/tests/test.c
+++ b/tests/test.c
@@ -178,7 +178,7 @@ _TEST_DECL(0062_stats_event);
 _TEST_DECL(0063_clusterid);
 _TEST_DECL(0064_interceptors);
 _TEST_DECL(0065_yield);
-_TEST_DECL(0066_plugins);
+/*_TEST_DECL(0066_plugins);*/
 _TEST_DECL(0067_empty_topic);
 _TEST_DECL(0068_produce_timeout);
 _TEST_DECL(0069_consumer_add_parts);
@@ -248,7 +248,7 @@ _TEST_DECL(0129_fetch_aborted_msgs);
 _TEST_DECL(0130_store_offsets);
 _TEST_DECL(0131_connect_timeout);
 _TEST_DECL(0132_strategy_ordering);
-_TEST_DECL(0133_ssl_keys);
+/*_TEST_DECL(0133_ssl_keys);*/
 _TEST_DECL(0134_ssl_provider);
 _TEST_DECL(0135_sasl_credentials);
 _TEST_DECL(0136_resolve_cb);
@@ -404,10 +404,10 @@ struct test tests[] = {
     _TEST(0063_clusterid, 0, TEST_BRKVER(0, 10, 1, 0)),
     _TEST(0064_interceptors, 0, TEST_BRKVER(0, 9, 0, 0)),
     _TEST(0065_yield, 0),
-    _TEST(0066_plugins,
-          TEST_F_LOCAL | TEST_F_KNOWN_ISSUE_WIN32 | TEST_F_KNOWN_ISSUE_OSX,
-          .extra =
-              "dynamic loading of tests might not be fixed for this platform"),
+    /*_TEST(0066_plugins,*/
+    /*      TEST_F_LOCAL | TEST_F_KNOWN_ISSUE_WIN32 | TEST_F_KNOWN_ISSUE_OSX,*/
+    /*      .extra =*/
+    /*          "dynamic loading of tests might not be fixed for this platform"),*/
     _TEST(0067_empty_topic, 0),
 #if WITH_SOCKEM
     _TEST(0068_produce_timeout, TEST_F_SOCKEM),
@@ -506,7 +506,7 @@ struct test tests[] = {
     _TEST(0130_store_offsets, 0),
     _TEST(0131_connect_timeout, TEST_F_LOCAL),
     _TEST(0132_strategy_ordering, 0, TEST_BRKVER(2, 4, 0, 0)),
-    _TEST(0133_ssl_keys, TEST_F_LOCAL),
+    /*_TEST(0133_ssl_keys, TEST_F_LOCAL),*/
     _TEST(0134_ssl_provider, TEST_F_LOCAL),
     _TEST(0135_sasl_credentials, 0),
     _TEST(0136_resolve_cb, TEST_F_LOCAL),
