From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Wed, 14 May 2025 01:38:22 +0200
Subject: [PATCH] HACK: Fix tests in the presence of -ffile-prefix-map

---
 tests/test_vars.in | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/tests/test_vars.in b/tests/test_vars.in
index a0b750b9fee9..4dea21eb01a0 100644
--- a/tests/test_vars.in
+++ b/tests/test_vars.in
@@ -5,16 +5,3 @@ EXEEXT=@EXEEXT@
 export EXEEXT
 HAVE_FORK=@HAVE_FORK@
 export HAVE_FORK
-
-# path of the tests directory
-if [ x"@srcdir@" != x"." ]; then
-    if [ -z "@IS_MSVC@" -o "@IS_MSVC@" != "1" ]; then
-       SRCDIR="@srcdir@/"
-    else
-       SRCDIR="@srcdir@\\"
-    fi
-else
-    SRCDIR=""
-fi
-
-export SRCDIR
