--- webrtc-audio-processing/webrtc/typedefs.h.orig	2019-02-20 16:32:17.346940276 +0100
+++ webrtc-audio-processing/webrtc/typedefs.h	2019-02-20 16:32:30.486986942 +0100
@@ -47,6 +47,12 @@
 #elif defined(__pnacl__)
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
+#elif defined(__powerpc64__)
+#define WEBRTC_ARCH_64_BITS
+#define WEBRTC_ARCH_LITTLE_ENDIAN
+#elif defined(__powerpc__)
+#define WEBRTC_ARCH_32_BITS
+#define WEBRTC_ARCH_BIG_ENDIAN
 #else
 #error Please add support for your architecture in typedefs.h
 #endif
