--- urllib3-2.5.0/pyproject.toml.orig	2025-09-23 12:14:24.870821705 +0200
+++ urllib3-2.5.0/pyproject.toml	2025-09-23 12:14:31.247870705 +0200
@@ -1,7 +1,7 @@
 # This file is protected via CODEOWNERS
 
 [build-system]
-requires = ["hatchling>=1.6.0,<2", "hatch-vcs>=0.4.0,<0.6.0", "setuptools-scm>=8,<9"]
+requires = ["hatchling>=1.6.0,<2", "hatch-vcs>=0.4.0,<0.6.0", "setuptools-scm>=8,<10"]
 build-backend = "hatchling.build"
 
 [project]
