diff -ruN konqueror-18.12.2.orig/CMakeLists.txt konqueror-18.12.2/CMakeLists.txt
--- konqueror-18.12.2.orig/CMakeLists.txt	2019-03-13 12:14:42.000000000 +0000
+++ konqueror-18.12.2/CMakeLists.txt	2019-03-13 12:36:38.000000000 +0000
@@ -23,7 +23,8 @@
 set(KONQUEROR_LIB_VERSION "5.0.97")
 set(KONQUEROR_VERSION "${KONQUEROR_LIB_VERSION}")
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets WebEngineWidgets)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets)
+find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS WebEngineWidgets)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS Parts KCMUtils KHtml KDELibs4Support Archive Crash)
 
 find_package(KF5 ${KF5_MIN_VERSION} COMPONENTS Activities DocTools) # Optional
@@ -55,7 +56,9 @@
 if(BUILD_TESTING)
     add_subdirectory( autotests )
 endif()
-add_subdirectory( webenginepart )
+if (Qt5WebEngineWidgets_FOUND)
+    add_subdirectory( webenginepart )
+endif()
 
 add_subdirectory( about )
 add_subdirectory( pics )
diff -ruN konqueror-18.12.2.orig/autotests/CMakeLists.txt konqueror-18.12.2/autotests/CMakeLists.txt
--- konqueror-18.12.2.orig/autotests/CMakeLists.txt	2019-03-13 12:14:42.000000000 +0000
+++ konqueror-18.12.2/autotests/CMakeLists.txt	2019-03-13 12:35:38.000000000 +0000
@@ -7,12 +7,14 @@
 
 ########### konqviewmgrtest ###############
 
-add_executable(konqviewmgrtest konqviewmgrtest.cpp)
-add_test(konqviewmgrtest konqviewmgrtest)
-ecm_mark_as_test(konqviewmgrtest)
-target_link_libraries(konqviewmgrtest kdeinit_konqueror Qt5::Core Qt5::Gui
-   # KF5::KHtml
-   kwebenginepartlib Qt5::WebEngineWidgets Qt5::Test)
+if (Qt5WebEngineWidgets_FOUND)
+  add_executable(konqviewmgrtest konqviewmgrtest.cpp)
+  add_test(konqviewmgrtest konqviewmgrtest)
+  ecm_mark_as_test(konqviewmgrtest)
+  target_link_libraries(konqviewmgrtest kdeinit_konqueror Qt5::Core Qt5::Gui
+     # KF5::KHtml
+     kwebenginepartlib Qt5::WebEngineWidgets Qt5::Test)
+endif()
 
 ########### historymanagertest ###############
 
@@ -30,10 +32,12 @@
 
 ########### konqhtmltest ###############
 
-add_executable(konqhtmltest konqhtmltest.cpp)
-add_test(konqhtmltest konqhtmltest)
-ecm_mark_as_test(konqhtmltest)
-target_link_libraries(konqhtmltest kdeinit_konqueror kwebenginepartlib Qt5::Core Qt5::Test)
+if (Qt5WebEngineWidgets_FOUND)
+  add_executable(konqhtmltest konqhtmltest.cpp)
+  add_test(konqhtmltest konqhtmltest)
+  ecm_mark_as_test(konqhtmltest)
+  target_link_libraries(konqhtmltest kdeinit_konqueror kwebenginepartlib Qt5::Core Qt5::Test)
+endif()
 
 ########### konqviewtest ###############
 
