From 2a1d7d7c346569c87c63436f52913de97d1d4a16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 22 Oct 2025 15:59:49 +0200
Subject: [PATCH] Don't overwrite PATH if already defined

---
 src/lxdm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lxdm.c b/src/lxdm.c
index dca2924..aac7de2 100644
--- a/src/lxdm.c
+++ b/src/lxdm.c
@@ -1481,8 +1481,8 @@ void lxdm_do_login(struct passwd *pw, char *session, char *lang, char *option)
 	path = g_key_file_get_string(config, "base", "path", 0);
 	if( G_UNLIKELY(path) && path[0] ) /* if PATH is specified in config file */
 		env=g_environ_setenv(env, "PATH", path, TRUE); /* override current $PATH with config value */
-	else /* don't use the global env, they are bad for user */
-		env=g_environ_setenv(env, "PATH", "/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin", TRUE); /* set proper default */
+	else if(!getenv("PATH")) /* if PATH is not set */
+		env=g_environ_setenv(env, "PATH", "/usr/local/bin:/usr/bin:/usr/local/sbin", TRUE); /* set proper default */
 	g_free(path);
 	/* optionally override $LANG, $LC_MESSAGES, and $LANGUAGE */
 	if( lang && lang[0] )
-- 
2.51.0

