diff --git a/setup.py b/setup.py
index a7b7e817..c7f0a308 100755
--- a/setup.py
+++ b/setup.py
@@ -22,6 +22,7 @@ import sys
 import sysconfig
 import tempfile
 import warnings
+from setuptools import find_packages
 
 
 with warnings.catch_warnings():
@@ -485,7 +486,7 @@ def main():
         url='https://github.com/giampaolo/psutil',
         platforms='Platform Independent',
         license='BSD-3-Clause',
-        packages=['psutil', 'psutil.tests'],
+        packages=find_packages(exclude=["psutil.tests", "psutil.tests.*"]),
         ext_modules=extensions,
         options=options,
         classifiers=[
