diff -ru stoken-0.92.orig/src/stc-nettle.c stoken-0.92/src/stc-nettle.c
--- stoken-0.92.orig/src/stc-nettle.c	2026-05-01 09:38:58.466337262 +0200
+++ stoken-0.92/src/stc-nettle.c	2026-05-01 09:41:24.039458066 +0200
@@ -35,7 +35,8 @@
 #include <nettle/cbc.h>
 #include <nettle/hmac.h>
 #include <nettle/rsa.h>
-#include <nettle/sha.h>
+#include <nettle/sha1.h>
+#include <nettle/version.h>
 
 #include "stoken-internal.h"
 
@@ -93,7 +94,11 @@
 		sha1_update(&md, in_len, in);
 	}
 	va_end(ap);
+#if NETTLE_VERSION_MAJOR > 3
+	sha1_digest(&md, out);
+#else
 	sha1_digest(&md, SHA1_DIGEST_SIZE, out);
+#endif
 }
 
 void stc_sha256_hash(uint8_t *out, ...)
@@ -113,7 +118,11 @@
 		sha256_update(&md, in_len, in);
 	}
 	va_end(ap);
+#if NETTLE_VERSION_MAJOR > 3
+	sha256_digest(&md, out);
+#else
 	sha256_digest(&md, SHA256_DIGEST_SIZE, out);
+#endif
 }
 
 int stc_b64_encode(const uint8_t *in,  unsigned long len,
