diff --git a/deps/v8/src/codegen/ppc/assembler-ppc.cc b/deps/v8/src/codegen/ppc/assembler-ppc.cc
index 2c9b33f3d84..694b750a1fa 100644
--- a/deps/v8/src/codegen/ppc/assembler-ppc.cc
+++ b/deps/v8/src/codegen/ppc/assembler-ppc.cc
@@ -114,7 +114,9 @@ void CpuFeatures::ProbeImpl(bool cross_compile) {
   // This variable is only used for certain archs to query SupportWasmSimd128()
   // at runtime in builtins using an extern ref. Other callers should use
   // CpuFeatures::SupportWasmSimd128().
-  CpuFeatures::supports_wasm_simd_128_ = CpuFeatures::SupportsWasmSimd128();
+  if (supported_.contains(PPC_9_PLUS)) {
+    CpuFeatures::supports_wasm_simd_128_ = CpuFeatures::SupportsWasmSimd128();
+  }
 }
 
 void CpuFeatures::PrintTarget() {
diff --git a/deps/v8/src/base/cpu.cc b/deps/v8/src/base/cpu.cc
index f64e1e71bed..5e4b2245e72 100644
--- a/deps/v8/src/base/cpu.cc
+++ b/deps/v8/src/base/cpu.cc
@@ -1018,6 +1018,8 @@ CPU::CPU()
       part_ = kPPCPower10;
     } else if (strcmp(auxv_cpu_type, "power9") == 0) {
       part_ = kPPCPower9;
+    } else if (strcmp(auxv_cpu_type, "power8") == 0) {
+      part_ = kPPCPower8;
     }
   }
 
diff --git a/deps/v8/src/base/cpu.h b/deps/v8/src/base/cpu.h
index 62c65e54031..a6f4971e079 100644
--- a/deps/v8/src/base/cpu.h
+++ b/deps/v8/src/base/cpu.h
@@ -64,7 +64,7 @@ class V8_BASE_EXPORT CPU final {
   static const int kNvidiaDenverV10 = 0x002;
 
   // PPC-specific part codes
-  enum { kPPCPower9, kPPCPower10, kPPCPower11 };
+  enum { kPPCPower8, kPPCPower9, kPPCPower10, kPPCPower11 };
 
   // General features
   bool has_fpu() const { return has_fpu_; }
