--- node-17.3.0/node.gypi.orig	2022-01-18 21:15:38.301884151 +0100
+++ node-17.3.0/node.gypi	2022-01-18 21:17:26.642868147 +0100
@@ -147,8 +147,15 @@
     [ 'node_shared_http_parser=="false"', {
       'dependencies': [
         'deps/llhttp/llhttp.gyp:llhttp'
+        'deps/http_parser/http_parser.gyp:http_parser'
       ],
     } ],
+ 
+    [ '"true"', { 'dependencies': [
+        'deps/llhttp/llhttp.gyp:llhttp'
+      ],
+    }],
+
 
     [ 'node_shared_cares=="false"', {
       'dependencies': [ 'deps/cares/cares.gyp:cares' ],
