diff --unified --recursive --text --new-file invoke-3.0.0.orig/tests/_util.py invoke-3.0.0/tests/_util.py
--- invoke-3.0.0.orig/tests/_util.py	2026-04-06 11:00:21.863961546 +0200
+++ invoke-3.0.0/tests/_util.py	2026-04-06 11:00:35.574159865 +0200
@@ -195,7 +195,7 @@
     def decorator(f):
         import fcntl
 
-        ioctl_patch = patch("invoke.runners.fcntl.ioctl", wraps=fcntl.ioctl)
+        ioctl_patch = patch("invoke.runners.fcntl.ioctl")
 
         @wraps(f)
         @patch("invoke.runners.pty")
