Description: Add missing header includes.
Author: Andrey Rakhmatullin <wrar@debian.org>
Bug-Debian: https://bugs.debian.org/1066208
Last-Update: 2024-03-16

Index: htdig-3.2.0b6/db/os_map.c
===================================================================
--- htdig-3.2.0b6.orig/db/os_map.c
+++ htdig-3.2.0b6/db/os_map.c
@@ -31,6 +31,8 @@ static const char sccsid[] = "@(#)os_map
 #endif
 
 #include "db_int.h"
+#include "db_page.h"
+#include "db_ext.h"
 #include "os_jump.h"
 
 #ifdef HAVE_MMAP
