--- htdig-3.2.0b6/htlib/md5.cc.orig	2025-04-06 17:24:48.122372440 +0200
+++ htdig-3.2.0b6/htlib/md5.cc	2025-04-06 17:25:01.730148954 +0200
@@ -1,6 +1,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <time.h>
+#include <cstring>
+
 extern "C" {
 #include "mhash_md5.h"
 }
--- htdig-3.2.0b6/htlib/mhash_md5.c.orig	2025-04-06 17:28:26.009161500 +0200
+++ htdig-3.2.0b6/htlib/mhash_md5.c	2025-04-06 17:28:56.550462772 +0200
@@ -32,6 +32,7 @@
    This is a slightly modified version 
  */
 
+#include <stdlib.h>
 #include "mhash_md5.h"
 
 /*
