From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Tue, 18 Jul 2023 03:46:01 +0200
Subject: [PATCH] Fix tests

---
 tests/gobject/docs/meson.build | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/tests/gobject/docs/meson.build b/tests/gobject/docs/meson.build
index a1dd66cadf09..2af2cf48caf5 100644
--- a/tests/gobject/docs/meson.build
+++ b/tests/gobject/docs/meson.build
@@ -5,6 +5,13 @@ gobject_test_source_dir = join_paths(
   'src'
 )
 
+gobject_test_examples_dir = join_paths(
+  srcdir,
+  'tests',
+  'gobject',
+  'examples'
+)
+
 gobject_test_docs_dir = join_paths(
   srcdir,
   'tests',
@@ -119,6 +126,7 @@ test(
     '--path=@0@'.format(':'.join([
       gobject_test_docs_dir,
       srcdir,
+      gobject_test_examples_dir,
       ])
     ),
   ],
