--- qtdeclarative/src/quick/scenegraph/qsgcontextplugin.cpp.orig	2024-10-28 15:29:27.507368927 +0100
+++ qtdeclarative/src/quick/scenegraph/qsgcontextplugin.cpp	2024-10-28 15:29:47.573289655 +0100
@@ -103,7 +103,7 @@
         // caused by run time hocus pocus. If one wants to use the software backend
         // in a GL or Vulkan capable Qt build (or on Windows or Apple platforms), it
         // has to be requested explicitly.
-#if !QT_CONFIG(opengl) && !QT_CONFIG(vulkan) && !QT_CONFIG(metal) && !defined(Q_OS_WIN)
+#if (!QT_CONFIG(opengl) && !QT_CONFIG(vulkan) && !QT_CONFIG(metal) && !defined(Q_OS_WIN)) || !defined(Q_LITTLE_ENDIAN)
         if (requestedBackend.isEmpty())
             requestedBackend = QLatin1String("software");
 #endif
