--- qtdeclarative-everywhere-src-6.6.1/src/quick/scenegraph/qsgcontextplugin.cpp.orig	2024-02-06 17:15:33.931417093 +0100
+++ qtdeclarative-everywhere-src-6.6.1/src/quick/scenegraph/qsgcontextplugin.cpp	2024-02-06 17:17:13.193370035 +0100
@@ -103,7 +103,7 @@
         // caused by run time hocus pocus. If one wants to use the software backend
         // in a GL or Vulkan capable Qt build (or on Windows or Apple platforms), it
         // has to be requested explicitly.
-#if !QT_CONFIG(opengl) && !QT_CONFIG(vulkan) && !defined(Q_OS_WIN) && !defined(Q_OS_MACOS) && !defined(Q_OS_IOS)
+#if (!QT_CONFIG(opengl) && !QT_CONFIG(vulkan) && !defined(Q_OS_WIN) && !defined(Q_OS_MACOS) && !defined(Q_OS_IOS)) || !defined(Q_LITTLE_ENDIAN)
         if (requestedBackend.isEmpty())
             requestedBackend = QLatin1String("software");
 #endif
