From: Celejar <celejar@gmail.com>
Date: Fri, 28 Feb 2020 05:30:01 +0000
Subject: Only warn once about the missing ICC support

---
 source/fitz/colorspace.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/source/fitz/colorspace.c b/source/fitz/colorspace.c
index d1a59ed..34ef00c 100644
--- a/source/fitz/colorspace.c
+++ b/source/fitz/colorspace.c
@@ -109,7 +109,12 @@ void fz_new_colorspace_context(fz_context *ctx)
 
 void fz_enable_icc(fz_context *ctx)
 {
-	fz_warn(ctx, "ICC support is not available");
+	static int warned = 0;
+
+	if (!warned) {
+		fz_warn(ctx, "ICC support is not available");
+		warned = 1;
+	}
 }
 
 void fz_disable_icc(fz_context *ctx)
