From: Kan-Ru Chen <koster@debian.org>
Date: Sat, 18 Nov 2023 17:30:57 +0900
Subject: MuPDF crossbuild use host cc for utils

---
 Makerules | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makerules b/Makerules
index 79fc615..b7c6014 100644
--- a/Makerules
+++ b/Makerules
@@ -10,6 +10,7 @@ ifeq ($(OS),)
 endif
 
 PKG_CONFIG ?= pkg-config
+CC_FOR_BUILD ?= $(CC)
 HAVE_LIBDL ?= yes
 WARNING_CFLAGS := -Wall -Wsign-compare
 
