From: Kan-Ru Chen <koster@debian.org>
Date: Mon, 12 Feb 2024 10:31:51 +0900
Subject: define SONAME header

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index bad2478..637f1f9 100644
--- a/Makefile
+++ b/Makefile
@@ -42,6 +42,7 @@ VERSION_PATCH = $(shell grep "define FZ_VERSION_PATCH" include/mupdf/fitz/versio
 
 ifeq ($(LINUX_OR_OPENBSD),yes)
   SO_VERSION = .$(VERSION_MINOR).$(VERSION_PATCH)
+  SONAME_LDFLAGS = -Wl,-soname,libmupdf.so$(SO_VERSION)
 endif
 
 # --- Commands ---
@@ -93,7 +94,7 @@ $(OUT)/%.exe: %.c
 	$(LINK_CMD)
 
 $(OUT)/%.$(SO)$(SO_VERSION):
-	$(LINK_CMD) $(LIB_LDFLAGS) $(THIRD_LIBS) $(LIBCRYPTO_LIBS)
+	$(LINK_CMD) $(SONAME_LDFLAGS) $(LIB_LDFLAGS) $(THIRD_LIBS) $(LIBCRYPTO_LIBS)
 ifneq ($(SO_VERSION),)
 	ln -sf $(notdir $@) $(patsubst %$(SO_VERSION), %, $@)
 endif
