--- kbd/src/libkeymap/dump.c.orig	2023-08-02 00:28:16.774469479 +0200
+++ kbd/src/libkeymap/dump.c	2023-08-02 00:28:18.254512204 +0200
@@ -72,14 +72,9 @@
 			continue;
 
 		for (j = 0; j < NR_KEYS / 2; j++) {
-			int value = lk_get_key(ctx, i, j);
+			int v = lk_get_key(ctx, i, j);
 
-			if (value < 0 || value > USHRT_MAX) {
-				ERR(ctx, _("can not bind key %d to value %d because it is too large"), j, value);
-				goto fail;
-			}
-
-			if (fwrite(&value, sizeof(unsigned short), 1, fd) != 1)
+			if (fwrite(&v, sizeof(v), 1, fd) != 1)
 				goto fail;
 		}
 	}
