--- libcmis-0.5.2/inc/libcmis-c/repository.h.orig	2019-05-29 07:32:24.070000000 +0000
+++ libcmis-0.5.2/inc/libcmis-c/repository.h	2019-05-29 07:32:53.440000000 +0000
@@ -28,15 +28,15 @@
 #ifndef _REPOSITORY_H_
 #define _REPOSITORY_H_
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <libxml/tree.h>
 
 #include "libcmis-c/libcmis-c-api.h"
 #include "libcmis-c/types.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 LIBCMIS_C_API void libcmis_vector_repository_free( libcmis_vector_Repository_Ptr vector );
 LIBCMIS_C_API size_t libcmis_vector_repository_size( libcmis_vector_Repository_Ptr vector );
 LIBCMIS_C_API libcmis_RepositoryPtr libcmis_vector_repository_get( libcmis_vector_Repository_Ptr vector, size_t i );
