--- messagelib-18.12.3/CMakeLists.txt.orig	2019-04-10 10:10:37.106327374 +0000
+++ messagelib-18.12.3/CMakeLists.txt	2019-04-10 10:11:07.603601792 +0000
@@ -69,7 +69,7 @@
 find_package(QGpgme ${GPGME_LIB_VERSION} CONFIG REQUIRED)
 
 if (NOT ${MIMETREEPARSER_ONLY_BUILD})
-   find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Widgets Network PrintSupport WebEngine WebEngineWidgets)
+   find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Widgets Network PrintSupport)
    find_package(KF5Archive ${KF5_VERSION} CONFIG REQUIRED)
    find_package(KF5Completion ${KF5_VERSION} CONFIG REQUIRED)
    find_package(KF5Config ${KF5_VERSION} CONFIG REQUIRED)
@@ -126,7 +126,6 @@
    add_subdirectory(messagecomposer)
    add_subdirectory(messagecore)
    add_subdirectory(messagelist)
-   add_subdirectory(webengineviewer)
 endif()
 
 install( FILES messagelib.renamecategories messagelib.categories DESTINATION ${KDE_INSTALL_CONFDIR} )
