diff -Naur pulseaudio-14.2.orig/src/modules/meson.build pulseaudio-14.2/src/modules/meson.build
--- pulseaudio-14.2.orig/src/modules/meson.build	2021-01-16 18:26:36.000000000 +0100
+++ pulseaudio-14.2/src/modules/meson.build	2021-05-08 20:23:23.605579000 +0200
@@ -268,6 +268,8 @@
 # FIXME: meson doesn't support multiple RPATH arguments currently
 rpath_dirs = join_paths(privlibdir) + ':' + join_paths(modlibexecdir)
 
+common_args = '-I@0@'.format(meson.current_build_dir())
+
 foreach m : all_modules
   name = m[0]
   sources = m[1]
@@ -280,7 +282,8 @@
     sources,
     headers,
     include_directories : [configinc, topinc],
-    c_args : [pa_c_args, server_c_args, '-DPA_MODULE_NAME=' + name.underscorify()] + extra_flags,
+    cpp_args : common_args,
+    c_args : [common_args, pa_c_args, server_c_args, '-DPA_MODULE_NAME=' + name.underscorify()] + extra_flags,
     install : true,
     install_rpath : rpath_dirs,
     install_dir : modlibexecdir,
