From 8dae947b024215e7a3a2d9ea55459c1bd77244ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Miro=20Hron=C4=8Dok?= <miro@hroncok.cz>
Date: Fri, 18 Oct 2024 15:06:33 +0200
Subject: [PATCH] Minimal fix for setuptools 74+

---
 nose/commands.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nose/commands.py b/nose/commands.py
index db9fe31..6c54408 100644
--- a/nose/commands.py
+++ b/nose/commands.py
@@ -142,7 +142,7 @@ else:
             if self.distribution.install_requires:
                 self.distribution.fetch_build_eggs(
                     self.distribution.install_requires)
-            if self.distribution.tests_require:
+            if getattr(self.distribution, 'tests_require', None):
                 self.distribution.fetch_build_eggs(
                     self.distribution.tests_require)
 
-- 
2.46.2

