diff --git a/files/fail2ban-tmpfiles.conf b/files/fail2ban-tmpfiles.conf
index 68f8e345..f1308582 100644
--- a/files/fail2ban-tmpfiles.conf
+++ b/files/fail2ban-tmpfiles.conf
@@ -1 +1,4 @@
-D /run/fail2ban 0755 root root -
\ No newline at end of file
+D /run/fail2ban            0755 root root
+d /etc/fail2ban/fail2ban.d 0755 root root
+d /etc/fail2ban/jail.d     0755 root root
+d /var/lib/fail2ban        0755 root root
