diff --git a/sympy/core/evalf.py b/sympy/core/evalf.py
index da173bf6d2..f6b2df9347 100644
--- a/sympy/core/evalf.py
+++ b/sympy/core/evalf.py
@@ -13,7 +13,7 @@
 from mpmath import inf as mpmath_inf
 from mpmath.libmp import (from_int, from_man_exp, from_rational, fhalf,
                           fnan, finf, fninf, fnone, fone, fzero, mpf_abs, mpf_add,
-                          mpf_atan, mpf_atan2, mpf_cmp, mpf_cos, mpf_e, mpf_exp, mpf_log, mpf_lt,
+                          mpf_atan, mpf_atan2, mpf_cmp, mpf_cos, mpf_e, mpf_exp, mpf_ln, mpf_lt,
                           mpf_mul, mpf_neg, mpf_pi, mpf_pow, mpf_pow_int, mpf_shift, mpf_sin,
                           mpf_sqrt, normalize, round_nearest, to_int, to_str, mpf_tan)
 from mpmath.libmp.backend import MPZ
@@ -991,7 +991,7 @@ def evalf_log(expr: 'log', prec: int, options: OPT_DICT) -> TMP_RES:
 
     imaginary_term = (mpf_cmp(xre, fzero) < 0)
 
-    re = mpf_log(mpf_abs(xre), prec, rnd)
+    re = mpf_ln(mpf_abs(xre), prec, rnd)
     size = fastlog(re)
     if prec - size > workprec and re != fzero:
         from .add import Add
@@ -1000,7 +1000,7 @@ def evalf_log(expr: 'log', prec: int, options: OPT_DICT) -> TMP_RES:
         xre, xim, _, _ = evalf_add(add, prec, options)
         prec2 = workprec - fastlog(xre)
         # xre is now x - 1 so we add 1 back here to calculate x
-        re = mpf_log(mpf_abs(mpf_add(xre, fone, prec2)), prec, rnd)
+        re = mpf_ln(mpf_abs(mpf_add(xre, fone, prec2)), prec, rnd)
 
     re_acc = prec
 
