diff --git a/Makefile b/Makefile
index 0adf886..fa74c9a 100644
--- a/Makefile
+++ b/Makefile
@@ -11,8 +11,8 @@ install: pdisk fdisk
 	-rm -f pmac-fdisk mac-fdisk
 	ln pdisk mac-fdisk
 	ln fdisk pmac-fdisk
-	install -o root -g root -m 0755 mac-fdisk ${DESTDIR}/sbin
-	install -o root -g root -m 0755 pmac-fdisk ${DESTDIR}/sbin
+	install -o root -g root -m 0755 mac-fdisk ${DESTDIR}/bin/mac-fdisk
+	install -o root -g root -m 0755 pmac-fdisk ${DESTDIR}/bin/pmac-fdisk
 
 distribution:
 	cd ..; tar cvf pdisk.src.tar.`date +%y%m%d` --files-from pdisk/list.src
