diff -x '*~' -purN a/modules/codec/vpx.c b/modules/codec/vpx.c
--- a/modules/codec/vpx.c	2018-03-29 14:04:52.000000000 +0200
+++ b/modules/codec/vpx.c	2019-02-10 16:29:58.878911991 +0100
@@ -116,18 +116,6 @@ static const struct
     { VLC_CODEC_I440, VPX_IMG_FMT_I440, 8, 0 },
 
     { VLC_CODEC_YV12, VPX_IMG_FMT_YV12, 8, 0 },
-    { VLC_CODEC_YUVA, VPX_IMG_FMT_444A, 8, 0 },
-    { VLC_CODEC_YUYV, VPX_IMG_FMT_YUY2, 8, 0 },
-    { VLC_CODEC_UYVY, VPX_IMG_FMT_UYVY, 8, 0 },
-    { VLC_CODEC_YVYU, VPX_IMG_FMT_YVYU, 8, 0 },
-
-    { VLC_CODEC_RGB15, VPX_IMG_FMT_RGB555, 8, 0 },
-    { VLC_CODEC_RGB16, VPX_IMG_FMT_RGB565, 8, 0 },
-    { VLC_CODEC_RGB24, VPX_IMG_FMT_RGB24, 8, 0 },
-    { VLC_CODEC_RGB32, VPX_IMG_FMT_RGB32, 8, 0 },
-
-    { VLC_CODEC_ARGB, VPX_IMG_FMT_ARGB, 8, 0 },
-    { VLC_CODEC_BGRA, VPX_IMG_FMT_ARGB_LE, 8, 0 },
 
     { VLC_CODEC_GBR_PLANAR, VPX_IMG_FMT_I444, 8, 1 },
     { VLC_CODEC_GBR_PLANAR_10L, VPX_IMG_FMT_I44416, 10, 1 },
