diff -ru aubio-0.4.9.orig/python/tests/test_cvec.py aubio-0.4.9/python/tests/test_cvec.py
--- aubio-0.4.9.orig/python/tests/test_cvec.py	2024-06-22 20:52:30.040448670 +0200
+++ aubio-0.4.9/python/tests/test_cvec.py	2024-06-22 20:52:51.850653850 +0200
@@ -43,7 +43,7 @@
         spec = cvec(1024)
         spec.phas[39:-1] = -np.pi
         assert_equal(spec.phas[0:39], 0)
-        assert_equal(spec.phas[39:-1], -np.pi)
+        assert_equal(spec.phas[39:-1], np.float32(-np.pi))
         assert_equal(spec.norm, 0)
 
     def test_assign_cvec_with_other_cvec(self):
