diff -ruN a/libffado/admin/pyuic.py b/libffado/admin/pyuic.py
--- a/libffado/admin/pyuic.py	2022-02-25 13:25:10.000000000 +0100
+++ b/libffado/admin/pyuic.py	2024-04-06 20:43:09.741707732 +0200
@@ -21,7 +21,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-import imp
+from importlib import util
 
 def pyuic_action( target, source, env ):
 	env.Execute( "pyuic " + str( source[0] ) + " > " + str( target[0] ) )
@@ -34,7 +34,7 @@
 	context.Message( "Checking for pyuic (by checking for the python module pyqtconfig) " )
 	ret = True
 	try:
-		imp.find_module( "pyqtconfig" )
+		util.find_spec( "pyqtconfig" )
 	except ImportError:
 		ret = False
 	context.Result( ret )
diff -ruN a/libffado/admin/pyuic4.py b/libffado/admin/pyuic4.py
--- a/libffado/admin/pyuic4.py	2022-02-25 13:25:10.000000000 +0100
+++ b/libffado/admin/pyuic4.py	2024-04-06 20:43:42.868641509 +0200
@@ -21,7 +21,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-import imp
+from importlib import util
 
 def pyuic4_action( target, source, env ):
 	env.Execute( "pyuic4 " + str( source[0] ) + " > " + str( target[0] ) )
@@ -34,7 +34,7 @@
 	context.Message( "Checking for pyuic4 (by checking for the python module pyqtconfig) " )
 	ret = True
 	try:
-		imp.find_module( "pyqtconfig" )
+		util.find_spec( "pyqtconfig" )
 	except ImportError:
 		ret = False
 	context.Result( ret )
diff -ruN a/libffado/admin/pyuic5.py b/libffado/admin/pyuic5.py
--- a/libffado/admin/pyuic5.py	2022-02-25 13:25:10.000000000 +0100
+++ b/libffado/admin/pyuic5.py	2024-04-06 20:42:25.674679876 +0200
@@ -22,7 +22,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-import imp
+from importlib import util
 
 def pyuic5_action( target, source, env ):
 	env.Execute( "pyuic5 " + str( source[0] ) + " > " + str( target[0] ) )
@@ -35,7 +35,7 @@
 	context.Message( "Checking for pyuic5 (by checking for the python module pyqtconfig) " )
 	ret = True
 	try:
-		imp.find_module( "pyqtconfig" )
+		util.find_spec( "pyqtconfig" )
 	except ImportError:
 		ret = False
 	context.Result( ret )
