--- wpewebkit-2.51.1/Source/WebKit/PlatformWPE.cmake.orig	2025-11-17 10:02:02.188000000 +0000
+++ wpewebkit-2.51.1/Source/WebKit/PlatformWPE.cmake	2025-11-17 10:02:38.575226748 +0000
@@ -368,7 +368,6 @@
 set(WebKitResources "")
 list(APPEND WebKitResources "        <file alias='css/wpe-theme.css'>wpe-theme.css</file>\n"
   "        <file alias='images/missingImage@2x'>missingImage@2x.png</file>\n"
-  "        <file alias='images/missingImage@3x'>missingImage@3x.png</file>\n"
   "        <file alias='images/missingImage'>missingImage.png</file>\n"
   "        <file alias='images/panIcon'>panIcon.png</file>\n"
   "        <file alias='images/textAreaResizeCorner@2x'>textAreaResizeCorner@2x.png</file>\n"
