diff -ruN kdepim-runtime-19.04.3.orig/CMakeLists.txt kdepim-runtime-19.04.3/CMakeLists.txt
--- kdepim-runtime-19.04.3.orig/CMakeLists.txt	2019-07-30 10:40:09.700000000 +0000
+++ kdepim-runtime-19.04.3/CMakeLists.txt	2019-07-30 10:40:00.860000000 +0000
@@ -98,7 +98,7 @@
 find_package(Qca-qt5)
 
 # QT5 package
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns DBus WebEngineWidgets NetworkAuth)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns DBus NetworkAuth)
 if (NOT Qca-qt5_FOUND)
     message(STATUS "QCA not found, public key authentication will not be supported")
 else()
diff -ruN kdepim-runtime-19.04.3.orig/resources/CMakeLists.txt kdepim-runtime-19.04.3/resources/CMakeLists.txt
--- kdepim-runtime-19.04.3.orig/resources/CMakeLists.txt	2019-07-30 10:40:09.700000000 +0000
+++ kdepim-runtime-19.04.3/resources/CMakeLists.txt	2019-07-30 10:40:00.860000000 +0000
@@ -45,7 +45,6 @@
 if (Libkolabxml_FOUND)
     add_subdirectory( kolab )
 endif()
-add_subdirectory( facebook )
 add_subdirectory( maildir )
 
 add_subdirectory( openxchange )
@@ -64,4 +63,3 @@
 add_subdirectory( vcard )
 add_subdirectory( folderarchivesettings )
 add_subdirectory( tomboynotes )
-add_subdirectory( ews )
diff -ruN kdepim-runtime-19.04.3.orig/resources/tomboynotes/CMakeLists.txt kdepim-runtime-19.04.3/resources/tomboynotes/CMakeLists.txt
--- kdepim-runtime-19.04.3.orig/resources/tomboynotes/CMakeLists.txt	2019-07-30 10:40:09.700000000 +0000
+++ kdepim-runtime-19.04.3/resources/tomboynotes/CMakeLists.txt	2019-07-30 10:40:00.860000000 +0000
@@ -61,7 +61,6 @@
     Qt5::DBus
     Qt5::Gui
     Qt5::Network
-    Qt5::WebEngineWidgets
     KF5::AkonadiAgentBase
     KF5::AkonadiNotes
     KF5::ConfigCore
diff -ruN kdepim-runtime-19.04.3.orig/resources/tomboynotes/tomboyserverauthenticatejob.cpp kdepim-runtime-19.04.3/resources/tomboynotes/tomboyserverauthenticatejob.cpp
--- kdepim-runtime-19.04.3.orig/resources/tomboynotes/tomboyserverauthenticatejob.cpp	2019-07-30 10:40:09.700000000 +0000
+++ kdepim-runtime-19.04.3/resources/tomboynotes/tomboyserverauthenticatejob.cpp	2019-07-30 10:40:18.550000000 +0000
@@ -25,19 +25,14 @@
 
 TomboyServerAuthenticateJob::TomboyServerAuthenticateJob(KIO::AccessManager *manager, QObject *parent)
     : TomboyJobBase(manager, parent)
-    , mWebView(new QWebEngineView(nullptr))
 {
-    mWebView->setContextMenuPolicy(Qt::NoContextMenu);
     // Connect the o2 authenfication signals
     connect(mO1, &O1::linkingFailed, this, &TomboyServerAuthenticateJob::onLinkingFailed);
     connect(mO1, &O1::linkingSucceeded, this, &TomboyServerAuthenticateJob::onLinkingSucceeded);
-    connect(mO1, &O1::openBrowser, this, &TomboyServerAuthenticateJob::onOpenBrowser);
-    connect(mO1, &O1::closeBrowser, mWebView, &QWebEngineView::close);
 }
 
 TomboyServerAuthenticateJob::~TomboyServerAuthenticateJob()
 {
-    delete mWebView;
 }
 
 void TomboyServerAuthenticateJob::start()
@@ -83,8 +78,6 @@
 
 void TomboyServerAuthenticateJob::onOpenBrowser(const QUrl &url)
 {
-    mWebView->setUrl(url);
-    mWebView->show();
 }
 
 void TomboyServerAuthenticateJob::onApiRequestFinished()
diff -ruN kdepim-runtime-19.04.3.orig/resources/tomboynotes/tomboyserverauthenticatejob.h kdepim-runtime-19.04.3/resources/tomboynotes/tomboyserverauthenticatejob.h
--- kdepim-runtime-19.04.3.orig/resources/tomboynotes/tomboyserverauthenticatejob.h	2019-07-30 10:40:09.700000000 +0000
+++ kdepim-runtime-19.04.3/resources/tomboynotes/tomboyserverauthenticatejob.h	2019-07-30 10:40:00.860000000 +0000
@@ -22,7 +22,6 @@
 
 #include "tomboyjobbase.h"
 #include <QString>
-#include <QWebEngineView>
 
 class TomboyServerAuthenticateJob : public TomboyJobBase
 {
@@ -47,8 +46,6 @@
     void onApiRequestFinished();
     void onUserRequestFinished();
     QString mUserURL;
-
-    QWebEngineView *mWebView = nullptr;
 };
 
 #endif // TOMBOYSERVERAUTHENTICATEJOB_H
