diff -ruN kdepim-runtime-21.08.0.orig/CMakeLists.txt kdepim-runtime-21.08.0/CMakeLists.txt
--- kdepim-runtime-21.08.0.orig/CMakeLists.txt	2021-08-27 10:36:36.223807726 +0200
+++ kdepim-runtime-21.08.0/CMakeLists.txt	2021-08-27 10:36:42.747161765 +0200
@@ -101,7 +101,7 @@
                    PURPOSE "Needed for ews resource.")
 option(KDEPIM_RUN_AKONADI_TEST "Enable autotest based on Akonadi." TRUE)
 # QT5 package
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns DBus WebEngineWidgets NetworkAuth)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns DBus NetworkAuth)
 
 
 # KF5 package
diff -ruN kdepim-runtime-21.08.0.orig/resources/CMakeLists.txt kdepim-runtime-21.08.0/resources/CMakeLists.txt
--- kdepim-runtime-21.08.0.orig/resources/CMakeLists.txt	2021-08-27 10:36:36.151814852 +0200
+++ kdepim-runtime-21.08.0/resources/CMakeLists.txt	2021-08-27 10:36:42.747161765 +0200
@@ -74,4 +74,3 @@
 add_subdirectory( vcard )
 add_subdirectory( folderarchivesettings )
 add_subdirectory( tomboynotes )
-add_subdirectory( ews )
diff -ruN kdepim-runtime-21.08.0.orig/resources/tomboynotes/CMakeLists.txt kdepim-runtime-21.08.0/resources/tomboynotes/CMakeLists.txt
--- kdepim-runtime-21.08.0.orig/resources/tomboynotes/CMakeLists.txt	2021-08-27 10:36:36.211808913 +0200
+++ kdepim-runtime-21.08.0/resources/tomboynotes/CMakeLists.txt	2021-08-27 10:36:59.385512468 +0200
@@ -68,7 +68,6 @@
     Qt::DBus
     Qt::Gui
     Qt::Network
-    Qt::WebEngineWidgets
     KF5::AkonadiAgentBase
     KF5::AkonadiNotes
     KF5::ConfigCore
diff -ruN kdepim-runtime-21.08.0.orig/resources/tomboynotes/tomboyserverauthenticatejob.cpp kdepim-runtime-21.08.0/resources/tomboynotes/tomboyserverauthenticatejob.cpp
--- kdepim-runtime-21.08.0.orig/resources/tomboynotes/tomboyserverauthenticatejob.cpp	2021-08-27 10:36:36.211808913 +0200
+++ kdepim-runtime-21.08.0/resources/tomboynotes/tomboyserverauthenticatejob.cpp	2021-08-27 10:36:42.747161765 +0200
@@ -12,19 +12,14 @@
 
 TomboyServerAuthenticateJob::TomboyServerAuthenticateJob(KIO::AccessManager *manager, QObject *parent)
     : TomboyJobBase(manager, parent)
-    , mWebView(new QWebEngineView(nullptr))
 {
-    mWebView->setContextMenuPolicy(Qt::NoContextMenu);
     // Connect the o2 authenfication signals
     connect(mO1, &O1::linkingFailed, this, &TomboyServerAuthenticateJob::onLinkingFailed);
     connect(mO1, &O1::linkingSucceeded, this, &TomboyServerAuthenticateJob::onLinkingSucceeded);
-    connect(mO1, &O1::openBrowser, this, &TomboyServerAuthenticateJob::onOpenBrowser);
-    connect(mO1, &O1::closeBrowser, mWebView, &QWebEngineView::close);
 }
 
 TomboyServerAuthenticateJob::~TomboyServerAuthenticateJob()
 {
-    delete mWebView;
 }
 
 void TomboyServerAuthenticateJob::start()
@@ -70,8 +65,6 @@
 
 void TomboyServerAuthenticateJob::onOpenBrowser(const QUrl &url)
 {
-    mWebView->setUrl(url);
-    mWebView->show();
 }
 
 void TomboyServerAuthenticateJob::onApiRequestFinished()
diff -ruN kdepim-runtime-21.08.0.orig/resources/tomboynotes/tomboyserverauthenticatejob.h kdepim-runtime-21.08.0/resources/tomboynotes/tomboyserverauthenticatejob.h
--- kdepim-runtime-21.08.0.orig/resources/tomboynotes/tomboyserverauthenticatejob.h	2021-08-27 10:36:36.211808913 +0200
+++ kdepim-runtime-21.08.0/resources/tomboynotes/tomboyserverauthenticatejob.h	2021-08-27 10:36:42.747161765 +0200
@@ -8,7 +8,6 @@
 
 #include "tomboyjobbase.h"
 #include <QString>
-#include <QWebEngineView>
 
 class TomboyServerAuthenticateJob : public TomboyJobBase
 {
@@ -33,7 +32,5 @@
     void onApiRequestFinished();
     void onUserRequestFinished();
     QString mUserURL;
-
-    QWebEngineView *const mWebView;
 };
 

