diff -ruN kdepim-runtime-21.04.2.orig/CMakeLists.txt kdepim-runtime-21.04.2/CMakeLists.txt
--- kdepim-runtime-21.04.2.orig/CMakeLists.txt	2021-07-14 21:41:46.262579726 +0200
+++ kdepim-runtime-21.04.2/CMakeLists.txt	2021-07-14 21:42:06.916521975 +0200
@@ -105,7 +105,7 @@
                    PURPOSE "Needed for ews resource.")
 option(KDEPIM_RUN_AKONADI_TEST "Enable autotest based on Akonadi." TRUE)
 # QT5 package
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns DBus WebEngineWidgets NetworkAuth)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns DBus NetworkAuth)
 
 
 # KF5 package
diff -ruN kdepim-runtime-21.04.2.orig/resources/CMakeLists.txt kdepim-runtime-21.04.2/resources/CMakeLists.txt
--- kdepim-runtime-21.04.2.orig/resources/CMakeLists.txt	2021-07-14 21:41:46.126593274 +0200
+++ kdepim-runtime-21.04.2/resources/CMakeLists.txt	2021-07-14 21:41:53.593849332 +0200
@@ -74,4 +74,3 @@
 add_subdirectory( vcard )
 add_subdirectory( folderarchivesettings )
 add_subdirectory( tomboynotes )
-add_subdirectory( ews )
diff -ruN kdepim-runtime-21.04.2.orig/resources/tomboynotes/CMakeLists.txt kdepim-runtime-21.04.2/resources/tomboynotes/CMakeLists.txt
--- kdepim-runtime-21.04.2.orig/resources/tomboynotes/CMakeLists.txt	2021-07-14 21:41:46.118594071 +0200
+++ kdepim-runtime-21.04.2/resources/tomboynotes/CMakeLists.txt	2021-07-14 21:41:53.597848934 +0200
@@ -68,7 +68,6 @@
     Qt5::DBus
     Qt5::Gui
     Qt5::Network
-    Qt5::WebEngineWidgets
     KF5::AkonadiAgentBase
     KF5::AkonadiNotes
     KF5::ConfigCore
diff -ruN kdepim-runtime-21.04.2.orig/resources/tomboynotes/tomboyserverauthenticatejob.cpp kdepim-runtime-21.04.2/resources/tomboynotes/tomboyserverauthenticatejob.cpp
--- kdepim-runtime-21.04.2.orig/resources/tomboynotes/tomboyserverauthenticatejob.cpp	2021-07-14 21:41:46.118594071 +0200
+++ kdepim-runtime-21.04.2/resources/tomboynotes/tomboyserverauthenticatejob.cpp	2021-07-14 21:41:53.597848934 +0200
@@ -12,19 +12,14 @@
 
 TomboyServerAuthenticateJob::TomboyServerAuthenticateJob(KIO::AccessManager *manager, QObject *parent)
     : TomboyJobBase(manager, parent)
-    , mWebView(new QWebEngineView(nullptr))
 {
-    mWebView->setContextMenuPolicy(Qt::NoContextMenu);
     // Connect the o2 authenfication signals
     connect(mO1, &O1::linkingFailed, this, &TomboyServerAuthenticateJob::onLinkingFailed);
     connect(mO1, &O1::linkingSucceeded, this, &TomboyServerAuthenticateJob::onLinkingSucceeded);
-    connect(mO1, &O1::openBrowser, this, &TomboyServerAuthenticateJob::onOpenBrowser);
-    connect(mO1, &O1::closeBrowser, mWebView, &QWebEngineView::close);
 }
 
 TomboyServerAuthenticateJob::~TomboyServerAuthenticateJob()
 {
-    delete mWebView;
 }
 
 void TomboyServerAuthenticateJob::start()
@@ -70,8 +65,6 @@
 
 void TomboyServerAuthenticateJob::onOpenBrowser(const QUrl &url)
 {
-    mWebView->setUrl(url);
-    mWebView->show();
 }
 
 void TomboyServerAuthenticateJob::onApiRequestFinished()
diff -ruN kdepim-runtime-21.04.2.orig/resources/tomboynotes/tomboyserverauthenticatejob.h kdepim-runtime-21.04.2/resources/tomboynotes/tomboyserverauthenticatejob.h
--- kdepim-runtime-21.04.2.orig/resources/tomboynotes/tomboyserverauthenticatejob.h	2021-07-14 21:41:46.118594071 +0200
+++ kdepim-runtime-21.04.2/resources/tomboynotes/tomboyserverauthenticatejob.h	2021-07-14 21:41:53.597848934 +0200
@@ -9,7 +9,6 @@
 
 #include "tomboyjobbase.h"
 #include <QString>
-#include <QWebEngineView>
 
 class TomboyServerAuthenticateJob : public TomboyJobBase
 {
@@ -34,8 +33,6 @@
     void onApiRequestFinished();
     void onUserRequestFinished();
     QString mUserURL;
-
-    QWebEngineView *const mWebView;
 };
 
 #endif // TOMBOYSERVERAUTHENTICATEJOB_H
