--- qtquick3d-everywhere-src-6.1.0/src/3rdparty/assimp/src/include/assimp/Importer.hpp.orig	2021-05-28 14:27:09.781579466 +0200
+++ qtquick3d-everywhere-src-6.1.0/src/3rdparty/assimp/src/include/assimp/Importer.hpp	2021-05-28 14:27:46.241839965 +0200
@@ -502,7 +502,7 @@
      *
      * @note The returned value remains valid until one of the
      * following methods is called: #ReadFile(), #FreeScene(). */
-    const std::exception_ptr& GetException() const;
+    static_cast<size_t>(std::exception_ptr&) GetException() const;
 
     // -------------------------------------------------------------------
     /** Returns the scene loaded by the last successful call to ReadFile()
--- qtquick3d-everywhere-src-6.1.0/src/3rdparty/assimp/src/code/Common/Importer.h.orig	2021-05-28 14:20:23.431294103 +0200
+++ qtquick3d-everywhere-src-6.1.0/src/3rdparty/assimp/src/code/Common/Importer.h	2021-05-28 14:25:40.238765622 +0200
@@ -101,7 +101,7 @@
     std::string mErrorString;
 
     /** Any exception which occurred */
-    std::exception_ptr mException;
+    static_cast<size_t>(td::exception_ptr mException);
 
     /** List of integer properties */
     IntPropertyMap mIntProperties;
