diff --git a/icmake/buildscripts/multicomp b/icmake/buildscripts/multicomp
index 2f2a7f96..5d914e6a 100644
--- a/icmake/buildscripts/multicomp
+++ b/icmake/buildscripts/multicomp
@@ -38,7 +38,7 @@ IFS='
     echo ": . tmp 0" >> jobs
     ls *.cc >> jobs
 
-    opts=" -Wall -Werror"
+    opts=" -Wall -Werror ${CXXFLAGS}"
 
     ../tmp/usr/libexec/icmake/icm-multicomp -q jobs \
         '/usr/bin/g++ -c -o $2 '${ICMAKE_CPPSTD}"${opts}"' $1' || exit 1
diff --git a/icmake/buildscripts/precomp b/icmake/buildscripts/precomp
index c977d28a..d2febede 100644
--- a/icmake/buildscripts/precomp
+++ b/icmake/buildscripts/precomp
@@ -5,7 +5,7 @@ precomp()
     echo "
 Buiding tmp/spch.gch and using .ih.gch files"
 
-    OPTS='g++ -Wall -Werror -O2 -c -o $2 '
+    OPTS='g++ -Wall -Werror -O2 ${CXXFLAGS} -c -o $2 '
     OPTS="$OPTS ${ICMAKE_CPPSTD}${opts} -x c++header \$1"
 
     ../tmp/usr/libexec/icmake/icm-spch $1 --all spch tmp/ "'${OPTS}'"
