diff --git a/icmake/buildscripts/multicomp b/icmake/buildscripts/multicomp
index 2f2a7f9..7cbc975 100644
--- a/icmake/buildscripts/multicomp
+++ b/icmake/buildscripts/multicomp
@@ -17,6 +17,7 @@ build()
                                # construct the jobs-file for icm-multicomp
     if [ -e CLASSES ] 
     then
+_OLDIFS=$IFS
 IFS='
 '
         for dir in `cat CLASSES` 
@@ -40,6 +41,7 @@ IFS='
 
     opts=" -Wall -Werror"
 
+IFS=$_OLDIFS
     ../tmp/usr/libexec/icmake/icm-multicomp -q jobs \
         '/usr/bin/g++ -c -o $2 '${ICMAKE_CPPSTD}"${opts}"' $1' || exit 1
  
