From 4d54c24e7637a82838b5b79caa8a09bbeb8f0d99 Mon Sep 17 00:00:00 2001
From: "A. Wilcox" <AWilcox@Wilcox-Tech.com>
Date: Sat, 11 Nov 2023 21:25:09 -0600
Subject: [PATCH 3/3] Disable Int10 support on all non-x86 arches

---
 src/r128_driver.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/r128_driver.c b/src/r128_driver.c
index 109c290..499ecd5 100644
--- a/src/r128_driver.c
+++ b/src/r128_driver.c
@@ -961,7 +961,7 @@ static Bool R128PreInitConfig(ScrnInfoPtr pScrn)
 
 static Bool R128PreInitDDC(ScrnInfoPtr pScrn, xf86Int10InfoPtr pInt10)
 {
-#if !defined(__powerpc__) && !defined(__alpha__) && !defined(__sparc__)
+#if defined(__i386__) || defined(__x86_64__)
     R128InfoPtr   info = R128PTR(pScrn);
     vbeInfoPtr pVbe;
 #endif
@@ -969,8 +969,8 @@ static Bool R128PreInitDDC(ScrnInfoPtr pScrn, xf86Int10InfoPtr pInt10)
     if (!xf86LoadSubModule(pScrn, "ddc")) return FALSE;
     if (!xf86LoadSubModule(pScrn, "i2c")) return FALSE;
 
-#if defined(__powerpc__) || defined(__alpha__) || defined(__sparc__)
-    /* Int10 is broken on PPC and some Alphas */
+#if !defined(__i386__) && !defined(__x86_64__)
+    /* Int10 is broken on non-x86 */
     return TRUE;
 #else
     if (xf86LoadSubModule(pScrn, "vbe")) {
@@ -1006,9 +1006,9 @@ static Bool R128PreInitCursor(ScrnInfoPtr pScrn)
 
 static Bool R128PreInitInt10(ScrnInfoPtr pScrn, xf86Int10InfoPtr *ppInt10)
 {
+#if defined(__i386__) || defined(__x86_64__)
     R128InfoPtr   info = R128PTR(pScrn);
-#if !defined(__powerpc__) && !defined(__alpha__)
-    /* int10 is broken on some Alphas and powerpc */
+    /* int10 is broken on non-x86 */
     if (xf86LoadSubModule(pScrn, "int10")) {
 	xf86DrvMsg(pScrn->scrnIndex,X_INFO,"initializing int10\n");
 	*ppInt10 = xf86InitInt10(info->pEnt->index);
-- 
2.39.3 (Apple Git-145)

