From 4f9f1ba9d55dd946b81c7bbb640a24010b4ce16c Mon Sep 17 00:00:00 2001
From: "A. Wilcox" <AWilcox@Wilcox-Tech.com>
Date: Sat, 11 Nov 2023 21:19:07 -0600
Subject: [PATCH 2/3] Only try to init XAA when XAA is available

---
 src/r128_driver.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/r128_driver.c b/src/r128_driver.c
index 213b948..109c290 100644
--- a/src/r128_driver.c
+++ b/src/r128_driver.c
@@ -1862,14 +1862,15 @@ Bool R128ScreenInit(SCREEN_INIT_ARGS_DECL)
 	        }
 
             if (!info->noAccel) {
+#ifdef HAVE_XAA_H
                 if (R128XAAAccelInit(pScreen)) {
                     info->accelOn = TRUE;
                     xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                                 "XAA acceleration enabled.\n");
-                } else {
+                } else
+#endif
                     xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                                 "Acceleration disabled.\n");
-                }
             }
         }
     }
-- 
2.39.3 (Apple Git-145)

