diff -aur six-1.16.0.old/test_six.py six-1.16.0/test_six.py
--- six-1.16.0.old/test_six.py	2024-11-16 14:59:08.926994916 +0100
+++ six-1.16.0/test_six.py	2024-11-16 14:59:45.527903915 +0100
@@ -127,6 +127,8 @@
         if item_name.startswith("tkinter"):
             if not have_tkinter:
                 pytest.skip("requires tkinter")
+            if item_name == "tkinter_tix" and sys.version_info >= (3, 13):
+                pytest.skip("tkinter.tix removed from Python 3.13")
         if item_name.startswith("dbm_gnu") and not have_gdbm:
             pytest.skip("requires gdbm")
         raise
