commit 458d9968b587da7b079d4ed95b8c4da9941dcbdc
Author: T.J. Townsend <blakkheim@archlinux.org>
Date:   Thu Dec 18 16:42:26 2025 -0500

    correct iptables location

diff --git a/src/util.py b/src/util.py
index 98e3c70..5b35cf9 100644
--- a/src/util.py
+++ b/src/util.py
@@ -721,9 +721,9 @@ def in_network(tested_add, tested_net, v6):
 # since these have historically used a default for 'exe'.
 def _find_system_iptables():
     exe = ""
-    for d in ["/sbin", "/bin",
-              "/usr/sbin", "/usr/bin",
-              "/usr/local/sbin", "/usr/local/bin"]:
+    for d in ["/bin", "/bin",
+              "/usr/bin", "/usr/bin",
+              "/usr/local/bin", "/usr/local/bin"]:
         exe = os.path.join(d, "iptables")
         if os.path.exists(exe):
             break
