commit 8578ff54588e3d5211922d2cda2f9e013d4e708d
Author: T.J. Townsend <blakkheim@archlinux.org>
Date:   Thu Dec 18 16:44:24 2025 -0500

    don't force-enable tcp selective acks

diff --git a/conf/sysctl.conf b/conf/sysctl.conf
index 4ea2032..042bf05 100644
--- a/conf/sysctl.conf
+++ b/conf/sysctl.conf
@@ -46,7 +46,7 @@ net/ipv4/conf/all/log_martians=0
 
 # normally allowing tcp_sack is ok, but if going through OpenBSD 3.8 RELEASE or
 # earlier pf firewall, should set this to 0
-net/ipv4/tcp_sack=1
+#net/ipv4/tcp_sack=1
 
 # Uncomment this to turn off ipv6 autoconfiguration
 #net/ipv6/conf/default/autoconf=0
