diff --git i/configure.ac w/configure.ac
index d9727be8..1d16c4d5 100644
--- i/configure.ac
+++ w/configure.ac
@@ -215,7 +215,7 @@ dnl Test for FLTK 1.3 library
 dnl -------------------------
 dnl
 dnl For debugging and to be user friendly
-AC_PATH_PROG(FLTK_CONFIG,fltk-config)
+AC_PATH_PROG(FLTK_CONFIG,fltk-config1.3)
 AC_MSG_CHECKING([FLTK 1.3])
 fltk_version="`$FLTK_CONFIG --version 2>/dev/null`"
 case $fltk_version in
@@ -228,7 +228,7 @@ case $fltk_version in
   ?*)    AC_MSG_RESULT(no)
          AC_MSG_ERROR(FLTK 1.3 required; version found: $fltk_version);;
   *)     AC_MSG_RESULT(no)
-         AC_MSG_ERROR(FLTK 1.3 required; fltk-config not found)
+         AC_MSG_ERROR(FLTK 1.3 required; fltk-config1.3 not found)
 esac
 
 dnl -----------------------------------
