From 55a4d30565b6b646c7324a24f11a03220e85aa83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 23 Oct 2025 01:31:13 +0200
Subject: [PATCH 1/2] Add MATE to NotShowIn for LXInput autostart

---
 src/lxinput.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxinput.c b/src/lxinput.c
index ee62cc8..de193cc 100644
--- a/src/lxinput.c
+++ b/src/lxinput.c
@@ -350,7 +350,7 @@ int main(int argc, char** argv)
                                   "Comment=%s\n"
                                   "NoDisplay=true\n"
                                   "Exec=sh -c 'xset m %d/10 %d r rate %d %d b %s%s'\n"
-                                  "NotShowIn=GNOME;KDE;XFCE;\n",
+                                  "NotShowIn=GNOME;KDE;XFCE;MATE;\n",
                                   _("LXInput autostart"),
                                   _("Setup keyboard and mouse using settings done in LXInput"),
                                   /* FIXME: how to setup left-handed mouse? */

From aae3857b404ff2861f1f9b44e394563b0a7df98c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 17 Jan 2026 19:02:15 +0100
Subject: [PATCH 2/2] Exclude more environments

These desktop environments have their own tools to configure the keyboard and mouse.
---
 data/lxinput.desktop.in | 2 +-
 src/lxinput.c           | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/lxinput.desktop.in b/data/lxinput.desktop.in
index 83d29b6..f314e3a 100644
--- a/data/lxinput.desktop.in
+++ b/data/lxinput.desktop.in
@@ -7,4 +7,4 @@ _Comment=Configure keyboard, mouse, and other input devices
 StartupNotify=true
 Terminal=false
 Categories=GTK;Settings;HardwareSettings;X-LXDE-Settings;
-NotShowIn=GNOME;KDE;XFCE;MATE;
+NotShowIn=GNOME;KDE;XFCE;MATE;Budgie;COSMIC;Enlightenment;LXQt;X-Cinnamon;
diff --git a/src/lxinput.c b/src/lxinput.c
index de193cc..be167b1 100644
--- a/src/lxinput.c
+++ b/src/lxinput.c
@@ -350,7 +350,7 @@ int main(int argc, char** argv)
                                   "Comment=%s\n"
                                   "NoDisplay=true\n"
                                   "Exec=sh -c 'xset m %d/10 %d r rate %d %d b %s%s'\n"
-                                  "NotShowIn=GNOME;KDE;XFCE;MATE;\n",
+                                  "NotShowIn=GNOME;KDE;XFCE;MATE;Budgie;COSMIC;Enlightenment;LXQt;X-Cinnamon;\n",
                                   _("LXInput autostart"),
                                   _("Setup keyboard and mouse using settings done in LXInput"),
                                   /* FIXME: how to setup left-handed mouse? */
