From 55a4d30565b6b646c7324a24f11a03220e85aa83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 23 Oct 2025 01:31:13 +0200
Subject: [PATCH] Add MATE to NotShowIn for LXInput autostart

---
 src/lxinput.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxinput.c b/src/lxinput.c
index ee62cc8..de193cc 100644
--- a/src/lxinput.c
+++ b/src/lxinput.c
@@ -350,7 +350,7 @@ int main(int argc, char** argv)
                                   "Comment=%s\n"
                                   "NoDisplay=true\n"
                                   "Exec=sh -c 'xset m %d/10 %d r rate %d %d b %s%s'\n"
-                                  "NotShowIn=GNOME;KDE;XFCE;\n",
+                                  "NotShowIn=GNOME;KDE;XFCE;MATE;\n",
                                   _("LXInput autostart"),
                                   _("Setup keyboard and mouse using settings done in LXInput"),
                                   /* FIXME: how to setup left-handed mouse? */
