--- linux-7.0.3/tools/objtool/disas.c	2026-05-03 17:14:28.199537969 +0000
+++ linux-7.0.3/tools/objtool/disas.c.orig	2026-05-03 16:57:37.309502395 +0000
@@ -109,7 +109,7 @@
 #define DINFO_FPRINTF(dinfo, ...)	\
 	((*(dinfo)->fprintf_func)((dinfo)->stream, __VA_ARGS__))
 #define bfd_vma_fmt			\
-	__builtin_choose_expr(sizeof(bfd_vma) == sizeof(unsigned long), "%#lx <%s>", "%#llx <%s>")
+	__builtin_choose_expr(sizeof(bfd_vma) == sizeof(unsigned long), "%#zx <%s>", "%#zx <%s>")
 
 static int disas_result_fprintf(struct disas_context *dctx,
 				const char *fmt, va_list ap)
