--- third_party/webrtc/BUILD.gn.orig	2020-08-19 01:16:33.415681318 +0200
+++ third_party/webrtc/BUILD.gn	2020-08-19 01:16:51.998818200 +0200
@@ -409,14 +409,6 @@
   }
 }
 
-config("common_objc") {
-  libs = [ "Foundation.framework" ]
-
-  if (rtc_use_metal_rendering) {
-    defines = [ "RTC_SUPPORTS_METAL" ]
-  }
-}
-
 if (!build_with_chromium) {
   # Target to build all the WebRTC production code.
   rtc_static_library("webrtc") {
