--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,7 +1,7 @@
 [build-system]
 requires = [
     "scikit-build-core>=0.11",
-    "Cython >=3.0.12, <3.2.0"
+    "Cython"
 ]
 build-backend = "scikit_build_core.build"
 
@@ -185,4 +185,4 @@ enable = ["cpython-freethreading", "pypy"]
 # in theory some tests require numpy / pandas as well. However those don't provider
 # wheels for all target platforms
 test-requires = ["pytest", "hypothesis"]
-test-command = "pytest {package}/tests"
\ No newline at end of file
+test-command = "pytest {package}/tests"
