--- goom2k4-0/configure.in.orig	2021-05-12 20:04:11.640971182 +0200
+++ goom2k4-0/configure.in	2021-05-12 20:08:00.893937178 +0200
@@ -19,17 +19,13 @@
 
 dnl *** check for xmms ***
 
-AM_PATH_XMMS2(0.9.5.1, HAVE_XMMS="yes", HAVE_XMMS="no")
+HAVE_XMMS="no"
 AM_CONDITIONAL(HAVE_XMMS,test "x$HAVE_XMMS" = "xyes")
 
 
 dnl *** SDL ***
 
-if test "x$HAVE_XMMS" = "xyes"; then
-  AM_PATH_SDL2(1.2.0, HAVE_SDL="yes", HAVE_SDL="no"])
-else
-  HAVE_SDL="no"
-fi
+HAVE_SDL="no"
 AM_CONDITIONAL(HAVE_SDL,test "x$HAVE_SDL" = "xyes")
 
 
--- goom2k4-0/Makefile.am.orig	2021-05-12 20:14:55.340304209 +0200
+++ goom2k4-0/Makefile.am	2021-05-12 20:15:24.561369200 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = src xmms-goom sdl-goom @MACFOLDER@
+SUBDIRS = src @MACFOLDER@
 
 #.pc file
 pkgconfigdir = $(libdir)/pkgconfig
--- goom2k4-0/src/Makefile.am.orig	2021-05-12 20:21:32.476419227 +0200
+++ goom2k4-0/src/Makefile.am	2021-05-12 20:21:43.923269726 +0200
@@ -16,7 +16,7 @@
 goom2_libdir = $(libdir)
 
 goom2_library_includedir=$(includedir)/goom
-goom2_library_include_HEADERS = goom.h goom_plugin_info.h goom_typedefs.h goom_graphic.h goom_config_param.h goom_visual_fx.h goom_filters.h goom_tools.h goomsl.h goomsl_hash.h goomsl_heap.h goom_tools.h goom_config.h
+goom2_library_include_HEADERS = goom.h goom_plugin_info.h goom_typedefs.h goom_graphic.h goom_config_param.h goom_visual_fx.h goom_filters.h goom_tools.h goomsl.h goomsl_hash.h goomsl_heap.h goom_config.h
 libgoom2_la_LDFLAGS = -export-dynamic -export-symbols-regex "goom.*" 
 libgoom2_la_SOURCES = \
 	goomsl_yacc.y goomsl_lex.l goomsl.c goomsl_hash.c goomsl_heap.c \
