diff --git a/src/sphinx_inline_tabs/_impl.py b/src/sphinx_inline_tabs/_impl.py
index 989ad4c..13b0e88 100644
--- a/src/sphinx_inline_tabs/_impl.py
+++ b/src/sphinx_inline_tabs/_impl.py
@@ -23,7 +23,7 @@ class _GeneralHTMLTagElement(nodes.Element, nodes.General):
         attributes.pop("classes")
         attributes.pop("names")
         attributes.pop("dupnames")
-        attributes.pop("backrefs")
+        attributes.pop("backrefs", None)
 
         text = translator.starttag(node, node.tagname, **attributes)
         translator.body.append(text.strip())
@@ -91,7 +91,7 @@ class TabHtmlTransform(SphinxPostTransform):
         self.counter = itertools.count(start=0, step=1)
 
         matcher = NodeMatcher(TabContainer)
-        for node in self.document.traverse(matcher):  # type: TabContainer
+        for node in self.document.findall(matcher):  # type: TabContainer
             self._process_one_node(node)
 
         while self.stack:
