diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 069a4060..f7fc1c49 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -228,7 +228,7 @@ if (LIBFACILE_FOUND)
 
    find_library(OCAML_ASMRUN_LIBRARY NAMES asmrun)
    if (OCAML_ASMRUN_LIBRARY)
-      target_link_libraries(kalzium ${OCAML_ASMRUN_LIBRARY})
+      target_link_libraries(kalzium ${OCAML_ASMRUN_LIBRARY} zstd)
    endif ()
 
    find_library(OCAML_STR_LIBRARY NAMES str)

