diff --git a/AMD/Config/AMDConfig.cmake.in b/AMD/Config/AMDConfig.cmake.in
index 9b26776d7..6d31d5c4e 100644
--- a/AMD/Config/AMDConfig.cmake.in
+++ b/AMD/Config/AMDConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/BTF/Config/BTFConfig.cmake.in b/BTF/Config/BTFConfig.cmake.in
index 26f707e14..c10e04d99 100644
--- a/BTF/Config/BTFConfig.cmake.in
+++ b/BTF/Config/BTFConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/CAMD/Config/CAMDConfig.cmake.in b/CAMD/Config/CAMDConfig.cmake.in
index 06e3d4cc9..52fd171d2 100644
--- a/CAMD/Config/CAMDConfig.cmake.in
+++ b/CAMD/Config/CAMDConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/CCOLAMD/Config/CCOLAMDConfig.cmake.in b/CCOLAMD/Config/CCOLAMDConfig.cmake.in
index 4152ada89..c0ba8ab5f 100644
--- a/CCOLAMD/Config/CCOLAMDConfig.cmake.in
+++ b/CCOLAMD/Config/CCOLAMDConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/CHOLMOD/Config/CHOLMODConfig.cmake.in b/CHOLMOD/Config/CHOLMODConfig.cmake.in
index 6b878cf2b..06a9ff350 100644
--- a/CHOLMOD/Config/CHOLMODConfig.cmake.in
+++ b/CHOLMOD/Config/CHOLMODConfig.cmake.in
@@ -90,7 +90,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/CHOLMOD/Config/CHOLMOD_CUDAConfig.cmake.in b/CHOLMOD/Config/CHOLMOD_CUDAConfig.cmake.in
index e49ce91a3..961ad3c2d 100644
--- a/CHOLMOD/Config/CHOLMOD_CUDAConfig.cmake.in
+++ b/CHOLMOD/Config/CHOLMOD_CUDAConfig.cmake.in
@@ -77,7 +77,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/COLAMD/Config/COLAMDConfig.cmake.in b/COLAMD/Config/COLAMDConfig.cmake.in
index a9597ee57..70d686239 100644
--- a/COLAMD/Config/COLAMDConfig.cmake.in
+++ b/COLAMD/Config/COLAMDConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/CXSparse/Config/CXSparseConfig.cmake.in b/CXSparse/Config/CXSparseConfig.cmake.in
index 1cb709acc..b3b517f39 100644
--- a/CXSparse/Config/CXSparseConfig.cmake.in
+++ b/CXSparse/Config/CXSparseConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/GPUQREngine/Config/GPUQREngineConfig.cmake.in b/GPUQREngine/Config/GPUQREngineConfig.cmake.in
index b0bd24f56..4994f3e0e 100644
--- a/GPUQREngine/Config/GPUQREngineConfig.cmake.in
+++ b/GPUQREngine/Config/GPUQREngineConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/GraphBLAS/CUDA/Config/GraphBLAS_CUDAConfig.cmake.in b/GraphBLAS/CUDA/Config/GraphBLAS_CUDAConfig.cmake.in
index 600d4d598..41db26531 100644
--- a/GraphBLAS/CUDA/Config/GraphBLAS_CUDAConfig.cmake.in
+++ b/GraphBLAS/CUDA/Config/GraphBLAS_CUDAConfig.cmake.in
@@ -78,7 +78,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/GraphBLAS/Config/GraphBLASConfig.cmake.in b/GraphBLAS/Config/GraphBLASConfig.cmake.in
index 9e1b2292e..3b552e828 100644
--- a/GraphBLAS/Config/GraphBLASConfig.cmake.in
+++ b/GraphBLAS/Config/GraphBLASConfig.cmake.in
@@ -91,7 +91,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/KLU/Config/KLUConfig.cmake.in b/KLU/Config/KLUConfig.cmake.in
index a1049ed13..12657689a 100644
--- a/KLU/Config/KLUConfig.cmake.in
+++ b/KLU/Config/KLUConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/KLU/Config/KLU_CHOLMODConfig.cmake.in b/KLU/Config/KLU_CHOLMODConfig.cmake.in
index 73a849d4a..aa1470e5a 100644
--- a/KLU/Config/KLU_CHOLMODConfig.cmake.in
+++ b/KLU/Config/KLU_CHOLMODConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/LDL/Config/LDLConfig.cmake.in b/LDL/Config/LDLConfig.cmake.in
index 948a34a58..0b791d458 100644
--- a/LDL/Config/LDLConfig.cmake.in
+++ b/LDL/Config/LDLConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/Mongoose/Config/MongooseConfig.cmake.in b/Mongoose/Config/MongooseConfig.cmake.in
index 84089876e..ed18cf143 100644
--- a/Mongoose/Config/MongooseConfig.cmake.in
+++ b/Mongoose/Config/MongooseConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/RBio/Config/RBioConfig.cmake.in b/RBio/Config/RBioConfig.cmake.in
index 2d0f259e0..a5daa8bc0 100644
--- a/RBio/Config/RBioConfig.cmake.in
+++ b/RBio/Config/RBioConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/SPEX/Config/SPEXConfig.cmake.in b/SPEX/Config/SPEXConfig.cmake.in
index 0b4d92267..83306b22f 100644
--- a/SPEX/Config/SPEXConfig.cmake.in
+++ b/SPEX/Config/SPEXConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/SPQR/Config/SPQRConfig.cmake.in b/SPQR/Config/SPQRConfig.cmake.in
index 195ab3d1b..fa430d048 100644
--- a/SPQR/Config/SPQRConfig.cmake.in
+++ b/SPQR/Config/SPQRConfig.cmake.in
@@ -91,7 +91,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/SPQR/Config/SPQR_CUDAConfig.cmake.in b/SPQR/Config/SPQR_CUDAConfig.cmake.in
index 108ee3b9e..83297a234 100644
--- a/SPQR/Config/SPQR_CUDAConfig.cmake.in
+++ b/SPQR/Config/SPQR_CUDAConfig.cmake.in
@@ -77,7 +77,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/SuiteSparse_GPURuntime/Config/SuiteSparse_GPURuntimeConfig.cmake.in b/SuiteSparse_GPURuntime/Config/SuiteSparse_GPURuntimeConfig.cmake.in
index fa6234140..e3d60d0e4 100644
--- a/SuiteSparse_GPURuntime/Config/SuiteSparse_GPURuntimeConfig.cmake.in
+++ b/SuiteSparse_GPURuntime/Config/SuiteSparse_GPURuntimeConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/SuiteSparse_config/Config/SuiteSparse_configConfig.cmake.in b/SuiteSparse_config/Config/SuiteSparse_configConfig.cmake.in
index c16ffd80d..cca8872da 100644
--- a/SuiteSparse_config/Config/SuiteSparse_configConfig.cmake.in
+++ b/SuiteSparse_config/Config/SuiteSparse_configConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
diff --git a/UMFPACK/Config/UMFPACKConfig.cmake.in b/UMFPACK/Config/UMFPACKConfig.cmake.in
index 0fd7f87e2..d97214e04 100644
--- a/UMFPACK/Config/UMFPACKConfig.cmake.in
+++ b/UMFPACK/Config/UMFPACKConfig.cmake.in
@@ -60,7 +60,7 @@ if ( TARGET ${_target_static} )
 endif ( )
 
 # Check for most common build types
-set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
+set ( _config_types "Debug" "Release" "RelWithDebInfo" "MinSizeRel" "None" )
 
 get_property ( _isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
 if ( _isMultiConfig )
