--- SuiteSparse-5.8.0/Makefile.orig	2020-07-04 11:08:32.938824768 +0200
+++ SuiteSparse-5.8.0/Makefile	2020-07-04 11:08:55.380537325 +0200
@@ -16,7 +16,7 @@
 # installs all libraries SuiteSparse/lib.
 go: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd Mongoose && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
+	( cd Mongoose && $(MAKE) library CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 	( cd AMD && $(MAKE) )
 	( cd BTF && $(MAKE) )
 	( cd CAMD && $(MAKE) )
