diff --git a/tests/src/test_TensorRestriction.cpp b/tests/src/test_TensorRestriction.cpp
index 43636678..5d2a2f08 100644
--- a/tests/src/test_TensorRestriction.cpp
+++ b/tests/src/test_TensorRestriction.cpp
@@ -189,7 +189,7 @@ void test_tensor_projection_identity(std::default_random_engine &generator,
          mgard::ShuffledTensorNodeRange(hierarchy, l - 1)) {
       // Encountered a handful of small errors.
       tracker += hierarchy.at(u, node.multiindex) ==
-                 Catch::Approx(p(coordinates(hierarchy, node))).epsilon(0.001);
+                 Catch::Approx(p(coordinates(hierarchy, node))).epsilon(0.002);
     }
     REQUIRE(tracker);
   }
