diff -ruN a/scripts/dblatex b/scripts/dblatex
--- a/scripts/dblatex	2020-11-29 18:50:08.000000000 +0100
+++ b/scripts/dblatex	2024-04-15 09:12:48.395963799 +0200
@@ -1,9 +1,4 @@
 #! /usr/bin/env python3
-import sys
-import os
-
-package_base = os.path.join(os.path.dirname(sys.argv[0]), "..")
-sys.path.insert(0, os.path.realpath(os.path.join(package_base, "lib")))
-
 from dbtexmf.dblatex import dblatex
-dblatex.main(base=package_base)
+
+dblatex.main(base="/usr/share/dblatex")
