From 2c0f63723fc756a74dc146668ec7786adf8d9f4f Mon Sep 17 00:00:00 2001
From: Dominik Heidler <dominik@heidler.eu>
Date: Thu, 31 Mar 2016 17:33:05 +0200
Subject: [PATCH] Use autoconf config.h for GPGME _FILE_OFFSET_BITS

fixes #779
---
 src/pgp/gpg.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/pgp/gpg.c b/src/pgp/gpg.c
index 8900915b5b..db87657b59 100644
--- a/src/pgp/gpg.c
+++ b/src/pgp/gpg.c
@@ -33,6 +33,7 @@
  */
 
 #include "prof_config.h"
+#include "config.h"
 
 #include <locale.h>
 #include <string.h>
