--- hipRAND-rocm-6.0.0/toolchain-linux.cmake.bak	2023-12-22 09:46:40.269044060 +0100
+++ hipRAND-rocm-6.0.0/toolchain-linux.cmake	2023-12-22 09:46:47.957023549 +0100
@@ -4,9 +4,9 @@
 #set(CMAKE_GENERATOR_PLATFORM x64)
 
 if (DEFINED ENV{ROCM_PATH})
-  set(rocm_bin "$ENV{ROCM_PATH}/hip/bin")
+  set(rocm_bin "$ENV{ROCM_PATH}/bin")
 else()
-  set(rocm_bin "/opt/rocm/hip/bin")
+  set(rocm_bin "/opt/rocm/bin")
 endif()
 
 
