diff --git a/tests/_towncrier_test.py b/tests/_towncrier_test.py
index 8b910dd..5b6f57c 100644
--- a/tests/_towncrier_test.py
+++ b/tests/_towncrier_test.py
@@ -1,14 +1,23 @@
 """Towncrier config reader tests."""
 
 
+from importlib.metadata import version as _get_installed_project_version
 from pathlib import Path
 from typing import Union
 
 import pytest
 
+from towncrier._settings.load import Config  # noqa: WPS436
+
 from sphinxcontrib.towncrier._towncrier import get_towncrier_config
 
 
+_TOWNCRIER_VERSION = _get_installed_project_version('towncrier')
+_TOWNCRIER_RELEASED_POST_2025_INCLUSIVE = int(  # noqa: WPS114
+    _TOWNCRIER_VERSION.split('.', 1)[0],
+) >= 25  # noqa: WPS432
+
+
 def test_towncrier_config_section_missing(
         monkeypatch: pytest.MonkeyPatch,
         tmp_path: Path,
@@ -25,6 +34,13 @@ def test_towncrier_config_section_missing(
         fr'`{empty_config_file !s}`: No \[tool\.towncrier\] section\.$'
     )
 
+    if _TOWNCRIER_RELEASED_POST_2025_INCLUSIVE:
+        assert isinstance(
+            get_towncrier_config(tmp_path, empty_config_file),
+            Config,
+        )
+        return
+
     with pytest.raises(LookupError, match=expected_error_msg):
         get_towncrier_config(tmp_path, empty_config_file)
 
