diff --git a/sound/soc/starfive/pwmdac.h b/sound/soc/starfive/pwmdac.h
index d4742f99a51b..4b1f4586e901 100644
--- a/sound/soc/starfive/pwmdac.h
+++ b/sound/soc/starfive/pwmdac.h
@@ -154,6 +154,9 @@ int sf_pwmdac_pcm_register(struct platform_device *pdev)
 {
 	return -EINVAL;
 }
+void pwmdac_dit_driver_init(void);
+void pwmdac_dit_driver_exit(void);
+
 #endif
 
 #endif
diff --git a/sound/soc/starfive/starfive_pwmdac.c b/sound/soc/starfive/starfive_pwmdac.c
index fba58e22d7ef..20cbf7083e10 100644
--- a/sound/soc/starfive/starfive_pwmdac.c
+++ b/sound/soc/starfive/starfive_pwmdac.c
@@ -968,14 +968,15 @@ static struct platform_driver sf_pwmdac_driver = {
 	},
 };
 
-
 static int __init pwmdac_driver_init(void)
 {
+    pwmdac_dit_driver_init();
 	return platform_driver_register(&sf_pwmdac_driver);
 }
 
 static void pwmdac_driver_exit(void)
 {
+    pwmdac_dit_driver_exit();
 	platform_driver_unregister(&sf_pwmdac_driver);
 }
 
diff --git a/sound/soc/starfive/starfive_pwmdac_transmitter.c b/sound/soc/starfive/starfive_pwmdac_transmitter.c
index 6d58fcfd5592..0c50c82c4c20 100755
--- a/sound/soc/starfive/starfive_pwmdac_transmitter.c
+++ b/sound/soc/starfive/starfive_pwmdac_transmitter.c
@@ -90,20 +90,16 @@ static struct platform_driver pwmdac_dit_driver = {
 	},
 };
 
-static int __init pwmdac_dit_driver_init(void)
+int __init pwmdac_dit_driver_init(void)
 {
 	return platform_driver_register(&pwmdac_dit_driver);
 }
 
-static void pwmdac_dit_driver_exit(void)
+void pwmdac_dit_driver_exit(void)
 {
 	platform_driver_unregister(&pwmdac_dit_driver);
 }
 
-late_initcall(pwmdac_dit_driver_init);
-module_exit(pwmdac_dit_driver_exit);
-
-
 MODULE_AUTHOR("curry.zhang <curry.zhang@starfivetech.com>");
 MODULE_DESCRIPTION("pwmdac dummy codec driver");
 MODULE_LICENSE("GPL v2");
