--- thunderbird-68.4.2/security/nss/lib/freebl/mpi/mpcpucache.c.orig	2020-02-02 08:28:25.908372279 +0100
+++ thunderbird-68.4.2/security/nss/lib/freebl/mpi/mpcpucache.c	2020-02-02 08:29:09.302883599 +0100
@@ -727,9 +727,12 @@
 dcbzl(char *array)
 {
     register char *a asm("r2") = array;
-    __asm__ __volatile__("dcbzl %0,r0"
-                         : "=r"(a)
-                         : "0"(a));
+   /* some older assemblers won't understand the dcbzl instruction
+    * variant, so we directly emit the instruction ourselves */
+    __asm__ __volatile__("mr 9, %0 ; .long 0x7C204FEC" /*dcbzl 0,9*/
+                         : /*out*/
+                         : "r" (a) /*in*/
+                         : "memory", "r9" /*clobber*/);
 }
 
 #define PPC_DO_ALIGN(x, y) ((char *)((((long long)(x)) + ((y)-1)) & ~((y)-1)))
