--- dhewm3-1.5.4/neo/framework/miniz/minizconf.h.orig	2024-11-27 17:39:30.848383871 +0100
+++ dhewm3-1.5.4/neo/framework/miniz/minizconf.h	2024-11-27 17:39:51.222300750 +0100
@@ -50,7 +50,7 @@
 #endif
 
 // FIXME: why not just set this to int64_t?
-#if !defined(_WIN32) && defined(Z_LARGE64)
+#if !defined(_WIN32) && defined(__USE_LARGEFILE64)
   #define z_off64_t off64_t
 #else
   #if defined(_WIN32)
