--- valgrind-3.21.0/VEX/priv/main_main.c.old	2023-01-07 14:48:48.000000000 +0000
+++ valgrind-3.21.0/VEX/priv/main_main.c	2024-10-01 21:53:47.879017945 +0000
@@ -2047,10 +2047,10 @@
          }
 
          /* ISA 3.0 not supported on 32-bit machines */
-         if ((hwcaps & VEX_HWCAPS_PPC32_ISA3_0) != 0) {
+         /*if ((hwcaps & VEX_HWCAPS_PPC32_ISA3_0) != 0) {
             invalid_hwcaps(arch, hwcaps,
                            "ISA 3.0 not supported in 32-bit mode \n");
-         }
+         }*/
          return;
       }
 
