diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08012267..b9c18f19 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,7 +58,7 @@ set(HAVE_OPENBABEL2 ${OPENBABEL2_FOUND})
 if(HAVE_OPENBABEL2 OR HAVE_OPENBABEL3)
     set(HAVE_OPENBABEL 1)
 endif()
-set(HAVE_EIGEN ${EIGEN3_FOUND})
+set(HAVE_EIGEN ${Eigen3_FOUND})
 set(HAVE_AVOGADRO ${AvogadroLibs_FOUND})
 
 if(HAVE_OPENBABEL3)
@@ -123,7 +123,7 @@ ecm_set_disabled_deprecation_versions(QT 5.14.0
 )
 
 
-if (HAVE_OPENBABEL AND AvogadroLibs_FOUND AND EIGEN3_FOUND)
+if (HAVE_OPENBABEL AND AvogadroLibs_FOUND AND Eigen3_FOUND)
   find_package(Qt6OpenGL ${QT_MIN_VERSION} REQUIRED)
   find_package(KF6NewStuff ${KF_MIN_VERSION} REQUIRED)
   # avoid compilerwarnings about redefinitions
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 97667169..5a94c777 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,7 +34,7 @@ if (HAVE_OPENBABEL)
    )
    target_include_directories(kalzium SYSTEM PRIVATE ${OPENBABEL_INCLUDE_DIR})
 
-   if (EIGEN3_FOUND AND AvogadroLibs_FOUND)
+   if (Eigen3_FOUND AND AvogadroLibs_FOUND)
       #include(${Avogadro_USE_FILE})
 
       target_sources(kalzium PRIVATE
@@ -207,7 +207,7 @@ target_link_libraries(kalzium
 
 if (HAVE_OPENBABEL)
    target_link_libraries(kalzium ${OPENBABEL_LIBRARIES})
-   if (EIGEN3_FOUND AND AvogadroLibs_FOUND)
+   if (Eigen3_FOUND AND AvogadroLibs_FOUND)
       target_link_libraries(kalzium
          KF6::NewStuffWidgets
          Qt::OpenGL
