diff -ruN mkinitcpio-25.orig/functions mkinitcpio-25/functions
--- mkinitcpio-25.orig/functions	2019-03-26 14:36:59.296803480 +0000
+++ mkinitcpio-25/functions	2019-03-26 14:40:27.013540092 +0000
@@ -150,6 +150,9 @@
     read kver _ < \
         <(dd if="$1" bs=1 count=127 skip=$(( offset + 0x200 )) 2>/dev/null)
 
+    # very ugly hack but works
+    [[ -z $kver ]] && kver=$(strings "$1" | grep modversions | awk '{print $1}')
+
     [[ $kver =~ $re ]] || return 1
 
     printf '%s' "$kver"
@@ -686,6 +689,10 @@
             ln -s "lib"     "$buildroot/usr/lib64"
             ln -s "usr/lib" "$buildroot/lib64"
             ;;
+        ppc64le)
+            ln -s "lib"     "$buildroot/usr/lib64"
+            ln -s "usr/lib" "$buildroot/lib64"
+            ;;
     esac
 
     # mkinitcpio version stamp
diff -ruN mkinitcpio-25.orig/mkinitcpio mkinitcpio-25/mkinitcpio
--- mkinitcpio-25.orig/mkinitcpio	2019-03-26 14:36:59.260806830 +0000
+++ mkinitcpio-25/mkinitcpio	2019-03-26 14:37:07.943998782 +0000
@@ -99,7 +99,7 @@
     fi
 
     arch=$(uname -m)
-    if [[ $arch != @(i?86|x86_64) ]]; then
+    if [[ $arch != @(i?86|x86_64|ppc64le) ]]; then
         error "kernel version extraction from image not supported for \`%s' architecture" "$arch"
         return 1
     fi
