diff -ruN mkinitcpio-29/functions mkinitcpio-29.ppc64le/functions
--- mkinitcpio-29/functions	2020-10-27 19:46:01.000000000 +0100
+++ mkinitcpio-29.ppc64le/functions	2020-11-27 09:44:04.814293021 +0100
@@ -150,6 +150,9 @@
     read kver _ < \
         <(dd if="$1" bs=1 count=127 skip=$(( offset + 0x200 )) 2>/dev/null)
 
+    # very ugly hack but works
+    [[ -z $kver ]] && kver=$(strings "$1" | grep '^Linux version [0-9]' | awk '{print $3}')
+
     [[ $kver =~ $re ]] || return 1
 
     printf '%s' "$kver"
@@ -692,6 +695,10 @@
             ln -s "lib"     "$buildroot/usr/lib64"
             ln -s "usr/lib" "$buildroot/lib64"
             ;;
+        ppc64*)
+            ln -s "lib"     "$buildroot/usr/lib64"
+            ln -s "usr/lib" "$buildroot/lib64"
+            ;;
     esac
 
     # mkinitcpio version stamp
diff -ruN mkinitcpio-29/man/lsinitcpio.1.txt mkinitcpio-29.ppc64le/man/lsinitcpio.1.txt
--- mkinitcpio-29/man/lsinitcpio.1.txt	2020-10-27 19:46:01.000000000 +0100
+++ mkinitcpio-29.ppc64le/man/lsinitcpio.1.txt	2020-11-26 15:33:10.306691263 +0100
@@ -53,7 +53,7 @@
 Bugs
 ----
 Upon writing this manpage, there were no noticeable bugs present. Please visit
-<http://bugs.archlinux.org/> for an up to date list.
+<https://github.com/kth5/archpower/issues> for an up to date list.
 
 See Also
 --------
diff -ruN mkinitcpio-29/man/mkinitcpio.8.txt mkinitcpio-29.ppc64le/man/mkinitcpio.8.txt
--- mkinitcpio-29/man/mkinitcpio.8.txt	2020-10-27 19:46:01.000000000 +0100
+++ mkinitcpio-29.ppc64le/man/mkinitcpio.8.txt	2020-11-26 15:33:10.306691263 +0100
@@ -359,7 +359,7 @@
 Bugs
 ----
 Upon writing this manpage, there were no noticeable bugs present. Please visit
-<http://bugs.archlinux.org/> for an up to date list.
+<https://github.com/kth5/archpower/issues> for an up to date list.
 
 Authors
 -------
diff -ruN mkinitcpio-29/mkinitcpio mkinitcpio-29.ppc64le/mkinitcpio
--- mkinitcpio-29/mkinitcpio	2020-10-27 19:46:01.000000000 +0100
+++ mkinitcpio-29.ppc64le/mkinitcpio	2020-11-26 15:33:10.307691160 +0100
@@ -102,7 +102,7 @@
     fi
 
     arch=$(uname -m)
-    if [[ $arch != @(i?86|x86_64) ]]; then
+    if [[ $arch != @(i?86|x86_64|ppc64le|ppc64|ppc) ]]; then
         error "kernel version extraction from image not supported for \`%s' architecture" "$arch"
         return 1
     fi
