diff -ruN mkinitcpio-25.orig/functions mkinitcpio-25/functions
--- mkinitcpio-25.orig/functions	2019-04-09 14:51:19.128762944 +0000
+++ mkinitcpio-25/functions	2019-04-09 14:51:24.908894498 +0000
@@ -150,6 +150,9 @@
     read kver _ < \
         <(dd if="$1" bs=1 count=127 skip=$(( offset + 0x200 )) 2>/dev/null)
 
+    # very ugly hack but works
+    [[ -z $kver ]] && kver=$(strings "$1" | grep modversions | awk '{print $1}')
+
     [[ $kver =~ $re ]] || return 1
 
     printf '%s' "$kver"
@@ -686,6 +689,10 @@
             ln -s "lib"     "$buildroot/usr/lib64"
             ln -s "usr/lib" "$buildroot/lib64"
             ;;
+        ppc64le)
+            ln -s "lib"     "$buildroot/usr/lib64"
+            ln -s "usr/lib" "$buildroot/lib64"
+            ;;
     esac
 
     # mkinitcpio version stamp
diff -ruN mkinitcpio-25.orig/man/lsinitcpio.1.txt mkinitcpio-25/man/lsinitcpio.1.txt
--- mkinitcpio-25.orig/man/lsinitcpio.1.txt	2019-04-09 14:51:19.128762944 +0000
+++ mkinitcpio-25/man/lsinitcpio.1.txt	2019-04-09 14:52:22.168222992 +0000
@@ -53,7 +53,7 @@
 Bugs
 ----
 Upon writing this manpage, there were no noticeable bugs present. Please visit
-<http://bugs.archlinux.org/> for an up to date list.
+<https://github.com/kth5/archpower/issues> for an up to date list.
 
 See Also
 --------
diff -ruN mkinitcpio-25.orig/man/mkinitcpio.8.txt mkinitcpio-25/man/mkinitcpio.8.txt
--- mkinitcpio-25.orig/man/mkinitcpio.8.txt	2019-04-09 14:51:19.128762944 +0000
+++ mkinitcpio-25/man/mkinitcpio.8.txt	2019-04-09 14:52:05.810462741 +0000
@@ -340,7 +340,7 @@
 Bugs
 ----
 Upon writing this manpage, there were no noticeable bugs present. Please visit
-<http://bugs.archlinux.org/> for an up to date list.
+<https://github.com/kth5/archpower/issues> for an up to date list.
 
 Authors
 -------
diff -ruN mkinitcpio-25.orig/mkinitcpio mkinitcpio-25/mkinitcpio
--- mkinitcpio-25.orig/mkinitcpio	2019-04-09 14:51:19.158763614 +0000
+++ mkinitcpio-25/mkinitcpio	2019-04-09 14:51:24.908894498 +0000
@@ -99,7 +99,7 @@
     fi
 
     arch=$(uname -m)
-    if [[ $arch != @(i?86|x86_64) ]]; then
+    if [[ $arch != @(i?86|x86_64|ppc64le) ]]; then
         error "kernel version extraction from image not supported for \`%s' architecture" "$arch"
         return 1
     fi
