--- mkinitcpio-25/mkinitcpio.orig	2019-03-19 11:09:56.122206605 +0000
+++ mkinitcpio-25/mkinitcpio	2019-03-19 11:10:05.497359443 +0000
@@ -99,7 +99,7 @@
     fi
 
     arch=$(uname -m)
-    if [[ $arch != @(i?86|x86_64) ]]; then
+    if [[ $arch != @(i?86|x86_64|ppc64le) ]]; then
         error "kernel version extraction from image not supported for \`%s' architecture" "$arch"
         return 1
     fi
--- mkinitcpio-25/functions.orig	2019-03-19 11:24:45.577631845 +0000
+++ mkinitcpio-25/functions	2019-03-19 11:25:15.774892292 +0000
@@ -150,6 +150,9 @@
     read kver _ < \
         <(dd if="$1" bs=1 count=127 skip=$(( offset + 0x200 )) 2>/dev/null)
 
+    # very ugly hack but works
+    [[ -z $kver ]] && kver=$(strings "$1" | grep modversions | awk '{print $1}')
+
     [[ $kver =~ $re ]] || return 1
 
     printf '%s' "$kver"
