--- a/pyproject.toml
+++ b/pyproject.toml
@@ -44,13 +44,13 @@ Source = "https://github.com/pypa/pip"
 Changelog = "https://pip.pypa.io/en/stable/news/"
 
 [build-system]
-requires = ["flit-core >=3.11,<4"]
+requires = ["flit-core"]
 build-backend = "flit_core.buildapi"
 
 [dependency-groups]
 test = [
     "cryptography",
-    "flit-core >= 3.11, < 4",
+    "flit-core",
     "freezegun",
     "installer",
     # pytest-subket requires 7.0+
@@ -68,7 +68,7 @@ test = [
 ]
 
 test-common-wheels = [
-    "flit-core >= 3.11, < 4",
+    "flit-core",
     # We pin setuptools<80 because our test suite currently
     # depends on setup.py develop to generate egg-link files.
     "setuptools >= 70.1.0, <80",
