diff --git i/fig2dev/tests/bitmaps.at w/fig2dev/tests/bitmaps.at
index 4b08454..99d4656 100644
--- i/fig2dev/tests/bitmaps.at
+++ w/fig2dev/tests/bitmaps.at
@@ -201,67 +201,3 @@ EOF`
 AT_CHECK([test $blackpixels -gt 1197 && test $blackpixels -lt 1243])
 AT_CLEANUP
 
-AT_SETUP([postscript, created by latex])
-AT_KEYWORDS(embed ps)
-AT_SKIP_IF([NO_GS || ! ppmhist -version])
-AT_DATA(latex.ps, [%!PS-Adobe-2.0
-%%Pages: 1
-%%PageOrder: Ascend
-%%BoundingBox: 0 0 420 596
-%%DocumentFonts: Helvetica
-%%DocumentPaperSizes: a5
-%%EndComments
-%%BeginProlog
-/m {moveto} bind def
-/s {show} bind def
-%%EndProlog
-%%BeginSetup
-%%Feature: *Resolution 600dpi
-%%BeginPaperSize: a5
-/setpagedevice where
-{ pop << /PageSize @<:@420 595@:>@ >> setpagedevice }
-{ /a5 where { pop a5 } if }
-ifelse
-%%EndPaperSize
-%%EndSetup
-%%Page: 1 1
-/Helvetica findfont 12 scalefont setfont
-24 559 m (Upper left) s
-336 559 m (Upper right) s
-24 24 m (Lower left) s
-336 24 m (Lower right) s
-126 340 m (Postscript as created by LaTeX) s
-%%Trailer
-userdict /end-hook known{end-hook}if
-%%EOF
-])
-
-blackpixels=`fig2dev -L ppm <<EOF | ppmhist -noheader | $SED -n '2p' | \
-tr -d '\t '
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.000 0 0 -1 0 0 5
-	0 latex.ps
-	 0 0 7000 0 7000 9917 0 9917 0 0
-4 0 0 50 -1 16 12 4.7124 4 210 2100 7225 1575 Included postscript file\001
-EOF`
-# $blackpixels are 1488 on archlinux with ghostscript 9.20,
-#		1433 on ubuntu 16.04.2 with ghostscript 9.18
-AT_CHECK([test $blackpixels -gt 1431 && test $blackpixels -lt 1498])
-AT_CLEANUP
-
-AT_SETUP([find /MediaBox in pdf file])
-AT_KEYWORDS(pdf)
-AT_CHECK(["$abs_builddir"/test2 "$srcdir/data/cross.pdf"], 0, ignore)
-AT_CLEANUP
-
-AT_SETUP([pdf])
-AT_KEYWORDS(pdf)
-AT_SKIP_IF([NO_GS])
-AT_CHECK([fig2dev -L pdf <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.000 0 0 -1 0 0 5
-	0 $srcdir/data/cross.pdf
-	 75 75 585 75 585 285 75 285 75 75
-EOF
-], 0, ignore)
-AT_CLEANUP
diff --git i/fig2dev/tests/input.at w/fig2dev/tests/input.at
index 63f82b9..396be0c 100644
--- i/fig2dev/tests/input.at
+++ w/fig2dev/tests/input.at
@@ -166,74 +166,6 @@ AT_CHECK([gzip -q -c $srcdir/data/line.eps >line.eps.Z && \
 ],0,ignore)
 AT_CLEANUP
 
-AT_SETUP([read compressed image files])
-AT_KEYWORDS(readpics gz)
-AT_SKIP_IF([NO_GZIP])
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.pdf
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.gif
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.jpg
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.pcx
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.png
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.ppm
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.tif
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.xbm
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CHECK([fig2dev -L eps <<EOF
-FIG_FILE_TOP
-2 5 0 1 0 -1 50 -1 -1 0.0 0 0 -1 0 0 5
-0 $srcdir/data/line.xpm
-0 0 510 0 510 210 0 210 0 0
-EOF
-],0,ignore)
-AT_CLEANUP
-
 AT_SETUP([absolute path in fig file, pipe ok])
 AT_KEYWORDS(readpics fullpath imgpipe)
 # if the path contains '%', the sed-command below fails
