Description: Insert trace headers at index 0
Author: David Bürgin <dbuergin@gluet.ch>
Bug: https://github.com/trusteddomainproject/OpenDKIM/pull/126

--- a/opendkim/opendkim.c
+++ b/opendkim/opendkim.c
@@ -3670,7 +3670,7 @@
 
 	if (ctx == NULL)
 		lua_pushnil(l);
-	else if (dkimf_insheader(ctx, 1, name, value) == MI_SUCCESS)
+	else if (dkimf_insheader(ctx, 0, name, value) == MI_SUCCESS)
 		lua_pushnumber(l, 1);
 	else
 		lua_pushnil(l);
@@ -4246,7 +4246,7 @@
 	assert(conf != NULL);
 	assert(ctx != NULL);
 
-	if (dkimf_insheader(ctx, 1, AUTHRESULTSHDR,
+	if (dkimf_insheader(ctx, 0, AUTHRESULTSHDR,
 	                    (char *) dfc->mctx_dkimar) == MI_FAILURE)
 	{
 		if (conf->conf_dolog)
@@ -13517,7 +13517,7 @@
 		         dkimf_lookup_inttostr(dfc->mctx_status,
 		                               dkimf_statusstrings));
 
-		if (dkimf_insheader(ctx, 1, AUTHRESULTSHDR,
+		if (dkimf_insheader(ctx, 0, AUTHRESULTSHDR,
 		                    (char *) header) == MI_FAILURE)
 		{
 			if (conf->conf_dolog)
@@ -14962,7 +14962,7 @@
 						        sizeof header);
 					}
 		
-					if (dkimf_insheader(ctx, 1,
+					if (dkimf_insheader(ctx, 0,
 					                    AUTHRESULTSHDR,
 					                    (char *) header) == MI_FAILURE)
 					{
@@ -15165,7 +15165,7 @@
 			dkimf_stripcr((char *) start);
 			dkimf_dstring_cat(dfc->mctx_tmpstr, start);
 
-			if (dkimf_insheader(ctx, 1, DKIM_SIGNHEADER,
+			if (dkimf_insheader(ctx, 0, DKIM_SIGNHEADER,
 			                    (char *) dkimf_dstring_get(dfc->mctx_tmpstr)) == MI_FAILURE)
 			{
 				if (conf->conf_dolog)
@@ -15201,7 +15201,7 @@
 		/* add VBR-Info header if generated */
 		if (dfc->mctx_vbrinfo != NULL)
 		{
-			if (dkimf_insheader(ctx, 1, VBR_INFOHEADER,
+			if (dkimf_insheader(ctx, 0, VBR_INFOHEADER,
 			                    dfc->mctx_vbrinfo) == MI_FAILURE)
 			{
 				if (conf->conf_dolog)
@@ -15251,7 +15251,7 @@
 		         dfc->mctx_jobid != NULL ? dfc->mctx_jobid
 		                                 : (u_char *) JOBIDUNKNOWN);
 
-		if (dkimf_insheader(ctx, 1, SWHEADERNAME, xfhdr) != MI_SUCCESS)
+		if (dkimf_insheader(ctx, 0, SWHEADERNAME, xfhdr) != MI_SUCCESS)
 		{
 			if (conf->conf_dolog)
 			{
