Description: Add missing ReplaceHeaders definition
Author: Toby Ovod-Everett
Bug: https://github.com/trusteddomainproject/OpenDKIM/pull/125

--- a/opendkim/opendkim-config.h
+++ b/opendkim/opendkim-config.h
@@ -147,6 +147,7 @@
 	{ "RemoveARFrom",		CONFIG_TYPE_STRING,	FALSE },
 	{ "RemoveOldSignatures",	CONFIG_TYPE_BOOLEAN,	FALSE },
 #ifdef _FFR_REPLACE_RULES
+	{ "ReplaceHeaders",		CONFIG_TYPE_STRING,	FALSE },
 	{ "ReplaceRules",		CONFIG_TYPE_STRING,	FALSE },
 #endif /* _FFR_REPLACE_RULES */
 	{ "ReportAddress",		CONFIG_TYPE_STRING,	FALSE },
