--- arch-nspawn.in.old	2021-03-16 23:23:38.512000000 +0300
+++ arch-nspawn.in	2021-03-16 23:23:47.328000000 +0300
@@ -127,7 +127,7 @@

 [[ -z $nosetarch ]] || unset CARCH

-exec ${CARCH:+setarch "$CARCH"} systemd-nspawn -q \
+exec systemd-nspawn -q \
 	-D "$working_dir" \
 	-E "PATH=/usr/local/sbin:/usr/local/bin:/usr/bin" \
 	--register=no --keep-unit --as-pid2 \

#This removes the setarch check because setarch does not recognize
#powerpc or powerpc64le to be a legitimate architecture, instead
#requiring ppc or ppc64le.
#This patch is lazy as it is named, and will be improved later.
