--- devtools-20230105/src/arch-nspawn.in.orig	2023-01-18 14:37:37.658643541 +0100
+++ devtools-20230105/src/arch-nspawn.in	2023-01-18 14:39:35.262844390 +0100
@@ -119,7 +119,12 @@
 if [[ -f "@pkgdatadir@/setarch-aliases.d/${CARCH}" ]]; then
 	read -r set_arch < "@pkgdatadir@/setarch-aliases.d/${CARCH}"
 else
-	set_arch="${CARCH}"
+	case "${CARCH}" in
+		powerpc)	set_arch='ppc' ;;
+		powerpc64)	set_arch='ppc64' ;;
+		powerpc64le)	set_arch='ppc64le' ;;
+		*)		set_arch="${CARCH}" ;;
+	esac
 fi
 
 exec ${CARCH:+setarch "$set_arch"} systemd-nspawn -q \
