diff -ru mongo-c-driver-2.3.2.orig/src/libmongoc/CMakeLists.txt mongo-c-driver-2.3.2/src/libmongoc/CMakeLists.txt
--- mongo-c-driver-2.3.2.orig/src/libmongoc/CMakeLists.txt	2026-07-10 23:16:53.049052070 +0200
+++ mongo-c-driver-2.3.2/src/libmongoc/CMakeLists.txt	2026-07-10 23:17:06.673746659 +0200
@@ -424,7 +424,7 @@
 
    TRY_COMPILE (RES ${CMAKE_CURRENT_BINARY_DIR}
    ${CMAKE_CURRENT_BINARY_DIR}/accept_test${VAR}.c CMAKE_FLAGS
-   "-Werror -DCMAKE_CXX_LINK_EXECUTABLE='echo not linking now...'" OUTPUT_VARIABLE LOG2)
+   -DCOMPILE_DEFINITIONS=-Werror -DCMAKE_CXX_LINK_EXECUTABLE='echo not linking now...' OUTPUT_VARIABLE LOG2)
 
    if (RES)
       message (
