diff --git a/tests/test_replwrap.py b/tests/test_replwrap.py
index 5ac782a..993e576 100644
--- a/tests/test_replwrap.py
+++ b/tests/test_replwrap.py
@@ -92,6 +92,8 @@ class REPLWrapTestCase(unittest.TestCase):
                                     "PS1='{0}' PS2='{1}' "
                                     "PROMPT_COMMAND=''")
 
+        repl.run_command("bind 'set enable-bracketed-paste off'")
+
         print(repl)
         res = repl.run_command("echo $HOME")
         print(res)
