diff -aur pyu2f-0.1.5.old/pyu2f/tests/customauthenticator_test.py pyu2f-0.1.5/pyu2f/tests/customauthenticator_test.py
--- pyu2f-0.1.5.old/pyu2f/tests/customauthenticator_test.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/pyu2f/tests/customauthenticator_test.py	2024-02-17 14:51:49.407867091 +0100
@@ -19,7 +19,7 @@
 import struct
 import sys
 
-import mock
+from unittest import mock
 from pyu2f import errors
 from pyu2f import model
 from pyu2f.convenience import customauthenticator
diff -aur pyu2f-0.1.5.old/pyu2f/tests/hardware_test.py pyu2f-0.1.5/pyu2f/tests/hardware_test.py
--- pyu2f-0.1.5.old/pyu2f/tests/hardware_test.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/pyu2f/tests/hardware_test.py	2024-02-17 14:51:45.747762615 +0100
@@ -16,7 +16,7 @@
 
 import sys
 
-import mock
+from unittest import mock
 
 from pyu2f import errors
 from pyu2f import hardware
diff -aur pyu2f-0.1.5.old/pyu2f/tests/hid/linux_test.py pyu2f-0.1.5/pyu2f/tests/hid/linux_test.py
--- pyu2f-0.1.5.old/pyu2f/tests/hid/linux_test.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/pyu2f/tests/hid/linux_test.py	2024-02-17 14:51:15.046880148 +0100
@@ -18,7 +18,7 @@
 import os
 import sys
 
-import mock
+from unittest import mock
 
 from pyu2f.hid import linux
 
diff -aur pyu2f-0.1.5.old/pyu2f/tests/hid/macos_test.py pyu2f-0.1.5/pyu2f/tests/hid/macos_test.py
--- pyu2f-0.1.5.old/pyu2f/tests/hid/macos_test.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/pyu2f/tests/hid/macos_test.py	2024-02-17 14:51:05.566605365 +0100
@@ -16,7 +16,7 @@
 
 import ctypes
 import sys
-import mock
+from unittest import mock
 
 from pyu2f import errors
 from pyu2f.hid import macos
diff -aur pyu2f-0.1.5.old/pyu2f/tests/hidtransport_test.py pyu2f-0.1.5/pyu2f/tests/hidtransport_test.py
--- pyu2f-0.1.5.old/pyu2f/tests/hidtransport_test.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/pyu2f/tests/hidtransport_test.py	2024-02-17 14:51:41.780982548 +0100
@@ -17,7 +17,7 @@
 from six.moves import range
 import sys
 
-import mock
+from unittest import mock
 
 from pyu2f import errors
 from pyu2f import hidtransport
diff -aur pyu2f-0.1.5.old/pyu2f/tests/localauthenticator_test.py pyu2f-0.1.5/pyu2f/tests/localauthenticator_test.py
--- pyu2f-0.1.5.old/pyu2f/tests/localauthenticator_test.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/pyu2f/tests/localauthenticator_test.py	2024-02-17 14:51:37.597529421 +0100
@@ -17,7 +17,7 @@
 import base64
 import sys
 
-import mock
+from unittest import mock
 from pyu2f import errors
 from pyu2f import model
 from pyu2f.convenience import localauthenticator
diff -aur pyu2f-0.1.5.old/pyu2f/tests/u2f_test.py pyu2f-0.1.5/pyu2f/tests/u2f_test.py
--- pyu2f-0.1.5.old/pyu2f/tests/u2f_test.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/pyu2f/tests/u2f_test.py	2024-02-17 14:47:31.997424901 +0100
@@ -16,7 +16,7 @@
 
 import sys
 
-import mock
+from unittest import mock
 
 from pyu2f import errors
 from pyu2f import model
diff -aur pyu2f-0.1.5.old/setup.py pyu2f-0.1.5/setup.py
--- pyu2f-0.1.5.old/setup.py	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/setup.py	2024-02-17 14:47:40.650948353 +0100
@@ -42,7 +42,6 @@
     tests_require=[
         'unittest2>=0.5.1',
         'pyfakefs>=2.4',
-        'mock>=1.0.1',
     ],
     include_package_data=True,
     platforms=["Windows", "Linux", "OS X", "macOS"],
diff -aur pyu2f-0.1.5.old/tox.ini pyu2f-0.1.5/tox.ini
--- pyu2f-0.1.5.old/tox.ini	2024-02-17 14:46:24.602609269 +0100
+++ pyu2f-0.1.5/tox.ini	2024-02-17 14:47:40.657615166 +0100
@@ -7,7 +7,6 @@
 [testenv]
 deps=pytest
      pyfakefs
-     mock
      six
 commands=py.test pyu2f/tests/
 
