diff --unified --recursive --text --new-file movit-1.7.2.orig/Makefile.in movit-1.7.2/Makefile.in
--- movit-1.7.2.orig/Makefile.in	2026-04-04 19:03:57.520198269 +0200
+++ movit-1.7.2/Makefile.in	2026-04-04 19:04:17.539027159 +0200
@@ -21,13 +21,13 @@
 
 CC=@CC@
 CXX=@CXX@
-CXXFLAGS=-Wall @CXXFLAGS@ -fvisibility-inlines-hidden -I$(GTEST_DIR)/include @SDL2_CFLAGS@ @Eigen3_CFLAGS@ @epoxy_CFLAGS@ @FFTW3_CFLAGS@ @benchmark_CFLAGS@
+CXXFLAGS=-Wall @CXXFLAGS@ -fvisibility-inlines-hidden @SDL2_CFLAGS@ @Eigen3_CFLAGS@ @epoxy_CFLAGS@ @FFTW3_CFLAGS@ @benchmark_CFLAGS@
 ifeq ($(with_benchmark),yes)
 CXXFLAGS += -DHAVE_BENCHMARK
 endif
 LDFLAGS=@LDFLAGS@
 LDLIBS=@epoxy_LIBS@ @FFTW3_LIBS@ -lpthread
-TEST_LDLIBS=@epoxy_LIBS@ @SDL2_LIBS@ @benchmark_LIBS@ -lpthread
+TEST_LDLIBS=@epoxy_LIBS@ @SDL2_LIBS@ @benchmark_LIBS@ -lgtest -lpthread
 DEMO_LDLIBS=@SDL2_image_LIBS@ -lrt -lpthread @libpng_LIBS@ @FFTW3_LIBS@
 SHELL=@SHELL@
 LIBTOOL=@LIBTOOL@ --tag=CXX
@@ -100,12 +100,10 @@
 
 # Google Test and other test library functions.
 OWN_TEST_OBJS = gtest_sdl_main.o test_util.o
-TEST_OBJS = gtest-all.o $(OWN_TEST_OBJS)
+TEST_OBJS = $(OWN_TEST_OBJS)
 
-gtest-all.o: $(GTEST_DIR)/src/gtest-all.cc
-	$(CXX) -MMD $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c $< -o $@
 gtest_sdl_main.o: gtest_sdl_main.cpp
-	$(CXX) -MMD $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c $< -o $@
+	$(CXX) -MMD $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
 
 # Unit tests.
 $(TESTS): %: %.o $(TEST_OBJS) libmovit.la
