Index: movit-1.6.3/util.cpp
===================================================================
--- movit-1.6.3.orig/util.cpp
+++ movit-1.6.3/util.cpp
@@ -231,7 +231,7 @@ string output_glsl_vec3(const string &na
 	return ss.str();
 }
 
-GLuint generate_vbo(GLint size, GLenum type, GLsizeiptr data_size, const GLvoid *data)
+GLuint generate_vbo(GLint size, GLenum type, long data_size, const GLvoid *data)
 {
 	GLuint vbo;
 	glGenBuffers(1, &vbo);
@@ -246,6 +246,11 @@ GLuint generate_vbo(GLint size, GLenum t
 	return vbo;
 }
 
+GLuint generate_vbo(GLint size, GLenum type, int data_size, const GLvoid *data)
+{
+	return generate_vbo(size, type, long{data_size}, data);
+}
+
 GLuint fill_vertex_attribute(GLuint glsl_program_num, const string &attribute_name, GLint size, GLenum type, GLsizeiptr data_size, const GLvoid *data)
 {
 	int attrib = glGetAttribLocation(glsl_program_num, attribute_name.c_str());
