diff --unified --recursive --text --new-file betamax-0.9.0.orig/src/betamax/util.py betamax-0.9.0/src/betamax/util.py
--- betamax-0.9.0.orig/src/betamax/util.py	2025-12-31 12:38:50.309928606 +0100
+++ betamax-0.9.0/src/betamax/util.py	2025-12-31 12:38:53.051492914 +0100
@@ -150,7 +150,8 @@
         reason=response.reason,
         headers=headers,
         preload_content=False,
-        original_response=MockHTTPResponse(headers)
+        original_response=MockHTTPResponse(headers),
+        enforce_content_length=False,
     )
     # NOTE(sigmavirus24):
     # urllib3 updated it's chunked encoding handling which breaks on recorded
