diff --git i/Reflector.py w/Reflector.py
--- i/Reflector.py
+++ w/Reflector.py
@@ -429,7 +429,8 @@ def _rate_threaded(mirrors, fmt, n_threads, kwargs):  # pylint: disable=too-many
 
     logger = get_logger()
     rates = {}
-    with multiprocessing.Pool(n_threads) as pool:
+    ctx = multiprocessing.get_context('fork')
+    with ctx.Pool(n_threads) as pool:
         for url, time_delta, ratio in pool.starmap(_rate_wrapper, args):
             kibps = ratio / 1024.0
             logger.info(fmt.format(url, kibps, time_delta))
