--- protobuf-21.3/src/google/protobuf/repeated_field.h.orig	2022-07-27 12:47:01.115178463 +0200
+++ protobuf-21.3/src/google/protobuf/repeated_field.h	2022-07-27 12:47:21.337117628 +0200
@@ -76,9 +76,6 @@
 
 template <typename T, int kRepHeaderSize>
 constexpr int RepeatedFieldLowerClampLimit() {
-  // The header is padded to be at least `sizeof(T)` when it would be smaller
-  // otherwise.
-  static_assert(sizeof(T) <= kRepHeaderSize, "");
   // We want to pad the minimum size to be a power of two bytes, including the
   // header.
   // The first allocation is kRepHeaderSize bytes worth of elements for a total
