diff --git i/src/pendulum/tz/__init__.py w/src/pendulum/tz/__init__.py
index 36c2c69..00bad08 100644
--- i/src/pendulum/tz/__init__.py
+++ w/src/pendulum/tz/__init__.py
@@ -1,7 +1,6 @@
 from __future__ import annotations
 
-from pathlib import Path
-from typing import cast
+from zoneinfo import available_timezones
 
 from pendulum.tz.local_timezone import get_local_timezone
 from pendulum.tz.local_timezone import set_local_timezone
@@ -9,7 +8,6 @@ from pendulum.tz.local_timezone import test_local_timezone
 from pendulum.tz.timezone import UTC
 from pendulum.tz.timezone import FixedTimezone
 from pendulum.tz.timezone import Timezone
-from pendulum.utils._compat import resources
 
 
 PRE_TRANSITION = "pre"
@@ -21,12 +19,11 @@ _timezones = None
 _tz_cache: dict[int, FixedTimezone] = {}
 
 
-def timezones() -> tuple[str, ...]:
+def timezones() -> set[str]:
     global _timezones
 
     if _timezones is None:
-        with cast(Path, resources.files("tzdata").joinpath("zones")).open() as f:
-            _timezones = tuple(tz.strip() for tz in f.readlines())
+        _timezones = available_timezones()
 
     return _timezones
 
