diff -aur testscenarios-0.5.0.old/testscenarios/tests/test_scenarios.py testscenarios-0.5.0/testscenarios/tests/test_scenarios.py
--- testscenarios-0.5.0.old/testscenarios/tests/test_scenarios.py	2025-12-16 20:55:13.726293614 +0100
+++ testscenarios-0.5.0/testscenarios/tests/test_scenarios.py	2025-12-16 20:55:33.572428369 +0100
@@ -193,7 +193,7 @@
             [self.SampleTest('test_nothing')],
             None)
         result_tests = list(testtools.iterate_tests(suite))
-        self.assertEquals(
+        self.assertEqual(
             2,
             len(result_tests),
             result_tests)
@@ -206,7 +206,7 @@
             unittest.TestLoader(),
             )
         result_tests = list(testtools.iterate_tests(suite))
-        self.assertEquals(
+        self.assertEqual(
             2,
             len(result_tests),
             result_tests)
