diff --git a/ansible_collections/dellemc/unity/plugins/module_utils/storage/dell/utils.py b/ansible_collections/dellemc/unity/plugins/module_utils/storage/dell/utils.py
index d96a27a7..78ab6b6b 100644
--- a/ansible_collections/dellemc/unity/plugins/module_utils/storage/dell/utils.py
+++ b/ansible_collections/dellemc/unity/plugins/module_utils/storage/dell/utils.py
@@ -69,8 +69,8 @@ except ImportError:
     HAS_UNITY_SDK, STOROPS_IMP_ERR = False, traceback.format_exc()
 
 try:
-    from pkg_resources import parse_version
-    import pkg_resources
+    from packaging.version import parse as parse_version
+    import importlib.metadata
 
     HAS_PKG_RESOURCE, PKG_RESOURCE_IMP_ERR = True, None
 except ImportError:
@@ -85,7 +85,7 @@ def ensure_required_libs(module):
                          exception=STOROPS_IMP_ERR)
 
     if not HAS_PKG_RESOURCE:
-        module.fail_json(msg=missing_required_lib("pkg_resources"),
+        module.fail_json(msg=missing_required_lib("importlib.metadata or packaging"),
                          exception=PKG_RESOURCE_IMP_ERR)
 
     if not HAS_URLLIB3:
@@ -94,7 +94,7 @@ def ensure_required_libs(module):
 
     min_ver = '1.2.12'
     try:
-        curr_version = pkg_resources.require("storops")[0].version
+        curr_version = importlib.metadata.version("storops")
     except Exception as err:
         module.fail_json(msg="Failed to get Storops SDK version - "
                              "{0}".format(str(err)))
