diff --git a/erlang.mk b/erlang.mk
index 1d2e3be2a9..ca4b12c09b 100644
--- a/erlang.mk
+++ b/erlang.mk
@@ -3810,7 +3810,6 @@ define dep_autopatch2
 	fi; \
 	$(call erlang,$(call dep_autopatch_appsrc.erl,$(1))); \
 	if [ -f $(DEPS_DIR)/$(1)/rebar -o -f $(DEPS_DIR)/$(1)/rebar.config -o -f $(DEPS_DIR)/$(1)/rebar.config.script -o -f $(DEPS_DIR)/$1/rebar.lock ]; then \
-		$(call dep_autopatch_fetch_rebar); \
 		$(call dep_autopatch_rebar,$(1)); \
 	else \
 		$(call dep_autopatch_gen,$(1)); \
