diff --git i/erlang.mk w/erlang.mk
index e6e7ea4ec9..e9fbf22191 100644
--- i/erlang.mk
+++ w/erlang.mk
@@ -708,7 +708,6 @@ define dep_autopatch_for_rebar3
 	fi; \
 	$(call erlang,$(call dep_autopatch_appsrc.erl,$1)); \
 	if [ -f $(DEPS_DIR)/$1/rebar -o -f $(DEPS_DIR)/$1/rebar.config -o -f $(DEPS_DIR)/$1/rebar.config.script -o -f $(DEPS_DIR)/$1/rebar.lock ]; then \
-		$(call dep_autopatch_fetch_rebar); \
 		$(call dep_autopatch_rebar,$1); \
 	else \
 		$(call dep_autopatch_gen,$1); \
