--- configure.py
+++ configure.py
@@ -848,7 +848,7 @@ def host_arch_cc():
     '__MIPSEL__'  : 'mipsel',
     '__mips__'    : 'mips',
     '__PPC64__'   : 'ppc64',
-    '__PPC__'     : 'ppc64',
+    '__PPC__'     : 'ppc',
     '__x86_64__'  : 'x64',
     '__s390__'    : 's390',
     '__s390x__'   : 's390x',
--- node.gyp
+++ node.gyp
@@ -479,6 +479,11 @@
       'msvs_disabled_warnings!': [4244],
 
       'conditions': [
+        [ 'host_arch=="mips" or host_arch=="mipsel" or host_arch=="ppc"', {
+	  'link_settings': {
+	    'libraries': [ '-latomic' ],
+	  },
+	}],
         [ 'node_code_cache_path!=""', {
           'sources': [ '<(node_code_cache_path)' ]
         }, {
