From 174fbe965c32412c95dccffd76ce196b678d3637 Mon Sep 17 00:00:00 2001
From: Nikita Ermakov <coffe92@gmail.com>
Date: Sat, 30 May 2020 16:40:25 +0300
Subject: [PATCH] Link with libatomic on riscv32/64

---
 src/qml/qml.pro | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/qml/qml.pro b/src/qml/qml.pro
index 14b46307aa..f6f7a35e52 100644
--- a/src/qml/qml.pro
+++ b/src/qml/qml.pro
@@ -19,6 +19,8 @@ solaris-cc*:QMAKE_CXXFLAGS_RELEASE -= -O2
 
 # Ensure this gcc optimization is switched off for mips platforms to avoid trouble with JIT.
 gcc:isEqual(QT_ARCH, "mips"): QMAKE_CXXFLAGS += -fno-reorder-blocks
+# Link with libatomic on riscv32/64
+gcc:if(isEqual(QT_ARCH, "riscv32")|isEqual(QT_ARCH, "riscv64")): LIBS += -latomic
 
 DEFINES += QT_NO_FOREACH
 
-- 
2.26.0


