--- qtbase/src/corelib/kernel/qtestsupport_core.cpp.orig	2026-05-04 07:09:44.044704280 +0000
+++ qtbase/src/corelib/kernel/qtestsupport_core.cpp	2026-05-04 07:20:49.638252529 +0000
@@ -7,9 +7,12 @@
 
 using namespace std::chrono_literals;
 
+// lazily hack around this not being lock-free on 32bit
+#ifdef defined(__powerpc__)
 // Assert that this instantiation of std::atomic is always lock-free so we
 // know that no code will execute on destruction.
 static_assert(std::atomic<std::chrono::milliseconds>::is_always_lock_free);
+#endif
 
 QT_BEGIN_NAMESPACE
 
