--- a/Makefile	2025-12-21 22:50:38.000000000 +0100
+++ b/Makefile.new	2025-12-22 13:47:05.936778630 +0100
@@ -40,8 +40,8 @@
 $(error "Platform '$(SYSTEM)' not supported")
 endif
 
-GLEW_PREFIX ?= /usr/local
-GLEW_DEST ?= /usr/local
+GLEW_PREFIX ?= /usr
+GLEW_DEST ?= /usr
 BINDIR    ?= $(GLEW_DEST)/bin
 LIBDIR    ?= $(GLEW_DEST)/lib
 INCDIR    ?= $(GLEW_DEST)/include/GL
@@ -104,8 +104,6 @@
 glew.lib.shared: lib $(LIB.SHARED.DIR) $(LIB.SHARED.DIR)/$(LIB.SHARED) glew.pc
 glew.lib.static: lib lib/$(LIB.STATIC) glew.pc
 
-.PHONY: glew.lib glew.lib.shared glew.lib.static
-
 lib:
 	mkdir lib
 
@@ -142,7 +142,7 @@
 glew.pc: glew.pc.in
 	sed \
 		-e "s|@prefix@|$(GLEW_PREFIX)|g" \
-		-e "s|@libdir@|$(LIBDIR)|g" \
+		-e "s|@libdir@|/usr/lib|g" \
 		-e "s|@exec_prefix@|$(BINDIR)|g" \
 		-e "s|@includedir@|$(INCDIR)|g" \
 		-e "s|@version@|$(GLEW_VERSION)|g" \
--- a/config/Makefile.linux	2025-12-22 13:56:23.000000000 +0100
+++ b/config/Makefile.linux.new	2025-12-22 14:02:44.132416535 +0100
@@ -13,8 +13,8 @@
   ARCH64 = true
 endif
 ifeq (${ARCH64},true)
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
-  LIBDIR = $(GLEW_DEST)/lib64
+  LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
+  LIBDIR = $(GLEW_DEST)/lib
 else
   LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
   LIBDIR = $(GLEW_DEST)/lib
