diff --git a/src/vim/syntax/PKGBUILD.vim b/src/vim/syntax/PKGBUILD.vim
index 37ce89e..f1c2a45 100644
--- a/src/vim/syntax/PKGBUILD.vim
+++ b/src/vim/syntax/PKGBUILD.vim
@@ -82,7 +82,7 @@ syn region pbBackupGroup start=/^backup=(/ end=/)/ contains=pb_k_backup,pbValidB
 
 " arch
 syn keyword pb_k_arch arch contained
-syn keyword pbArch i686 x86_64 x86_64_v3 ppc pentium4 armv7h aarch64 loong64 any contained
+syn keyword pbArch i686 x86_64 x86_64_v3 ppc pentium4 armv7h aarch64 loong64 powerpc64le powerpc64 powerpc espresso any contained
 syn match pbIllegalArch /[^='"() ]/ contained contains=pbArch
 syn region pbArchGroup start=/^arch=(/ end=/)/ contains=pb_k_arch,pbArch,pbIllegalArch,pbComment
 
diff --git a/src/vim/syntax/SRCINFO.vim b/src/vim/syntax/SRCINFO.vim
index e26fa8d..0eed0d5 100644
--- a/src/vim/syntax/SRCINFO.vim
+++ b/src/vim/syntax/SRCINFO.vim
@@ -31,7 +31,11 @@ syn keyword srcInfoField
             \ source_x86_64
             \ source_x86_64_v3
             \ source_i686
+            \ source_espresso
             \ source_pentium4
+            \ source_powerpc
+            \ source_powerpc64
+            \ source_powerpc64le
             \ source_ppc
             \ source_aarch64
             \ source_armv7h
@@ -40,7 +44,11 @@ syn keyword srcInfoField
             \ depends_x86_64
             \ depends_x86_64_v3
             \ depends_i686
+            \ depends_espresso
             \ depends_pentium4
+            \ depends_powerpc
+            \ depends_powerpc64
+            \ depends_powerpc64le
             \ depends_ppc
             \ depends_aarch64
             \ depends_armv7h
@@ -49,7 +57,11 @@ syn keyword srcInfoField
             \ checkdepends_x86_64
             \ checkdepends_x86_64_v3
             \ checkdepends_i686
+            \ checkdepends_espresso
             \ checkdepends_pentium4
+            \ checkdepends_powerpc
+            \ checkdepends_powerpc64
+            \ checkdepends_powerpc64le
             \ checkdepends_ppc
             \ checkdepends_aarch64
             \ checkdepends_armv7h
@@ -58,7 +70,11 @@ syn keyword srcInfoField
             \ makedepends_x86_64
             \ makedepends_x86_64_v3
             \ makedepends_i686
+            \ makedepends_espresso
             \ makedepends_pentium4
+            \ makedepends_powerpc
+            \ makedepends_powerpc64
+            \ makedepends_powerpc64le
             \ makedepends_ppc
             \ makedepends_aarch64
             \ makedepends_armv7h
@@ -67,7 +83,11 @@ syn keyword srcInfoField
             \ optdepends_x86_64
             \ optdepends_x86_64_v3
             \ optdepends_i686
+            \ optdepends_espresso
             \ optdepends_pentium4
+            \ optdepends_powerpc
+            \ optdepends_powerpc64
+            \ optdepends_powerpc64le
             \ optdepends_ppc
             \ optdepends_aarch64
             \ optdepends_armv7h
@@ -76,7 +96,11 @@ syn keyword srcInfoField
             \ provides_x86_64
             \ provides_x86_64_v3
             \ provides_i686
+            \ provides_espresso
             \ provides_pentium4
+            \ provides_powerpc
+            \ provides_powerpc64
+            \ provides_powerpc64le
             \ provides_ppc
             \ provides_aarch64
             \ provides_armv7h
@@ -85,7 +109,11 @@ syn keyword srcInfoField
             \ conflicts_x86_64
             \ conflicts_x86_64_v3
             \ conflicts_i686
+            \ conflicts_espresso
             \ conflicts_pentium4
+            \ conflicts_powerpc
+            \ conflicts_powerpc64
+            \ conflicts_powerpc64le
             \ conflicts_ppc
             \ conflicts_aarch64
             \ conflicts_armv7h
@@ -94,7 +122,11 @@ syn keyword srcInfoField
             \ replaces_x86_64
             \ replaces_x86_64_v3
             \ replaces_i686
+            \ replaces_espresso
             \ replaces_pentium4
+            \ replaces_powerpc
+            \ replaces_powerpc64
+            \ replaces_powerpc64le
             \ replaces_ppc
             \ replaces_aarch64
             \ replaces_armv7h
@@ -103,7 +135,11 @@ syn keyword srcInfoField
             \ md5sums_x86_64
             \ md5sums_x86_64_v3
             \ md5sums_i686
+            \ md5sums_espresso
             \ md5sums_pentium4
+            \ md5sums_powerpc
+            \ md5sums_powerpc64
+            \ md5sums_powerpc64le
             \ md5sums_ppc
             \ md5sums_aarch64
             \ md5sums_armv7h
@@ -112,7 +148,11 @@ syn keyword srcInfoField
             \ sha1sums_x86_64
             \ sha1sums_x86_64_v3
             \ sha1sums_i686
+            \ sha1sums_espresso
             \ sha1sums_pentium4
+            \ sha1sums_powerpc
+            \ sha1sums_powerpc64
+            \ sha1sums_powerpc64le
             \ sha1sums_ppc
             \ sha1sums_aarch64
             \ sha1sums_armv7h
@@ -121,7 +161,11 @@ syn keyword srcInfoField
             \ sha224sums_x86_64
             \ sha224sums_x86_64_v3
             \ sha224sums_i686
+            \ sha224sums_espresso
             \ sha224sums_pentium4
+            \ sha224sums_powerpc
+            \ sha224sums_powerpc64
+            \ sha224sums_powerpc64le
             \ sha224sums_ppc
             \ sha224sums_aarch64
             \ sha224sums_armv7h
@@ -130,7 +174,11 @@ syn keyword srcInfoField
             \ sha256sums_x86_64
             \ sha256sums_x86_64_v3
             \ sha256sums_i686
+            \ sha256sums_espresso
             \ sha256sums_pentium4
+            \ sha256sums_powerpc
+            \ sha256sums_powerpc64
+            \ sha256sums_powerpc64le
             \ sha256sums_ppc
             \ sha256sums_aarch64
             \ sha256sums_armv7h
@@ -139,7 +187,11 @@ syn keyword srcInfoField
             \ sha384sums_x86_64
             \ sha384sums_x86_64_v3
             \ sha384sums_i686
+            \ sha384sums_espresso
             \ sha384sums_pentium4
+            \ sha384sums_powerpc
+            \ sha384sums_powerpc64
+            \ sha384sums_powerpc64le
             \ sha384sums_ppc
             \ sha384sums_aarch64
             \ sha384sums_armv7h
@@ -148,7 +200,11 @@ syn keyword srcInfoField
             \ sha512sums_x86_64
             \ sha512sums_x86_64_v3
             \ sha512sums_i686
+            \ sha512sums_espresso
             \ sha512sums_pentium4
+            \ sha512sums_powerpc
+            \ sha512sums_powerpc64
+            \ sha512sums_powerpc64le
             \ sha512sums_ppc
             \ sha512sums_aarch64
             \ sha512sums_armv7h
@@ -157,7 +213,11 @@ syn keyword srcInfoField
             \ b2sums_x86_64
             \ b2sums_x86_64_v3
             \ b2sums_i686
+            \ b2sums_espresso
             \ b2sums_pentium4
+            \ b2sums_powerpc
+            \ b2sums_powerpc64
+            \ b2sums_powerpc64le
             \ b2sums_ppc
             \ b2sums_aarch64
             \ b2sums_armv7h
