diff --git i/tests/test_config.py w/tests/test_config.py
index 8ae94ff..e17ed4c 100644
--- i/tests/test_config.py
+++ w/tests/test_config.py
@@ -1,6 +1,6 @@
 import os
 
-import mock
+from unittest import mock
 import pytest
 from sqlalchemy.pool import NullPool
 
diff --git i/tests/test_utils.py w/tests/test_utils.py
index 4c673f9..6c53f07 100644
--- i/tests/test_utils.py
+++ w/tests/test_utils.py
@@ -1,4 +1,4 @@
-import mock
+from unittest import mock
 
 from flask_sqlalchemy import utils
 
