diff -ruN pydantic-core-2.15.0.orig/Cargo.toml pydantic-core-2.15.0/Cargo.toml
--- pydantic-core-2.15.0.orig/Cargo.toml	2024-01-13 20:51:03.248020576 +0100
+++ pydantic-core-2.15.0/Cargo.toml	2024-01-13 20:51:30.749157039 +0100
@@ -26,7 +26,7 @@
 ]
 
 [dependencies]
-pyo3 = { version = "0.20.1", features = ["generate-import-lib", "num-bigint"] }
+pyo3 = { version = "0.20.2", features = ["generate-import-lib", "num-bigint"] }
 regex = "1.10.2"
 strum = { version = "0.25.0", features = ["derive"] }
 strum_macros = "0.25.3"
