--- pydantic-core-2.16.3/Cargo.toml.orig	2024-03-11 14:21:45.007490906 +0100
+++ pydantic-core-2.16.3/Cargo.toml	2024-03-11 14:21:59.577986703 +0100
@@ -26,7 +26,7 @@
 ]
 
 [dependencies]
-pyo3 = { version = "0.20.2", features = ["generate-import-lib", "num-bigint"] }
+pyo3 = { version = "0.20.3", features = ["generate-import-lib", "num-bigint"] }
 regex = "1.10.2"
 strum = { version = "0.25.0", features = ["derive"] }
 strum_macros = "0.25.3"
@@ -70,12 +70,12 @@
 strip = false
 
 [dev-dependencies]
-pyo3 = { version = "0.20.2", features = ["auto-initialize"] }
+pyo3 = { version = "0.20.3", features = ["auto-initialize"] }
 
 [build-dependencies]
 version_check = "0.9.4"
 # used where logic has to be version/distribution specific, e.g. pypy
-pyo3-build-config = { version = "0.20.2" }
+pyo3-build-config = { version = "0.20.3" }
 
 [lints.clippy]
 dbg_macro = "warn"
