diff --git a/CMakeLists.txt b/CMakeLists.txt
index 39b136d..5ebbb46 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,9 @@ find_package(Qt6XdgIconLoader ${QTXDG_MINIMUM_VERSION} REQUIRED)
 find_package(lxqt2-build-tools ${LXQTBT_MINIMUM_VERSION} REQUIRED)
 find_package(dbusmenu-lxqt ${DBUSMENU_MINIMUM_VERSION} REQUIRED)
 find_package(fm-qt6 ${FMQT_MINIMUM_VERSION} REQUIRED)
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
+endif()
 
 get_target_property(LIB_FM_QT_CONFIGURATIONS fm-qt6 IMPORTED_CONFIGURATIONS)
 if (LIB_FM_QT_CONFIGURATIONS)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 008e2be..aec8e3b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -38,6 +38,9 @@ target_link_libraries(qtlxqt
     dbusmenu-lxqt
     Qt6XdgIconLoader
 )
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  target_link_libraries(qtlxqt Qt6::GuiPrivate)
+endif()
 
 lxqt_query_qt(_QT_PLUGINS_DIR QT_INSTALL_PLUGINS)
 
