--- kwallet-5.101.0.orig/src/runtime/kwalletd/backend/backendpersisthandler.cpp	2023-01-01 22:29:39.893983790 +0100
+++ kwallet-5.101.0/src/runtime/kwalletd/backend/backendpersisthandler.cpp	2023-01-01 22:31:21.423278675 +0100
@@ -639,13 +639,13 @@
                                               btnRetry,
                                               KStandardGuiItem::cancel());
         if (userChoice == KMessageBox::PrimaryAction) {
-            decryptedData.seek(0, SEEK_SET);
+            decryptedData.rewind();
             goto retry_label;
         }
         return -1;
     }
 
-    decryptedData.seek(0, SEEK_SET);
+    decryptedData.rewind();
     QByteArray dataBuffer;
     while ((bytes = decryptedData.read(buffer, sizeof(buffer) / sizeof(buffer[0])))) {
         dataBuffer.append(buffer, bytes);
