--- kwallet-5.95.0/src/runtime/kwalletd/backend/backendpersisthandler.cpp.orig	2022-06-24 19:01:42.165582800 +0200
+++ kwallet-5.95.0/src/runtime/kwalletd/backend/backendpersisthandler.cpp	2022-06-24 19:04:13.182476072 +0200
@@ -565,7 +565,7 @@
 
     char buffer[4096];
     ssize_t bytes = 0;
-    encryptedData.seek(0, SEEK_SET);
+    encryptedData.rewind();
     while ((bytes = encryptedData.read(buffer, sizeof(buffer) / sizeof(buffer[0]))) > 0) {
         if (sf.write(buffer, bytes) != bytes) {
             KMessageBox::errorWId(w,
@@ -621,7 +621,7 @@
     }
 
     GpgME::Data decryptedData;
-    encryptedData.seek(0, SEEK_SET);
+    encryptedData.rewind();
     GpgME::DecryptionResult res = ctx->decrypt(encryptedData, decryptedData);
     if (res.error()) {
         qCDebug(KWALLETBACKEND_LOG) << "Error decrypting message: " << res.error().asString() << ", code " << res.error().code() << ", source "
@@ -638,13 +638,13 @@
                                                       btnRetry,
                                                       KStandardGuiItem::cancel());
         if (userChoice == KMessageBox::Yes) {
-            decryptedData.seek(0, SEEK_SET);
+            decryptedData.rewind();
             goto retry_label;
         }
         return -1;
     }
 
-    decryptedData.seek(0, SEEK_SET);
+    decryptedData.rewind();
     QByteArray dataBuffer;
     while (bytes = decryptedData.read(buffer, sizeof(buffer) / sizeof(buffer[0]))) {
         dataBuffer.append(buffer, bytes);

