--- libcss-0.9.1/Makefile.orig	2022-04-27 11:26:55.640384439 +0200
+++ libcss-0.9.1/Makefile	2022-04-27 11:27:05.303373667 +0200
@@ -21,12 +21,6 @@
 WARNFLAGS := -Wall -W -Wundef -Wpointer-arith -Wcast-align \
 	-Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wnested-externs
-# BeOS/Haiku/AmigaOS4 standard library headers create warnings
-ifneq ($(HOST),i586-pc-haiku)
-  ifneq ($(findstring amigaos,$(HOST)),amigaos)
-    WARNFLAGS := $(WARNFLAGS) -Werror
-  endif
-endif
 
 CFLAGS := -D_DEFAULT_SOURCE -D_DEFAULT_SOURCE \
 	-I$(CURDIR)/include/ -I$(CURDIR)/src \
