From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sat, 21 Jun 2025 09:19:16 +0200
Subject: [PATCH] Adapt to pkgconf-2.5.0

---
 LibPkgConf.xs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/LibPkgConf.xs b/LibPkgConf.xs
index 57e6892996ee..5fa1c00dca4c 100644
--- a/LibPkgConf.xs
+++ b/LibPkgConf.xs
@@ -334,16 +334,20 @@ _package_from_file(self, filename)
     FILE *fp;
     pkgconf_pkg_t *package;
   CODE:
+#if LIBPKGCONF_VERSION >= 20500
+    RETVAL = PTR2IV(pkgconf_pkg_new_from_path(&self->client, filename, 0));
+#else
     fp = fopen(filename, "r");
     if(fp != NULL) {
 #if LIBPKGCONF_VERSION >= 10900
       package = pkgconf_pkg_new_from_file(&self->client, filename, fp, 0);
 #else
       package = pkgconf_pkg_new_from_file(&self->client, filename, fp);
 #endif
       RETVAL = PTR2IV(package);
     } else
       RETVAL = 0;
+#endif
   OUTPUT:
     RETVAL
 
