diff -Nurp PkgConfig-LibPkgConf-0.11/t/client.t PkgConfig-LibPkgConf-0.11-2.0.3/t/client.t
--- PkgConfig-LibPkgConf-0.11/t/client.t	2023-11-22 19:46:56.204983196 +0100
+++ PkgConfig-LibPkgConf-0.11-2.0.3/t/client.t	2023-11-22 19:48:31.990180955 +0100
@@ -216,13 +216,13 @@ subtest 'path attributes' => sub {
 
     _is_deeply
       [PkgConfig::LibPkgConf::Client->new->path],
-      [map { path_relocate "$root$_" } qw( /foo /bar /baz /ralph )];
+      [map { path_relocate "$root$_" } qw( /bar /foo /ralph /baz )];
     _is_deeply
       [PkgConfig::LibPkgConf::Client->new(path => join($sep, map { "$root$_" } qw( /trans /formers )))->path],
-      [map { path_relocate "$root$_" } qw( /trans /formers )];
+      [map { path_relocate "$root$_" } qw( /formers /trans )];
     _is_deeply
       [PkgConfig::LibPkgConf::Client->new(path => [map { "$root$_" } qw( /trans /formers )])->path],
-      [map { path_relocate "$root$_" } qw( /trans /formers )];
+      [map { path_relocate "$root$_" } qw( /formers /trans )];
 
   };
 
@@ -232,10 +232,10 @@ subtest 'path attributes' => sub {
 
     _is_deeply
       [PkgConfig::LibPkgConf::Client->new->filter_lib_dirs],
-      [map { path_relocate "$root$_" } qw( /foo/lib /bar/lib )];
+      [map { path_relocate "$root$_" } qw( /bar/lib /foo/lib )];
     _is_deeply
       [PkgConfig::LibPkgConf::Client->new(filter_lib_dirs => [map { "$root$_" } qw( /trans/lib /formers/lib )])->filter_lib_dirs],
-      [map { path_relocate "$root$_" } qw( /trans/lib /formers/lib )];
+      [map { path_relocate "$root$_" } qw( /formers/lib /trans/lib )];
 
   };
 
@@ -245,10 +245,10 @@ subtest 'path attributes' => sub {
 
     _is_deeply
       [PkgConfig::LibPkgConf::Client->new->filter_include_dirs],
-      [map { path_relocate "$root$_" } qw( /foo/include /bar/include )];
+      [map { path_relocate "$root$_" } qw( /bar/include /foo/include )];
     _is_deeply
       [PkgConfig::LibPkgConf::Client->new(filter_include_dirs => [map { "$root$_" } qw( /trans/include /formers/include )])->filter_include_dirs],
-      [map { path_relocate "$root$_" } qw( /trans/include /formers/include )];
+      [map { path_relocate "$root$_" } qw( /formers/include /trans/include )];
 
   };
 
