diff --git a/tool/lemon.c b/tool/lemon.c
index 324dda0c5f..0842a07332 100644
--- a/tool/lemon.c
+++ b/tool/lemon.c
@@ -3763,6 +3763,7 @@ PRIVATE void tplt_skip_header(FILE *in, int *lineno)
 PRIVATE FILE *tplt_open(struct lemon *lemp)
 {
   static char templatename[] = "lempar.c";
+  static char system_templatename[] = "/usr/share/lemon/lempar.c";
   char buf[1000];
   FILE *in;
   char *tpltname;
@@ -3797,6 +3798,8 @@ PRIVATE FILE *tplt_open(struct lemon *lemp)
     tpltname = buf;
   }else if( access(templatename,004)==0 ){
     tpltname = templatename;
+  }else if( access(system_templatename,004)==0 ){
+    tpltname = system_templatename;
   }else{
     toFree = tpltname = pathsearch(lemp->argv[0],templatename,0);
   }
