--- assimp-5.2.0/code/CMakeLists.txt.orig	2022-02-03 10:41:53.120702607 +0100
+++ assimp-5.2.0/code/CMakeLists.txt	2022-02-03 10:42:23.897591554 +0100
@@ -1173,11 +1173,9 @@
 
 TARGET_USE_COMMON_OUTPUT_DIRECTORY(assimp)
 
-# enable warnings as errors ########################################
+# enable warnings as errors  only with MSVC ###########
 IF (MSVC)
   TARGET_COMPILE_OPTIONS(assimp PRIVATE /WX)
-ELSE()
-  TARGET_COMPILE_OPTIONS(assimp PRIVATE -Werror)
 ENDIF()
 
 # adds C_FLAGS required to compile zip.c on old GCC 4.x compiler
