--- a/custom/conf/app.example.ini
+++ b/custom/conf/app.example.ini
@@ -51,14 +51,14 @@ APP_NAME = ; Forgejo: Beyond coding. We Forge.
 ;APP_DISPLAY_NAME_FORMAT = {APP_NAME}: {APP_SLOGAN}
 ;;
 ;; RUN_USER will automatically detect the current user - but you can set it here change it if you run locally
-RUN_USER = ; git
+RUN_USER = forgejo
 ;;
 ;; Application run mode, affects performance and debugging: "dev" or "prod", default is "prod"
 ;; Mode "dev" makes Forgejo easier to develop and debug, values other than "dev" are treated as "prod" which is for production use.
 ;RUN_MODE = prod
 ;;
 ;; The working directory, see the comment of AppWorkPath above
-;WORK_PATH =
+WORK_PATH = /var/lib/forgejo
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
@@ -577,14 +577,14 @@ ENABLED = true
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Root path for the log files - defaults to %(GITEA_WORK_DIR)/log
-;ROOT_PATH =
+ROOT_PATH = /var/log/forgejo
 ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Main Logger
 ;;
 ;; Either "console", "file" or "conn", default is "console"
 ;; Use comma to separate multiple modes, e.g. "console, file"
-MODE = console
+MODE = file
 ;;
 ;; Either "Trace", "Debug", "Info", "Warn", "Error" or "None", default is "Info"
 LEVEL = Info
@@ -639,9 +639,9 @@ LEVEL = Info
 ;[log.console]
 ;STDERR = false
 ;;
-;[log.file]
+[log.file]
 ;; Set the file_name for the logger. If this is a relative path this will be relative to ROOT_PATH
-;FILE_NAME =
+FILE_NAME = forgejo.log
 ;; This enables automated log rotate(switch of following options), default is true
 ;LOG_ROTATE = true
 ;; Max size shift of a single file, default is 28 means 1 << 28, 256MB
@@ -957,7 +957,7 @@ LEVEL = Info
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;[repository]
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; Root path for storing all repository data. By default, it is set to %(APP_DATA_PATH)s/gitea-repositories.
+;; Root path for storing all repository data. By default, it is set to %(APP_DATA_PATH)s/forgejo-repositories.
 ;; A relative path is interpreted as _`AppWorkPath`_/%(ROOT)s
 ;ROOT =
 ;;
