--- libgovirt-v0.3.9/meson.build.orig	2022-12-06 15:25:32.329050738 +0100
+++ libgovirt-v0.3.9/meson.build	2022-12-06 15:25:43.843829095 +0100
@@ -59,7 +59,6 @@
                         '-DGLIB_VERSION_MAX_ALLOWED=@0@'.format(glib_encoded_version),
                         '-Wall',
                         '-Wextra',
-                        '-Werror',
                         '-Wcast-align',
                         '-Wmissing-declarations',
                         '-Wpointer-arith',
