diff -ruN pbbuttonsd-0.8.1a.orig/pbbuttonsd.cnf.in pbbuttonsd-0.8.1a/pbbuttonsd.cnf.in
--- pbbuttonsd-0.8.1a.orig/pbbuttonsd.cnf.in	2021-04-05 23:27:07.235623187 +0200
+++ pbbuttonsd-0.8.1a/pbbuttonsd.cnf.in	2021-04-05 23:32:46.808512578 +0200
@@ -35,7 +35,7 @@
 
 [MODULE MIXER ALSA]
 Card=default
-Channels=Master, PC Speaker
+Channels=Master, Beep 
 
 [MODULE PMAC]
 TPModeUpKey=225 + alt
diff -ruN pbbuttonsd-0.8.1a.orig/scripts/pmcs-config pbbuttonsd-0.8.1a/scripts/pmcs-config
--- pbbuttonsd-0.8.1a.orig/scripts/pmcs-config	2021-04-05 23:27:07.267624401 +0200
+++ pbbuttonsd-0.8.1a/scripts/pmcs-config	2021-04-05 23:28:16.870265840 +0200
@@ -1,7 +1,7 @@
 # Configuration file for the PowerManagement Control Scripts
 
 # Section for general options
-general_HDDevice="/dev/hda"	 # hdsetup
+general_HDDevice="/dev/sda"	 # hdsetup
 general_PCMCIAShutdown="suspend" # pcmcia
 
 # Path of laptop_mode.sh delivered with the kernel source tree
diff -ruN pbbuttonsd-0.8.1a.orig/scripts/scripts.d/laptopmode.sh pbbuttonsd-0.8.1a/scripts/scripts.d/laptopmode.sh
--- pbbuttonsd-0.8.1a.orig/scripts/scripts.d/laptopmode.sh	2021-04-05 23:27:07.267624401 +0200
+++ pbbuttonsd-0.8.1a/scripts/scripts.d/laptopmode.sh	2021-04-05 23:38:31.789613403 +0200
@@ -121,15 +121,6 @@
 	       echo $a.$b
 	     )
 	 )
-case "$KLEVEL" in
-	"2.4"|"2.6")
-		true
-		;;
-	*)
-		echo "Unhandled kernel version: $KLEVEL ('uname -r' = '$(uname -r)')"
-		exit 1
-		;;
-esac
 
 # Shall we remount journaled fs. with appropiate commit interval? (1=yes)
 DO_REMOUNTS=1
