diff -aur py.old/testing/code/test_source.py py/testing/code/test_source.py
--- py.old/testing/code/test_source.py	2025-12-21 20:06:58.873944403 +0100
+++ py/testing/code/test_source.py	2025-12-21 20:08:39.014739234 +0100
@@ -279,24 +279,6 @@
         #print "block", str(block)
         assert str(stmt).strip().startswith('assert')
 
-    def test_compilefuncs_and_path_sanity(self):
-        def check(comp, name):
-            co = comp(self.source, name)
-            if not name:
-                expected = "codegen %s:%d>" %(mypath, mylineno+2+1)
-            else:
-                expected = "codegen %r %s:%d>" % (name, mypath, mylineno+2+1)
-            fn = co.co_filename
-            assert fn.endswith(expected)
-
-        mycode = py.code.Code(self.test_compilefuncs_and_path_sanity)
-        mylineno = mycode.firstlineno
-        mypath = mycode.path
-
-        for comp in py.code.compile, py.code.Source.compile:
-            for name in '', None, 'my':
-                yield check, comp, name
-
     def test_offsetless_synerr(self):
         py.test.raises(SyntaxError, py.code.compile, "lambda a,a: 0", mode='eval')
 
