awoo..?

--- a/llvm/lib/Target/PowerPC/PPCISelLowering.cpp 2024-06-16 03:21:32.000000000 +1000
+++ b/llvm/lib/Target/PowerPC/PPCISelLowering.cpp 2025-02-23 17:44:28.962696748 +1100
@@ -11943,6 +11943,9 @@ PPCTargetLowering::EmitAtomicBinary(Mach
     BB->addSuccessor(exitMBB);
     BB = loop2MBB;
   }
+  BuildMI(BB, dl, TII->get(PPC::DCBST))
+    .addReg(ptrA)
+    .addReg(ptrB);
   BuildMI(BB, dl, TII->get(StoreMnemonic))
     .addReg(TmpReg).addReg(ptrA).addReg(ptrB);
   BuildMI(BB, dl, TII->get(PPC::BCC))
@@ -12200,6 +12202,9 @@ MachineBasicBlock *PPCTargetLowering::Em
     BB = loop2MBB;
   }
   BuildMI(BB, dl, TII->get(PPC::OR), Tmp4Reg).addReg(Tmp3Reg).addReg(Tmp2Reg);
+  BuildMI(BB, dl, TII->get(PPC::DCBST))
+      .addReg(ZeroReg)
+      .addReg(PtrReg);
   BuildMI(BB, dl, TII->get(PPC::STWCX))
       .addReg(Tmp4Reg)
       .addReg(ZeroReg)
@@ -13018,6 +13023,9 @@ PPCTargetLowering::EmitInstrWithCustomIn
     BB->addSuccessor(exitMBB);
 
     BB = loop2MBB;
+    BuildMI(BB, dl, TII->get(PPC::DCBST))
+       .addReg(ptrA)
+       .addReg(ptrB);
     BuildMI(BB, dl, TII->get(StoreMnemonic))
         .addReg(newval)
         .addReg(ptrA)
@@ -13191,6 +13199,9 @@ PPCTargetLowering::EmitInstrWithCustomIn
     BuildMI(BB, dl, TII->get(PPC::OR), Tmp4Reg)
         .addReg(Tmp2Reg)
         .addReg(NewVal3Reg);
+    BuildMI(BB, dl, TII->get(PPC::DCBST))
+       .addReg(ZeroReg)
+       .addReg(PtrReg);
     BuildMI(BB, dl, TII->get(PPC::STWCX))
         .addReg(Tmp4Reg)
         .addReg(ZeroReg)
