--- a/drivers/clk/starfive/clk-starfive-jh7110-pll.c	2023-10-12 18:01:17.405091587 +0200
+++ b/drivers/clk/starfive/clk-starfive-jh7110-pll.c	2023-10-12 18:01:35.667256650 +0200
@@ -280,7 +280,7 @@
 	debugfs_create_regset32("registers", 0400, dentry, regset);
 }
 #else
-#define jh7110_clk_debug_init NULL
+#define jh7110_clk_pll_debug_init NULL
 #endif
 
 static const struct clk_ops jh7110_clk_pll_ops = {
