--- gcc-9.2.0/libphobos/src/std/math.d.orig	2019-09-09 14:45:19.921226009 +0200
+++ gcc-9.2.0/libphobos/src/std/math.d	2019-09-09 14:45:42.258908286 +0200
@@ -246,7 +246,7 @@
 version (LittleEndian)
 {
     static assert(real.mant_dig == 53 || real.mant_dig == 64
-               || real.mant_dig == 113,
+               || real.mant_dig == 113 || real.mant_dig == 106,
       "Only 64-bit, 80-bit, and 128-bit reals"~
       " are supported for LittleEndian CPUs");
 }
