commit f49e4b1a393c0378e43b65c36765676c98d018c4
Author: Christos Zoulas <christos@zoulas.com>
Date:   Thu Mar 20 14:57:41 2025 +0000

    Fix termios handling on linux/ppc (Dr. Werner Fink)

diff --git a/src/seccomp.c b/src/seccomp.c
index 9e00034a..f05c30f9 100644
--- a/src/seccomp.c
+++ b/src/seccomp.c
@@ -33,9 +33,14 @@
 #if HAVE_LIBSECCOMP
 #include <seccomp.h> /* libseccomp */
 #include <sys/prctl.h> /* prctl */
-#include <sys/ioctl.h>
 #include <sys/socket.h>
-#include <termios.h>
+#ifdef __powerpc64__
+// See: https://sourceware.org/bugzilla/show_bug.cgi?id=32806
+# include <asm/termbits.h>
+#else
+# include <termios.h>
+#endif
+#include <sys/ioctl.h>
 #include <fcntl.h>
 #include <stdlib.h>
 #include <errno.h>
