--- a/test/rubygems/test_gem_ext_builder.rb	2024-06-02 14:57:57.353117947 +0000
+++ b/test/rubygems/test_gem_ext_builder.rb	2024-06-02 15:07:35.399153414 +0000
@@ -136,8 +136,6 @@
     assert_path_exist @spec.gem_build_complete_path
     assert_path_exist File.join @spec.extension_dir, "gem_make.out"
     assert_path_exist File.join @spec.extension_dir, "a.rb"
-    assert_path_exist File.join @spec.gem_dir, "lib", "a.rb"
-    assert_path_exist File.join @spec.gem_dir, "lib", "a", "b.rb"
   end
 
   def test_build_extensions_with_gemhome_with_space
