diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index ec1bc0cd8ee8..fae5e2ab56b0 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -58,9 +58,7 @@ endif
 
 include $(srctree)/arch/riscv/Makefile.isa
 
-# Remove F,D,V from isa string for all. Keep extensions between "fd" and "v" by
-# matching non-v and non-multi-letter extensions out with the filter ([^v_]*)
-KBUILD_CFLAGS += -march=$(shell echo $(riscv-march-y) | sed -E 's/(rv32ima|rv64ima)fd([^v_]*)v?/\1\2/')
+KBUILD_CFLAGS += -march=$(riscv-march-y)
 
 KBUILD_AFLAGS += -march=$(riscv-march-y)
 
