--- pygtk-2.24.0/pango.c.orig	2019-10-16 16:20:57.310593640 +0200
+++ pygtk-2.24.0/pango.c	2019-10-16 16:21:09.529309109 +0200
@@ -3819,7 +3819,7 @@
         return NULL;
     }
     klass = g_type_class_ref(pyg_type_from_object(cls));
-    if (PANGO_FONT_CLASS(klass)->find_shaper)
+    if (PANGO_FONT_CLASS(klass)->pango_font_find_shaper)
         ret = PANGO_FONT_CLASS(klass)->find_shaper(PANGO_FONT(self->obj), lang, ch);
     else {
         PyErr_SetString(PyExc_NotImplementedError, "virtual method Pango.Font.find_shaper not implemented");
