--- chmlib-0.40/src/chm_lib.c.orig	2019-05-05 20:55:19.050000000 +0000
+++ chmlib-0.40/src/chm_lib.c	2019-05-05 20:56:01.950000000 +0000
@@ -164,7 +164,7 @@
 
 /* x86-64 */
 /* Note that these may be appropriate for other 64-bit machines. */
-#elif __x86_64__ || __ia64__
+#elif __x86_64__ || __ia64__ || __powerpc64__ || __powerpc__ || (__riscv && (__riscv_xlen == 64))
 typedef unsigned char           UChar;
 typedef short                   Int16;
 typedef unsigned short          UInt16;
