Description: Fix implicit declaration after armhf time_t transition
Author: Zixing Liu <zixing.liu@canonical.com>
Forwarded: no
Last-Update: 2024-04-03

Index: libcdio/lib/driver/_cdio_stdio.h
===================================================================
--- libcdio.orig/lib/driver/_cdio_stdio.h
+++ libcdio/lib/driver/_cdio_stdio.h
@@ -22,6 +22,12 @@
 
 #include "_cdio_stream.h"
 
+#include <features.h>
+#if defined(_FILE_OFFSET_BITS) && defined(__REDIRECT) && (_FILE_OFFSET_BITS == 64)
+#define lseek64 lseek
+#define fseeko64 fseeko
+#endif
+
 /*!
   Initialize a new stdio stream reading from pathname.
   A pointer to the stream is returned or NULL if there was an error.
