
diff --git a/pr/include/md/_linux.cfg b/pr/include/md/_linux.cfg
--- a/pr/include/md/_linux.cfg
+++ b/pr/include/md/_linux.cfg
@@ -494,17 +494,21 @@
 #define PR_ALIGN_OF_WORD    4
 
 #define PR_BYTES_PER_WORD_LOG2   2
 #define PR_BYTES_PER_DWORD_LOG2  3
 
 #elif defined(__mips__)
 
 /* For _ABI64 */
+#if defined(__GLIBC__) || defined(__UCLIBC__)
+#include <sgidefs.h>
+#else
 #include <asm/sgidefs.h>
+#endif
 
 #ifdef __MIPSEB__
 #define IS_BIG_ENDIAN 1
 #undef  IS_LITTLE_ENDIAN
 #elif defined(__MIPSEL__)
 #define IS_LITTLE_ENDIAN 1
 #undef  IS_BIG_ENDIAN
 #else




