diff --git a/support/hoe/deprecated_gem.rb b/support/hoe/deprecated_gem.rb
index dc266bd..b44dce7 100644
--- a/support/hoe/deprecated_gem.rb
+++ b/support/hoe/deprecated_gem.rb
@@ -1,12 +1,28 @@
-
 # A Hoe plug-in to provide a second, linked gemspec, for a gem that has been
 # deprecated in favour of a modern name. (The name is an artifact of Hoe's
 # plugin loading.)
-module Hoe::Deprecated_Gem # rubocop:disable Style/ClassAndModuleCamelCase
+module Hoe::Deprecated_Gem # standard:disable Naming/ClassAndModuleCamelCase
   def linked_spec(spec)
-    atm = YAML.load(YAML.dump(spec))
-    atm.name = 'archive-tar-minitar'
-    d = %Q('#{atm.name}' has been deprecated; just install '#{spec.name}'.)
+    permitted_classes = %w[
+      Symbol Time Date Gem::Dependency Gem::Platform Gem::Requirement
+      Gem::Specification Gem::Version Gem::Version::Requirement
+      YAML::Syck::DefaultKey Syck::DefaultKey
+    ]
+    permitted_symbols = %w[development runtime]
+    atm = begin
+      YAML.safe_load(
+        YAML.dump(spec),
+        :permitted_classes => permitted_classes,
+        :permitted_symbols => permitted_symbols,
+        :aliases => true
+      )
+    rescue
+      YAML.safe_load(
+        YAML.dump(spec), permitted_classes, permitted_symbols, true
+      )
+    end
+    atm.name = "archive-tar-minitar"
+    d = %('#{atm.name}' has been deprecated; just install '#{spec.name}'.)
     atm.description = "#{d} #{spec.description}"
     atm.summary = atm.post_install_message = d
     atm.files.delete_if do |f|
@@ -16,13 +32,13 @@ module Hoe::Deprecated_Gem # rubocop:disable Style/ClassAndModuleCamelCase
     atm.rdoc_options.clear
     atm.dependencies.clear
 
-    version = Gem::Version.new(spec.version.segments.first(2).join('.'))
+    version = Gem::Version.new(spec.version.segments.first(2).join("."))
 
     atm.add_dependency(spec.name, "~> #{version}")
-    atm.add_dependency(%Q(#{spec.name}-cli), "~> #{version}")
+    atm.add_dependency(%(#{spec.name}-cli), "~> #{version}")
 
     unless @include_all
-      [ :signing_key, :cert_chain ].each { |name|
+      [:signing_key, :cert_chain].each { |name|
         atm.send("#{name}=".to_sym, atm.default_value(name))
       }
     end
@@ -31,11 +47,11 @@ module Hoe::Deprecated_Gem # rubocop:disable Style/ClassAndModuleCamelCase
   end
 
   def define_deprecated_gem_tasks
-    gemspec = spec.name + '.gemspec'
-    atmspec = 'archive-tar-minitar.gemspec'
+    gemspec = spec.name + ".gemspec"
+    atmspec = "archive-tar-minitar.gemspec"
 
     file atmspec => gemspec do
-      open(atmspec, 'w') { |f| f.write(linked_spec(spec).to_ruby) }
+      File.open(atmspec, "w") { |f| f.write(linked_spec(spec).to_ruby) }
     end
 
     task :gemspec => atmspec
