diff -Naur a/drivers/ps3/ps3stor_lib.c b/drivers/ps3/ps3stor_lib.c
--- a/drivers/ps3/ps3stor_lib.c	2024-03-17 19:50:30.722503769 +0700
+++ b/drivers/ps3/ps3stor_lib.c	2024-03-17 19:50:34.599417581 +0700
@@ -329,8 +329,13 @@
 		return -1;
 	}
 
-	wait_for_completion(&dev->done);
-	if (dev->lv1_status) {
+	res = wait_for_completion_timeout(&dev->done, msecs_to_jiffies(2000));
+	if (res == 0) {
+		dev_err(&dev->sbd.core,
+			"%s:%u: send_device_command 0x%llx timed out\n",
+			__func__, __LINE__, cmd);
+		return -1;
+	} else if (dev->lv1_status) {
 		dev_dbg(&dev->sbd.core, "%s:%u: command 0x%llx failed 0x%llx\n",
 			__func__, __LINE__, cmd, dev->lv1_status);
 		return dev->lv1_status;
