diff -Naur a/drivers/ps3/ps3-sys-manager.c b/drivers/ps3/ps3-sys-manager.c
--- a/drivers/ps3/ps3-sys-manager.c	2024-03-17 19:51:07.053701057 +0700
+++ b/drivers/ps3/ps3-sys-manager.c	2024-03-17 19:51:08.173676485 +0700
@@ -645,7 +645,7 @@
 	ps3_vuart_cancel_async(dev);
 
 	ps3_sys_manager_send_attr(dev, 0);
-	ps3_sys_manager_send_next_op(dev, PS3_SM_NEXT_OP_SYS_REBOOT,
+	ps3_sys_manager_send_next_op(dev, PS3_SM_NEXT_OP_LPAR_REBOOT,
 		user_wake_sources);
 
 	ps3_sys_manager_fin(dev);
