diff -Naur a/arch/powerpc/include/asm/lv1call.h b/arch/powerpc/include/asm/lv1call.h
--- a/arch/powerpc/include/asm/lv1call.h	2024-03-17 19:51:55.230652903 +0700
+++ b/arch/powerpc/include/asm/lv1call.h	2024-03-17 19:51:58.029592547 +0700
@@ -251,10 +251,10 @@
 LV1_CALL(configure_execution_time_variable,             1, 0,  77 )
 LV1_CALL(get_spe_irq_outlet,                            2, 1,  78 )
 LV1_CALL(set_spe_privilege_state_area_1_register,       3, 0,  79 )
-LV1_CALL(create_repository_node,                        6, 0,  90 )
+LV1_CALL(create_repository_node,                        7, 0,  90 )
 LV1_CALL(read_repository_node,                          5, 2,  91 )
-LV1_CALL(write_repository_node,                         6, 0,  92 )
-LV1_CALL(delete_repository_node,                        4, 0,  93 )
+LV1_CALL(write_repository_node,                         7, 0,  92 )
+LV1_CALL(delete_repository_node,                        5, 0,  93 )
 LV1_CALL(read_htab_entries,                             2, 5,  95 )
 LV1_CALL(set_dabr,                                      2, 0,  96 )
 LV1_CALL(get_total_execution_time,                      2, 1, 103 )
