diff -Naur a/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c b/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c
--- a/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c	2024-03-17 19:52:55.396367644 +0700
+++ b/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c	2024-03-17 19:52:56.736339299 +0700
@@ -171,7 +171,7 @@
 				      &cmd->tag, &cmd->size);
 	if (cmd->status) {
 		complete(&cmd->done);
-		pr_info("%s: cmd issue failed\n", __func__);
+		pr_info("%s: cmd (%d) issue failed (%d)\n", __func__, cmd->cmd, cmd->status);
 		return;
 	}
 
