diff -Naur a/arch/powerpc/platforms/ps3/interrupt.c b/arch/powerpc/platforms/ps3/interrupt.c
--- a/arch/powerpc/platforms/ps3/interrupt.c	2024-03-17 19:53:18.963870784 +0700
+++ b/arch/powerpc/platforms/ps3/interrupt.c	2024-03-17 19:53:21.646814444 +0700
@@ -375,6 +375,7 @@
 {
 	return lv1_send_event_locally(virq_to_hw(virq));
 }
+EXPORT_SYMBOL_GPL(ps3_event_receive_port_destroy);
 
 /**
  * ps3_sb_event_receive_port_setup - Setup a system bus event receive port.
