diff -Naur a/arch/powerpc/include/asm/ps3.h b/arch/powerpc/include/asm/ps3.h
--- a/arch/powerpc/include/asm/ps3.h	2024-03-17 19:54:42.242141737 +0700
+++ b/arch/powerpc/include/asm/ps3.h	2024-03-17 19:54:45.076083579 +0700
@@ -340,6 +340,7 @@
 #define PS3_MODULE_ALIAS_STOR_NOR_FLASH	"ps3:12:0"
 #define PS3_MODULE_ALIAS_DISP_MANAGER	"ps3:13:0"
 #define PS3_MODULE_ALIAS_STOR_ENCDEC	"ps3:14:0"
+#define PS3_MODULE_ALIAS_JUPITER	"ps3:15:0"
 
 enum ps3_system_bus_device_type {
 	PS3_DEVICE_TYPE_IOC0 = 1,
diff -Naur a/drivers/net/wireless/ps3jupiter/ps3_jupiter.c b/drivers/net/wireless/ps3jupiter/ps3_jupiter.c
--- a/drivers/net/wireless/ps3jupiter/ps3_jupiter.c	2024-03-17 19:54:43.865108425 +0700
+++ b/drivers/net/wireless/ps3jupiter/ps3_jupiter.c	2024-03-17 19:54:45.076083579 +0700
@@ -1238,8 +1238,8 @@
 module_init(ps3_jupiter_init);
 module_exit(ps3_jupiter_exit);
 
-MODULE_SUPPORTED_DEVICE("PS3 Jupiter");
 MODULE_DEVICE_TABLE(usb, ps3_jupiter_devtab);
 MODULE_DESCRIPTION("PS3 Jupiter");
 MODULE_AUTHOR("glevand");
 MODULE_LICENSE("GPL");
+MODULE_ALIAS(PS3_MODULE_ALIAS_JUPITER);
diff -Naur a/drivers/net/wireless/ps3jupiter/ps3_jupiter_sta.c b/drivers/net/wireless/ps3jupiter/ps3_jupiter_sta.c
--- a/drivers/net/wireless/ps3jupiter/ps3_jupiter_sta.c	2024-03-17 19:54:43.865108425 +0700
+++ b/drivers/net/wireless/ps3jupiter/ps3_jupiter_sta.c	2024-03-17 19:54:45.077083558 +0700
@@ -2927,8 +2927,8 @@
 module_init(ps3_jupiter_sta_init);
 module_exit(ps3_jupiter_sta_exit);
 
-MODULE_SUPPORTED_DEVICE("PS3 Jupiter STA");
 MODULE_DEVICE_TABLE(usb, ps3_jupiter_sta_devtab);
 MODULE_DESCRIPTION("PS3 Jupiter STA");
 MODULE_AUTHOR("glevand");
 MODULE_LICENSE("GPL");
+MODULE_ALIAS("ps3:15:0");
