diff -Naur a/include/linux/mmzone.h b/include/linux/mmzone.h
--- a/include/linux/mmzone.h	2024-03-17 19:54:52.606929234 +0700
+++ b/include/linux/mmzone.h	2024-03-17 19:54:53.553909847 +0700
@@ -1751,7 +1751,7 @@
 #define SECTION_BLOCKFLAGS_BITS \
 	((1UL << (PFN_SECTION_SHIFT - pageblock_order)) * NR_PAGEBLOCK_BITS)
 
-#if (MAX_PAGE_ORDER + PAGE_SHIFT) > SECTION_SIZE_BITS
+#if (MAX_PAGE_ORDER - 1 + PAGE_SHIFT) > SECTION_SIZE_BITS
 #error Allocator MAX_PAGE_ORDER exceeds SECTION_SIZE
 #endif
 
