diff -Naur a/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c b/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c
--- a/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c	2024-03-17 19:55:08.860597111 +0700
+++ b/drivers/net/ethernet/toshiba/ps3_gelic_wireless.c	2024-03-17 19:55:10.188570035 +0700
@@ -1479,7 +1479,6 @@
 		wl->scan_stat = GELIC_WL_SCAN_STAT_INIT;
 		complete(&wl->scan_done);
 		ret = -ENOMEM;
-		goto out;
 	}
 	kfree(cmd);
 out:
