--- a/arch/powerpc/platforms/ps3/Kconfig	2025-04-24 16:11:44.725214501 -0700
+++ b/arch/powerpc/platforms/ps3/Kconfig	2025-04-24 16:12:57.954980861 -0700
@@ -181,7 +181,7 @@
 
 config PS3_VRAM
 	tristate "PS3 Video RAM Storage Driver"
-	depends on FB_PS3=y && BLOCK && m
+	depends on FB_PS3=y && BLOCK
 	help
 	  This driver allows you to use excess PS3 video RAM as volatile
 	  storage or system swap.
