--- linux-6.14.7/arch/powerpc/boot/Makefile.orig	2025-05-28 17:08:16.831578224 +0200
+++ linux-6.14.7/arch/powerpc/boot/Makefile	2025-05-28 17:09:02.097547379 +0200
@@ -33,12 +33,14 @@
 endif
 
 ifdef CONFIG_PPC64_BOOT_WRAPPER
+BOOTTARGETFLAGS	+= -std=gnu11
 BOOTTARGETFLAGS	+= -m64
 BOOTTARGETFLAGS	+= -mabi=elfv2
 ifdef CONFIG_PPC64_ELF_ABI_V2
 BOOTTARGETFLAGS	+= $(call cc-option,-mabi=elfv2)
 endif
 else
+BOOTTARGETFLAGS	+= -std=gnu11
 BOOTTARGETFLAGS	:= -m32
 endif
 
