diff --git a/tests/run/ufunc.pyx b/tests/run/ufunc.pyx
index f489a07d5..47bb0bfbc 100644
--- a/tests/run/ufunc.pyx
+++ b/tests/run/ufunc.pyx
@@ -125,7 +125,7 @@ def test_plus_one_twice():
     >>> plus_one_twice(int_arr_1d)  # doctest: +ELLIPSIS
     (array([ 1,  5,  9, 13, 17]...), array([ 1,  5,  9, 13, 17]...))
     >>> plus_one_twice(1.j)
-    ((1+1j), (1+1j))
+    (np.complex128(1+1j), np.complex128(1+1j))
 
     2D variant skipped because it's hard to sensible doctest
     """
