--- clit18/clit.c.orig	2024-08-09 13:40:30.898055167 +0200
+++ clit18/clit.c	2024-08-09 13:41:48.526460124 +0200
@@ -125,9 +125,9 @@
             dir_program[i+1] = '\0'; break;
         }
     }
-    printf(sTitle);
+    printf("%s", sTitle);
     if (argc < 3) { 
-        printf(sUsage);
+        printf("%s", sUsage);
         return -1;
     }
     base = 1;
@@ -189,7 +189,7 @@
         lit_close(&lit);
         exit(-1);
     }
-    printf("LIT INFORMATION.........\n");
+    printf("%s", "LIT INFORMATION.........\n");
     if (lit.drmlevel >= 0) 
         printf("DRM         =  %d    \n", lit.drmlevel);
     printf("Timestamp   =  %08lx \n", lit.timestamp);
