diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca73b5b..859ee45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,6 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     I18n
     Config
     CoreAddons
-    Crash
     DBusAddons
     DNSSD
     DocTools
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c15c72..5f95f20 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,7 +83,6 @@ if(KPipeWire_FOUND AND PlasmaWaylandProtocols_FOUND)
     find_package(KWayland REQUIRED)
     find_package(QtWaylandScanner REQUIRED)
     find_package(Qt6WaylandClient)
-    find_package(Qt6XkbCommonSupport)
     find_package(Wayland REQUIRED COMPONENTS Client)
 endif()
 
