diff -r 7aafcc60b4f5 passlib/handlers/bcrypt.py
--- a/passlib/handlers/bcrypt.py	Thu Oct 08 15:01:32 2020 -0400
+++ b/passlib/handlers/bcrypt.py	Sun Jun 09 23:25:35 2024 +0200
@@ -609,7 +609,7 @@
         except ImportError: # pragma: no cover
             return False
         try:
-            version = _bcrypt.__about__.__version__
+            version = _bcrypt.__version__
         except:
             log.warning("(trapped) error reading bcrypt version", exc_info=True)
             version = '<unknown>'
