--- a/jdk/src/share/classes/javax/swing/JTable.java
+++ b/jdk/src/share/classes/javax/swing/JTable.java
@@ -8224,7 +8224,12 @@ public class JTable extends JComponent i
              * @return the <code>AccessibleAction</code>, or <code>null</code>
              */
             public AccessibleAction getAccessibleAction() {
-                return getCurrentAccessibleContext().getAccessibleAction();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleAction();
+                } else {
+                    return null;
+                }
             }
 
             /**
@@ -8246,7 +8251,12 @@ public class JTable extends JComponent i
              *    <code>null</code>
              */
             public AccessibleSelection getAccessibleSelection() {
-                return getCurrentAccessibleContext().getAccessibleSelection();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleSelection();
+                } else {
+                    return null;
+                }
             }
 
             /**
@@ -8256,7 +8266,12 @@ public class JTable extends JComponent i
              * @return the <code>AccessibleText</code>, or <code>null</code>
              */
             public AccessibleText getAccessibleText() {
-                return getCurrentAccessibleContext().getAccessibleText();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleText();
+                } else {
+                    return null;
+                }
             }
 
             /**
@@ -8266,7 +8281,12 @@ public class JTable extends JComponent i
              * @return the <code>AccessibleValue</code>, or <code>null</code>
              */
             public AccessibleValue getAccessibleValue() {
-                return getCurrentAccessibleContext().getAccessibleValue();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleValue();
+                } else {
+                    return null;
+                }
             }
 
 
@@ -9025,7 +9045,12 @@ public class JTable extends JComponent i
              * @return the <code>AccessibleAction</code>, or <code>null</code>
              */
             public AccessibleAction getAccessibleAction() {
-                return getCurrentAccessibleContext().getAccessibleAction();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleAction();
+                } else {
+                    return null;
+                }
             }
 
             /**
@@ -9047,7 +9072,12 @@ public class JTable extends JComponent i
              *    <code>null</code>
              */
             public AccessibleSelection getAccessibleSelection() {
-                return getCurrentAccessibleContext().getAccessibleSelection();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleSelection();
+                } else {
+                    return null;
+                }
             }
 
             /**
@@ -9057,7 +9087,12 @@ public class JTable extends JComponent i
              * @return the <code>AccessibleText</code>, or <code>null</code>
              */
             public AccessibleText getAccessibleText() {
-                return getCurrentAccessibleContext().getAccessibleText();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleText();
+                } else {
+                    return null;
+                }
             }
 
             /**
@@ -9067,7 +9102,12 @@ public class JTable extends JComponent i
              * @return the <code>AccessibleValue</code>, or <code>null</code>
              */
             public AccessibleValue getAccessibleValue() {
-                return getCurrentAccessibleContext().getAccessibleValue();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+		    return ac.getAccessibleValue();
+                } else {
+                    return null;
+                }
             }
 
 
--- a/jdk/src/share/classes/javax/swing/JTree.java
+++ b/jdk/src/share/classes/javax/swing/JTree.java
@@ -4969,7 +4969,7 @@ public class JTree extends JComponent im
             public AccessibleSelection getAccessibleSelection() {
                 AccessibleContext ac = getCurrentAccessibleContext();
                 if (ac != null && isLeaf) {
-                    return getCurrentAccessibleContext().getAccessibleSelection();
+                    return ac.getAccessibleSelection();
                 } else {
                     return this;
                 }
@@ -4984,7 +4984,7 @@ public class JTree extends JComponent im
             public AccessibleText getAccessibleText() {
                 AccessibleContext ac = getCurrentAccessibleContext();
                 if (ac != null) {
-                    return getCurrentAccessibleContext().getAccessibleText();
+                    return ac.getAccessibleText();
                 } else {
                     return null;
                 }
@@ -4999,7 +4999,7 @@ public class JTree extends JComponent im
             public AccessibleValue getAccessibleValue() {
                 AccessibleContext ac = getCurrentAccessibleContext();
                 if (ac != null) {
-                    return getCurrentAccessibleContext().getAccessibleValue();
+                    return ac.getAccessibleValue();
                 } else {
                     return null;
                 }
--- a/jdk/src/share/classes/javax/swing/table/JTableHeader.java
+++ b/jdk/src/share/classes/javax/swing/table/JTableHeader.java
@@ -1088,7 +1088,12 @@ public class JTableHeader extends JCompo
             }
 
             public AccessibleAction getAccessibleAction() {
-                return getCurrentAccessibleContext().getAccessibleAction();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+			return ac.getAccessibleAction();
+		} else {
+			return null;
+		}
             }
 
            /**
@@ -1104,15 +1109,30 @@ public class JTableHeader extends JCompo
             }
 
             public AccessibleSelection getAccessibleSelection() {
-                return getCurrentAccessibleContext().getAccessibleSelection();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+			return ac.getAccessibleSelection();
+		} else {
+			return null;
+		}
             }
 
             public AccessibleText getAccessibleText() {
-                return getCurrentAccessibleContext().getAccessibleText();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+			return ac.getAccessibleText();
+		} else {
+			return null;
+		}
             }
 
             public AccessibleValue getAccessibleValue() {
-                return getCurrentAccessibleContext().getAccessibleValue();
+                AccessibleContext ac = getCurrentAccessibleContext();
+                if (ac != null) {
+			return ac.getAccessibleValue();
+		} else {
+			return null;
+		}
             }
 
 
--- a/jdk/src/windows/classes/com/sun/java/accessibility/AccessBridge.java
+++ b/jdk/src/windows/classes/com/sun/java/accessibility/AccessBridge.java
@@ -6658,7 +6658,7 @@ final public class AccessBridge extends
         public AccessibleSelection getAccessibleSelection() {
             AccessibleContext ac = getCurrentAccessibleContext();
             if (ac != null && isLeaf) {
-                return getCurrentAccessibleContext().getAccessibleSelection();
+                return ac.getAccessibleSelection();
             } else {
                 return this;
             }
@@ -6673,7 +6673,7 @@ final public class AccessBridge extends
         public AccessibleText getAccessibleText() {
             AccessibleContext ac = getCurrentAccessibleContext();
             if (ac != null) {
-                return getCurrentAccessibleContext().getAccessibleText();
+                return ac.getAccessibleText();
             } else {
                 return null;
             }
@@ -6688,7 +6688,7 @@ final public class AccessBridge extends
         public AccessibleValue getAccessibleValue() {
             AccessibleContext ac = getCurrentAccessibleContext();
             if (ac != null) {
-                return getCurrentAccessibleContext().getAccessibleValue();
+                return ac.getAccessibleValue();
             } else {
                 return null;
             }
