Description: fix not accounting for time_t ≠ int in printflike
Origin: https://github.com/openjdk/jdk/commit/d41d2a7a82cb6eff17396717e2e14139ad8179ba.patch
Bug: https://bugs.openjdk.org/browse/JDK-8334502
Forwarded: not-needed (I think/hope)
From: Vladimir Petko <vpetko@openjdk.org>

--- a/hotspot/src/share/vm/runtime/os.cpp
+++ b/hotspot/src/share/vm/runtime/os.cpp
@@ -181,9 +181,9 @@ char* os::iso8601_time(char* buffer, siz
     abs_local_to_UTC = -(abs_local_to_UTC);
   }
   // Convert time zone offset seconds to hours and minutes.
-  const time_t zone_hours = (abs_local_to_UTC / seconds_per_hour);
-  const time_t zone_min =
-    ((abs_local_to_UTC % seconds_per_hour) / seconds_per_minute);
+  const int zone_hours = static_cast<int>(abs_local_to_UTC / seconds_per_hour);
+  const int zone_min =
+    static_cast<int>((abs_local_to_UTC % seconds_per_hour) / seconds_per_minute);
 
   // Print an ISO 8601 date and time stamp into the buffer
   const int year = 1900 + time_struct.tm_year;
