--- a/hotspot/agent/src/os/linux/Makefile
+++ b/hotspot/agent/src/os/linux/Makefile
@@ -81,7 +81,7 @@ LDFLAGS_NO_EXEC_STACK="-Wl,-z,noexecstac
 LFLAGS_LIBSA += $(LDFLAGS_NO_EXEC_STACK)
 
 $(LIBSA): $(ARCH) $(OBJS) mapfile
-        $(GCC) -shared $(LFLAGS_LIBSA) -o $(LIBSA) $(OBJS) $(LIBS)
+        $(GCC) -shared -Wl,-Bsymbolic-functions $(LFLAGS_LIBSA) -o $(LIBSA) $(OBJS) $(LIBS)
 
 test.o: test.c
 	$(GCC) -c -o test.o -g -D_GNU_SOURCE -D$(ARCH) $(INCLUDES) test.c
--- a/hotspot/make/linux/makefiles/gcc.make
+++ b/hotspot/make/linux/makefiles/gcc.make
@@ -306,6 +306,7 @@ endif
 
 # Enable linker optimization
 LFLAGS += -Xlinker -O1
+LFLAGS += -Wl,-Bsymbolic-functions
 
 ifeq ($(USE_CLANG),)
   # If this is a --hash-style=gnu system, use --hash-style=both
