--- ./src/formats/CMakeLists.txt.orig	2016-10-18 23:51:09.619157899 +0200
+++ ./src/formats/CMakeLists.txt	2016-10-18 23:51:39.203157505 +0200
@@ -67,18 +67,6 @@
   )
 endif(MSVC OR HAVE_REGEX_H)
 
-# Support JSON format with MSVC only if JSON_LIBRARY is specified
-if(NOT MSVC OR JSON_LIBRARY)
-  set(formats_json
-    chemdoodlejsonformat
-    pubchemjsonformat
-  )
-  set(json_additional_sources json/customwriter.cpp)
-  if(NOT MSVC) # With MSVC, we link to the .dll rather than compiling it in
-    set(json_additional_sources json/jsoncpp.cpp ${json_additional_sources})
-  endif()
-endif()
-
 set(formats_misc
       APIInterface
       CSRformat
@@ -340,18 +328,5 @@
                         SUFFIX ${MODULE_EXTENSION})
 endforeach(format)
 
-foreach(format ${formats_json})
-  add_library(${format} ${PLUGIN_TYPE} json/${format}.cpp ${json_additional_sources}
-              "${openbabel_BINARY_DIR}/include/openbabel/babelconfig.h")
-  target_link_libraries(${format} ${libs} openbabel)
-  install(TARGETS ${format}
-          RUNTIME DESTINATION ${BIN_INSTALL_DIR}
-          LIBRARY DESTINATION ${OB_PLUGIN_INSTALL_DIR}
-          ARCHIVE DESTINATION ${OB_PLUGIN_INSTALL_DIR})
-  set_target_properties(${format} PROPERTIES
-                        OUTPUT_NAME ${format}
-                        PREFIX ""
-                        SUFFIX ${MODULE_EXTENSION})
-endforeach(format)
 endif(MSVC)
 endif(NOT MINIMAL_BUILD)
