--- libavutil/hwcontext_vaapi.c.orig    2024-06-09 21:22:38.289653589 +0200
+++ libavutil/hwcontext_vaapi.c 2024-06-09 21:26:27.703209698 +0200
@@ -1088,7 +1088,7 @@
     buffer_desc.width        = src_fc->width;
     buffer_desc.height       = src_fc->height;
     buffer_desc.data_size    = desc->objects[0].size;
-    buffer_desc.buffers      = &buffer_handle;
+    buffer_desc.buffers      = (uintptr_t *)&buffer_handle;
     buffer_desc.num_buffers  = 1;
     buffer_desc.flags        = 0;

@@ -1355,7 +1355,7 @@
     }

     av_log(hwfc, AV_LOG_DEBUG, "DRM PRIME fd is %ld.\n",
-           mapping->buffer_info.handle);
+           (long int)mapping->buffer_info.handle);

     mapping->drm_desc.nb_objects = 1;
     mapping->drm_desc.objects[0] = (AVDRMObjectDescriptor) {

