diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4df28710..8ffffbcd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -177,7 +177,7 @@ if(WIN32)
 endif(WIN32)
 
 find_package(Threads)
-find_package(Boost 1.42 COMPONENTS program_options thread system iostreams filesystem REQUIRED)
+find_package(Boost 1.42 COMPONENTS program_options thread iostreams filesystem REQUIRED)
 
 if(Boost_FOUND AND Boost_PROGRAM_OPTIONS_FOUND)
   include_directories(${Boost_INCLUDE_DIRS})
--- a/cmake/liblas-config.cmake.in
+++ b/cmake/liblas-config.cmake.in
@@ -15,7 +15,7 @@
 endif ()
 
 include (CMakeFindDependencyMacro)
-find_dependency (Boost @Boost_VERSION_STRING@ EXACT COMPONENTS program_options thread system iostreams filesystem)
+find_dependency (Boost @Boost_VERSION_STRING@ EXACT COMPONENTS program_options thread iostreams filesystem)
 
 # Tell the user project where to find our headers and libraries
 get_filename_component (_DIR ${CMAKE_CURRENT_LIST_FILE} PATH)
