diff --git a/src/gt_citation.cpp b/src/gt_citation.cpp
index dc39adde..035efe08 100644
--- a/src/gt_citation.cpp
+++ b/src/gt_citation.cpp
@@ -114,15 +114,15 @@ char* ImagineCitationTranslation(char* psCitation, geokey_t keyID)
       p1 = p + strlen(p);
       char *p2 = strchr(p, '\n');
       if(p2)
-        p1 = MIN(p1, p2);
+        p1 = CPL_MIN(p1, p2);
       p2 = strchr(p, '\0');
       if(p2)
-        p1 = MIN(p1, p2);
+        p1 = CPL_MIN(p1, p2);
       for(i=0; keyNames[i]!=NULL; i++)
       {
         p2 = strstr(p, keyNames[i]);
         if(p2)
-          p1 = MIN(p1, p2);
+          p1 = CPL_MIN(p1, p2);
       }
     }
 
@@ -180,15 +180,15 @@ char* ImagineCitationTranslation(char* psCitation, geokey_t keyID)
         p1 = p + strlen(p);
         char *p2 = strchr(p, '\n');
         if(p2)
-          p1 = MIN(p1, p2);
+          p1 = CPL_MIN(p1, p2);
         p2 = strchr(p, '\0');
         if(p2)
-          p1 = MIN(p1, p2);
+          p1 = CPL_MIN(p1, p2);
         for(int j=0; keyNames[j]!=NULL; j++)
         {
           p2 = strstr(p, keyNames[j]);
           if(p2)
-            p1 = MIN(p1, p2);
+            p1 = CPL_MIN(p1, p2);
         }
       }
       if(p && p1 && p1>p)
diff --git a/src/gt_wkt_srs.cpp b/src/gt_wkt_srs.cpp
index 9d222834..8b2ed4fa 100755
--- a/src/gt_wkt_srs.cpp
+++ b/src/gt_wkt_srs.cpp
@@ -666,7 +666,7 @@ char *GTIFGetOGISDefn( GTIF *hGTIF, GTIFDefn * psDefn )
         double		adfParm[10];
         int		i;
 
-        for( i = 0; i < MIN(10,psDefn->nParms); i++ )
+        for( i = 0; i < CPL_MIN(10,psDefn->nParms); i++ )
             adfParm[i] = psDefn->ProjParm[i];
 
         for( ; i < 10; i++ )
@@ -1296,7 +1296,7 @@ int GTIFSetFromOGISDefn( GTIF * psGTIF, const char *pszOGCWKT )
         nUOMLengthCode = 9002; /* international foot */
     else if( (pszLinearUOMName != NULL
               && EQUAL(pszLinearUOMName,SRS_UL_US_FOOT))
-             || ABS(dfLinearUOM-GTIFAtof(SRS_UL_US_FOOT_CONV)) < 0.0000001 )
+             || CPL_ABS(dfLinearUOM-GTIFAtof(SRS_UL_US_FOOT_CONV)) < 0.0000001 )
         nUOMLengthCode = 9003; /* us survey foot */
     else if( fabs(dfLinearUOM-1.0) > 0.00000001 )
         nUOMLengthCode = KvUserDefined;
@@ -2475,7 +2475,7 @@ CPLErr GTIFWktFromMemBufEx( int nSize, unsigned char *pabyBuffer,
         && nCount >= 2 )
     {
         padfGeoTransform[1] = padfScale[0];
-        padfGeoTransform[5] = - ABS(padfScale[1]);
+        padfGeoTransform[5] = - CPL_ABS(padfScale[1]);
 
         if( TIFFGetField(hTIFF,TIFFTAG_GEOTIEPOINTS,&nCount,&padfTiePoints )
             && nCount >= 6 )
@@ -2628,7 +2628,7 @@ CPLErr GTIFMemBufFromWktEx( const char *pszWKT, const double *padfGeoTransform,
 
     if( padfGeoTransform[0] != 0.0 || padfGeoTransform[1] != 1.0
         || padfGeoTransform[2] != 0.0 || padfGeoTransform[3] != 0.0
-        || padfGeoTransform[4] != 0.0 || ABS(padfGeoTransform[5]) != 1.0 )
+        || padfGeoTransform[4] != 0.0 || CPL_ABS(padfGeoTransform[5]) != 1.0 )
     {
 
         if( padfGeoTransform[2] == 0.0 && padfGeoTransform[4] == 0.0 )
