diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fca91757..a1e8641d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -219,11 +219,13 @@ add_library(las SHARED ${LIBLAS_SOURCES})
 add_library(las_c SHARED ${LIBLAS_C_SOURCES})
 
 target_link_libraries(las
+  PUBLIC
   ${TIFF_LIBRARY}
   ${GEOTIFF_LIBRARY}
   ${GDAL_LIBRARY}
   ${LASZIP_LIBRARY}
   ${Boost_LIBRARIES}
+  PRIVATE
   ${PROJ_LIBRARIES})
 
 target_link_libraries(las_c
