--- rustc-1.66.0-src/compiler/rustc_target/src/abi/call/powerpc64.rs.orig	2023-01-17 15:34:17.094202392 +0100
+++ rustc-1.66.0-src/compiler/rustc_target/src/abi/call/powerpc64.rs	2023-01-17 15:35:49.820776341 +0100
@@ -119,14 +119,7 @@
     Ty: TyAbiInterface<'a, C> + Copy,
     C: HasDataLayout + HasTargetSpec,
 {
-    let abi = if cx.target_spec().env == "musl" {
-        ELFv2
-    } else {
-        match cx.data_layout().endian {
-            Endian::Big => ELFv1,
-            Endian::Little => ELFv2,
-        }
-    };
+    let abi = ELFv2;
 
     if !fn_abi.ret.is_ignore() {
         classify_ret(cx, &mut fn_abi.ret, abi);
