--- rustc-1.48.0-src/src/tools/rls/rls/src/cmd.rs.orig	2021-04-08 11:10:17.600000000 +0000
+++ rustc-1.48.0-src/src/tools/rls/rls/src/cmd.rs	2021-04-08 19:35:15.592000000 +0000
@@ -7,7 +7,7 @@
 use crate::server::{self, LsService, Notification, Request, RequestId};
 use rls_analysis::{AnalysisHost, Target};
 use rls_vfs::Vfs;
-use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::atomic::{AtomicU32, Ordering};
 
 use lsp_types::{
     ClientCapabilities, CodeActionContext, CodeActionParams, CompletionItem,
@@ -316,8 +316,8 @@
 }
 
 fn next_id() -> RequestId {
-    static ID: AtomicU64 = AtomicU64::new(1);
-    RequestId::Num(ID.fetch_add(1, Ordering::SeqCst))
+    static ID: AtomicU32 = AtomicU32::new(1);
+    RequestId::Num(ID.fetch_add(1, Ordering::SeqCst).into())
 }
 
 // Custom reader and output for the RLS server.
--- rustc-1.48.0-src/src/tools/rls/rls/src/server/io.rs.orig	2021-04-08 11:02:27.604000000 +0000
+++ rustc-1.48.0-src/src/tools/rls/rls/src/server/io.rs	2021-04-08 19:45:19.404000000 +0000
@@ -5,7 +5,7 @@
 
 use std::fmt;
 use std::io::{self, BufRead, Write};
-use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::atomic::{AtomicU32, Ordering};
 use std::sync::Arc;
 
 use jsonrpc_core::{self as jsonrpc, response, version, Id};
@@ -169,13 +169,13 @@
 /// An output that sends notifications and responses on `stdout`.
 #[derive(Clone)]
 pub(super) struct StdioOutput {
-    next_id: Arc<AtomicU64>,
+    next_id: Arc<AtomicU32>,
 }
 
 impl StdioOutput {
     /// Constructs a new `stdout` output.
     pub(crate) fn new() -> StdioOutput {
-        StdioOutput { next_id: Arc::new(AtomicU64::new(1)) }
+        StdioOutput { next_id: Arc::new(AtomicU32::new(1).into()) }
     }
 }
 
@@ -192,7 +192,7 @@
     }
 
     fn provide_id(&self) -> RequestId {
-        RequestId::Num(self.next_id.fetch_add(1, Ordering::SeqCst))
+        RequestId::Num(self.next_id.fetch_add(1, Ordering::SeqCst).into())
     }
 }
 
