--- rustc-1.76.0-src/compiler/rustc_target/src/abi/call/powerpc64.rs.vanilla	2024-02-20 12:22:13.165372817 +0100
+++ rustc-1.76.0-src/compiler/rustc_target/src/abi/call/powerpc64.rs	2024-02-20 12:23:04.119367907 +0100
@@ -131,7 +131,7 @@
         ELFv2
     } else {
         match cx.data_layout().endian {
-            Endian::Big => ELFv1,
+            Endian::Big => ELFv2,
             Endian::Little => ELFv2,
         }
     };
