--- a/examples/keystore/webapi.py
+++ b/examples/keystore/webapi.py
@@ -1,12 +1,17 @@
 
 from __future__ import absolute_import
-import pkg_resources
+
+try:
+    from importlib.resources import files
+except ImportError:
+    from importlib_resources import files
+
 import keystore.web
 import twisted.web.server
 
 
 def read_resource(path):
-    return pkg_resources.resource_string('keystore', path)
+    return files("keystore").joinpath(path).read_bytes()
 
 
 class KeyResource(keystore.web.DeferrableResource):
