--- a/lib/rbs/annotate/rdoc_source.rb
+++ b/lib/rbs/annotate/rdoc_source.rb
@@ -23,7 +23,7 @@
         @stores.clear()
 
         RDoc::RI::Paths.each(with_system_dir, with_site_dir, with_home_dir, with_gems_dir ? :latest : false, *extra_dirs.map(&:to_s)) do |path, type|
-          store = RDoc::Store.new(path, type)
+          store = RDoc::Store.new(RDoc::Options.new, path: path, type: type)
           store.load_all
 
           @stores << store
--- a/test/rbs/rdoc/rbs_parser_test.rb
+++ b/test/rbs/rdoc/rbs_parser_test.rb
@@ -5,7 +5,7 @@
 class RDocPluginParserTest < Test::Unit::TestCase
   def parser(content)
     top_level = RDoc::TopLevel.new("a.rbs")
-    top_level.store = RDoc::Store.new()
+    top_level.store = RDoc::Store.new(RDoc::Options.new)
 
     RBS::RDocPlugin::Parser.new(top_level, content)
   end
