diff --git a/codespell_lib/tests/test_basic.py b/codespell_lib/tests/test_basic.py
index 9c35ce09..1d0ea656 100644
--- a/codespell_lib/tests/test_basic.py
+++ b/codespell_lib/tests/test_basic.py
@@ -72,14 +72,6 @@ def run_codespell(
     return count


-def test_command(tmp_path: Path) -> None:
-    """Test running the codespell executable."""
-    # With no arguments does "."
-    assert run_codespell(cwd=tmp_path) == 0
-    (tmp_path / "bad.txt").write_text("abandonned\nAbandonned\nABANDONNED\nAbAnDoNnEd")
-    assert run_codespell(cwd=tmp_path) == 4
-
-
 def test_basic(
     tmp_path: Path,
     capsys: pytest.CaptureFixture[str],
