--- OpenXR-SDK-Source-release-1.1.54/src/common/platform_utils.hpp.orig	2026-02-17 09:23:53.471252619 +0100
+++ OpenXR-SDK-Source-release-1.1.54/src/common/platform_utils.hpp	2026-02-17 09:29:18.645704142 +0100
@@ -53,7 +53,9 @@
 #define XR_ARCH_ABI "mips64"
 #elif defined(__mips)
 #define XR_ARCH_ABI "mips"
-#elif defined(__powerpc64__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#elif defined(__powerpc__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define XR_ARCH_ABI "ppc"
+#elif defined(__powerpc64__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #define XR_ARCH_ABI "ppc64"
 #elif defined(__powerpc__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #define XR_ARCH_ABI "ppc64el"
