--- OpenXR-SDK-Source-release-1.0.30/src/common/platform_utils.hpp.orig	2023-10-03 21:43:53.659446585 +0200
+++ OpenXR-SDK-Source-release-1.0.30/src/common/platform_utils.hpp	2023-10-03 21:44:23.040523141 +0200
@@ -57,6 +57,8 @@
 #define XR_ARCH_ABI "ppc64"
 #elif defined(__powerpc__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #define XR_ARCH_ABI "ppc64el"
+#elif defined(__powerpc__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define XR_ARCH_ABI "ppc"
 #elif defined(__s390x__) || defined(__zarch__)
 #define XR_ARCH_ABI "s390x"
 #elif defined(__hppa__)
