--- wxPython-4.2.1/wscript.orig	2024-05-25 12:40:13.339500355 +0200
+++ wxPython-4.2.1/wscript	2024-05-25 12:40:19.942850204 +0200
@@ -40,8 +40,6 @@
 
     opt.add_option('--debug', dest='debug', action='store_true', default=False,
                    help='Turn on debugger-related compile options.')
-    opt.add_option('--python', dest='python', default='', action='store',
-                   help='Full path to the Python executable to use.')
     opt.add_option('--wx_config', dest='wx_config', default='wx-config', action='store',
                    help='Full path to the wx-config script to be used for this build.')
     opt.add_option('--no_magic', dest='no_magic', action='store_true', default=False,
