diff --git a/pytest_tornado/plugin.py b/pytest_tornado/plugin.py
index b231ee4..d4225c2 100644
--- a/pytest_tornado/plugin.py
+++ b/pytest_tornado/plugin.py
@@ -3,7 +3,7 @@ import sys
 import types
 import inspect
 import datetime
-import pkg_resources
+import packaging.version
 import pytest
 import tornado
 import tornado.gen
@@ -16,7 +16,7 @@ if sys.version_info[:2] >= (3, 5):
 else:
     iscoroutinefunction = lambda f: False
 
-_PYTEST_VERSION = pkg_resources.parse_version(pytest.__version__)
+_PYTEST_VERSION = packaging.version.parse(pytest.__version__)
 
 
 def _get_async_test_timeout():
@@ -70,7 +70,7 @@ def _timeout(item):
 @pytest.mark.tryfirst
 def pytest_pycollect_makeitem(collector, name, obj):
     if collector.funcnamefilter(name) and inspect.isgeneratorfunction(obj):
-        if _PYTEST_VERSION >= pkg_resources.parse_version("5.4.0"):
+        if _PYTEST_VERSION >= packaging.version.parse("5.4.0"):
             item = pytest.Function.from_parent(collector, name=name)
         else:
             item = pytest.Function(name, parent=collector)
diff --git a/setup.py b/setup.py
index 2272c95..18792cb 100644
--- a/setup.py
+++ b/setup.py
@@ -36,7 +36,7 @@ setup(
     keywords=('pytest py.test tornado async asynchronous '
               'testing unit tests plugin'),
     packages=find_packages(exclude=["tests.*", "tests"]),
-    install_requires=['pytest>=3.6', 'tornado>=4.1', 'setuptools'],
+    install_requires=['pytest>=3.6', 'tornado>=4.1', 'setuptools', 'packaging'],
     entry_points={
         'pytest11': ['tornado = pytest_tornado.plugin'],
     },
