diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 14499af..6850d76 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -19,12 +19,6 @@ if (SPGLIB_INSTALL)
 				LIBRARY DESTINATION ${Python_INSTALL_DIR} COMPONENT Spglib_Runtime
 				PUBLIC_HEADER DESTINATION ${Python_INSTALL_DIR} COMPONENT Spglib_Runtime
 				RUNTIME DESTINATION ${Python_INSTALL_DIR} COMPONENT Spglib_Runtime)
-	else ()
-		# TODO: Cmake forces to install PUBLIC_HEADER when defined
-		# https://gitlab.kitware.com/cmake/cmake/-/issues/24326
-		install(TARGETS Spglib_symspg
-				LIBRARY DESTINATION ${Python_INSTALL_DIR} COMPONENT Spglib_Runtime
-				PUBLIC_HEADER DESTINATION ${Python_INSTALL_DIR} COMPONENT Spglib_Runtime)
 	endif ()
 	install(TARGETS Spglib_python
 			LIBRARY DESTINATION ${Python_INSTALL_DIR} COMPONENT Spglib_Runtime)
