diff -ru libaec-v1.1.3.orig/CMakeLists.txt libaec-v1.1.3/CMakeLists.txt
--- libaec-v1.1.3.orig/CMakeLists.txt	2025-04-24 17:09:53.165430706 +0200
+++ libaec-v1.1.3/CMakeLists.txt	2025-04-24 17:10:48.724394700 +0200
@@ -71,9 +71,9 @@
 set(libaec_CONFIG_VERSION_OUT ${CMAKE_CURRENT_BINARY_DIR}/cmake/libaec-config-version.cmake)
 configure_file(${libaec_CONFIG_VERSION_IN} ${libaec_CONFIG_VERSION_OUT} @ONLY)
 install(FILES ${libaec_CONFIG_OUT}
-        DESTINATION cmake)
+        DESTINATION cmake/libaec)
 install(FILES ${libaec_CONFIG_VERSION_OUT}
-        DESTINATION cmake)
+        DESTINATION cmake/libaec)
 
 # Cpack configuration mainly for Windows installer
 set(CPACK_PACKAGE_NAME "libaec")
