diff -aur logutils-0.3.5.old/tests/test_dictconfig.py logutils-0.3.5/tests/test_dictconfig.py
--- logutils-0.3.5.old/tests/test_dictconfig.py	2024-04-13 19:55:08.372365822 +0200
+++ logutils-0.3.5/tests/test_dictconfig.py	2024-04-13 19:55:19.222580688 +0200
@@ -568,7 +568,7 @@
             raise RuntimeError()
         except RuntimeError:
             logging.exception("just testing")
-        self.assertEquals(h.formatted[0],
+        self.assertEqual(h.formatted[0],
             "ERROR:root:just testing\nGot a [RuntimeError]")
 
     def test_config4a_ok(self):
@@ -580,7 +580,7 @@
             raise RuntimeError()
         except RuntimeError:
             logging.exception("just testing")
-        self.assertEquals(h.formatted[0],
+        self.assertEqual(h.formatted[0],
             "ERROR:root:just testing\nGot a [RuntimeError]")
 
     def test_config5_ok(self):
