diff --git a/test/spec_headers.rb b/test/spec_headers.rb
index f22680b77..af96315c3 100644
--- a/test/spec_headers.rb
+++ b/test/spec_headers.rb
@@ -220,8 +220,8 @@ def test_has_value
 
   def test_inspect
     %i'inspect to_s'.each do |meth|
-      assert_equal '{}', @h.send(meth)
-      assert_equal '{"ab"=>"1", "cd"=>"2", "3"=>"4"}', @fh.send(meth)
+      assert_equal({}.inspect, @h.send(meth))
+      assert_equal({"ab"=>"1", "cd"=>"2", "3"=>"4"}.inspect, @fh.send(meth))
     end
   end
 
