--- Cairo-1.109/Makefile.PL.orig	2025-09-03 16:50:46.819431328 +0200
+++ Cairo-1.109/Makefile.PL	2025-09-03 16:51:27.506408789 +0200
@@ -595,7 +595,7 @@
 	print "Compiling Cairo with FreeType support\n";
 	push @xs_files, 'CairoFt.xs';
 	my %ft_cfg = ExtUtils::PkgConfig->find ('cairo-ft');
-	$cairo_cfg{cflags} .= ' '.$ft_cfg{cflags};
+	$cairo_cfg{cflags} .= ' -Wno-incompatible-pointer-types '.$ft_cfg{cflags};
 	$cairo_cfg{libs} .= ' '.$ft_cfg{libs};
 }
 
