diff --git a/include_server/c_extensions/distcc_pump_c_extensions_module.c b/include_server/c_extensions/distcc_pump_c_extensions_module.c
index 763dd42..b00ffec 100644
--- a/include_server/c_extensions/distcc_pump_c_extensions_module.c
+++ b/include_server/c_extensions/distcc_pump_c_extensions_module.c
@@ -33,6 +33,10 @@ const char *rs_program_name = "distcc_include_server";
 #include "distcc.h"
 #include "rpc.h"
 
+#ifndef PY_SSIZE_T_CLEAN
+#define PY_SSIZE_T_CLEAN
+#endif /* PY_SSIZE_T_CLEAN */
+
 static PyObject *distcc_pump_c_extensionsError;
 void initdistcc_pump_c_extensions(void);
 
