diff --unified --recursive --text --new-file fixtures-4.2.3.orig/pyproject.toml fixtures-4.2.3/pyproject.toml
--- fixtures-4.2.3.orig/pyproject.toml	2025-03-04 13:34:59.280487591 +0100
+++ fixtures-4.2.3/pyproject.toml	2025-03-04 13:33:36.042932264 +0100
@@ -48,3 +48,6 @@
 
 [tool.hatch.build.targets.sdist]
 include = ["fixtures*"]
+
+[tool.hatch.build.targets.wheel]
+exclude = ["fixtures/tests"]
