--- a/src/usage.c
+++ b/src/usage.c
@@ -903,8 +903,6 @@ void usage(struct lsof_context *ctx, /* context */
             (void)fprintf(stderr, "    compiler: %s\n", cp);
         if ((cp = isnullstr(LSOF_CCV)))
             (void)fprintf(stderr, "    compiler version: %s\n", cp);
-        if ((cp = isnullstr(LSOF_CCFLAGS)))
-            (void)fprintf(stderr, "    compiler flags: %s\n", cp);
         if ((cp = isnullstr(LSOF_LDFLAGS)))
             (void)fprintf(stderr, "    loader flags: %s\n", cp);
         if ((cp = isnullstr(LSOF_SYSINFO)))
