diff --git i/oslo_config/tests/test_cfg.py w/oslo_config/tests/test_cfg.py
index b2f7146..ab0a052 100644
--- i/oslo_config/tests/test_cfg.py
+++ w/oslo_config/tests/test_cfg.py
@@ -4383,7 +4383,10 @@ class SubCommandTestCase(BaseTestCase):
         self.conf.register_cli_opt(cfg.SubCommandOpt('cmd1'))
         self.conf.register_cli_opt(cfg.SubCommandOpt('cmd2'))
         self.useFixture(fixtures.MonkeyPatch('sys.stderr', io.StringIO()))
-        if sys.version_info >= (3, 12, 5):
+        if sys.version_info >= (3, 14):
+            self.assertRaisesRegex(ValueError, 'multiple',
+                                   self.conf, [])
+        elif sys.version_info >= (3, 12, 5):
             self.assertRaisesRegex(argparse.ArgumentError, 'multiple',
                                    self.conf, [])
         else:
