diff -ruN PackageKit-1.1.13.orig/backends/alpm/pk-alpm-config.c PackageKit-1.1.13/backends/alpm/pk-alpm-config.c
--- PackageKit-1.1.13.orig/backends/alpm/pk-alpm-config.c	2020-10-20 15:57:23.674000000 +0000
+++ PackageKit-1.1.13/backends/alpm/pk-alpm-config.c	2020-10-20 15:59:40.166000000 +0000
@@ -606,7 +606,6 @@
 pk_alpm_config_initialize_alpm (PkAlpmConfig *config, GError **error)
 {
 	alpm_handle_t *handle;
-	alpm_errno_t errno;
 	gsize dir = 1;
 
 	g_return_val_if_fail (config != NULL, FALSE);
diff -ruN PackageKit-1.1.13.orig/backends/alpm/pk-alpm-databases.c PackageKit-1.1.13/backends/alpm/pk-alpm-databases.c
--- PackageKit-1.1.13.orig/backends/alpm/pk-alpm-databases.c	2020-10-20 15:57:23.673000000 +0000
+++ PackageKit-1.1.13/backends/alpm/pk-alpm-databases.c	2020-10-20 16:05:59.453000000 +0000
@@ -40,7 +40,7 @@
 	const alpm_list_t *i;
 
 	if (alpm_unregister_all_syncdbs (priv->alpm) < 0) {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error_literal (error, PK_ALPM_ERROR, errno,
 				     alpm_strerror (errno));
 		return FALSE;
@@ -59,7 +59,7 @@
 
 		db = alpm_register_syncdb (priv->alpm, repo->name, level);
 		if (db == NULL) {
-			alpm_errno_t errno = alpm_errno (priv->alpm);
+			errno = alpm_errno (priv->alpm);
 			g_set_error (error, PK_ALPM_ERROR, errno, "[%s]: %s",
 				     repo->name, alpm_strerror (errno));
 			return FALSE;
diff -ruN PackageKit-1.1.13.orig/backends/alpm/pk-alpm-install.c PackageKit-1.1.13/backends/alpm/pk-alpm-install.c
--- PackageKit-1.1.13.orig/backends/alpm/pk-alpm-install.c	2020-10-20 15:57:23.673000000 +0000
+++ PackageKit-1.1.13/backends/alpm/pk-alpm-install.c	2020-10-20 16:02:04.567000000 +0000
@@ -62,7 +62,7 @@
 
 	for (; *paths != NULL; ++paths) {
 		if (pk_alpm_install_add_file (job, *paths) < 0) {
-			alpm_errno_t errno = alpm_errno (priv->alpm);
+			errno = alpm_errno (priv->alpm);
 			g_set_error (error, PK_ALPM_ERROR, errno, "%s: %s",
 				     *paths, alpm_strerror (errno));
 			return FALSE;
diff -ruN PackageKit-1.1.13.orig/backends/alpm/pk-alpm-remove.c PackageKit-1.1.13/backends/alpm/pk-alpm-remove.c
--- PackageKit-1.1.13.orig/backends/alpm/pk-alpm-remove.c	2020-10-20 15:57:23.673000000 +0000
+++ PackageKit-1.1.13/backends/alpm/pk-alpm-remove.c	2020-10-20 15:57:35.874000000 +0000
@@ -42,7 +42,7 @@
 
 		alpm_pkg_t *pkg = alpm_db_get_pkg (priv->localdb, name);
 		if (pkg == NULL || alpm_remove_pkg (priv->alpm, pkg) < 0) {
-			alpm_errno_t errno = alpm_errno (priv->alpm);
+			errno = alpm_errno (priv->alpm);
 			g_set_error (error, PK_ALPM_ERROR, errno, "%s: %s", name,
 				     alpm_strerror (errno));
 			return FALSE;
diff -ruN PackageKit-1.1.13.orig/backends/alpm/pk-alpm-sync.c PackageKit-1.1.13/backends/alpm/pk-alpm-sync.c
--- PackageKit-1.1.13.orig/backends/alpm/pk-alpm-sync.c	2020-10-20 15:57:23.673000000 +0000
+++ PackageKit-1.1.13/backends/alpm/pk-alpm-sync.c	2020-10-20 16:05:06.745000000 +0000
@@ -52,7 +52,7 @@
 		}
 
 		if (i == NULL) {
-			alpm_errno_t errno = ALPM_ERR_DB_NOT_FOUND;
+			errno = ALPM_ERR_DB_NOT_FOUND;
 			g_set_error (error, PK_ALPM_ERROR, errno, "%s/%s: %s",
 				     repo, name, alpm_strerror (errno));
 			return FALSE;
@@ -60,7 +60,7 @@
 
 		pkg = alpm_db_get_pkg (i->data, name);
 		if (pkg == NULL || alpm_add_pkg (priv->alpm, pkg) < 0) {
-			alpm_errno_t errno = alpm_errno (priv->alpm);
+			errno = alpm_errno (priv->alpm);
 			g_set_error (error, PK_ALPM_ERROR, errno, "%s/%s: %s",
 				     repo, name, alpm_strerror (errno));
 			return FALSE;
diff -ruN PackageKit-1.1.13.orig/backends/alpm/pk-alpm-transaction.c PackageKit-1.1.13/backends/alpm/pk-alpm-transaction.c
--- PackageKit-1.1.13.orig/backends/alpm/pk-alpm-transaction.c	2020-10-20 15:57:23.673000000 +0000
+++ PackageKit-1.1.13/backends/alpm/pk-alpm-transaction.c	2020-10-20 16:01:03.554000000 +0000
@@ -755,7 +755,7 @@
 	PkBackendAlpmPrivate *priv = pk_backend_get_user_data (backend);
 
 	if (alpm_trans_init (priv->alpm, flags) < 0) {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error_literal (error, PK_ALPM_ERROR, errno,
 				     alpm_strerror (errno));
 		return FALSE;
@@ -956,11 +956,11 @@
 	}
 
 	if (prefix != NULL) {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error (error, PK_ALPM_ERROR, errno, "%s: %s", prefix,
 			     alpm_strerror (errno));
 	} else {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error_literal (error, PK_ALPM_ERROR, errno,
 				     alpm_strerror (errno));
 	}
@@ -1066,11 +1066,11 @@
 	}
 
 	if (prefix != NULL) {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error (error, PK_ALPM_ERROR, errno, "%s: %s", prefix,
 			     alpm_strerror (errno));
 	} else {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error_literal (error, PK_ALPM_ERROR, errno,
 				     alpm_strerror (errno));
 	}
@@ -1100,7 +1100,7 @@
 	pkalpm_current_job = NULL;
 
 	if (alpm_trans_release (priv->alpm) < 0) {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error_literal (error, PK_ALPM_ERROR, errno,
 				     alpm_strerror (errno));
 		return FALSE;
diff -ruN PackageKit-1.1.13.orig/backends/alpm/pk-backend-alpm.c PackageKit-1.1.13/backends/alpm/pk-backend-alpm.c
--- PackageKit-1.1.13.orig/backends/alpm/pk-backend-alpm.c	2020-10-20 15:57:23.673000000 +0000
+++ PackageKit-1.1.13/backends/alpm/pk-backend-alpm.c	2020-10-20 16:03:06.037000000 +0000
@@ -90,7 +90,7 @@
 
 	priv->localdb = alpm_get_localdb (priv->alpm);
 	if (priv->localdb == NULL) {
-		alpm_errno_t errno = alpm_errno (priv->alpm);
+		errno = alpm_errno (priv->alpm);
 		g_set_error (error, PK_ALPM_ERROR, errno, "[%s]: %s", "local",
 			     alpm_strerror (errno));
 	}
