--- PackageKit-1.2.6/backends/alpm/pk-alpm-update.c.orig	2023-06-13 00:34:01.160534562 +0200
+++ PackageKit-1.2.6/backends/alpm/pk-alpm-update.c	2023-06-13 00:34:40.844611653 +0200
@@ -71,9 +71,7 @@
 pk_alpm_pkg_build_urls (alpm_pkg_t *pkg)
 {
 	gchar **urls = g_new0 (gchar *, 2);
-	urls[0] = g_strdup_printf ("https://archlinux.org/packages/%s/%s/%s/",
-				   alpm_db_get_name (alpm_pkg_get_db (pkg)),
-				   alpm_pkg_get_arch (pkg),
+	urls[0] = g_strdup_printf ("https://github.com/kth5/archpower/tree/master/%s",
 				   alpm_pkg_get_name (pkg));
 	return urls;
 }
