diff --git i/fluidsynth.service.in w/fluidsynth.service.in
index 21f1c860..8ae54b30 100644
--- i/fluidsynth.service.in
+++ w/fluidsynth.service.in
@@ -4,8 +4,6 @@ Documentation=man:fluidsynth(1)
 After=sound.target
 After=pipewire.service pulseaudio.service
 Wants=pipewire.service pulseaudio.service
-# If you need more than one instance, use `systemctl edit` to override this:
-ConditionPathExists=!/run/lock/fluidsynth/fluidsynth.lock
 ConditionUser=!@system
 
 [Service]
@@ -27,8 +25,6 @@ Environment=OTHER_OPTS= SOUND_FONT=
 EnvironmentFile=@FLUID_DAEMON_ENV_FILE@
 EnvironmentFile=-%h/.config/fluidsynth
 ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/fluidsynth -is $OTHER_OPTS "${SOUND_FONT}"
-ExecStartPre=touch /run/lock/fluidsynth/fluidsynth.lock
-ExecStopPost=rm -f /run/lock/fluidsynth/fluidsynth.lock
 
 [Install]
 WantedBy=default.target
diff --git i/fluidsynth.tmpfiles.in w/fluidsynth.tmpfiles.in
index eb119f8b..0cb0ebf6 100644
--- i/fluidsynth.tmpfiles.in
+++ w/fluidsynth.tmpfiles.in
@@ -1 +1 @@
-d /run/lock/fluidsynth 0777 root root
+d /run/lock/fluidsynth 0755 root root
