--- linux-6.14.7/arch/powerpc/boot/Makefile.orig	2025-05-28 18:13:17.941781651 +0200
+++ linux-6.14.7/arch/powerpc/boot/Makefile	2025-05-28 18:13:49.141536694 +0200
@@ -33,13 +33,15 @@
 endif
 
 ifdef CONFIG_PPC64_BOOT_WRAPPER
+BOOTTARGETFLAGS	+= -std=gnu11
 BOOTTARGETFLAGS	+= -m64
 BOOTTARGETFLAGS	+= -mabi=elfv2
 ifdef CONFIG_PPC64_ELF_ABI_V2
 BOOTTARGETFLAGS	+= $(call cc-option,-mabi=elfv2)
 endif
 else
-BOOTTARGETFLAGS	:= -m32
+BOOTTARGETFLAGS	:= -std=gnu11
+BOOTTARGETFLAGS	+= -m32
 endif
 
 ifdef CONFIG_TARGET_CPU_BOOL
