diff --git a/configure.ac b/configure.ac
index 9fa9b868b..24018b755 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2898,7 +2898,7 @@ if test "$with_uhdr" != 'no'; then
     AC_LANG_PUSH([C++])
     AC_CHECK_HEADER([ultrahdr_api.h],[passed=`expr $passed + 1`],[failed=`expr $failed + 1`])
     AC_CHECK_LIB([uhdr],[uhdr_create_decoder],[passed=`expr $passed + 1`],[failed=`expr $failed + 1`],[])
-
+    AC_LANG_POP
     AC_MSG_CHECKING([if google ultrahdr package is complete])
 
     if test $passed -gt 0; then
