diff -ru gpsd-3.26.1.orig/SConscript gpsd-3.26.1/SConscript
--- gpsd-3.26.1.orig/SConscript	2025-07-02 09:19:57.165645187 +0200
+++ gpsd-3.26.1/SConscript	2025-07-02 09:29:30.695141072 +0200
@@ -2321,7 +2321,7 @@
     ('@GPSAPIVERMAJ@', api_version_major),
     ('@GPSAPIVERMIN@', api_version_minor),
     ('@GPSPACKET@',  packet_ffi_shared[0].get_path()),
-    ('@GROUP@',  def_group),
+    ('@GROUP@',  env['gpsd_group']),
     ('@ICONPATH@',   installdir('icondir', add_destdir=False)),
     ('@INCLUDEDIR@', installdir('includedir', add_destdir=False)),
     ('@IRCCHAN@',    ircchan),
