diff -Naur a/js/src/old-configure.in b/js/src/old-configure.in
--- a/js/src/old-configure.in	2023-02-20 09:44:55.284485086 +0200
+++ b/js/src/old-configure.in	2023-02-20 09:47:08.337905001 +0200
@@ -239,19 +239,6 @@
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    # While sha1 is deterministic, it is slower.
-    if test -z "$DEVELOPER_OPTIONS" -o "$OS_TARGET" = "Android"; then
-        build_id=sha1
-    else
-        build_id=uuid
-    fi
-    AC_MSG_CHECKING([for --build-id=$build_id option to ld])
-    _SAVE_LDFLAGS=$LDFLAGS
-    LDFLAGS="$LDFLAGS -Wl,--build-id=$build_id"
-    AC_TRY_LINK(,,AC_MSG_RESULT([yes]),
-                  AC_MSG_RESULT([no])
-                  LDFLAGS=$_SAVE_LDFLAGS)
-
     _DEFINES_CFLAGS="-include $jsconfdefs -DMOZILLA_CLIENT"
 fi
 
diff -Naur a/old-configure.in b/old-configure.in
--- a/old-configure.in	2023-02-20 09:44:50.341395195 +0200
+++ b/old-configure.in	2023-02-20 09:47:26.498236118 +0200
@@ -253,19 +253,6 @@
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    # While sha1 is deterministic, it is slower.
-    if test -z "$DEVELOPER_OPTIONS" -o "$OS_TARGET" = "Android"; then
-        build_id=sha1
-    else
-        build_id=uuid
-    fi
-    AC_MSG_CHECKING([for --build-id=$build_id option to ld])
-    _SAVE_LDFLAGS=$LDFLAGS
-    LDFLAGS="$LDFLAGS -Wl,--build-id=$build_id"
-    AC_TRY_LINK(,,AC_MSG_RESULT([yes]),
-                  AC_MSG_RESULT([no])
-                  LDFLAGS=$_SAVE_LDFLAGS)
-
     AC_MSG_CHECKING([for --ignore-unresolved-symbol option to ld])
     HAVE_LINKER_SUPPORT_IGNORE_UNRESOLVED=
     _SAVE_LDFLAGS=$LDFLAGS
