--- a/security/sandbox/linux/Sandbox.cpp  2023-07-10 20:29:48.000000000 +0100
+++ b/security/sandbox/linux/Sandbox.cpp      2023-07-15 09:13:43.561724668 +0100
@@ -18,6 +18,8 @@
 #include <dirent.h>
 #ifdef NIGHTLY_BUILD
 #  include "dlfcn.h"
+#else
+#  include <dlfcn.h>
 #endif
 #include <errno.h>
 #include <fcntl.h>
@@ -685,7 +687,7 @@
   files->Add("/proc/stat", SandboxOpenedFile::Error{});
   files->Add("/proc/net/unix", SandboxOpenedFile::Error{});
   files->Add("/proc/self/maps", SandboxOpenedFile::Error{});
-
+  Unused << dlopen("libgcc_s.so.1", RTLD_GLOBAL|RTLD_LAZY);
   // Finally, start the sandbox.
   SetCurrentProcessSandbox(GetMediaSandboxPolicy(files));
 }