diff -Naur a/media/ffvpx/libavcodec/moz.build b/media/ffvpx/libavcodec/moz.build
--- a/media/ffvpx/libavcodec/moz.build	2024-06-07 00:33:58.000000000 +0300
+++ b/media/ffvpx/libavcodec/moz.build	2024-06-10 21:13:44.142620997 +0300
@@ -143,6 +143,10 @@
 c11_flags = ["-std=gnu11"]
 if CONFIG["CC_TYPE"] == "clang-cl":
     c11_flags.insert(0, "-Xclang")
+
+if CONFIG["CC_TYPE"] == "gcc" and int(CONFIG["CC_VERSION"].split(".")[0]) >= 14:
+	c11_flags.insert(0, "-Wno-error=incompatible-pointer-types")
+
 CFLAGS += c11_flags
 
 if not CONFIG["MOZ_SYSTEM_LIBVPX"]:
