--- build/moz.configure/lto-pgo.configure.orig	2020-12-15 21:24:23.531655536 +0100
+++ build/moz.configure/lto-pgo.configure	2020-12-15 21:24:50.581884685 +0100
@@ -87,7 +87,7 @@
         return namespace(
             gen_cflags=["-fprofile-generate"],
             gen_ldflags=["-fprofile-generate"],
-            use_cflags=["-fprofile-use", "-fprofile-correction", "-Wcoverage-mismatch"],
+            use_cflags=["-fprofile-use", "-fprofile-correction", "-Wcoverage-mismatch", "-Wno-coverage-mismatch"],
             use_ldflags=["-fprofile-use"],
         )
 
--- build/pgo/profileserver.py.orig	2020-12-15 21:27:11.481452600 +0100
+++ build/pgo/profileserver.py	2020-12-15 21:27:36.634876307 +0100
@@ -212,6 +212,9 @@
             print("Firefox exited successfully, but produced a crashreport")
             sys.exit(1)
 
+        os.system("gccpgostub.sh")
+        sys.exit(0)
+
         llvm_profdata = env.get("LLVM_PROFDATA")
         if llvm_profdata:
             profraw_files = glob.glob("*.profraw")
--- mfbt/tests/moz.build.orig	2020-12-15 21:28:27.762659242 +0100
+++ mfbt/tests/moz.build	2020-12-15 21:29:01.669213598 +0100
@@ -79,11 +79,11 @@
 # Not to be unified with the rest, because this test
 # sets MOZ_PRETEND_NO_JSRUST, which changes the behavior
 # of the included headers.
-CppUnitTests(
-    [
-        "TestUtf8",
-    ]
-)
+#CppUnitTests(
+#    [
+#        "TestUtf8",
+#    ]
+#)
 
 if not CONFIG["MOZ_ASAN"]:
     CppUnitTests(
