--- mozilla-unified/memory/build/mozjemalloc.cpp.orig	2019-02-28 07:39:08.000000000 +0000
+++ mozilla-unified/memory/build/mozjemalloc.cpp	2019-02-28 07:39:30.000000000 +0000
@@ -182,7 +182,7 @@
 // Debug builds are opted out too, for test coverage.
 #ifndef MOZ_DEBUG
 #if !defined(__ia64__) && !defined(__sparc__) && !defined(__mips__) && \
-    !defined(__aarch64__)
+    !defined(__aarch64__) && !defined(__powerpc__)
 #define MALLOC_STATIC_PAGESIZE 1
 #endif
 #endif
