--- firefox-110.0.1/js/src/irregexp/imported/regexp-parser.cc.orig	2023-03-07 09:03:21.965148217 +0100
+++ firefox-110.0.1/js/src/irregexp/imported/regexp-parser.cc	2023-03-07 09:03:42.914963494 +0100
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#pragma GCC diagnostic ignored "-Wnonnull"
+
 #include "irregexp/imported/regexp-parser.h"
 
 #include "irregexp/imported/regexp-ast.h"
