--- firefox-110.0.1/third_party/libwebrtc/moz.build.orig	2023-03-07 09:09:36.206079325 +0100
+++ firefox-110.0.1/third_party/libwebrtc/moz.build	2023-03-07 09:10:35.471885182 +0100
@@ -624,7 +624,8 @@
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_decoder_isac_float_gn",
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_encoder_isac_float_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_c_gn",
-        "/third_party/libwebrtc/modules/audio_coding/isac_gn"
+        "/third_party/libwebrtc/modules/audio_coding/isac_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn"
     ]
 
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
