--- firefox-142.0.1/third_party/abseil-cpp/absl/copts/configure_copts.bzl.orig	2025-08-28 22:26:16.983031174 +0200
+++ firefox-142.0.1/third_party/abseil-cpp/absl/copts/configure_copts.bzl	2025-08-28 22:26:34.420193603 +0200
@@ -51,7 +51,6 @@
     ":cpu_x64_windows_msvc": ABSL_RANDOM_HWAES_MSVC_X64_FLAGS,
     ":cpu_x64_windows": ABSL_RANDOM_HWAES_MSVC_X64_FLAGS,
     ":cpu_k8": ABSL_RANDOM_HWAES_X64_FLAGS,
-    ":cpu_ppc": ["-mcrypto"],
     ":cpu_aarch64": ABSL_RANDOM_HWAES_ARM64_FLAGS,
 
     # Supported by default or unsupported.
