diff -up a/toolkit/moz.configure.wasi b/toolkit/moz.configure
--- a/toolkit/moz.configure.wasi
+++ b/toolkit/moz.configure
@@ -2663,7 +2663,7 @@ with only_when(requires_wasm_sandboxing
     def wasi_sysroot_flags(wasi_sysroot):
         if wasi_sysroot:
             log.info("Using wasi sysroot in %s", wasi_sysroot)
-            return ["--sysroot=%s" % wasi_sysroot]
+            return ["--sysroot=%s" % wasi_sysroot, "-nodefaultlibs",  "-lc", "-lwasi-emulated-process-clocks", "-lc++", "-lc++abi", "%%PORTAGE_WORKDIR%%/wasi-sdk-%%WASI_SDK_VER%%-%%WASI_ARCH%%-linux/lib/clang/%%WASI_SDK_LLVM_VER%%/lib/wasm32-unknown-wasi/libclang_rt.builtins.a"]
         return []
 
     set_config("WASI_SYSROOT", wasi_sysroot)
