diff -aur nodeenv-1.8.0.old/tests/nodeenv_test.py nodeenv-1.8.0/tests/nodeenv_test.py
--- nodeenv-1.8.0.old/tests/nodeenv_test.py	2024-11-19 21:21:09.713406747 +0100
+++ nodeenv-1.8.0/tests/nodeenv_test.py	2024-11-19 21:21:33.096670234 +0100
@@ -1,13 +1,13 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+import sys
 import os.path
 if sys.version_info < (3, 3):
     from pipes import quote as _quote
 else:
     from shlex import quote as _quote
 import subprocess
-import sys
 import sysconfig
 import platform
 
