From 7dc2d5f5f5cd0e20fdf538295023e9ddc116a0b1 Mon Sep 17 00:00:00 2001
From: Antoine Martin <dev@ayakael.net>
Date: Sat, 13 Jan 2024 19:40:05 -0500
Subject: [PATCH 1/2] src/constants.c: use KeyDataEc rather than KetDataEcdsa

---
 src/constants.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/constants.c b/src/constants.c
index 34c81b2..3e698e8 100644
--- a/src/constants.c
+++ b/src/constants.c
@@ -447,7 +447,7 @@ int PyXmlSec_ConstantsModule_Init(PyObject* package) {
 #endif
 #if XMLSEC_VERSION_HEX > 0x10212
     // from version 1.2.19
-    PYXMLSEC_ADD_KEYDATA_CONSTANT(KeyDataEcdsa, "ECDSA")
+    PYXMLSEC_ADD_KEYDATA_CONSTANT(KeyDataEc, "ECDSA")
 #endif
     PYXMLSEC_ADD_KEYDATA_CONSTANT(KeyDataHmac, "HMAC")
     PYXMLSEC_ADD_KEYDATA_CONSTANT(KeyDataRsa, "RSA")

From 7b573378854883ffca6b089639989ea908ac796a Mon Sep 17 00:00:00 2001
From: Antoine Martin <dev@ayakael.net>
Date: Sat, 13 Jan 2024 19:40:28 -0500
Subject: [PATCH 2/2] Use xmlSecCryptoAppKeyLoadEx instead of
 xmlSecCryptoAppKeyLoad

---
 src/keys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/keys.c b/src/keys.c
index 1362b12..47678a5 100644
--- a/src/keys.c
+++ b/src/keys.c
@@ -163,7 +163,7 @@ static PyObject* PyXmlSec_KeyFromFile(PyObject* self, PyObject* args, PyObject*
     if (is_content) {
         key->handle = xmlSecCryptoAppKeyLoadMemory((const xmlSecByte*)data, (xmlSecSize)data_size, format, password, NULL, NULL);
     } else {
-        key->handle = xmlSecCryptoAppKeyLoad(data, format, password, NULL, NULL);
+        key->handle = xmlSecCryptoAppKeyLoadEx(data, xmlSecKeyDataTypePrivate, format, password, NULL, NULL);
     }
     Py_END_ALLOW_THREADS;
 
