diff -aur factory_boy-3.3.0.old/tests/test_mongoengine.py factory_boy-3.3.0/tests/test_mongoengine.py
--- factory_boy-3.3.0.old/tests/test_mongoengine.py	2024-06-30 16:10:49.129025030 +0200
+++ factory_boy-3.3.0/tests/test_mongoengine.py	2024-06-30 16:11:30.689954127 +0200
@@ -10,6 +10,8 @@
 except ImportError:
     raise unittest.SkipTest("mongodb tests disabled.")
 
+import mongomock
+
 import factory
 from factory.mongoengine import MongoEngineFactory
 
@@ -57,6 +59,7 @@
             # PyMongo>=2.1 has a 20s timeout, use 100ms instead
             serverselectiontimeoutms=cls.server_timeout_ms,
             uuidRepresentation='standard',
+            mongo_client_class=mongomock.MongoClient,
         )
 
     @classmethod
