diff --git i/CMakeLists.txt w/CMakeLists.txt
index b01acc3bc..f47d1f814 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -923,10 +923,9 @@ if(SDL_ASSEMBLY)
       if(COMPILER_SUPPORTS_ALTIVEC)
         set(HAVE_ALTIVEC TRUE)
         set(SDL_ALTIVEC_BLITTERS 1)
-        sdl_compile_options(PRIVATE "-maltivec")
-        sdl_compile_options(PRIVATE "-fno-tree-vectorize")
         set_property(SOURCE "${SDL3_SOURCE_DIR}/src/video/SDL_blit_N.c" APPEND PROPERTY COMPILE_DEFINITIONS "SDL_ENABLE_ALTIVEC")
         set_property(SOURCE "${SDL3_SOURCE_DIR}/src/video/SDL_blit_N.c" PROPERTY SKIP_PRECOMPILE_HEADERS 1)
+        set_source_files_properties(src/video/SDL_blit_N.c PROPERTIES COMPILE_FLAGS "-maltivec -fno-tree-vectorize")
       endif()
     endif()
 
