diff --git i/misc/irqbalance.service w/misc/irqbalance.service
index 87e19c1..ed2522d 100644
--- i/misc/irqbalance.service
+++ w/misc/irqbalance.service
@@ -7,8 +7,8 @@ ConditionCPUs=>1
 
 [Service]
 EnvironmentFile=-/usr/lib/irqbalance/defaults.env
-EnvironmentFile=-/path/to/irqbalance.env
-ExecStart=/usr/sbin/irqbalance $IRQBALANCE_ARGS
+EnvironmentFile=-/etc/irqbalance.env
+ExecStart=/usr/bin/irqbalance $IRQBALANCE_ARGS
 CapabilityBoundingSet=
 NoNewPrivileges=yes
 ProtectSystem=strict
