diff -ruN a/Makefile b/Makefile
--- a/Makefile	2024-03-19 16:31:58.000000000 +0100
+++ b/Makefile	2024-03-21 16:22:41.092197215 +0100
@@ -637,24 +637,22 @@
 	false
 endif
 
-install-shared-c: install-shared-check shared install-headers
-	install -d $(DESTDIR)$(libdir)
-	install -m $(SO_INSTALL_MODE) $(OUT)/libmupdf.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/
+install-shared-c: install-shared-check install-headers
+	install -vDm $(SO_INSTALL_MODE) $(OUT)/libmupdf.$(SO)$(SO_VERSION) -t $(DESTDIR)$(libdir)/
 ifneq ($(OS),OpenBSD)
 	ln -sf libmupdf.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdf.$(SO)
 endif
 
-install-shared-c++: install-shared-c c++
-	install -m 644 platform/c++/include/mupdf/*.h $(DESTDIR)$(incdir)/mupdf
-	install -m $(SO_INSTALL_MODE) $(OUT)/libmupdfcpp.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/
+install-shared-c++: install-shared-check
+	install -vDm 644 platform/c++/include/mupdf/*.h -t $(DESTDIR)$(incdir)/mupdf
+	install -vDm $(SO_INSTALL_MODE) $(OUT)/libmupdfcpp.$(SO)$(SO_VERSION) -t $(DESTDIR)$(libdir)/
 ifneq ($(OS),OpenBSD)
 	ln -sf libmupdfcpp.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdfcpp.$(SO)
 endif
 
-install-shared-python: install-shared-c++ python
-	install -d $(DESTDIR)$(pydir)/mupdf
-	install -m $(SO_INSTALL_MODE) $(OUT)/_mupdf.$(SO) $(DESTDIR)$(pydir)/mupdf
-	install -m 644 $(OUT)/mupdf.py $(DESTDIR)$(pydir)/mupdf/__init__.py
+install-shared-python:
+	install -vDm $(SO_INSTALL_MODE) $(OUT)/_mupdf.$(SO) -t $(DESTDIR)$(pydir)/mupdf
+	install -vDm 644 $(OUT)/mupdf.py $(DESTDIR)$(pydir)/mupdf/__init__.py
 
 else
 
