diff -ruN a/Makefile b/Makefile
--- a/Makefile	2024-01-10 16:00:05.000000000 +0100
+++ b/Makefile	2024-01-26 20:17:17.921194085 +0100
@@ -600,11 +600,11 @@
 
 # C++, Python and C# shared libraries.
 c++-%: shared-%
-	./scripts/mupdfwrap.py --venv -d $(OUT) -b 01
+	./scripts/mupdfwrap.py -d $(OUT) -b 01
 python-%: c++-%
-	./scripts/mupdfwrap.py --venv -d $(OUT) -b 23
+	./scripts/mupdfwrap.py -d $(OUT) -b 23
 csharp-%: c++-%
-	./scripts/mupdfwrap.py --venv -d $(OUT) -b --csharp 23
+	./scripts/mupdfwrap.py -d $(OUT) -b --csharp 23
 
 
 # Installs of C, C++, Python and C# shared libraries
