From cbb9a8198920beb9b8ed3e36e6a569668b17cf3e Mon Sep 17 00:00:00 2001
From: "LevitatingBusinessMan (Rein Fernhout)" <me@levitati.ng>
Date: Tue, 18 Mar 2025 14:42:26 +0100
Subject: [PATCH] correctly link ruby bindings

fixes #1279
---
 bindings/ruby/extconf.rb | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/bindings/ruby/extconf.rb b/bindings/ruby/extconf.rb
index 882b0b071..de0171dc6 100644
--- a/bindings/ruby/extconf.rb
+++ b/bindings/ruby/extconf.rb
@@ -13,6 +13,10 @@
    $LDFLAGS += ' -blibpath:$(EPREFIX)/lib'
 end
 
-dir_config("rrd",['$(ABS_TOP_BUILDDIR)/src', '$(ABS_TOP_SRCDIR)/src'],'$(ABS_TOP_BUILDDIR)/src/.libs')
+ABS_TOP_BUILDDIR = ENV['ABS_TOP_BUILDDIR'] || '../..'
+ABS_TOP_SRCDIR = ENV['ABS_TOP_SRCDIR'] || '../..'
+
+
+dir_config("rrd", ["#{ABS_TOP_BUILDDIR}/src", "#{ABS_TOP_SRCDIR}/src"], "#{ABS_TOP_BUILDDIR}/src/.libs")
 have_library("rrd", "rrd_create")
 create_makefile("RRD")
