--- make/autoconf/flags-cflags.m4	2025-11-02 21:14:17.343800757 +0100
+++ make/autoconf/flags-cflags.m4.orig	2025-11-02 21:13:41.833617237 +0100
@@ -732,7 +732,7 @@
       $1_CFLAGS_CPU_JVM="-mno-multiple -mno-string"
       if test "x$FLAGS_CPU" = xppc64; then
         # -mminimal-toc fixes `relocation truncated to fit' error for gcc 4.1.
-        $1_CFLAGS_CPU="-mcpu=power8 -mtune=power8"
+        $1_CFLAGS_CPU="-mcpu=powerpc64 -mtune=power5 -DABI_ELFv2"
         $1_CFLAGS_CPU_JVM="${$1_CFLAGS_CPU_JVM} -mminimal-toc"
       elif test "x$FLAGS_CPU" = xppc64le; then
         # Little endian machine uses ELFv2 ABI.
