diff -aur parameterized-0.9.0/parameterized/test.py parameterized-0.9.0.old/parameterized/test.py
--- parameterized-0.9.0/parameterized/test.py	2023-03-27 03:58:07.000000000 +0200
+++ parameterized-0.9.0.old/parameterized/test.py	2024-04-03 22:24:06.396241531 +0200
@@ -2,7 +2,7 @@
 
 import inspect
 import sys
-import mock
+from unittest import mock
 from functools import wraps
 from unittest import TestCase
 try:
