diff -aur src/parameterized-0.8.1/parameterized/test.py src/parameterized-0.8.1.new/parameterized/test.py
--- src/parameterized-0.8.1/parameterized/test.py	2021-01-09 21:33:37.000000000 +0100
+++ src/parameterized-0.8.1.new/parameterized/test.py	2023-09-25 16:26:46.261621406 +0200
@@ -1,7 +1,7 @@
 # coding=utf-8
 
 import inspect
-import mock
+from unittest import mock
 from unittest import TestCase
 from nose.tools import assert_equal, assert_raises
 
diff -aur src/parameterized-0.8.1/tox.ini src/parameterized-0.8.1.new/tox.ini
--- src/parameterized-0.8.1/tox.ini	2021-01-09 21:33:37.000000000 +0100
+++ src/parameterized-0.8.1.new/tox.ini	2023-09-25 16:26:38.834909152 +0200
@@ -3,7 +3,6 @@
 [testenv]
 deps=
     nose
-    mock
     nose2: nose2
     pytest2: pytest>=2,<3
     pytest3: pytest>=3,<4
Only in src/parameterized-0.8.1.new: tox.ini.rej
