diff -aur parameterized-0.9.0/parameterized/test.py parameterized-0.9.0.orig/parameterized/test.py
--- parameterized-0.9.0/parameterized/test.py	2024-04-03 22:28:48.619573176 +0200
+++ parameterized-0.9.0.orig/parameterized/test.py	2024-04-03 22:27:26.374309509 +0200
@@ -35,7 +35,7 @@
     def func_decorator(func):
         @wraps(func)
         def wrapper(self, *args, **kwargs):
-            with self.assertRaisesRegex(expected_exception, expected_regexp):
+            with self.assertRaisesRegexp(expected_exception, expected_regexp):
                 func(self, *args, **kwargs)
 
         return wrapper
