diff --git a/docs/meson.build b/docs/meson.build
index f24ae6dd..19264cf0 100644
--- a/docs/meson.build
+++ b/docs/meson.build
@@ -86,7 +86,7 @@ if build_doc
     command: [sphinx_p,
       '-q',                       # quiet
       '-E',                       # rebuild from scratch
-      '-j', 'auto',               # parallel build
+      '-j', '1',                  # non-parallel build
       '-d', '@PRIVATE_DIR@',      # doctrees dir
       '-c', '@OUTDIR@',           # conf.py dir
       '@CURRENT_SOURCE_DIR@/rst', # source dir
