diff -ru SuperTux-v0.6.3-Source.orig/external/discord-sdk/thirdparty/rapidjson-1.1.0/include/rapidjson/document.h SuperTux-v0.6.3-Source/external/discord-sdk/thirdparty/rapidjson-1.1.0/include/rapidjson/document.h
--- SuperTux-v0.6.3-Source.orig/external/discord-sdk/thirdparty/rapidjson-1.1.0/include/rapidjson/document.h	2024-08-31 15:37:22.681399878 +0200
+++ SuperTux-v0.6.3-Source/external/discord-sdk/thirdparty/rapidjson-1.1.0/include/rapidjson/document.h	2024-08-31 15:37:53.831860837 +0200
@@ -316,8 +316,6 @@
 
     GenericStringRef(const GenericStringRef& rhs) : s(rhs.s), length(rhs.length) {}
 
-    GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
-
     //! implicit conversion to plain CharType pointer
     operator const Ch *() const { return s; }
 
