--- at-spi2-core/dbind/dbtest.c.orig	2019-02-18 19:15:05.059846039 +0100
+++ at-spi2-core/dbind/dbtest.c	2019-02-18 19:15:20.166525995 +0100
@@ -17,6 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include <linux/prctl.h>
 #include <stdio.h>
 #include <glib.h>
 #include <string.h>
--- at-spi2-core/bus/at-spi-bus-launcher.c.orig	2019-02-18 18:25:32.000000000 +0000
+++ at-spi2-core/bus/at-spi-bus-launcher.c	2019-02-18 18:25:41.000000000 +0000
@@ -25,11 +25,9 @@
 #include <unistd.h>
 #include <string.h>
 #include <signal.h>
-#ifdef __linux
 #include <sys/prctl.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#endif
 #include <sys/wait.h>
 #include <errno.h>
 #include <stdio.h>
