--- chromium-138.0.7204.183/net/filter/filter_source_stream_test_util.h.orig	2025-08-10 12:05:14.062164144 +0200
+++ chromium-138.0.7204.183/net/filter/filter_source_stream_test_util.h	2025-08-10 12:05:26.109848947 +0200
@@ -5,6 +5,7 @@
 #ifndef NET_FILTER_FILTER_SOURCE_STREAM_TEST_UTIL_H_
 #define NET_FILTER_FILTER_SOURCE_STREAM_TEST_UTIL_H_
 
+#include <stdint.h>
 #include <string_view>
 #include <vector>
 
--- chromium-138.0.7204.183/third_party/perfetto/src/trace_processor/dataframe/dataframe.h.orig	2025-08-12 13:15:22.761916293 +0200
+++ chromium-138.0.7204.183/third_party/perfetto/src/trace_processor/dataframe/dataframe.h	2025-08-12 13:15:42.460752502 +0200
@@ -103,7 +103,7 @@
       static const Type kDouble = base::variant_index<FilterValue, double>();
       static const Type kString =
           base::variant_index<FilterValue, const char*>();
-      static const Type kNull = base::variant_index<FilterValue, nullptr_t>();
+      static const Type kNull = base::variant_index<FilterValue, std::nullptr_t>();
       int64_t GetInt64Value(uint32_t col) const {
         return base::unchecked_get<int64_t>(filter_values_[col]);
       }
--- chromium-138.0.7204.183/third_party/perfetto/src/trace_processor/dataframe/impl/types.h.orig	2025-08-12 13:17:58.797042812 +0200
+++ chromium-138.0.7204.183/third_party/perfetto/src/trace_processor/dataframe/impl/types.h	2025-08-12 13:18:11.740117089 +0200
@@ -139,7 +139,7 @@
   using Double = FlexVector<double>;
   using String = FlexVector<StringPool::Id>;
 
-  using DataPointer = std::variant<nullptr_t,
+  using DataPointer = std::variant<std::nullptr_t,
                                    const uint32_t*,
                                    const int32_t*,
                                    const int64_t*,
--- chromium-138.0.7204.183/chrome/browser/privacy_sandbox/notice/desktop_view_manager_interface.h.orig	2025-08-12 17:58:55.086734829 +0200
+++ chromium-138.0.7204.183/chrome/browser/privacy_sandbox/notice/desktop_view_manager_interface.h	2025-08-12 17:59:02.985626207 +0200
@@ -4,6 +4,7 @@
 #ifndef CHROME_BROWSER_PRIVACY_SANDBOX_NOTICE_DESKTOP_VIEW_MANAGER_INTERFACE_H_
 #define CHROME_BROWSER_PRIVACY_SANDBOX_NOTICE_DESKTOP_VIEW_MANAGER_INTERFACE_H_
 #include <vector>
+#include <optional>
 
 #include "base/memory/raw_ptr.h"
 #include "chrome/browser/privacy_sandbox/notice/notice.mojom-forward.h"
