--- a/src/gftp-2.9.1b/src/gtk/misc-gtk.c 2025-11-04 20:41:30.917151346 +0200
+++ b/src/gftp-2.9.1b/src/gtk/misc-gtk.c 2025-11-04 20:41:57.781084453 +0200
@@ -618,9 +618,9 @@
   }
   /* run yesfunc or nofunc */
   if (responseID == GTK_RESPONSE_YES && ddata->yesfunc) {
-     ddata->yesfunc (ddata->yespointer, ddata);
+     ddata->yesfunc ();
   } else if (ddata->nofunc) {
-     ddata->nofunc (ddata->nopointer, ddata);
+     ddata->nofunc ();
   }
   /* free ddata */
   if (ddata->entry_text) g_free (ddata->entry_text);
