--- a/src/gftp-2.9.1b/src/gtk/gftp-gtk.h	2025-11-11 00:18:02.984615210 +0200
+++ b/src/gftp-2.9.1b/src/gtk/gftp-gtk.h	2025-11-11 00:18:34.684536275 +0200
@@ -104,10 +104,10 @@
   char * entry_text;
   int checkbox_is_ticked;
 
-  void (*yesfunc) ();
+  void (*yesfunc) (gpointer yespointer, gpointer data);
   gpointer yespointer;
 
-  void (*nofunc) ();
+  void (*nofunc) (gpointer nopointer, gpointer data);
   gpointer nopointer;
 } gftp_dialog_data;
 
