--- sed/sed/utils.c.orig	2026-09-03 15:40:20.840123056 +0200
+++ sed/sed/utils.c	2026-09-03 15:40:31.589053775 +0200
@@ -230,8 +230,8 @@
 {
   clearerr (stream);
   if (size && fwrite (ptr, size, nmemb, stream) != nmemb)
-    panic (ngettext ("couldn't write %jd item to %s: %s",
-                     "couldn't write %jd items to %s: %s", nmemb),
+    panic (ngettext ("couldn't write %td item to %s: %s",
+                     "couldn't write %td items to %s: %s", nmemb),
           nmemb, utils_fp_name (stream),
           strerror (errno));
 }
