diff --git a/src/osm/io/CMakeLists.txt b/src/osm/io/CMakeLists.txt
index bf1d02c..ec95f7e 100644
--- a/src/osm/io/CMakeLists.txt
+++ b/src/osm/io/CMakeLists.txt
@@ -1,7 +1,7 @@
 # SPDX-FileCopyrightText: 2020-2022 Volker Krause <vkrause@kde.org>
 # SPDX-License-Identifier: BSD-2-Clause
 
-if (Protobuf_FOUND AND Protobuf_PROTOC_EXECUTABLE)
+if (Protobuf_FOUND AND TARGET protobuf::protoc)
     PROTOBUF_GENERATE_CPP(pbf_srcs pbf_hdrs
         ../pbf/fileformat.proto
         ../pbf/osmformat.proto
@@ -19,7 +19,7 @@ if (Protobuf_FOUND AND Protobuf_PROTOC_EXECUTABLE)
     set_target_properties(KOSM_pbfioplugin PROPERTIES POSITION_INDEPENDENT_CODE ON)
     target_link_libraries(KOSM_pbfioplugin
         PUBLIC KOSM
-        PRIVATE ${Protobuf_LIBRARIES} ZLIB::ZLIB
+        PRIVATE protobuf::libprotobuf ZLIB::ZLIB
     )
 endif()
 
