diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1665707..e33eafa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.20)
 
 project(libappimage)
 
diff --git a/cmake/Modules/Findsquashfuse.cmake b/cmake/Modules/Findsquashfuse.cmake
index 24529a4..8679188 100644
--- a/cmake/Modules/Findsquashfuse.cmake
+++ b/cmake/Modules/Findsquashfuse.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.20)
 
 find_path(squashfuse_H_DIR
     NAMES squashfuse.h
diff --git a/cmake/dependencies.cmake b/cmake/dependencies.cmake
index 7cce2db..e10e78e 100644
--- a/cmake/dependencies.cmake
+++ b/cmake/dependencies.cmake
@@ -1,5 +1,5 @@
 # >= 3.2 required for ExternalProject_Add_StepDependencies
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.20)
 
 include(${CMAKE_CURRENT_LIST_DIR}/scripts.cmake)
 
diff --git a/cmake/reproducible_builds.cmake b/cmake/reproducible_builds.cmake
index 7fefaf4..c190fd8 100644
--- a/cmake/reproducible_builds.cmake
+++ b/cmake/reproducible_builds.cmake
@@ -2,7 +2,7 @@
 # it will replace such paths with relative ones
 # see https://reproducible-builds.org/docs/build-path/ for more information
 
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.20)
 
 include(CheckCCompilerFlag)
 
diff --git a/cmake/scripts.cmake b/cmake/scripts.cmake
index f792d1b..a0bbbda 100644
--- a/cmake/scripts.cmake
+++ b/cmake/scripts.cmake
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.20)
 
 include(ExternalProject)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9737205..96708f1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,5 @@
 # required for pkg-config to create PkgConfig::<prefix> imported library targets
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.20)
 
 add_subdirectory(xdg-basedir)
 add_subdirectory(libappimage_hashlib)
diff --git a/src/libappimage/CMakeLists.txt b/src/libappimage/CMakeLists.txt
index 0926933..58439df 100644
--- a/src/libappimage/CMakeLists.txt
+++ b/src/libappimage/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.20)
 
 add_subdirectory(core)
 add_subdirectory(utils)
diff --git a/src/xdg-basedir/CMakeLists.txt b/src/xdg-basedir/CMakeLists.txt
index 064260d..f0672ed 100644
--- a/src/xdg-basedir/CMakeLists.txt
+++ b/src/xdg-basedir/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.20)
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index cb48305..74f9889 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.20)
 
 # global definitions
 add_definitions(
diff --git a/tests/client_app/CMakeLists.txt b/tests/client_app/CMakeLists.txt
index b5ed361..bb3408a 100644
--- a/tests/client_app/CMakeLists.txt
+++ b/tests/client_app/CMakeLists.txt
@@ -1,5 +1,5 @@
 project(client_app)
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.20)
 
 find_package(libappimage REQUIRED)
 
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.20)
 
 include(CTest)
 
