--- source/CMakeLists.txt.orig	2022-04-04 20:03:06.107355488 +0200
+++ source/CMakeLists.txt	2022-04-04 20:03:18.450147724 +0200
@@ -43,7 +43,7 @@
 set(ARM_ALIASES armv6l armv7l aarch64)
 list(FIND X86_ALIASES "${SYSPROC}" X86MATCH)
 list(FIND ARM_ALIASES "${SYSPROC}" ARMMATCH)
-set(POWER_ALIASES ppc64 ppc64le)
+set(POWER_ALIASES ppc64 ppc64le powerpc64 powerpc64le)
 list(FIND POWER_ALIASES "${SYSPROC}" POWERMATCH)
 if("${SYSPROC}" STREQUAL "" OR X86MATCH GREATER "-1")
     set(X86 1)
