diff --git a/setuptools/tests/fixtures.py b/setuptools/tests/fixtures.py
index a5472984b..e2868182a 100644
--- a/setuptools/tests/fixtures.py
+++ b/setuptools/tests/fixtures.py
@@ -84,6 +84,7 @@ def _build_distributions(tmp_path_factory, request):
             sys.executable,
             "-m",
             "build",
+            "--no-isolation",
             "--outdir",
             str(tmp),
             str(request.config.rootdir),
