--- linux-6.1.141/arch/powerpc/boot/Makefile.orig	2025-06-12 10:00:59.911964366 +0200
+++ linux-6.1.141/arch/powerpc/boot/Makefile	2025-06-12 10:01:55.585787148 +0200
@@ -36,7 +36,7 @@
 		 -fno-strict-aliasing -O2 -msoft-float -mno-altivec -mno-vsx \
 		 $(call cc-option,-mno-spe) $(call cc-option,-mspe=no) \
 		 -pipe -fomit-frame-pointer -fno-builtin -fPIC -nostdinc \
-		 $(LINUXINCLUDE)
+		 -std=gnu11 $(LINUXINCLUDE)
 
 ifdef CONFIG_PPC64_BOOT_WRAPPER
 BOOTCFLAGS	+= -m64
