diff -ru libfm-qt-2.2.0.orig/CMakeLists.txt libfm-qt-2.2.0/CMakeLists.txt
--- libfm-qt-2.2.0.orig/CMakeLists.txt	2025-10-05 10:26:29.458168696 +0200
+++ libfm-qt-2.2.0/CMakeLists.txt	2025-10-05 10:28:37.795058864 +0200
@@ -34,6 +34,9 @@
 
 find_package(Qt6Widgets "${QT_MINIMUM_VERSION}" REQUIRED)
 find_package(Qt6LinguistTools "${QT_MINIMUM_VERSION}" REQUIRED)
+if(Qt6Widgets_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
+endif()
 
 find_package(lxqt2-build-tools "${LXQTBT_MINIMUM_VERSION}" REQUIRED)
 find_package(GLIB "${GLIB_MINIMUM_VERSION}" REQUIRED COMPONENTS gio gio-unix gobject gthread)
diff -ru libfm-qt-2.2.0.orig/src/CMakeLists.txt libfm-qt-2.2.0/src/CMakeLists.txt
--- libfm-qt-2.2.0.orig/src/CMakeLists.txt	2025-10-05 10:26:29.459338134 +0200
+++ libfm-qt-2.2.0/src/CMakeLists.txt	2025-10-05 10:27:51.230780249 +0200
@@ -142,6 +142,10 @@
     ${EXIF_LIBRARIES}
 )
 
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+    target_link_libraries(${LIBFM_QT_LIBRARY_NAME} Qt6::GuiPrivate)
+endif()
+
 # set libtool soname
 set_target_properties(${LIBFM_QT_LIBRARY_NAME} PROPERTIES
     VERSION ${LIBFM_QT_ABI_VERSION}
