--- ./subprojects/base-services/src/main/java/org/gradle/internal/os/OperatingSystem.java.orig	2020-06-02 11:52:39.474417315 +0200
+++ ./subprojects/base-services/src/main/java/org/gradle/internal/os/OperatingSystem.java	2020-06-02 11:55:17.734489160 +0200
@@ -358,6 +358,12 @@
             if ("powerpc".equals(arch)) {
                 arch = "ppc";
             }
+	    if ("powerpc64".equals(arch)) {
+		arch = "ppc64";
+	    }
+	    if ("powerpc64le".equals(arch)) {
+		arch = "ppc64";
+	    }
             return arch;
         }
 
