diff --git a/src/requests/certs.py b/src/requests/certs.py
index be422c3e..9bfe1c80 100644
--- a/src/requests/certs.py
+++ b/src/requests/certs.py
@@ -11,7 +11,8 @@ If you are packaging Requests, e.g., for a Linux distribution or a managed
 environment, you can change the definition of where() to return a separately
 packaged CA bundle.
 """
-from certifi import where
+def where():
+    return "/etc/ssl/certs/ca-certificates.crt"
 
 if __name__ == "__main__":
     print(where())
