diff --git a/examples/sdl-draw.c b/examples/sdl-draw.c
index e62db2764..a8a6e5a7f 100644
--- a/examples/sdl-draw.c
+++ b/examples/sdl-draw.c
@@ -1,6 +1,7 @@
 
 #include <gegl.h>
 #include <SDL.h>
+#include <stdio.h>
 #include <math.h>
 
 typedef struct
diff --git a/operations/external/sdl2-display.c b/operations/external/sdl2-display.c
index a56542a43..a9b6ddbe6 100644
--- a/operations/external/sdl2-display.c
+++ b/operations/external/sdl2-display.c
@@ -35,6 +35,7 @@ property_string (window_title, _("Window title"), "window_title")
 
 #include "gegl-op.h"
 #include <SDL.h>
+#include <stdio.h>
 
 typedef struct {
   SDL_Window   *window;
