diff --git a/configure b/configure
index 3209e225..f03c66b3 100755
--- a/configure
+++ b/configure
@@ -661,6 +661,31 @@ if ! compile_prog "" "" "C11 atomics"; then
 fi
 
 
+##########################################
+# 64 bit atomics support
+atomic_64="no"
+cat > $TMPC << EOF
+#include <stdatomic.h>
+#include <stdint.h>
+
+int main(void)
+{
+  _Atomic uint64_t v;
+  atomic_store_explicit(&v, 1, memory_order_release);
+  return 0;
+}
+EOF
+if compile_prog "" "" "64 bit atomics"; then
+  atomic_64="yes"
+elif compile_prog "" "-latomic" "64 bit atomics require libatomic"; then
+  LIBS="-latomic $LIBS"
+  atomic_64="yes"
+else
+  fatal "64 bit atomics not supported"
+fi
+print_config "64 bit atomics" "$atomic_64"
+
+
 ##########################################
 # check for wordsize
 wordsize="0"
