--- qtwebengine-everywhere-src-5.15.0/src/3rdparty/chromium/third_party/angle/include/platform/Platform.h.orig	2020-07-20 08:04:39.056484819 +0200
+++ qtwebengine-everywhere-src-5.15.0/src/3rdparty/chromium/third_party/angle/include/platform/Platform.h	2020-07-20 08:05:00.739296824 +0200
@@ -9,6 +9,7 @@
 #ifndef ANGLE_PLATFORM_H
 #define ANGLE_PLATFORM_H
 
+#include <stddef.h>
 #include <stdint.h>
 #include <array>
 
