--- src/3rdparty/chromium/base/i18n/string_compare.cc.orig	2020-11-07 21:57:18.831945969 +0100
+++ src/3rdparty/chromium/base/i18n/string_compare.cc	2020-11-07 21:57:36.681099042 +0100
@@ -18,8 +18,8 @@
                                              StringPiece16 rhs) {
   UErrorCode error = U_ZERO_ERROR;
   UCollationResult result = collator.compare(
-      icu::UnicodeString(FALSE, lhs.data(), static_cast<int>(lhs.length())),
-      icu::UnicodeString(FALSE, rhs.data(), static_cast<int>(rhs.length())),
+      icu::UnicodeString(false, lhs.data(), static_cast<int>(lhs.length())),
+      icu::UnicodeString(false, rhs.data(), static_cast<int>(rhs.length())),
       error);
   DCHECK(U_SUCCESS(error));
   return result;
--- src/3rdparty/chromium/base/i18n/time_formatting.cc.orig	2020-11-07 22:00:51.929875115 +0100
+++ src/3rdparty/chromium/base/i18n/time_formatting.cc	2020-11-07 22:01:08.508157940 +0100
@@ -240,7 +240,7 @@
   icu::FieldPosition ignore(icu::FieldPosition::DONT_CARE);
   measure_format.formatMeasures(measures, 3, formatted, ignore, status);
   *out = i18n::UnicodeStringToString16(formatted);
-  return U_SUCCESS(status) == TRUE;
+  return U_SUCCESS(status) == true;
 }
 
 string16 DateIntervalFormat(const Time& begin_time,
--- src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/text_codec_icu.cc.orig	2020-11-07 22:36:40.984864086 +0100
+++ src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/text_codec_icu.cc	2020-11-07 22:36:51.320814410 +0100
@@ -327,7 +327,7 @@
   DLOG_IF(ERROR, err == U_AMBIGUOUS_ALIAS_WARNING)
       << "ICU ambiguous alias warning for encoding: " << encoding_.GetName();
   if (converter_icu_)
-    ucnv_setFallback(converter_icu_, TRUE);
+    ucnv_setFallback(converter_icu_, true);
 }
 
 int TextCodecICU::DecodeToBuffer(UChar* target,
--- src/3rdparty/chromium/components/signin/public/identity_manager/identity_utils.cc.orig	2020-11-08 00:53:12.624659420 +0100
+++ src/3rdparty/chromium/components/signin/public/identity_manager/identity_utils.cc	2020-11-08 00:53:18.887036727 +0100
@@ -32,7 +32,7 @@
 
   // See if the username matches the policy-provided pattern.
   UErrorCode status = U_ZERO_ERROR;
-  const icu::UnicodeString icu_pattern(FALSE, utf16_pattern.data(),
+  const icu::UnicodeString icu_pattern(false, utf16_pattern.data(),
                                        utf16_pattern.length());
   icu::RegexMatcher matcher(icu_pattern, UREGEX_CASE_INSENSITIVE, status);
   if (!U_SUCCESS(status)) {
--- src/3rdparty/chromium/components/url_formatter/spoof_checks/idn_spoof_checker.cc.orig	2020-11-08 10:17:09.710294110 +0100
+++ src/3rdparty/chromium/components/url_formatter/spoof_checks/idn_spoof_checker.cc	2020-11-08 10:17:38.904204303 +0100
@@ -368,7 +368,7 @@
     return Result::kICUSpoofChecks;
   }
 
-  icu::UnicodeString label_string(FALSE /* isTerminated */, label.data(),
+  icu::UnicodeString label_string(false /* isTerminated */, label.data(),
                                   base::checked_cast<int32_t>(label.size()));
 
   // A punycode label with 'xn--' prefix is not subject to the URL
@@ -711,7 +711,7 @@
     base::StringPiece tld,
     base::StringPiece16 tld_unicode) {
   icu::UnicodeString tld_string(
-      FALSE /* isTerminated */, tld_unicode.data(),
+      false /* isTerminated */, tld_unicode.data(),
       base::checked_cast<int32_t>(tld_unicode.size()));
   // Allow if the TLD contains any letter from the script, in which case it's
   // likely to be a TLD in that script.
--- src/3rdparty/chromium/third_party/blink/renderer/platform/text/text_break_iterator_icu.cc.orig	2020-11-08 15:36:02.750895160 +0100
+++ src/3rdparty/chromium/third_party/blink/renderer/platform/text/text_break_iterator_icu.cc	2020-11-08 15:36:51.338117758 +0100
@@ -312,13 +312,13 @@
       text->chunkOffset = offset <= std::numeric_limits<int32_t>::max()
                               ? static_cast<int32_t>(offset)
                               : 0;
-      is_accessible = TRUE;
+      is_accessible = true;
       return true;
     }
     if (native_index >= native_length &&
         text->chunkNativeLimit == native_length) {
       text->chunkOffset = text->chunkLength;
-      is_accessible = FALSE;
+      is_accessible = false;
       return true;
     }
   } else {
@@ -331,12 +331,12 @@
       text->chunkOffset = offset <= std::numeric_limits<int32_t>::max()
                               ? static_cast<int32_t>(offset)
                               : 0;
-      is_accessible = TRUE;
+      is_accessible = true;
       return true;
     }
     if (native_index <= 0 && !text->chunkNativeStart) {
       text->chunkOffset = 0;
-      is_accessible = FALSE;
+      is_accessible = false;
       return true;
     }
   }
@@ -347,7 +347,7 @@
                               int64_t native_index,
                               UBool forward) {
   if (!text->context)
-    return FALSE;
+    return false;
   int64_t native_length = TextNativeLength(text);
   UBool is_accessible;
   if (TextInChunkOrOutOfRange(text, native_index, native_length, forward,
@@ -371,7 +371,7 @@
     DCHECK_EQ(new_context, kPriorContext);
     TextLatin1SwitchToPriorContext(text, native_index, native_length, forward);
   }
-  return TRUE;
+  return true;
 }
 
 static const struct UTextFuncs kTextLatin1Funcs = {
@@ -511,7 +511,7 @@
 
 static UBool TextUTF16Access(UText* text, int64_t native_index, UBool forward) {
   if (!text->context)
-    return FALSE;
+    return false;
   int64_t native_length = TextNativeLength(text);
   UBool is_accessible;
   if (TextInChunkOrOutOfRange(text, native_index, native_length, forward,
@@ -533,7 +533,7 @@
     DCHECK_EQ(new_context, kPriorContext);
     TextUTF16SwitchToPriorContext(text, native_index, native_length, forward);
   }
-  return TRUE;
+  return true;
 }
 
 static const struct UTextFuncs kTextUTF16Funcs = {
--- src/3rdparty/chromium/third_party/blink/renderer/platform/text/locale_icu.cc.orig	2020-11-08 15:43:18.148837903 +0100
+++ src/3rdparty/chromium/third_party/blink/renderer/platform/text/locale_icu.cc	2020-11-08 15:43:31.348524958 +0100
@@ -169,12 +169,12 @@
     return g_empty_string;
 
   UErrorCode status = U_ZERO_ERROR;
-  int32_t length = udat_toPattern(date_format, TRUE, nullptr, 0, &status);
+  int32_t length = udat_toPattern(date_format, true, nullptr, 0, &status);
   if (status != U_BUFFER_OVERFLOW_ERROR || !length)
     return g_empty_string;
   StringBuffer<UChar> buffer(length);
   status = U_ZERO_ERROR;
-  udat_toPattern(date_format, TRUE, buffer.Characters(), length, &status);
+  udat_toPattern(date_format, true, buffer.Characters(), length, &status);
   if (U_FAILURE(status))
     return g_empty_string;
   return String::Adopt(buffer);
--- src/3rdparty/chromium/third_party/blink/renderer/platform/text/unicode_utilities.cc.orig	2020-11-08 15:45:53.399372206 +0100
+++ src/3rdparty/chromium/third_party/blink/renderer/platform/text/unicode_utilities.cc	2020-11-08 15:46:02.903423863 +0100
@@ -300,7 +300,7 @@
   DCHECK(U_SUCCESS(status));
   int32_t input_length = static_cast<int32_t>(length);
   // copy-on-write.
-  icu::UnicodeString normalized(FALSE, characters, input_length);
+  icu::UnicodeString normalized(false, characters, input_length);
   // In the vast majority of cases, input is already NFC. Run a quick check
   // to avoid normalizing the entire input unnecessarily.
   int32_t normalized_prefix_length =
--- src/3rdparty/chromium/components/spellcheck/renderer/spellcheck_worditerator.cc.orig	2020-11-08 16:01:32.646165648 +0100
+++ src/3rdparty/chromium/components/spellcheck/renderer/spellcheck_worditerator.cc	2020-11-08 16:01:41.751273928 +0100
@@ -442,7 +442,7 @@
   // spellchecker and we need manual normalization as well. The normalized
   // text does not have to be NUL-terminated since its characters are copied to
   // string16, which adds a NUL character when we need.
-  icu::UnicodeString input(FALSE, &text_[input_start],
+  icu::UnicodeString input(false, &text_[input_start],
                            base::checked_cast<int32_t>(input_length));
   UErrorCode status = U_ZERO_ERROR;
   icu::UnicodeString output;
--- src/3rdparty/chromium/ui/base/l10n/formatter.cc.orig	2020-11-09 10:01:54.473988587 +0100
+++ src/3rdparty/chromium/ui/base/l10n/formatter.cc	2020-11-09 10:02:03.833015819 +0100
@@ -232,7 +232,7 @@
                        int value,
                        icu::UnicodeString* formatted_string) const {
   DCHECK(simple_format_[unit]);
-  DCHECK(formatted_string->isEmpty() == TRUE);
+  DCHECK(formatted_string->isEmpty() == true);
   UErrorCode error = U_ZERO_ERROR;
   FormatNumberInPlural(*simple_format_[unit],
                         value, formatted_string, &error);
@@ -248,7 +248,7 @@
       << "Detailed() not implemented for your (format, length) combination!";
   DCHECK(detailed_format_[units][1])
       << "Detailed() not implemented for your (format, length) combination!";
-  DCHECK(formatted_string->isEmpty() == TRUE);
+  DCHECK(formatted_string->isEmpty() == true);
   UErrorCode error = U_ZERO_ERROR;
   FormatNumberInPlural(*detailed_format_[units][0], value_1,
                        formatted_string, &error);
@@ -281,7 +281,7 @@
     base::string16 pattern = l10n_util::GetStringUTF16(pluralities.id);
     UErrorCode error = U_ZERO_ERROR;
     std::unique_ptr<icu::MessageFormat> format(new icu::MessageFormat(
-        icu::UnicodeString(FALSE, pattern.data(), pattern.length()), error));
+        icu::UnicodeString(false, pattern.data(), pattern.length()), error));
     DCHECK(U_SUCCESS(error));
     if (format.get())
       return format;
--- src/3rdparty/chromium/components/autofill/core/common/autofill_regexes.cc.orig	2020-11-09 10:08:56.724081510 +0100
+++ src/3rdparty/chromium/components/autofill/core/common/autofill_regexes.cc	2020-11-09 10:09:14.733208031 +0100
@@ -43,7 +43,7 @@
 icu::RegexMatcher* AutofillRegexes::GetMatcher(const base::string16& pattern) {
   auto it = matchers_.find(pattern);
   if (it == matchers_.end()) {
-    const icu::UnicodeString icu_pattern(FALSE, pattern.data(),
+    const icu::UnicodeString icu_pattern(false, pattern.data(),
                                          pattern.length());
 
     UErrorCode status = U_ZERO_ERROR;
@@ -70,20 +70,20 @@
   base::AutoLock lock(*g_lock);
 
   icu::RegexMatcher* matcher = g_autofill_regexes->GetMatcher(pattern);
-  icu::UnicodeString icu_input(FALSE, input.data(), input.length());
+  icu::UnicodeString icu_input(false, input.data(), input.length());
   matcher->reset(icu_input);
 
   UErrorCode status = U_ZERO_ERROR;
   UBool matched = matcher->find(0, status);
   DCHECK(U_SUCCESS(status));
 
-  if (matched == TRUE && match) {
+  if (matched == true && match) {
     icu::UnicodeString match_unicode = matcher->group(0, status);
     DCHECK(U_SUCCESS(status));
     *match = base::i18n::UnicodeStringToString16(match_unicode);
   }
 
-  return matched == TRUE;
+  return matched == true;
 }
 
 }  // namespace autofill
--- src/3rdparty/chromium/services/service_manager/zygote/zygote_linux.cc.orig	2020-11-09 13:15:17.703116760 +0100
+++ src/3rdparty/chromium/services/service_manager/zygote/zygote_linux.cc	2020-11-09 13:15:27.232144346 +0100
@@ -564,7 +564,7 @@
   if (!iter.ReadString16(&timezone_id))
     return -1;
   icu::TimeZone::adoptDefault(icu::TimeZone::createTimeZone(
-      icu::UnicodeString(FALSE, timezone_id.data(), timezone_id.length())));
+      icu::UnicodeString(false, timezone_id.data(), timezone_id.length())));
 
   if (!iter.ReadInt(&numfds))
     return -1;
--- src/3rdparty/chromium/components/url_formatter/spoof_checks/skeleton_generator.cc.orig	2020-11-24 07:47:49.886520642 +0100
+++ src/3rdparty/chromium/components/url_formatter/spoof_checks/skeleton_generator.cc	2020-11-24 07:48:05.950846117 +0100
@@ -117,7 +117,7 @@
 Skeletons SkeletonGenerator::GetSkeletons(base::StringPiece16 hostname) {
   Skeletons skeletons;
   size_t hostname_length = hostname.length() - (hostname.back() == '.' ? 1 : 0);
-  icu::UnicodeString host(FALSE, hostname.data(), hostname_length);
+  icu::UnicodeString host(false, hostname.data(), hostname_length);
   // If input has any characters outside Latin-Greek-Cyrillic and [0-9._-],
   // there is no point in getting rid of diacritics because combining marks
   // attached to non-LGC characters are already blocked.
