diff --git a/src/core/configure.cmake b/src/core/configure.cmake
index 7a0543619..a755c573a 100644
--- a/src/core/configure.cmake
+++ b/src/core/configure.cmake
@@ -13,14 +13,14 @@ set_property(CACHE INPUT_qt3d_assimp PROPERTY STRINGS undefined qt system no)
 
 #### Libraries
 
-qt_find_package(WrapQt3DAssimp 5 PROVIDED_TARGETS WrapQt3DAssimp::WrapQt3DAssimp MODULE_NAME 3dcore QMAKE_LIB qt3d-assimp)
+qt_find_package(WrapQt3DAssimp 6 PROVIDED_TARGETS WrapQt3DAssimp::WrapQt3DAssimp MODULE_NAME 3dcore QMAKE_LIB qt3d-assimp)
 
 qt_config_compile_test("assimp"
                    LABEL ""
                    PROJECT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../config.tests/assimp"
 # special case begin
                    LIBRARIES WrapQt3DAssimp::WrapQt3DAssimp
-                   PACKAGES PACKAGE WrapQt3DAssimp 5
+                   PACKAGES PACKAGE WrapQt3DAssimp 6
 # special case end
 )
 
