--- webkitgtk-2.42.1/Tools/MiniBrowser/gtk/BrowserWindow.c.orig	2023-11-12 09:55:06.198656295 +0100
+++ webkitgtk-2.42.1/Tools/MiniBrowser/gtk/BrowserWindow.c	2023-11-12 09:55:45.106611911 +0100
@@ -254,7 +254,7 @@
 #endif
 #undef MAX_TITLE
 
-        char *actionName = g_strdup_printf("action-%lu", ++actionId);
+        char *actionName = g_strdup_printf("action-%lu", (long unsigned int)++actionId);
         GSimpleAction *action = g_simple_action_new(actionName, NULL);
         g_object_set_data_full(G_OBJECT(action), "back-forward-list-item", g_object_ref(item), g_object_unref);
         g_signal_connect_swapped(action, "activate", G_CALLBACK(browserWindowHistoryItemActivated), window);
