--- Source/cmake/WebKitCompilerFlags.cmake.orig	2021-05-23 16:22:36.959801395 +0200
+++ Source/cmake/WebKitCompilerFlags.cmake	2021-05-23 16:22:45.338938211 +0200
@@ -158,13 +158,6 @@
     WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-Wall -Wextra)
 endif ()
 
-# Ninja tricks compilers into turning off color support.
-if (CMAKE_GENERATOR STREQUAL "Ninja")
-    WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-fcolor-diagnostics
-                                         -fdiagnostics-color=always)
-endif ()
-
-
 string(TOLOWER ${CMAKE_HOST_SYSTEM_PROCESSOR} LOWERCASE_CMAKE_HOST_SYSTEM_PROCESSOR)
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND NOT "${LOWERCASE_CMAKE_HOST_SYSTEM_PROCESSOR}" MATCHES "x86_64")
     # To avoid out of memory when building with debug option in 32bit system.
