From 26155f0997470d4ac10b0295ed5c1c16e070734a Mon Sep 17 00:00:00 2001
From: Sean Whitton <spwhitton@spwhitton.name>
Date: Tue, 29 Aug 2023 09:08:55 +0100
Subject: Change native-comp-async-jobs-number default to 1

---
 lisp/emacs-lisp/comp.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/emacs-lisp/comp.el b/lisp/emacs-lisp/comp.el
index d2c6dbe0c05..b64190a35aa 100644
--- a/lisp/emacs-lisp/comp.el
+++ b/lisp/emacs-lisp/comp.el
@@ -117,7 +117,7 @@ native-comp-never-optimize-functions
   :type '(repeat symbol)
   :version "28.1")
 
-(defcustom native-comp-async-jobs-number 0
+(defcustom native-comp-async-jobs-number 1
   "Default number of subprocesses used for async native compilation.
 Value of zero means to use half the number of the CPU's execution units,
 or one if there's just one execution unit."
