--- intern/numaapi/source/build_config.h.orig	2019-10-06 09:39:39.100352487 +0200
+++ intern/numaapi/source/build_config.h	2019-10-06 09:40:50.342911199 +0200
@@ -307,6 +307,15 @@
 #    define ARCH_CPU_32_BITS 1
 #    define ARCH_CPU_BIG_ENDIAN 1
 #  endif
+#elif defined(__powerpc64__)
+#  define ARCH_CPU_PPC_FAMILY 1
+#  define ARCH_CPU_PPC 1
+#  define ARCH_CPU_64_BITS 1
+#  if defined(__BIG_ENDIAN__)
+#    define ARCH_CPU_BIG_ENDIAN 1
+#  else
+#    define ARCH_CPU_LITTLE_ENDIAN 1
+#  endif
 #else
 #  error Please add support for your architecture in build_config.h
 #endif
@@ -338,6 +347,14 @@
 #  define ARCH_CPU_MIPS64_FAMILY 0
 #endif
 
+#if !defined(ARCH_CPU_PPC_FAMILY)
+#  define ARCH_CPU_PPC_FAMILY 0
+#endif
+
+#if !defined(ARCH_CPU_PPC64_FAMILY)
+#  define ARCH_CPU_PPC64_FAMILY 0
+#endif
+
 ////////////////////////////////////////////////////////////////////////////////
 // Sizes of platform-dependent types.
 
