diff --git a/ThirdParty/xdmf3/update.sh b/ThirdParty/xdmf3/update.sh
index 01dfc959e9bfb930ddd617267b08f0cb225aaab5..3542eee09194cf0c93f19bc2b1632868d2a9c1a7 100755
--- a/ThirdParty/xdmf3/update.sh
+++ b/ThirdParty/xdmf3/update.sh
@@ -8,7 +8,7 @@ readonly name="xdmf3"
 readonly ownership="XDMF Upstream <kwrobot@kitware.com>"
 readonly subtree="ThirdParty/$name/vtk$name"
 readonly repo="https://gitlab.kitware.com/third-party/xdmf.git"
-readonly tag="for/vtk-20231124-master-gfe7dd1ca"
+readonly tag="for/vtk-20231212-master-gfe7dd1ca"
 readonly paths="
 .gitattributes
 CMake/XdmfFunctions.cmake
diff --git a/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp b/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp
index d833c258237b72dc0fce72912eaed6728a7934bf..cb5dd509efc219471ea3e2144529f6e61c60be38 100644
--- a/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp
+++ b/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.cpp
@@ -180,6 +180,14 @@ public:
     return XdmfArrayType::UInt32();
   }
 
+#if defined(_WIN32) || defined(__APPLE__)
+  shared_ptr<const XdmfArrayType>
+  getArrayType(const unsigned long * const) const
+  {
+    return XdmfArrayType::UInt64();
+  }
+#endif
+
   shared_ptr<const XdmfArrayType>
   getArrayType(const uint64_t * const) const
   {
diff --git a/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.hpp b/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.hpp
index b92b1954d825fa5110b4717ee4f8da9b8cce2f1f..703e98af32916f2842d6082718177ddfead4ed18 100644
--- a/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.hpp
+++ b/ThirdParty/xdmf3/vtkxdmf3/core/XdmfArray.hpp
@@ -1527,6 +1527,7 @@ private:
     shared_ptr<std::vector<unsigned char> >,
     shared_ptr<std::vector<unsigned short> >,
     shared_ptr<std::vector<unsigned int> >,
+    shared_ptr<std::vector<unsigned long> >,
     shared_ptr<std::vector<uint64_t> >,
     shared_ptr<std::vector<std::string> >,
     boost::shared_array<const char>,
@@ -1538,6 +1539,7 @@ private:
     boost::shared_array<const unsigned char>,
     boost::shared_array<const unsigned short>,
     boost::shared_array<const unsigned int>,
+    boost::shared_array<const unsigned long>,
     boost::shared_array<const uint64_t>  > ArrayVariant;
 
   unsigned int mArrayPointerNumValues;
