Author: Jordi Mallach <jordi@debian.org>
Description: do not assume python points to python3.
Origin: Debian
Forwarded: no

Index: mame/makefile
===================================================================
--- mame.orig/makefile
+++ mame/makefile
@@ -1030,7 +1030,7 @@ GIT_AVAILABLE    := $(shell git --versio
 else
 GCC_VERSION      := $(shell $(TOOLCHAIN)$(subst @,,$(CC)) -dumpfullversion 2> NUL)
 CLANG_VERSION    := $(shell $(TOOLCHAIN)$(subst @,,$(CC)) --version 2> NUL| head -n 1 | grep clang | sed "s/^.*[^0-9]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\).*$$/\1/" | head -n 1)
-PYTHON_AVAILABLE := $(shell $(PYTHON) --version > NUL 2>&1 && echo python)
+PYTHON_AVAILABLE := $(shell $(PYTHON) --version > NUL 2>&1 && echo python3)
 GIT_AVAILABLE    := $(shell git --version > NUL 2>&1 && echo git)
 endif
 ifdef MSBUILD
@@ -1060,7 +1060,7 @@ CLANG_VERSION    := $(shell $(TOOLCHAIN)
 else
 CLANG_VERSION    := $(shell $(TOOLCHAIN)$(subst @,,$(CC))  --version  2> /dev/null | head -n 1 | grep -e 'version [0-9]\+\.[0-9]\+\(\.[0-9]\+\)\?' -o | grep -e '[0-9]\+\.[0-9]\+\(\.[0-9]\+\)\?' -o | tail -n 1)
 endif
-PYTHON_AVAILABLE := $(shell $(PYTHON) --version > /dev/null 2>&1 && echo python)
+PYTHON_AVAILABLE := $(shell $(PYTHON) --version > /dev/null 2>&1 && echo python3)
 GIT_AVAILABLE := $(shell git --version > /dev/null 2>&1 && echo git)
 endif
 
@@ -1081,7 +1081,7 @@ endif
 endif
 endif
 
-ifneq ($(PYTHON_AVAILABLE),python)
+ifneq ($(PYTHON_AVAILABLE),python3)
 $(error Python is not available in path)
 endif
 
