Description: Fix ftbfs with GCC-10
Author: Sudip Mukherjee <sudipm.mukherjee@gmail.com>
Bug-Debian: https://bugs.debian.org/957992

---

--- xplc-0.3.13.orig/xplc/moduleloader.cpp
+++ xplc-0.3.13/xplc/moduleloader.cpp
@@ -62,7 +62,7 @@ Module* Module::loadModule(const char* m
 
   switch(moduleinfo->version_major) {
 #ifdef UNSTABLE
-  case -1:
+  case (unsigned int)-1:
     /* nothing to do */
     break;
 #endif
