--- xwayland-24.1.0/os/osdep.h.orig	2024-06-20 08:53:34.859276429 +0200
+++ xwayland-24.1.0/os/osdep.h	2024-06-20 08:53:50.945689830 +0200
@@ -95,7 +95,7 @@
     ConnectionInputPtr input;
     ConnectionOutputPtr output;
     XID auth_id;                /* authorization id */
-    CARD32 conn_time;           /* timestamp if not established, else 0  */
+    Atom conn_time;             /* timestamp if not established, else 0  */
     struct _XtransConnInfo *trans_conn; /* transport connection object */
     int flags;
 } OsCommRec, *OsCommPtr;
