--- konsole-24.01.95/src/profile/Profile.cpp.orig	2024-02-18 22:44:22.720998395 +0100
+++ konsole-24.01.95/src/profile/Profile.cpp	2024-02-18 22:44:28.488432793 +0100
@@ -129,7 +129,7 @@
     {ReverseUrlHints, "ReverseUrlHints", TERMINAL_GROUP, false},
     {BlinkingTextEnabled, "BlinkingTextEnabled", TERMINAL_GROUP, true},
     {FlowControlEnabled, "FlowControlEnabled", TERMINAL_GROUP, true},
-    {BidiRenderingEnabled, "BidiRenderingEnabled", TERMINAL_GROUP, true},
+    {BidiRenderingEnabled, "BidiRenderingEnabled", TERMINAL_GROUP, false},
     {BidiLineLTR, "BidiLineLTR", TERMINAL_GROUP, true},
     {BidiTableDirOverride, "BidiTableDirOverride", TERMINAL_GROUP, true},
     {BlinkingCursorEnabled, "BlinkingCursorEnabled", TERMINAL_GROUP, false},
