diff --git a/scripts/adjust_autoksyms.sh b/scripts/adjust_autoksyms.sh
index 6e6d639..2b30884 100755
--- a/scripts/adjust_autoksyms.sh
+++ b/scripts/adjust_autoksyms.sh
@@ -59,7 +59,7 @@ EOT
 [ "$(ls -A "$MODVERDIR")" ] &&
 for mod in "$MODVERDIR"/*.mod; do
        sed -n -e '3{s/ /\n/g;/^$/!p;}' "$mod"
-done | sort -u |
+done | grep -Fv ".TOC." | sort -u |
 while read sym; do
        echo "#define __KSYM_${sym} 1"
 done >> "$new_ksyms_file"
