--- mesa-24.0.1/src/gallium/auxiliary/gallivm/lp_bld_init.c.orig	2024-02-26 12:35:35.455352462 +0100
+++ mesa-24.0.1/src/gallium/auxiliary/gallivm/lp_bld_init.c	2024-02-26 12:36:11.187758364 +0100
@@ -47,7 +47,7 @@
 #include <llvm-c/Transforms/PassBuilder.h>
 #elif GALLIVM_HAVE_CORO == 1
 #include <llvm-c/Transforms/Scalar.h>
-#if LLVM_VERSION_MAJOR >= 7
+#if LLVM_VERSION_MAJOR >= 7 && LLVM_VERSION_MAJOR <17
 #include <llvm-c/Transforms/Utils.h>
 #endif
 #if LLVM_VERSION_MAJOR <= 8 && (DETECT_ARCH_AARCH64 || DETECT_ARCH_ARM || DETECT_ARCH_S390 || DETECT_ARCH_MIPS64)
