diff --git i/src/wheel/_bdist_wheel.py w/src/wheel/_bdist_wheel.py
index 88973eb..e79f7fd 100644
--- i/src/wheel/_bdist_wheel.py
+++ w/src/wheel/_bdist_wheel.py
@@ -27,8 +27,8 @@ from setuptools import Command
 from . import __version__ as wheel_version
 from .metadata import pkginfo_to_metadata
 from .util import log
-from .vendored.packaging import tags
-from .vendored.packaging import version as _packaging_version
+from packaging import tags
+from packaging import version as _packaging_version
 from .wheelfile import WheelFile
 
 if TYPE_CHECKING:
diff --git i/src/wheel/cli/convert.py w/src/wheel/cli/convert.py
index ec5ecf3..2727106 100644
--- i/src/wheel/cli/convert.py
+++ w/src/wheel/cli/convert.py
@@ -15,7 +15,7 @@ from zipfile import ZipFile
 
 from .. import __version__
 from ..metadata import generate_requirements
-from ..vendored.packaging.tags import parse_tag
+from packaging.tags import parse_tag
 from ..wheelfile import WheelFile
 
 egg_filename_re = re.compile(
diff --git i/src/wheel/metadata.py w/src/wheel/metadata.py
index b8098fa..3a17175 100644
--- i/src/wheel/metadata.py
+++ w/src/wheel/metadata.py
@@ -13,7 +13,7 @@ from email.message import Message
 from email.parser import Parser
 from typing import Generator, Iterable, Iterator, Literal
 
-from .vendored.packaging.requirements import Requirement
+from packaging.requirements import Requirement
 
 
 def _nonblank(str: str) -> bool | Literal[""]:
diff --git i/tests/test_bdist_wheel.py w/tests/test_bdist_wheel.py
index fcb2dfc..38d5429 100644
--- i/tests/test_bdist_wheel.py
+++ w/tests/test_bdist_wheel.py
@@ -23,7 +23,7 @@ from wheel._bdist_wheel import (
     remove_readonly,
     remove_readonly_exc,
 )
-from wheel.vendored.packaging import tags
+from packaging import tags
 from wheel.wheelfile import WheelFile
 
 DEFAULT_FILES = {
