--- NVIDIA-Linux-ppc64le-418.67/kernel/nvidia-uvm/uvm8_tools.c.orig	2019-07-09 10:04:29.050000000 +0000
+++ NVIDIA-Linux-ppc64le-418.67/kernel/nvidia-uvm/uvm8_tools.c	2019-07-09 10:04:46.350000000 +0000
@@ -206,13 +206,6 @@
     return event_tracker != NULL && !event_tracker->is_queue;
 }
 
-static void put_user_pages(struct page **pages, NvU64 page_count)
-{
-    NvU64 i;
-    for (i = 0; i < page_count; i++)
-        put_page(pages[i]);
-}
-
 static void unmap_user_pages(struct page **pages, void *addr, NvU64 size)
 {
     size = DIV_ROUND_UP(size, PAGE_SIZE);
