--- multipath-tools.org/create-config.mk	2025-02-09 15:13:02.727244979 +0100
+++ multipath-tools/create-config.mk	2025-02-09 15:14:32.754157269 +0100
@@ -64,7 +64,7 @@
 # gcc 4.8 compiles blacklist.c only with -Wno-missing-field-initializers
 TEST_MISSING_INITIALIZERS = $(shell \
 	echo 'struct A {int a, b;}; struct B {struct A a; int b;} b = {.a.a=1};' | \
-		$(CC) -c -Werror -Wmissing-field-initializers -o /dev/null -xc - >/dev/null 2>&1 \
+		$(CC) -c -Wmissing-field-initializers -o /dev/null -xc - >/dev/null 2>&1 \
 	|| echo -Wno-missing-field-initializers)
 
 # gcc 4.8.4 and certain versions of liburcu fail to compile this with -Werror=type-limits
@@ -136,7 +136,7 @@
 # Evaluates to "option" if yes, and "fallback" otherwise.
 TEST_CC_OPTION = $(shell \
 	if echo 'int main(void){return 0;}' | \
-		$(CC) -o /dev/null -c -Werror "$(1)" -xc - >/dev/null 2>&1; \
+		$(CC) -o /dev/null -c "$(1)" -xc - >/dev/null 2>&1; \
 	then \
 		echo "$(1)"; \
 	else \
@@ -148,11 +148,11 @@
 # but it doesn't seem to make a difference wrt the compilation result.
 FORTIFY_OPT := $(shell \
 	if printf '$(__HASH__)include <string.h>\nint main(void) { return 0; }\n' | \
-		$(CC) -o /dev/null $(OPTFLAGS) -c -Werror -D_FORTIFY_SOURCE=3 -xc - 2>/dev/null; \
+		$(CC) -o /dev/null $(OPTFLAGS) -c -D_FORTIFY_SOURCE=3 -xc - 2>/dev/null; \
 	then \
 		echo "-D_FORTIFY_SOURCE=3"; \
 	elif printf '$(__HASH__)include <string.h>\nint main(void) { return 0; }\n' | \
-		$(CC) -o /dev/null $(OPTFLAGS) -c -Werror -D_FORTIFY_SOURCE=2 -xc - 2>/dev/null; \
+		$(CC) -o /dev/null $(OPTFLAGS) -c -D_FORTIFY_SOURCE=2 -xc - 2>/dev/null; \
 	then \
 		echo "-D_FORTIFY_SOURCE=2"; \
 	fi)
--- multipath-tools.org/Makefile.inc	2025-02-09 15:13:02.727244979 +0100
+++ multipath-tools/Makefile.inc	2025-02-09 15:25:08.074051437 +0100
@@ -105,8 +105,8 @@
 WARN_ONLY       :=
 ERROR           := $(if $(WARN_ONLY),,error=)
 WERROR          := $(if $(WARN_ONLY),,-Werror)
-WARNFLAGS	:= $(WERROR) -Wall -Wextra -Wformat=2 $(WFORMATOVERFLOW) -W$(ERROR)implicit-int \
-		  -W$(ERROR)implicit-function-declaration -W$(ERROR)format-security \
+WARNFLAGS	:= $(WERROR) -Wall -Wextra -Wformat=2 $(WFORMATOVERFLOW) -W$(ERROR)format-security \
+                  -Wno-implicit-function-declaration \
 		  $(WNOCLOBBERED) -W$(ERROR)cast-qual $(ERROR_DISCARDED_QUALIFIERS) $(W_URCU_TYPE_LIMITS)
 
 CPPFLAGS	:= $(FORTIFY_OPT) $(CPPFLAGS) $(D_URCU_VERSION) \
