--- propcache-0.4.1/packaging/pep517_backend/_backend.py.orig	2025-12-10 15:54:35.971215051 +0100
+++ propcache-0.4.1/packaging/pep517_backend/_backend.py	2025-12-10 15:54:49.642763041 +0100
@@ -376,7 +376,7 @@
     if is_pure_python_build:
         c_ext_build_deps = []
     else:
-        c_ext_build_deps = ['Cython ~= 3.1.0']
+        c_ext_build_deps = ['Cython ~= 3.2.0']
 
     return _setuptools_get_requires_for_build_wheel(
         config_settings=config_settings,
