--- texlive-bin-2025.20250325.74682.orig/texk/ps2pk/fontfcn.h
+++ texlive-bin-2025.20250325.74682/texk/ps2pk/fontfcn.h
@@ -64,7 +64,7 @@
 		             FontScalablePtr, fsBitmapFormat,
 		             fsBitmapFormatMask, DOUBLE, DOUBLE);
 extern void Type1RegisterFontFileFunctions(void);
-extern void Type1CloseFont();
+extern void Type1CloseFont(FontPtr);
 extern int CheckFSFormat(int format,int fmask,int *bit,int *byte,int *scan,int *glyph,int *image);
  
 /***================================================================***/
