--- texlive-bin-2025.20250325.74682.orig/texk/web2c/luatexdir/unilib/ustring.c
+++ texlive-bin-2025.20250325.74682/texk/web2c/luatexdir/unilib/ustring.c
@@ -396,7 +396,7 @@
     char buf[60], *pt, *ret;
     const unichar_t *upt;
     double val;
-    extern double strtod();		/* Please don't delete this, not all of us have good ansi headers */
+    /* extern double strtod(); */		/* Please don't delete this, not all of us have good ansi headers */
 
     for ( upt=str, pt=buf; *upt<128 && *upt!='\0' && pt-buf<sizeof(buf)-1; )
 	*pt++ = *upt++;
@@ -415,7 +415,7 @@
     char buf[60], *pt, *ret;
     const unichar_t *upt;
     long val;
-    extern long strtol();		/* Please don't delete this, not all of us have good ansi headers */
+    /* extern long strtol(); */		/* Please don't delete this, not all of us have good ansi headers */
 
     for ( upt=str, pt=buf; *upt<128 && *upt!='\0' && pt<buf+sizeof(buf)-1; )
 	*pt++ = *upt++;
