Index: texlive-bin-2025.20250325.74682/texk/web2c/ChangeLog
===================================================================
Index: texlive-bin-2025.20250325.74682/texk/web2c/ctangleboot.cin
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/ctangleboot.cin	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/ctangleboot.cin	(revision 74892)
@@ -2,7 +2,7 @@
 #line 66 "cwebdir/ctangle.w"
 
 /*5:*/
-#line 54 "cwebdir/comm-w2c.h"
+#line 53 "cwebdir/comm-w2c.h"
 
 #include <ctype.h>  
 #include <stdbool.h>  
@@ -15,15 +15,15 @@
 #ifndef HAVE_GETTEXT
 #define HAVE_GETTEXT 0
 #endif
-#line 66 "cwebdir/comm-w2c.h"
+#line 65 "cwebdir/comm-w2c.h"
 
 #if HAVE_GETTEXT
 #include <libintl.h> 
 #else
-#line 70 "cwebdir/comm-w2c.h"
+#line 69 "cwebdir/comm-w2c.h"
 #define gettext(a) a
 #endif
-#line 72 "cwebdir/comm-w2c.h"
+#line 71 "cwebdir/comm-w2c.h"
 
 /*:5*/
 #line 67 "cwebdir/ctangle.w"
@@ -142,7 +142,6 @@
 /*3:*/
 #line 32 "cwebdir/comm-w2c.h"
 
-typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
 typedef enum{
@@ -152,7 +151,7 @@
 extern int phase;
 
 /*:3*//*6:*/
-#line 94 "cwebdir/comm-w2c.h"
+#line 93 "cwebdir/comm-w2c.h"
 
 extern char section_text[];
 extern char*section_text_end;
@@ -160,7 +159,7 @@
 extern char*id_loc;
 
 /*:6*//*7:*/
-#line 112 "cwebdir/comm-w2c.h"
+#line 111 "cwebdir/comm-w2c.h"
 
 extern char buffer[];
 extern char*buffer_end;
@@ -168,7 +167,7 @@
 extern char*limit;
 
 /*:7*//*8:*/
-#line 129 "cwebdir/comm-w2c.h"
+#line 128 "cwebdir/comm-w2c.h"
 
 extern int include_depth;
 extern FILE*file[];
@@ -180,20 +179,20 @@
 extern int line[];
 extern int change_line;
 extern int change_depth;
-extern boolean input_has_ended;
-extern boolean changing;
-extern boolean web_file_open;
+extern bool input_has_ended;
+extern bool changing;
+extern bool web_file_open;
 
 /*:8*//*10:*/
-#line 150 "cwebdir/comm-w2c.h"
+#line 149 "cwebdir/comm-w2c.h"
 
 extern sixteen_bits section_count;
-extern boolean changed_section[];
-extern boolean change_pending;
-extern boolean print_where;
+extern bool changed_section[];
+extern bool change_pending;
+extern bool print_where;
 
 /*:10*//*11:*/
-#line 165 "cwebdir/comm-w2c.h"
+#line 164 "cwebdir/comm-w2c.h"
 
 typedef struct name_info{
 char*byte_start;
@@ -218,12 +217,12 @@
 extern hash_pointer hash_ptr;
 
 /*:11*//*13:*/
-#line 210 "cwebdir/comm-w2c.h"
+#line 209 "cwebdir/comm-w2c.h"
 
 extern int history;
 
 /*:13*//*15:*/
-#line 231 "cwebdir/comm-w2c.h"
+#line 230 "cwebdir/comm-w2c.h"
 
 extern int argc;
 extern char**argv;
@@ -232,11 +231,11 @@
 extern char idx_file_name[];
 extern char scn_file_name[];
 extern char check_file_name[];
-extern boolean flags[];
+extern bool flags[];
 extern const char*use_language;
 
 /*:15*//*16:*/
-#line 247 "cwebdir/comm-w2c.h"
+#line 246 "cwebdir/comm-w2c.h"
 
 extern FILE*C_file;
 extern FILE*tex_file;
@@ -308,7 +307,7 @@
 #line 457 "cwebdir/ctangle.w"
 
 static eight_bits out_state;
-static boolean protect;
+static bool protect;
 
 /*:42*//*45:*/
 #line 488 "cwebdir/ctangle.w"
@@ -321,7 +320,7 @@
 /*:45*//*53:*/
 #line 584 "cwebdir/ctangle.w"
 
-static boolean output_defs_seen= false;
+static bool output_defs_seen= false;
 
 /*:53*//*57:*/
 #line 694 "cwebdir/ctangle.w"
@@ -336,13 +335,13 @@
 /*:62*//*66:*/
 #line 827 "cwebdir/ctangle.w"
 
-static boolean comment_continues= false;
+static bool comment_continues= false;
 
 /*:66*//*68:*/
 #line 864 "cwebdir/ctangle.w"
 
 static name_pointer cur_section_name;
-static boolean no_where;
+static bool no_where;
 
 /*:68*//*82:*/
 #line 1182 "cwebdir/ctangle.w"
@@ -354,33 +353,33 @@
 #line 71 "cwebdir/ctangle.w"
 
 /*4:*/
-#line 43 "cwebdir/comm-w2c.h"
+#line 42 "cwebdir/comm-w2c.h"
 
 extern void common_init(void);
 
 /*:4*//*9:*/
-#line 144 "cwebdir/comm-w2c.h"
+#line 143 "cwebdir/comm-w2c.h"
 
-extern boolean get_line(void);
+extern bool get_line(void);
 extern void check_complete(void);
 extern void reset_input(void);
 
 /*:9*//*12:*/
-#line 188 "cwebdir/comm-w2c.h"
+#line 187 "cwebdir/comm-w2c.h"
 
 extern name_pointer id_lookup(const char*,const char*,eight_bits);
 
-extern name_pointer section_lookup(char*,char*,boolean);
+extern name_pointer section_lookup(char*,char*,bool);
 extern void print_prefix_name(name_pointer);
 extern void print_section_name(name_pointer);
 extern void sprint_section_name(char*,name_pointer);
 
-extern boolean names_match(name_pointer,const char*,size_t,eight_bits);
+extern bool names_match(name_pointer,const char*,size_t,eight_bits);
 
 extern void init_node(name_pointer);
 
 /*:12*//*14:*/
-#line 213 "cwebdir/comm-w2c.h"
+#line 212 "cwebdir/comm-w2c.h"
 
 extern int wrap_up(void);
 extern void err_print(const char*);
@@ -399,7 +398,7 @@
 #line 347 "cwebdir/ctangle.w"
 
 static void push_level(name_pointer);
-static void pop_level(boolean);
+static void pop_level(bool);
 static void get_output(void);
 
 /*:37*//*44:*/
@@ -417,7 +416,7 @@
 #line 807 "cwebdir/ctangle.w"
 
 static eight_bits skip_ahead(void);
-static boolean skip_comment(boolean);
+static bool skip_comment(bool);
 
 /*:65*//*70:*/
 #line 915 "cwebdir/ctangle.w"
@@ -523,7 +522,7 @@
 /*:2*//*24:*/
 #line 155 "cwebdir/ctangle.w"
 
-boolean names_match(
+bool names_match(
 name_pointer p,
 const char*first,
 size_t l,
@@ -580,7 +579,7 @@
 
 static void
 pop_level(
-boolean flag)
+bool flag)
 {
 if(flag&&cur_repl->text_link<section_flag){
 cur_repl= cur_repl->text_link+text_info;
@@ -722,7 +721,7 @@
 /*107:*/
 #line 431 "cwebdir/ctang-w2c.ch"
 
-boolean comparison= false;
+bool comparison= false;
 
 if((check_file= fopen(check_file_name,"r"))==NULL)
 fatal(_("! Cannot open output file "),check_file_name);
@@ -828,7 +827,7 @@
 
 char in_buf[BUFSIZ+1];
 int in_size;
-boolean comparison= true;
+bool comparison= true;
 if((check_file= fopen(check_file_name,"r"))==NULL)
 fatal(_("! Cannot open output file "),check_file_name);
 
@@ -869,7 +868,7 @@
 
 char in_buf[BUFSIZ+1];
 int in_size;
-boolean comparison= true;
+bool comparison= true;
 if((check_file= fopen(check_file_name,"r"))==NULL)
 fatal(_("! Cannot open output file "),check_file_name);
 
@@ -905,7 +904,7 @@
 /*114:*/
 #line 522 "cwebdir/ctang-w2c.ch"
 {
-boolean comparison= true;
+bool comparison= true;
 /*111:*/
 #line 486 "cwebdir/ctang-w2c.ch"
 
@@ -929,7 +928,7 @@
 /*107:*/
 #line 431 "cwebdir/ctang-w2c.ch"
 
-boolean comparison= false;
+bool comparison= false;
 
 if((check_file= fopen(check_file_name,"r"))==NULL)
 fatal(_("! Cannot open output file "),check_file_name);
@@ -1166,8 +1165,8 @@
 /*:64*//*67:*/
 #line 830 "cwebdir/ctangle.w"
 
-static boolean skip_comment(
-boolean is_long_comment)
+static bool skip_comment(
+bool is_long_comment)
 {
 char c;
 while(true){
@@ -1207,7 +1206,7 @@
 static eight_bits
 get_next(void)
 {
-static boolean preprocessing= false;
+static bool preprocessing= false;
 eight_bits c;
 while(true){
 if(loc> limit){
@@ -1248,7 +1247,7 @@
 if(xisdigit(c)||c=='.')/*73:*/
 #line 956 "cwebdir/ctangle.w"
 {
-boolean hex_flag= false;
+bool hex_flag= false;
 id_first= loc-1;
 if(*id_first=='.'&&!xisdigit(*loc))goto mistake;
 if(*id_first=='0'){
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebboot.cin
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebboot.cin	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebboot.cin	(revision 74892)
@@ -2,7 +2,7 @@
 #line 62 "cwebdir/common.w"
 
 /*4:*/
-#line 54 "cwebdir/comm-w2c.h"
+#line 53 "cwebdir/comm-w2c.h"
 
 #include <ctype.h>  
 #include <stdbool.h>  
@@ -15,15 +15,15 @@
 #ifndef HAVE_GETTEXT
 #define HAVE_GETTEXT 0
 #endif
-#line 66 "cwebdir/comm-w2c.h"
+#line 65 "cwebdir/comm-w2c.h"
 
 #if HAVE_GETTEXT
 #include <libintl.h> 
 #else
-#line 70 "cwebdir/comm-w2c.h"
+#line 69 "cwebdir/comm-w2c.h"
 #define gettext(a) a
 #endif
-#line 72 "cwebdir/comm-w2c.h"
+#line 71 "cwebdir/comm-w2c.h"
 
 /*:4*//*91:*/
 #line 739 "cwebdir/comm-w2c.ch"
@@ -41,8 +41,6 @@
 /*:91*//*93:*/
 #line 786 "cwebdir/comm-w2c.ch"
 
-typedef bool boolean;
-#define HAVE_BOOLEAN 1
 #include <kpathsea/kpathsea.h>  
 
 #include <w2c/config.h>  
@@ -49,7 +47,7 @@
 #include <lib/lib.h>  
 
 /*:93*//*96:*/
-#line 825 "cwebdir/comm-w2c.ch"
+#line 823 "cwebdir/comm-w2c.ch"
 
 #define CWEB
 #include "help.h" 
@@ -154,7 +152,6 @@
 /*2:*/
 #line 32 "cwebdir/comm-w2c.h"
 
-typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
 typedef enum{
@@ -164,7 +161,7 @@
 extern int phase;
 
 /*:2*//*5:*/
-#line 94 "cwebdir/comm-w2c.h"
+#line 93 "cwebdir/comm-w2c.h"
 
 extern char section_text[];
 extern char*section_text_end;
@@ -172,7 +169,7 @@
 extern char*id_loc;
 
 /*:5*//*6:*/
-#line 112 "cwebdir/comm-w2c.h"
+#line 111 "cwebdir/comm-w2c.h"
 
 extern char buffer[];
 extern char*buffer_end;
@@ -180,7 +177,7 @@
 extern char*limit;
 
 /*:6*//*7:*/
-#line 129 "cwebdir/comm-w2c.h"
+#line 128 "cwebdir/comm-w2c.h"
 
 extern int include_depth;
 extern FILE*file[];
@@ -192,20 +189,20 @@
 extern int line[];
 extern int change_line;
 extern int change_depth;
-extern boolean input_has_ended;
-extern boolean changing;
-extern boolean web_file_open;
+extern bool input_has_ended;
+extern bool changing;
+extern bool web_file_open;
 
 /*:7*//*9:*/
-#line 150 "cwebdir/comm-w2c.h"
+#line 149 "cwebdir/comm-w2c.h"
 
 extern sixteen_bits section_count;
-extern boolean changed_section[];
-extern boolean change_pending;
-extern boolean print_where;
+extern bool changed_section[];
+extern bool change_pending;
+extern bool print_where;
 
 /*:9*//*10:*/
-#line 165 "cwebdir/comm-w2c.h"
+#line 164 "cwebdir/comm-w2c.h"
 
 typedef struct name_info{
 char*byte_start;
@@ -230,12 +227,12 @@
 extern hash_pointer hash_ptr;
 
 /*:10*//*12:*/
-#line 210 "cwebdir/comm-w2c.h"
+#line 209 "cwebdir/comm-w2c.h"
 
 extern int history;
 
 /*:12*//*14:*/
-#line 231 "cwebdir/comm-w2c.h"
+#line 230 "cwebdir/comm-w2c.h"
 
 extern int argc;
 extern char**argv;
@@ -244,11 +241,11 @@
 extern char idx_file_name[];
 extern char scn_file_name[];
 extern char check_file_name[];
-extern boolean flags[];
+extern bool flags[];
 extern const char*use_language;
 
 /*:14*//*15:*/
-#line 247 "cwebdir/comm-w2c.h"
+#line 246 "cwebdir/comm-w2c.h"
 
 extern FILE*C_file;
 extern FILE*tex_file;
@@ -303,9 +300,9 @@
 int line[max_include_depth];
 int change_line;
 int change_depth;
-boolean input_has_ended;
-boolean changing;
-boolean web_file_open= false;
+bool input_has_ended;
+bool changing;
+bool web_file_open= false;
 
 /*:25*//*26:*/
 #line 216 "cwebdir/common.w"
@@ -317,10 +314,10 @@
 #line 545 "cwebdir/common.w"
 
 sixteen_bits section_count;
-boolean changed_section[max_sections];
-boolean change_pending;
+bool changed_section[max_sections];
+bool change_pending;
 
-boolean print_where= false;
+bool print_where= false;
 
 /*:42*//*43:*/
 #line 590 "cwebdir/common.w"
@@ -360,7 +357,7 @@
 char scn_file_name[max_file_name_length];
 char check_file_name[max_file_name_length];
 #line 1131 "cwebdir/common.w"
-boolean flags[128];
+bool flags[128];
 
 /*:73*//*83:*/
 #line 1267 "cwebdir/common.w"
@@ -396,33 +393,33 @@
 #line 66 "cwebdir/common.w"
 
 /*3:*/
-#line 43 "cwebdir/comm-w2c.h"
+#line 42 "cwebdir/comm-w2c.h"
 
 extern void common_init(void);
 
 /*:3*//*8:*/
-#line 144 "cwebdir/comm-w2c.h"
+#line 143 "cwebdir/comm-w2c.h"
 
-extern boolean get_line(void);
+extern bool get_line(void);
 extern void check_complete(void);
 extern void reset_input(void);
 
 /*:8*//*11:*/
-#line 188 "cwebdir/comm-w2c.h"
+#line 187 "cwebdir/comm-w2c.h"
 
 extern name_pointer id_lookup(const char*,const char*,eight_bits);
 
-extern name_pointer section_lookup(char*,char*,boolean);
+extern name_pointer section_lookup(char*,char*,bool);
 extern void print_prefix_name(name_pointer);
 extern void print_section_name(name_pointer);
 extern void sprint_section_name(char*,name_pointer);
 
-extern boolean names_match(name_pointer,const char*,size_t,eight_bits);
+extern bool names_match(name_pointer,const char*,size_t,eight_bits);
 
 extern void init_node(name_pointer);
 
 /*:11*//*13:*/
-#line 213 "cwebdir/comm-w2c.h"
+#line 212 "cwebdir/comm-w2c.h"
 
 extern int wrap_up(void);
 extern void err_print(const char*);
@@ -435,7 +432,7 @@
 
 /*:13*//*24:*/
 #line 176 "cwebdir/common.w"
-static boolean input_ln(FILE*);
+static bool input_ln(FILE*);
 
 /*:24*//*28:*/
 #line 237 "cwebdir/common.w"
@@ -449,8 +446,8 @@
 #line 760 "cwebdir/common.w"
 
 static int web_strcmp(char*,size_t,char*,size_t);
-static name_pointer add_section_name(name_pointer,int,char*,char*,boolean);
-static void extend_section_name(name_pointer,char*,char*,boolean);
+static name_pointer add_section_name(name_pointer,int,char*,char*,bool);
+static void extend_section_name(name_pointer,char*,char*,bool);
 
 /*:55*//*64:*/
 #line 987 "cwebdir/common.w"
@@ -462,7 +459,7 @@
 
 #line 514 "cwebdir/comm-w2c.ch"
 /*:76*//*98:*/
-#line 838 "cwebdir/comm-w2c.ch"
+#line 836 "cwebdir/comm-w2c.ch"
 
 static void cb_usage(const_string str);
 static void cb_usagehelp(const_string*message);
@@ -490,7 +487,7 @@
 #line 78 "cwebdir/comm-w2c.ch"
 
 /*94:*/
-#line 800 "cwebdir/comm-w2c.ch"
+#line 798 "cwebdir/comm-w2c.ch"
 
 kpse_set_program_name(argv[0],"cweb");
 
@@ -592,7 +589,7 @@
 /*:20*//*23:*/
 #line 156 "cwebdir/common.w"
 
-static boolean input_ln(
+static bool input_ln(
 FILE*fp)
 {
 int c= EOF;
@@ -750,7 +747,7 @@
 /*:32*//*35:*/
 #line 362 "cwebdir/common.w"
 
-boolean get_line(void)
+bool get_line(void)
 {
 restart:
 if(changing&&include_depth==change_depth)
@@ -1081,7 +1078,7 @@
 int c,
 char*first,
 char*last,
-boolean ispref)
+bool ispref)
 {
 name_pointer p= name_ptr;
 char*s= first_chunk(p);
@@ -1112,7 +1109,7 @@
 name_pointer p,
 char*first,
 char*last,
-boolean ispref)
+bool ispref)
 {
 char*s;
 name_pointer q= p+1;
@@ -1138,7 +1135,7 @@
 name_pointer
 section_lookup(
 char*first,char*last,
-boolean ispref)
+bool ispref)
 {
 int c= less;
 name_pointer p= root;
@@ -1249,7 +1246,7 @@
 name_pointer q= r+1;
 char*ss,*s= first_chunk(r);
 int c= less;
-boolean ispref;
+bool ispref;
 while(true){
 ss= (r+1)->byte_start-1;
 if(*ss==' '&&ss>=r->byte_start)ispref= true,q= q->link;
@@ -1391,7 +1388,7 @@
 char*dot_pos;
 char*name_pos;
 char*s;
-boolean found_web= false,found_change= false,found_out= false;
+bool found_web= false,found_change= false,found_out= false;
 
 
 #line 486 "cwebdir/comm-w2c.ch"
@@ -1416,7 +1413,7 @@
 if(strcmp("-help",*argv)==0||strcmp("--help",*argv)==0)
 
 /*97:*/
-#line 829 "cwebdir/comm-w2c.ch"
+#line 827 "cwebdir/comm-w2c.ch"
 
 cb_usagehelp(program==ctangle?CTANGLEHELP:
 program==cweave?CWEAVEHELP:CTWILLHELP);
@@ -1428,7 +1425,7 @@
 if(strcmp("-version",*argv)==0||strcmp("--version",*argv)==0)
 
 /*100:*/
-#line 875 "cwebdir/comm-w2c.ch"
+#line 873 "cwebdir/comm-w2c.ch"
 
 printversionandexit(cb_banner,
 program==ctwill?"Donald E. Knuth":"Silvio Levy and Donald E. Knuth",
@@ -1602,7 +1599,7 @@
 }
 
 /*:75*//*99:*/
-#line 842 "cwebdir/comm-w2c.ch"
+#line 840 "cwebdir/comm-w2c.ch"
 
 static void cb_usage(const_string str)
 {
@@ -1634,7 +1631,7 @@
 }
 
 /*:99*//*101:*/
-#line 883 "cwebdir/comm-w2c.ch"
+#line 881 "cwebdir/comm-w2c.ch"
 
 void cb_show_banner(void)
 {
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ChangeLog
===================================================================
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-bs.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-bs.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-bs.ch	(revision 74892)
@@ -141,7 +141,7 @@
 int c, /* right or left? */
 char *first, /* first character of section name */
 char *last, /* last character of section name, plus one */
-boolean ispref) /* are we adding a prefix or a full name? */
+bool ispref) /* are we adding a prefix or a full name? */
 {
   name_pointer p=name_ptr; /* new node */
   char *s=first_chunk(p);
@@ -152,7 +152,7 @@
 int c, /* right or left? */
 char huge* first, /* first character of section name */
 char huge* last, /* last character of section name, plus one */
-boolean ispref) /* are we adding a prefix or a full name? */
+bool ispref) /* are we adding a prefix or a full name? */
 {
   name_pointer p=name_ptr; /* new node */
   char huge* s=first_chunk(p);
@@ -165,7 +165,7 @@
 name_pointer p, /* name to be extended */
 char *first, /* beginning of extension text */
 char *last, /* one beyond end of extension text */
-boolean ispref) /* are we adding a prefix or a full name? */
+bool ispref) /* are we adding a prefix or a full name? */
 {
   char *s;
 @y
@@ -174,7 +174,7 @@
 name_pointer p, /* name to be extended */
 char huge* first, /* beginning of extension text */
 char huge* last, /* one beyond end of extension text */
-boolean ispref) /* are we adding a prefix or a full name? */
+bool ispref) /* are we adding a prefix or a full name? */
 {
   char huge* s;
 @z
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-mac.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-mac.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-mac.ch	(revision 74892)
@@ -15,7 +15,7 @@
 @^system dependencies@>
 
 @c
-static boolean input_ln( /* copies a line into |buffer| or returns |false| */
+static bool input_ln( /* copies a line into |buffer| or returns |false| */
 FILE *fp) /* what file to read from */
 {
   int c=EOF; /* character read; initialized so some compilers won't complain */
@@ -43,7 +43,7 @@
 @^system dependencies@>
 
 @c
-static boolean input_ln( /* copies a line into |buffer| or returns |false| */
+static bool input_ln( /* copies a line into |buffer| or returns |false| */
 FILE *fp) /* what file to read from */
 {
   int  c=EOF; /* character read; initialized so some compilers won't complain */
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-mini.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-mini.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-mini.ch	(revision 74892)
@@ -58,11 +58,11 @@
 Section 24.
 
 @x
-@ @<Predecl...@>=@+static boolean input_ln(FILE *);
+@ @<Predecl...@>=@+static bool input_ln(FILE *);
 @y
-@ @<Predecl...@>=@+static boolean input_ln(FILE *);
+@ @<Predecl...@>=@+static bool input_ln(FILE *);
 @-input_ln@>
-@$input_ln {COMMON}23 \&{static} \&{boolean} (\,)@>
+@$input_ln {COMMON}23 \&{static} \&{bool} (\,)@>
 @z
 
 ẞection 28.
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.ch	(revision 74892)
@@ -67,7 +67,7 @@
 @z
 
 @x [1.18] l.80
-boolean program; /* \.{CWEAVE} or \.{CTANGLE}? */
+bool program; /* \.{CWEAVE} or \.{CTANGLE}? */
 @y
 cweb program; /* \.{CTANGLE} or \.{CWEAVE} or \.{CTWILL}? */
 @z
@@ -784,8 +784,6 @@
 @d kpse_find_cweb(name) kpse_find_file(name,kpse_cweb_format,true)
 
 @<Include files@>=
-typedef bool boolean;
-#define HAVE_BOOLEAN 1
 #include <kpathsea/kpathsea.h> /* include every \Kpathsea/ header;
   |@!kpathsea_debug|, |@!const_string|, |@!string| */
 #include <w2c/config.h> /* \&{integer} */
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.h
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.h	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.h	(revision 74892)
@@ -30,7 +30,6 @@
 @i iso_types.w
 
 @<Common code...@>=@^system dependencies@>
-typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
 typedef enum {
@@ -137,12 +136,12 @@
 extern int line[]; /* number of current line in the stacked files */
 extern int change_line; /* number of current line in change file */
 extern int change_depth; /* where \.{@@y} originated during a change */
-extern boolean input_has_ended; /* if there is no more input */
-extern boolean changing; /* if the current line is from |change_file| */
-extern boolean web_file_open; /* if the web file is being read */
+extern bool input_has_ended; /* if there is no more input */
+extern bool changing; /* if the current line is from |change_file| */
+extern bool web_file_open; /* if the web file is being read */
 
 @ @<Predecl...@>=
-extern boolean get_line(void); /* inputs the next line */
+extern bool get_line(void); /* inputs the next line */
 extern void check_complete(void); /* checks that all changes were picked up */
 extern void reset_input(void); /* initialize to read the web file and change file */
 
@@ -149,9 +148,9 @@
 @ Code related to section numbers:
 @<Common code...@>=
 extern sixteen_bits section_count; /* the current section number */
-extern boolean changed_section[]; /* is the section changed? */
-extern boolean change_pending; /* is a decision about change still unclear? */
-extern boolean print_where; /* tells \.{CTANGLE} to print line and file info */
+extern bool changed_section[]; /* is the section changed? */
+extern bool change_pending; /* is a decision about change still unclear? */
+extern bool print_where; /* tells \.{CTANGLE} to print line and file info */
 
 @ Code related to identifier and section name storage:
 @d length(c) (size_t)((c+1)->byte_start-(c)->byte_start) /* the length of a name */
@@ -188,12 +187,12 @@
 @ @<Predecl...@>=
 extern name_pointer id_lookup(const char *,const char *,eight_bits);
    /* looks up a string in the identifier table */
-extern name_pointer section_lookup(char *,char *,boolean); /* finds section name */
+extern name_pointer section_lookup(char *,char *,bool); /* finds section name */
 extern void print_prefix_name(name_pointer);@/
 extern void print_section_name(name_pointer);@/
 extern void sprint_section_name(char *,name_pointer);
 @#
-extern boolean names_match(name_pointer,const char *,size_t,eight_bits);
+extern bool names_match(name_pointer,const char *,size_t,eight_bits);
 /* two routines defined in \.{ctangle.w} and \.{cweave.w} */
 extern void init_node(name_pointer);
 
@@ -236,7 +235,7 @@
 extern char idx_file_name[]; /* name of |idx_file| */
 extern char scn_file_name[]; /* name of |scn_file| */
 extern char check_file_name[]; /* name of |check_file| */
-extern boolean flags[]; /* an option for each 7-bit code */
+extern bool flags[]; /* an option for each 7-bit code */
 extern const char *use_language; /* prefix to \.{cwebmac.tex} in \TEX/ output */
 
 @ Code related to output:
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w32.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w32.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w32.ch	(revision 74892)
@@ -35,9 +35,9 @@
 set of programs using the __fastcall convention.
 
 @x section 11
-extern boolean names_match(name_pointer,const char *,size_t,eight_bits);
+extern bool names_match(name_pointer,const char *,size_t,eight_bits);
 @y
-extern boolean __cdecl names_match(name_pointer,const char *,size_t,eight_bits);
+extern bool __cdecl names_match(name_pointer,const char *,size_t,eight_bits);
 @z
 
 @x section 75
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.bux
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.bux	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.bux	(revision 74892)
@@ -17,7 +17,7 @@
 @$xref {CWEAVE.W}24 \\{equiv\_or\_xref}@>
 
 @$init_node {CWEAVE.W}32 \&{void} (\,)@>
-@$names_match {CWEAVE.W}32 \&{boolean} (\,)@>
+@$names_match {CWEAVE.W}32 \&{bool} (\,)@>
 
 @$print_stats {CWEAVE.W}270 \&{void} (\,)@>
 
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.c
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.c	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.c	(revision 74892)
@@ -2,7 +2,7 @@
 #line 62 "common.w"
 
 /*4:*/
-#line 48 "common.h"
+#line 47 "common.h"
 
 #include <ctype.h>  
 #include <stdbool.h>  
@@ -103,14 +103,13 @@
 /*2:*/
 #line 35 "common.h"
 
-typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
-extern boolean program;
+extern bool program;
 extern int phase;
 
 /*:2*//*5:*/
-#line 78 "common.h"
+#line 77 "common.h"
 
 extern char section_text[];
 extern char*section_text_end;
@@ -118,7 +117,7 @@
 extern char*id_loc;
 
 /*:5*//*6:*/
-#line 96 "common.h"
+#line 95 "common.h"
 
 extern char buffer[];
 extern char*buffer_end;
@@ -126,7 +125,7 @@
 extern char*limit;
 
 /*:6*//*7:*/
-#line 113 "common.h"
+#line 112 "common.h"
 
 extern int include_depth;
 extern FILE*file[];
@@ -137,20 +136,20 @@
 extern int line[];
 extern int change_line;
 extern int change_depth;
-extern boolean input_has_ended;
-extern boolean changing;
-extern boolean web_file_open;
+extern bool input_has_ended;
+extern bool changing;
+extern bool web_file_open;
 
 /*:7*//*9:*/
-#line 133 "common.h"
+#line 132 "common.h"
 
 extern sixteen_bits section_count;
-extern boolean changed_section[];
-extern boolean change_pending;
-extern boolean print_where;
+extern bool changed_section[];
+extern bool change_pending;
+extern bool print_where;
 
 /*:9*//*10:*/
-#line 148 "common.h"
+#line 147 "common.h"
 
 typedef struct name_info{
 char*byte_start;
@@ -175,12 +174,12 @@
 extern hash_pointer hash_ptr;
 
 /*:10*//*12:*/
-#line 193 "common.h"
+#line 192 "common.h"
 
 extern int history;
 
 /*:12*//*14:*/
-#line 211 "common.h"
+#line 210 "common.h"
 
 extern int argc;
 extern char**argv;
@@ -188,10 +187,10 @@
 extern char tex_file_name[];
 extern char idx_file_name[];
 extern char scn_file_name[];
-extern boolean flags[];
+extern bool flags[];
 
 /*:14*//*15:*/
-#line 225 "common.h"
+#line 224 "common.h"
 
 extern FILE*C_file;
 extern FILE*tex_file;
@@ -205,7 +204,7 @@
 /*18:*/
 #line 79 "common.w"
 
-boolean program;
+bool program;
 
 /*:18*//*19:*/
 #line 88 "common.w"
@@ -241,9 +240,9 @@
 int line[max_include_depth];
 int change_line;
 int change_depth;
-boolean input_has_ended;
-boolean changing;
-boolean web_file_open= false;
+bool input_has_ended;
+bool changing;
+bool web_file_open= false;
 
 /*:25*//*26:*/
 #line 216 "common.w"
@@ -255,10 +254,10 @@
 #line 545 "common.w"
 
 sixteen_bits section_count;
-boolean changed_section[max_sections];
-boolean change_pending;
+bool changed_section[max_sections];
+bool change_pending;
 
-boolean print_where= false;
+bool print_where= false;
 
 /*:42*//*43:*/
 #line 590 "common.w"
@@ -295,7 +294,7 @@
 char tex_file_name[max_file_name_length];
 char idx_file_name[max_file_name_length];
 char scn_file_name[max_file_name_length];
-boolean flags[128];
+bool flags[128];
 
 /*:73*//*83:*/
 #line 1267 "common.w"
@@ -310,33 +309,33 @@
 #line 66 "common.w"
 
 /*3:*/
-#line 43 "common.h"
+#line 42 "common.h"
 
 extern void common_init(void);
 
 /*:3*//*8:*/
-#line 127 "common.h"
+#line 126 "common.h"
 
-extern boolean get_line(void);
+extern bool get_line(void);
 extern void check_complete(void);
 extern void reset_input(void);
 
 /*:8*//*11:*/
-#line 171 "common.h"
+#line 170 "common.h"
 
 extern name_pointer id_lookup(const char*,const char*,eight_bits);
 
-extern name_pointer section_lookup(char*,char*,boolean);
+extern name_pointer section_lookup(char*,char*,bool);
 extern void print_prefix_name(name_pointer);
 extern void print_section_name(name_pointer);
 extern void sprint_section_name(char*,name_pointer);
 
-extern boolean names_match(name_pointer,const char*,size_t,eight_bits);
+extern bool names_match(name_pointer,const char*,size_t,eight_bits);
 
 extern void init_node(name_pointer);
 
 /*:11*//*13:*/
-#line 196 "common.h"
+#line 195 "common.h"
 
 extern int wrap_up(void);
 extern void err_print(const char*);
@@ -347,7 +346,7 @@
 
 /*:13*//*24:*/
 #line 176 "common.w"
-static boolean input_ln(FILE*);
+static bool input_ln(FILE*);
 
 /*:24*//*28:*/
 #line 237 "common.w"
@@ -361,8 +360,8 @@
 #line 760 "common.w"
 
 static int web_strcmp(char*,size_t,char*,size_t);
-static name_pointer add_section_name(name_pointer,int,char*,char*,boolean);
-static void extend_section_name(name_pointer,char*,char*,boolean);
+static name_pointer add_section_name(name_pointer,int,char*,char*,bool);
+static void extend_section_name(name_pointer,char*,char*,bool);
 
 /*:55*//*64:*/
 #line 987 "common.w"
@@ -423,7 +422,7 @@
 /*:20*//*23:*/
 #line 156 "common.w"
 
-static boolean input_ln(
+static bool input_ln(
 FILE*fp)
 {
 int c= EOF;
@@ -565,7 +564,7 @@
 /*:32*//*35:*/
 #line 362 "common.w"
 
-boolean get_line(void)
+bool get_line(void)
 {
 restart:
 if(changing&&include_depth==change_depth)
@@ -878,7 +877,7 @@
 int c,
 char*first,
 char*last,
-boolean ispref)
+bool ispref)
 {
 name_pointer p= name_ptr;
 char*s= first_chunk(p);
@@ -907,7 +906,7 @@
 name_pointer p,
 char*first,
 char*last,
-boolean ispref)
+bool ispref)
 {
 char*s;
 name_pointer q= p+1;
@@ -929,7 +928,7 @@
 name_pointer
 section_lookup(
 char*first,char*last,
-boolean ispref)
+bool ispref)
 {
 int c= less;
 name_pointer p= root;
@@ -1028,7 +1027,7 @@
 name_pointer q= r+1;
 char*ss,*s= first_chunk(r);
 int c= less;
-boolean ispref;
+bool ispref;
 while(true){
 ss= (r+1)->byte_start-1;
 if(*ss==' '&&ss>=r->byte_start)ispref= true,q= q->link;
@@ -1145,7 +1144,7 @@
 char*dot_pos;
 char*name_pos;
 char*s;
-boolean found_web= false,found_change= false,found_out= false;
+bool found_web= false,found_change= false,found_out= false;
 
 
 strcpy(change_file_name,"/dev/null");
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.h
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.h	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.h	(revision 74892)
@@ -33,10 +33,9 @@
 @d cweave true
 
 @<Common code...@>=@^system dependencies@>
-typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
-extern boolean program; /* \.{CWEAVE} or \.{CTANGLE}? */
+extern bool program; /* \.{CWEAVE} or \.{CTANGLE}? */
 extern int phase; /* which phase are we in? */
 
 @ The procedure that gets everything rolling:
@@ -120,12 +119,12 @@
 extern int line[]; /* number of current line in the stacked files */
 extern int change_line; /* number of current line in change file */
 extern int change_depth; /* where \.{@@y} originated during a change */
-extern boolean input_has_ended; /* if there is no more input */
-extern boolean changing; /* if the current line is from |change_file| */
-extern boolean web_file_open; /* if the web file is being read */
+extern bool input_has_ended; /* if there is no more input */
+extern bool changing; /* if the current line is from |change_file| */
+extern bool web_file_open; /* if the web file is being read */
 
 @ @<Predecl...@>=
-extern boolean get_line(void); /* inputs the next line */
+extern bool get_line(void); /* inputs the next line */
 extern void check_complete(void); /* checks that all changes were picked up */
 extern void reset_input(void); /* initialize to read the web file and change file */
 
@@ -132,9 +131,9 @@
 @ Code related to section numbers:
 @<Common code...@>=
 extern sixteen_bits section_count; /* the current section number */
-extern boolean changed_section[]; /* is the section changed? */
-extern boolean change_pending; /* is a decision about change still unclear? */
-extern boolean print_where; /* tells \.{CTANGLE} to print line and file info */
+extern bool changed_section[]; /* is the section changed? */
+extern bool change_pending; /* is a decision about change still unclear? */
+extern bool print_where; /* tells \.{CTANGLE} to print line and file info */
 
 @ Code related to identifier and section name storage:
 @d length(c) (size_t)((c+1)->byte_start-(c)->byte_start) /* the length of a name */
@@ -171,12 +170,12 @@
 @ @<Predecl...@>=
 extern name_pointer id_lookup(const char *,const char *,eight_bits);
    /* looks up a string in the identifier table */
-extern name_pointer section_lookup(char *,char *,boolean); /* finds section name */
+extern name_pointer section_lookup(char *,char *,bool); /* finds section name */
 extern void print_prefix_name(name_pointer);@/
 extern void print_section_name(name_pointer);@/
 extern void sprint_section_name(char *,name_pointer);
 @#
-extern boolean names_match(name_pointer,const char *,size_t,eight_bits);
+extern bool names_match(name_pointer,const char *,size_t,eight_bits);
 /* two routines defined in \.{ctangle.w} and \.{cweave.w} */
 extern void init_node(name_pointer);
 
@@ -215,7 +214,7 @@
 extern char tex_file_name[]; /* name of |tex_file| */
 extern char idx_file_name[]; /* name of |idx_file| */
 extern char scn_file_name[]; /* name of |scn_file| */
-extern boolean flags[]; /* an option for each 7-bit code */
+extern bool flags[]; /* an option for each 7-bit code */
 
 @ Code related to output:
 @d update_terminal() fflush(stdout) /* empty the terminal output buffer */
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.w
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.w	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/common.w	(revision 74892)
@@ -77,7 +77,7 @@
 |program|.
 
 @<Global var...@>=
-boolean program; /* \.{CWEAVE} or \.{CTANGLE}? */
+bool program; /* \.{CWEAVE} or \.{CTANGLE}? */
 
 @ \.{CWEAVE} operates in three phases: First it inputs the source
 file and stores cross-reference data, then it inputs the source once again and
@@ -154,7 +154,7 @@
 @^system dependencies@>
 
 @c
-static boolean input_ln( /* copies a line into |buffer| or returns |false| */
+static bool input_ln( /* copies a line into |buffer| or returns |false| */
 FILE *fp) /* what file to read from */
 {
   int c=EOF; /* character read; initialized so some compilers won't complain */
@@ -173,7 +173,7 @@
   return true;
 }
 
-@ @<Predecl...@>=@+static boolean input_ln(FILE *);
+@ @<Predecl...@>=@+static bool input_ln(FILE *);
 
 @* File handling. Now comes the problem of deciding which file to read from
 next.  Recall that the actual text that \.{CWEB} should process comes from two
@@ -198,9 +198,9 @@
 int line[max_include_depth]; /* number of current line in the stacked files */
 int change_line; /* number of current line in change file */
 int change_depth; /* where \.{@@y} originated during a change */
-boolean input_has_ended; /* if there is no more input */
-boolean changing; /* if the current line is from |change_file| */
-boolean web_file_open=false; /* if the web file is being read */
+bool input_has_ended; /* if there is no more input */
+bool changing; /* if the current line is from |change_file| */
+bool web_file_open=false; /* if the web file is being read */
 
 @ When |changing==false|, the next line of |change_file| is kept in
 |change_buffer|, for purposes of comparison with the next
@@ -360,7 +360,7 @@
 information in the \CEE/ file by means of the |print_where| flag.
 
 @c
-boolean get_line(void) /* inputs the next line */
+bool get_line(void) /* inputs the next line */
 {
   restart:
   if (changing && include_depth==change_depth)
@@ -544,10 +544,10 @@
 
 @ @<Global var...@>=
 sixteen_bits section_count; /* the current section number */
-boolean changed_section[max_sections]; /* is the section changed? */
-boolean change_pending; /* if the current change is not yet recorded in
+bool changed_section[max_sections]; /* is the section changed? */
+bool change_pending; /* if the current change is not yet recorded in
   |changed_section[section_count]| */
-boolean print_where=false; /* should \.{CTANGLE} print line and file info? */
+bool print_where=false; /* should \.{CTANGLE} print line and file info? */
 
 @* Storage of names and strings.
 Both \.{CWEAVE} and \.{CTANGLE} store identifiers, section names and
@@ -759,8 +759,8 @@
 
 @<Predecl...@>=
 static int web_strcmp(char *,size_t,char *,size_t);@/
-static name_pointer add_section_name(name_pointer,int,char *,char *,boolean);@/
-static void extend_section_name(name_pointer,char *,char *,boolean);
+static name_pointer add_section_name(name_pointer,int,char *,char *,bool);@/
+static void extend_section_name(name_pointer,char *,char *,bool);
 
 @ @c
 static int web_strcmp( /* fuller comparison than |strcmp| */
@@ -798,7 +798,7 @@
 int c, /* right or left? */
 char *first, /* first character of section name */
 char *last, /* last character of section name, plus one */
-boolean ispref) /* are we adding a prefix or a full name? */
+bool ispref) /* are we adding a prefix or a full name? */
 {
   name_pointer p=name_ptr; /* new node */
   char *s=first_chunk(p);
@@ -825,7 +825,7 @@
 name_pointer p, /* name to be extended */
 char *first, /* beginning of extension text */
 char *last, /* one beyond end of extension text */
-boolean ispref) /* are we adding a prefix or a full name? */
+bool ispref) /* are we adding a prefix or a full name? */
 {
   char *s;
   name_pointer q=p+1;
@@ -851,7 +851,7 @@
 name_pointer
 section_lookup( /* find or install section name in tree */
 char *first,char *last, /* first and last characters of new name */
-boolean ispref) /* is the new name a prefix or a full name? */
+bool ispref) /* is the new name a prefix or a full name? */
 {
   int c=less; /* comparison between two names; initialized so some compilers won't complain */
   name_pointer p=root; /* current node of the search tree */
@@ -962,7 +962,7 @@
   name_pointer q=r+1; /* access to subsequent chunks */
   char *ss, *s=first_chunk(r);
   int c=less; /* comparison */
-  boolean ispref; /* is chunk |r| a prefix? */
+  bool ispref; /* is chunk |r| a prefix? */
   while (true) {
     ss=(r+1)->byte_start-1;
     if (*ss==' ' && ss>=r->byte_start) ispref=true,q=q->link;
@@ -1128,7 +1128,7 @@
 char tex_file_name[max_file_name_length]; /* name of |tex_file| */
 char idx_file_name[max_file_name_length]; /* name of |idx_file| */
 char scn_file_name[max_file_name_length]; /* name of |scn_file| */
-boolean flags[128]; /* an option for each 7-bit code */
+bool flags[128]; /* an option for each 7-bit code */
 
 @ The |flags| will be initially |false|. Some of them are set to~|true| before
 scanning the arguments; if additional flags are |true| by default they
@@ -1159,7 +1159,7 @@
   char *dot_pos; /* position of |'.'| in the argument */
   char *name_pos; /* file name beginning, sans directory */
   char *s; /* pointer for scanning strings */
-  boolean found_web=false,found_change=false,found_out=false;
+  bool found_web=false,found_change=false,found_out=false;
              /* have these names been seen? */
 
   strcpy(change_file_name,"/dev/null");
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctang-w2c.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctang-w2c.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctang-w2c.ch	(revision 74892)
@@ -429,7 +429,7 @@
   rename(check_file_name,C_file_name); /* This was the first run */
 
 @ @<Set up the comparison of temporary output@>=
-  boolean comparison=false;
+  bool comparison=false;
 
   if((check_file=fopen(check_file_name,"r"))==NULL)
     fatal(_("! Cannot open output file "),check_file_name);
@@ -520,7 +520,7 @@
 @ No copying necessary, just remove the temporary output file.
 
 @<Redirect temporary output to \.{/dev/null}@>={
-  boolean comparison=true;
+  bool comparison=true;
   @<Create the secondary output...@>@;
 }
 
@@ -527,7 +527,7 @@
 @ @<Setup system redirection@>=
 char in_buf[BUFSIZ+1];
 int in_size;
-boolean comparison=true;
+bool comparison=true;
 if((check_file=fopen(check_file_name,"r"))==NULL)
   fatal(_("! Cannot open output file "),check_file_name);
 @.Cannot open output file@>
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctang-w32.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctang-w32.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctang-w32.ch	(revision 74892)
@@ -10,13 +10,13 @@
 @z
 
 @x section 24
-boolean names_match(
+bool names_match(
 name_pointer p, /* points to the proposed match */
 const char *first, /* position of first character of string */
 size_t l, /* length of identifier */
 eight_bits t) /* not used by \.{CTANGLE} */
 @y
-boolean __cdecl names_match(
+bool __cdecl names_match(
 name_pointer p, /* points to the proposed match */
 const char *first, /* position of first character of string */
 size_t l, /* length of identifier */
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctangle.c
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctangle.c	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctangle.c	(revision 74892)
@@ -2,7 +2,7 @@
 #line 66 "ctangle.w"
 
 /*5:*/
-#line 48 "common.h"
+#line 47 "common.h"
 
 #include <ctype.h>  
 #include <stdbool.h>  
@@ -127,14 +127,13 @@
 /*3:*/
 #line 35 "common.h"
 
-typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
-extern boolean program;
+extern bool program;
 extern int phase;
 
 /*:3*//*6:*/
-#line 78 "common.h"
+#line 77 "common.h"
 
 extern char section_text[];
 extern char*section_text_end;
@@ -142,7 +141,7 @@
 extern char*id_loc;
 
 /*:6*//*7:*/
-#line 96 "common.h"
+#line 95 "common.h"
 
 extern char buffer[];
 extern char*buffer_end;
@@ -150,7 +149,7 @@
 extern char*limit;
 
 /*:7*//*8:*/
-#line 113 "common.h"
+#line 112 "common.h"
 
 extern int include_depth;
 extern FILE*file[];
@@ -161,20 +160,20 @@
 extern int line[];
 extern int change_line;
 extern int change_depth;
-extern boolean input_has_ended;
-extern boolean changing;
-extern boolean web_file_open;
+extern bool input_has_ended;
+extern bool changing;
+extern bool web_file_open;
 
 /*:8*//*10:*/
-#line 133 "common.h"
+#line 132 "common.h"
 
 extern sixteen_bits section_count;
-extern boolean changed_section[];
-extern boolean change_pending;
-extern boolean print_where;
+extern bool changed_section[];
+extern bool change_pending;
+extern bool print_where;
 
 /*:10*//*11:*/
-#line 148 "common.h"
+#line 147 "common.h"
 
 typedef struct name_info{
 char*byte_start;
@@ -199,12 +198,12 @@
 extern hash_pointer hash_ptr;
 
 /*:11*//*13:*/
-#line 193 "common.h"
+#line 192 "common.h"
 
 extern int history;
 
 /*:13*//*15:*/
-#line 211 "common.h"
+#line 210 "common.h"
 
 extern int argc;
 extern char**argv;
@@ -212,10 +211,10 @@
 extern char tex_file_name[];
 extern char idx_file_name[];
 extern char scn_file_name[];
-extern boolean flags[];
+extern bool flags[];
 
 /*:15*//*16:*/
-#line 225 "common.h"
+#line 224 "common.h"
 
 extern FILE*C_file;
 extern FILE*tex_file;
@@ -280,7 +279,7 @@
 #line 457 "ctangle.w"
 
 static eight_bits out_state;
-static boolean protect;
+static bool protect;
 
 /*:42*//*45:*/
 #line 488 "ctangle.w"
@@ -293,7 +292,7 @@
 /*:45*//*53:*/
 #line 584 "ctangle.w"
 
-static boolean output_defs_seen= false;
+static bool output_defs_seen= false;
 
 /*:53*//*57:*/
 #line 694 "ctangle.w"
@@ -308,13 +307,13 @@
 /*:62*//*66:*/
 #line 827 "ctangle.w"
 
-static boolean comment_continues= false;
+static bool comment_continues= false;
 
 /*:66*//*68:*/
 #line 864 "ctangle.w"
 
 static name_pointer cur_section_name;
-static boolean no_where;
+static bool no_where;
 
 /*:68*//*82:*/
 #line 1182 "ctangle.w"
@@ -326,33 +325,33 @@
 #line 71 "ctangle.w"
 
 /*4:*/
-#line 43 "common.h"
+#line 42 "common.h"
 
 extern void common_init(void);
 
 /*:4*//*9:*/
-#line 127 "common.h"
+#line 126 "common.h"
 
-extern boolean get_line(void);
+extern bool get_line(void);
 extern void check_complete(void);
 extern void reset_input(void);
 
 /*:9*//*12:*/
-#line 171 "common.h"
+#line 170 "common.h"
 
 extern name_pointer id_lookup(const char*,const char*,eight_bits);
 
-extern name_pointer section_lookup(char*,char*,boolean);
+extern name_pointer section_lookup(char*,char*,bool);
 extern void print_prefix_name(name_pointer);
 extern void print_section_name(name_pointer);
 extern void sprint_section_name(char*,name_pointer);
 
-extern boolean names_match(name_pointer,const char*,size_t,eight_bits);
+extern bool names_match(name_pointer,const char*,size_t,eight_bits);
 
 extern void init_node(name_pointer);
 
 /*:12*//*14:*/
-#line 196 "common.h"
+#line 195 "common.h"
 
 extern int wrap_up(void);
 extern void err_print(const char*);
@@ -369,7 +368,7 @@
 #line 347 "ctangle.w"
 
 static void push_level(name_pointer);
-static void pop_level(boolean);
+static void pop_level(bool);
 static void get_output(void);
 
 /*:37*//*44:*/
@@ -387,7 +386,7 @@
 #line 807 "ctangle.w"
 
 static eight_bits skip_ahead(void);
-static boolean skip_comment(boolean);
+static bool skip_comment(bool);
 
 /*:65*//*70:*/
 #line 915 "ctangle.w"
@@ -485,7 +484,7 @@
 /*:2*//*24:*/
 #line 155 "ctangle.w"
 
-boolean names_match(
+bool names_match(
 name_pointer p,
 const char*first,
 size_t l,
@@ -538,7 +537,7 @@
 
 static void
 pop_level(
-boolean flag)
+bool flag)
 {
 if(flag&&cur_repl->text_link<section_flag){
 cur_repl= cur_repl->text_link+text_info;
@@ -868,8 +867,8 @@
 /*:64*//*67:*/
 #line 830 "ctangle.w"
 
-static boolean skip_comment(
-boolean is_long_comment)
+static bool skip_comment(
+bool is_long_comment)
 {
 char c;
 while(true){
@@ -905,7 +904,7 @@
 static eight_bits
 get_next(void)
 {
-static boolean preprocessing= false;
+static bool preprocessing= false;
 eight_bits c;
 while(true){
 if(loc> limit){
@@ -946,7 +945,7 @@
 if(xisdigit(c)||c=='.')/*73:*/
 #line 956 "ctangle.w"
 {
-boolean hex_flag= false;
+bool hex_flag= false;
 id_first= loc-1;
 if(*id_first=='.'&&!xisdigit(*loc))goto mistake;
 if(*id_first=='0'){
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctangle.w
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctangle.w	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctangle.w	(revision 74892)
@@ -153,7 +153,7 @@
 starting at position |first| equals the identifier pointed to by |p|:
 
 @c
-boolean names_match(
+bool names_match(
 name_pointer p, /* points to the proposed match */
 const char *first, /* position of first character of string */
 size_t l, /* length of identifier */
@@ -334,7 +334,7 @@
 @c
 static void
 pop_level( /* do this when |cur_byte| reaches |cur_end| */
-boolean flag) /* |flag==false| means we are in |output_defs| */
+bool flag) /* |flag==false| means we are in |output_defs| */
 {
   if (flag && cur_repl->text_link<section_flag) { /* link to a continuation */
     cur_repl=cur_repl->text_link+text_info; /* stay on the same level */
@@ -346,7 +346,7 @@
 
 @ @<Predecl...@>=
 static void push_level(name_pointer);@/
-static void pop_level(boolean);@/
+static void pop_level(bool);@/
 static void get_output(void);
 
 @ The heart of the output procedure is the function |get_output|,
@@ -456,7 +456,7 @@
 
 @<Private...@>=
 static eight_bits out_state; /* current status of partial output */
-static boolean protect; /* should newline characters be quoted? */
+static bool protect; /* should newline characters be quoted? */
 
 @ Here is a routine that is invoked when we want to output the current line.
 During the output process, |cur_line| equals the number of the next line
@@ -582,7 +582,7 @@
     output_defs();
 
 @ @<Private...@>=
-static boolean output_defs_seen=false;
+static bool output_defs_seen=false;
 
 @ @d C_printf(c,a) fprintf(C_file,c,a)
 @d C_putc(c) fputc((int)(c),C_file) /* isn't \CEE/ wonderfully consistent? */
@@ -806,7 +806,7 @@
 
 @ @<Predecl...@>=
 static eight_bits skip_ahead(void);@/
-static boolean skip_comment(boolean);
+static bool skip_comment(bool);
 
 @ The |skip_comment| procedure reads through the input at somewhat high
 speed in order to pass over comments, which \.{CTANGLE} does not transmit
@@ -825,11 +825,11 @@
 No comment, long or short, is allowed to contain `\.{@@\ }' or `\.{@@*}'.
 
 @<Private...@>=
-static boolean comment_continues=false; /* are we scanning a comment? */
+static bool comment_continues=false; /* are we scanning a comment? */
 
 @ @c
-static boolean skip_comment( /* skips over comments */
-boolean is_long_comment)
+static bool skip_comment( /* skips over comments */
+bool is_long_comment)
 {
   char c; /* current character */
   while (true) {
@@ -863,7 +863,7 @@
 
 @<Private...@>=
 static name_pointer cur_section_name; /* name of section just scanned */
-static boolean no_where; /* suppress |print_where|? */
+static bool no_where; /* suppress |print_where|? */
 
 @ As one might expect, |get_next| consists mostly of a big switch
 that branches to the various special cases that can arise.
@@ -872,7 +872,7 @@
 static eight_bits
 get_next(void) /* produces the next input token */
 {
-  static boolean preprocessing=false;
+  static bool preprocessing=false;
   eight_bits c; /* the current character */
   while (true) {
     if (loc>limit) {
@@ -954,7 +954,7 @@
 }
 
 @ @<Get a constant@>= {
-  boolean hex_flag = false; /* are we reading a hexadecimal literal? */
+  bool hex_flag = false; /* are we reading a hexadecimal literal? */
   id_first=loc-1;
   if (*id_first=='.' && !xisdigit(*loc)) goto mistake; /* not a constant */
   if (*id_first=='0') {
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-hint.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-hint.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-hint.ch	(revision 74892)
@@ -662,10 +662,10 @@
 
 @x
 finish_C( /* finishes a definition or a \CEE/ part */
-  boolean visible) /* |true| if we should produce \TeX\ output */
+  bool visible) /* |true| if we should produce \TeX\ output */
 @y
 finish_C( /* finishes a definition or a \CEE/ part */
-  boolean visible) /* |true| if we should produce \TeX\ output */
+  bool visible) /* |true| if we should produce \TeX\ output */
 @-finish_C@>
 @$finish_C {CTWILL}234 \&{static} \&{void} (\,)@>
 @z
@@ -673,9 +673,9 @@
 Section 235.
 
 @x
-@ @<Predecl...@>=@+static void finish_C(boolean);
+@ @<Predecl...@>=@+static void finish_C(bool);
 @y
-@ @<Predecl...@>=@+static void finish_C(boolean);
+@ @<Predecl...@>=@+static void finish_C(bool);
 @-finish_C@>
 @z
 
@@ -877,9 +877,9 @@
 Section 282.
 
 @x
-@ @<Predec...@>=@+static boolean app_supp(text_pointer);
+@ @<Predec...@>=@+static bool app_supp(text_pointer);
 @y
-@ @<Predec...@>=@+static boolean app_supp(text_pointer);
+@ @<Predec...@>=@+static bool app_supp(text_pointer);
 @-app_supp@>
 @z
 
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-mini.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-mini.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-mini.ch	(revision 74892)
@@ -1912,10 +1912,10 @@
 
 @x
 finish_C( /* finishes a definition or a \CEE/ part */
-  boolean visible) /* |true| if we should produce \TeX\ output */
+  bool visible) /* |true| if we should produce \TeX\ output */
 @y
 finish_C( /* finishes a definition or a \CEE/ part */
-  boolean visible) /* |true| if we should produce \TeX\ output */
+  bool visible) /* |true| if we should produce \TeX\ output */
 @-finish_C@>
 @$finish_C {CTWILL}251 \&{static} \&{void} (\,)@>
 @z
@@ -1923,9 +1923,9 @@
 Section 252.
 
 @x
-@ @<Predecl...@>=@+static void finish_C(boolean);
+@ @<Predecl...@>=@+static void finish_C(bool);
 @y
-@ @<Predecl...@>=@+static void finish_C(boolean);
+@ @<Predecl...@>=@+static void finish_C(bool);
 @-finish_C@>
 @z
 
@@ -2223,9 +2223,9 @@
 Section 300.
 
 @x
-@ @<Predec...@>=@+static boolean app_supp(text_pointer);
+@ @<Predec...@>=@+static bool app_supp(text_pointer);
 @y
-@ @<Predec...@>=@+static boolean app_supp(text_pointer);
+@ @<Predec...@>=@+static bool app_supp(text_pointer);
 @-app_supp@>
 @z
 
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-w2c.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-w2c.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-w2c.ch	(revision 74892)
@@ -127,7 +127,7 @@
 turned on during the first phase.
 
 @<Private...@>=
-static boolean change_exists; /* has any section changed? */
+static bool change_exists; /* has any section changed? */
 @y
 turned on during the first phase---NOT!
 @z
@@ -950,14 +950,14 @@
 @z
 
 @x [12.229] l.4212
-static boolean group_found=false; /* has a starred section occurred? */
+static bool group_found=false; /* has a starred section occurred? */
 
 @ @<Translate the \9{c}current section@>= {
   section_count++;
 @y
-static boolean group_found=false; /* has a starred section occurred? */
-static boolean right_start_switch; /* has `\.{@@r}' occurred recently? */
-static boolean temp_switch; /* has `\.{@@\%}' occurred recently? */
+static bool group_found=false; /* has a starred section occurred? */
+static bool right_start_switch; /* has `\.{@@r}' occurred recently? */
+static bool temp_switch; /* has `\.{@@\%}' occurred recently? */
 
 @ @d usage_sentinel (struct perm_meaning *)1
 @<Translate the \9{c}current section@>= {
@@ -1199,9 +1199,9 @@
 @z
 
 @x [13.262] l.4776
-    else {@+boolean all_caps=true;@+@t}\6{@>
+    else {@+bool all_caps=true;@+@t}\6{@>
 @y
-    else {@+boolean all_caps=true;@+char *p;
+    else {@+bool all_caps=true;@+char *p;
       /* index into |byte_mem| */ @+@t}\6{@>
 @z
 
@@ -1486,7 +1486,7 @@
 static char *ministring_buf_end=ministring_buf+max_tex_chars-1;
   /* end of |ministring_buf| */
 static char *ministring_ptr; /* first available slot in |ministring_buf| */
-static boolean ms_mode; /* are we outputting to |ministring_buf|? */
+static bool ms_mode; /* are we outputting to |ministring_buf|? */
 
 @q Section 27->274. @>
 @ @<Set init...@>=
@@ -1584,10 +1584,10 @@
 should set |ident_seen=false| first. (This is admittedly tricky.)
 
 @<Private var...@>=
-static boolean ident_seen;
+static bool ident_seen;
 
 @ @c
-static boolean app_supp(
+static bool app_supp(
   text_pointer p)
 { token_pointer j;
   if (ident_seen && **p>=tok_flag)
@@ -1607,7 +1607,7 @@
 }
 
 @q Section 282. @>
-@ @<Predec...@>=@+static boolean app_supp(text_pointer);
+@ @<Predec...@>=@+static bool app_supp(text_pointer);
 
 @q Section 142->283. @>
 @ The trickiest part of \.{CTWILL} is the procedure |make_ministring(pp+l)|,
@@ -1660,7 +1660,7 @@
   text_pointer q=(p-1)->trans, r;
   token t;
   int ast_count=0; /* asterisks preceding the expression */
-  boolean non_ast_seen=false; /* have we seen a non-asterisk? */
+  bool non_ast_seen=false; /* have we seen a non-asterisk? */
   while (true) {
     if (*(q+1)==*q+1) {
       r=q;@+break; /* e.g., \&{struct}; we're doing production 45 or 46 */
@@ -1699,8 +1699,8 @@
 
 @q Section 253->288. @>
 @ @<Private...@>=
-static boolean is_macro; /* it's a macro def, not a format def */
-static boolean def_diff; /* |false| iff the current macro has parameters */
+static bool is_macro; /* it's a macro def, not a format def */
+static bool def_diff; /* |false| iff the current macro has parameters */
 static name_pointer id_being_defined; /* the definee */
 
 @q Section 257->289. @>
@@ -1832,7 +1832,7 @@
 switch (cur_name->ilk) {@+char *p; /* index into |byte_mem| */@+@t}\6{\4@>
   case normal: case func_template:
     if (is_tiny(cur_name)) out_str("\\|");
-    else {@+boolean all_caps=true;@+@t}\6{@>
+    else {@+bool all_caps=true;@+@t}\6{@>
       for (p=cur_name->byte_start;p<(cur_name+1)->byte_start;p++)
         if (xislower(*p)) { /* not entirely uppercase */
           all_caps=false; break;
@@ -1942,7 +1942,7 @@
 
 @<Update the result...@>= {
 if((tex_file=fopen(tex_file_name,"r"))!=NULL) {
-  boolean comparison=false;
+  bool comparison=false;
 
   if((check_file=fopen(check_file_name,"r"))==NULL)
     fatal(_("! Cannot open output file "),check_file_name);
@@ -2015,8 +2015,6 @@
 @d kpse_find_cweb(name) kpse_find_file(name,kpse_cweb_format,true)
 
 @<Include files@>=
-typedef bool boolean;
-#define HAVE_BOOLEAN 1
 #include <kpathsea/tex-file.h> /* |@!kpse_find_file| */
 
 @ @<Set up |cur_file_name|...@>=
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill.bux
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill.bux	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill.bux	(revision 74892)
@@ -19,11 +19,11 @@
 @$line {COMMON.W}25 \&{int} [\,]@>
 @$change_line {COMMON.W}25 \&{int}@>
 @$change_depth {COMMON.W}25 \&{int}@>
-@$input_has_ended {COMMON.W}25 \&{boolean}@>
-@$changing {COMMON.W}25 \&{boolean}@>
-@$web_file_open {COMMON.W}25 \&{boolean}@>
+@$input_has_ended {COMMON.W}25 \&{bool}@>
+@$changing {COMMON.W}25 \&{bool}@>
+@$web_file_open {COMMON.W}25 \&{bool}@>
 
-@$get_line {COMMON.W}35 \&{boolean} (\,)@>
+@$get_line {COMMON.W}35 \&{bool} (\,)@>
 
 @$check_complete {COMMON.W}39 \&{void} (\,)@>
 
@@ -30,9 +30,9 @@
 @$reset_input {COMMON.W}40 \&{void} (\,)@>
 
 @$section_count {COMMON.W}42 \&{sixteen\_bits}@>
-@$changed_section {COMMON.W}42 \&{boolean} [\,]@>
-@$change_pending {COMMON.W}42 \&{boolean}@>
-@$print_where {COMMON.W}42 \&{boolean}@>
+@$changed_section {COMMON.W}42 \&{bool} [\,]@>
+@$change_pending {COMMON.W}42 \&{bool}@>
+@$print_where {COMMON.W}42 \&{bool}@>
 
 @$byte_mem {COMMON.W}43 \&{char} [\,]@>
 @$byte_mem_end {COMMON.W}43 \&{char} ${*}$@>
@@ -70,7 +70,7 @@
 @$idx_file_name {COMMON.W}73 \&{char} [\,]@>
 @$scn_file_name {COMMON.W}73 \&{char} [\,]@>
 @$check_file_name {COMMON.W}73 \&{char} [\,]@>
-@$flags {COMMON.W}73 \&{boolean} [\,]@>
+@$flags {COMMON.W}73 \&{bool} [\,]@>
 
 @$C_file {COMMON.W}83 \&{FILE} ${*}$@>
 @$tex_file {COMMON.W}83 \&{FILE} ${*}$@>
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweav-w2c.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweav-w2c.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweav-w2c.ch	(revision 74892)
@@ -642,7 +642,7 @@
 
 @<Update the result...@>= {
 if((tex_file=fopen(tex_file_name,"r"))!=NULL) {
-  boolean comparison=false;
+  bool comparison=false;
 
   if((check_file=fopen(check_file_name,"r"))==NULL)
     fatal(_("! Cannot open output file "),check_file_name);
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweav-w32.ch
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweav-w32.ch	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweav-w32.ch	(revision 74892)
@@ -8,13 +8,13 @@
 @z
 
 @x section 32
-boolean names_match(
+bool names_match(
 name_pointer p, /* points to the proposed match */
 const char *first, /* position of first character of string */
 size_t l, /* length of identifier */
 eight_bits t) /* desired |ilk| */
 @y
-boolean __cdecl names_match(
+bool __cdecl names_match(
 name_pointer p, /* points to the proposed match */
 const char *first, /* position of first character of string */
 size_t l, /* length of identifier */
Index: texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweave.w
===================================================================
--- texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweave.w	(revision 74891)
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/cweave.w	(revision 74892)
@@ -182,7 +182,7 @@
 turned on during the first phase.
 
 @<Private...@>=
-static boolean change_exists; /* has any section changed? */
+static bool change_exists; /* has any section changed? */
 
 @ The other large memory area in \.{CWEAVE} keeps the cross-reference data.
 All uses of the name |p| are recorded in a linked list beginning at
@@ -355,7 +355,7 @@
 
 @ Here are the two procedures needed to complete |id_lookup|:
 @c
-boolean names_match(
+bool names_match(
 name_pointer p, /* points to the proposed match */
 const char *first, /* position of first character of string */
 size_t l, /* length of identifier */
@@ -730,7 +730,7 @@
 @d right_preproc 0217 /* ends a preprocessor command */
 
 @<Private...@>=
-static boolean preprocessing=false; /* are we scanning a preprocessor command? */
+static bool preprocessing=false; /* are we scanning a preprocessor command? */
 
 @ @<Raise prep...@>= {
   preprocessing=true;
@@ -743,7 +743,7 @@
 name as a string.
 
 @<Private...@>=
-static boolean sharp_include_line=false;
+static bool sharp_include_line=false;
   /* are we scanning a \#\&{include} line? */
 
 @ @<Check if next token is |include|@>=
@@ -1140,7 +1140,7 @@
     if (next_control!=begin_comment && next_control!=begin_short_comment)
       C_xref(ignore);
     else {
-      boolean is_long_comment=(next_control==begin_comment);
+      bool is_long_comment=(next_control==begin_comment);
       bal=copy_comment(is_long_comment,1); next_control='|';
       while (bal>0) {
         C_xref(section_name); /* do not reference section names in comments */
@@ -1282,7 +1282,7 @@
 
 @<Private...@>=
 static xref_pointer cur_xref; /* temporary cross-reference pointer */
-static boolean an_output; /* did |file_flag| precede |cur_xref|? */
+static bool an_output; /* did |file_flag| precede |cur_xref|? */
 
 @ The following recursive procedure
 walks through the tree of section names and prints out anomalies.
@@ -1348,7 +1348,7 @@
 @d tex_puts(c) fputs(c,active_file)
 
 @<Predecl...@>=
-static void flush_buffer(char *,boolean,boolean);@/
+static void flush_buffer(char *,bool,bool);@/
 static void finish_line(void);
 
 @ @c
@@ -1355,7 +1355,7 @@
 static void
 flush_buffer(
 char *b, /* outputs from |out_buf+1| to |b|, where |b<=out_ptr| */
-boolean per_cent,boolean carryover)
+bool per_cent,bool carryover)
 {
   char *j=b; /* pointer into |out_buf| */
   if (! per_cent) /* remove trailing blanks */
@@ -1472,7 +1472,7 @@
 
 @<Predecl...@>=
 static void out_section(sixteen_bits);@/
-static void out_name(name_pointer,boolean);
+static void out_name(name_pointer,bool);
 
 @ @c
 static void
@@ -1492,7 +1492,7 @@
 static void
 out_name(
 name_pointer p,
-boolean quote_xalpha)
+bool quote_xalpha)
 {
   char *k; /* pointer into |byte_mem| */
   out('{');
@@ -1519,7 +1519,7 @@
 @<Predecl...@>=
 static void copy_limbo(void);@/
 static eight_bits copy_TeX(void);@/
-static int copy_comment(boolean,int);
+static int copy_comment(bool,int);
 
 @ @c
 static void
@@ -1585,7 +1585,7 @@
 
 @c
 static int copy_comment( /* copies \TeX\ code in comments */
-boolean is_long_comment, /* is this a traditional \CEE/ comment? */
+bool is_long_comment, /* is this a traditional \CEE/ comment? */
 int bal) /* brace balance */
 {
   char c; /* current character being copied */
@@ -3627,7 +3627,7 @@
 token list; it also builds a new scrap if |scrapping==true|.
 
 @<Predec...@>=
-static void app_cur_id(boolean);@/
+static void app_cur_id(bool);@/
 static text_pointer C_translate(void);@/
 static void outer_parse(void);
 
@@ -3634,7 +3634,7 @@
 @ @c
 static void
 app_cur_id(
-boolean scrapping) /* are we making this into a scrap? */
+bool scrapping) /* are we making this into a scrap? */
 {
   name_pointer p=id_lookup(id_first,id_loc,normal);
   if (p->ilk<=custom) { /* not a reserved word */
@@ -3698,7 +3698,7 @@
     if (next_control!=begin_comment && next_control!=begin_short_comment)
       C_parse(ignore);
     else {
-      boolean is_long_comment=(next_control==begin_comment);
+      bool is_long_comment=(next_control==begin_comment);
       @<Make sure that there is room for the new...@>@;
       app(cancel); app(inserted);
       if (is_long_comment) app_str("\\C{"@q}@>);
@@ -3911,8 +3911,8 @@
   char delim; /* first and last character of string being copied */
   char *save_loc, *save_limit; /* |loc| and |limit| to be restored */
   name_pointer cur_section_name; /* name of section being output */
-  boolean save_mode; /* value of |cur_mode| before a sequence of breaks */
-  boolean dindent_pending=false; /* should a |dindent| be output? */
+  bool save_mode; /* value of |cur_mode| before a sequence of breaks */
+  bool dindent_pending=false; /* should a |dindent| be output? */
   app(end_translation); /* append a sentinel */
   freeze_text(); push_level(text_ptr-1);
   while (true) {
@@ -4206,10 +4206,10 @@
 static int save_line; /* former value of |out_line| */
 static char *save_place; /* former value of |out_ptr| */
 static int sec_depth; /* the integer, if any, following \.{@@*} */
-static boolean space_checked; /* have we done |emit_space_if_needed|? */
-static boolean format_visible; /* should the next format declaration be output? */
-static boolean doing_format=false; /* are we outputting a format declaration? */
-static boolean group_found=false; /* has a starred section occurred? */
+static bool space_checked; /* have we done |emit_space_if_needed|? */
+static bool format_visible; /* should the next format declaration be output? */
+static bool doing_format=false; /* are we outputting a format declaration? */
+static bool group_found=false; /* has a starred section occurred? */
 
 @ @<Translate the \9{c}current section@>= {
   section_count++;
@@ -4305,7 +4305,7 @@
 @c
 static void
 finish_C( /* finishes a definition or a \CEE/ part */
-  boolean visible) /* |true| if we should produce \TeX\ output */
+  bool visible) /* |true| if we should produce \TeX\ output */
 {
   text_pointer p; /* translation of the scraps */
   if (visible) {
@@ -4330,7 +4330,7 @@
     /* forget the tokens and the scraps */
 }
 
-@ @<Predecl...@>=@+static void finish_C(boolean);
+@ @<Predecl...@>=@+static void finish_C(bool);
 
 @ Keeping in line with the conventions of the \CEE/ preprocessor (and
 otherwise contrary to the rules of \.{CWEB}) we distinguish here
@@ -4773,7 +4773,7 @@
 switch (cur_name->ilk) {@+char *p; /* index into |byte_mem| */@+@t}\6{\4@>
   case normal: case func_template:
     if (is_tiny(cur_name)) out_str("\\|");
-    else {@+boolean all_caps=true;@+@t}\6{@>
+    else {@+bool all_caps=true;@+@t}\6{@>
       for (p=cur_name->byte_start;p<(cur_name+1)->byte_start;p++)
         if (xislower(*p)) { /* not entirely uppercase */
           all_caps=false; break;
