--- src/firefox-91.12.0/gfx/webrender_bindings/webrender_ffi.h.orig	2022-08-16 10:02:32.775661134 +0200
+++ src/firefox-91.12.0/gfx/webrender_bindings/webrender_ffi.h	2022-08-16 10:02:45.690394969 +0200
@@ -77,8 +77,6 @@
 struct WrPipelineIdAndEpoch;
 using WrPipelineIdEpochs = nsTArray<WrPipelineIdAndEpoch>;
 
-const uint64_t ROOT_CLIP_CHAIN = ~0;
-
 }  // namespace wr
 }  // namespace mozilla
 
