diff -aur testrepository-0.0.21.old/testrepository/tests/test_repository.py testrepository-0.0.21/testrepository/tests/test_repository.py
--- testrepository-0.0.21.old/testrepository/tests/test_repository.py	2025-12-18 21:37:16.617154784 +0100
+++ testrepository-0.0.21/testrepository/tests/test_repository.py	2025-12-18 21:37:43.415060086 +0100
@@ -358,33 +358,6 @@
         run = repo.get_test_run(inserted)
         self.assertEqual(inserted, run.get_id())
 
-    def test_get_test_run_preserves_time(self):
-        self.skip('Fix me before releasing.')
-        # The test run outputs the time events that it received.
-        now = datetime(2001, 1, 1, 0, 0, 0, tzinfo=iso8601.UTC)
-        second = timedelta(seconds=1)
-        repo = self.repo_impl.initialise(self.sample_url)
-        test_id = self.getUniqueString()
-        test = make_test(test_id, True)
-        result = repo.get_inserter()
-        result.startTestRun()
-        result.status(timestamp=now, test_id=test_id, test_status='inprogress')
-        result.status(timestamp=(now + 1 * second), test_id=test_id, test_status='success')
-        inserted = result.stopTestRun()
-        run = repo.get_test_run(inserted)
-        result = ExtendedTestResult()
-        run.get_test().run(result)
-        self.assertEqual(
-            [('time', now),
-             ('tags', set(), set()),
-             ('startTest', Wildcard),
-             ('time', now + 1 * second),
-             ('addSuccess', Wildcard),
-             ('stopTest', Wildcard),
-             ('tags', set(), set()),
-             ],
-            result._events)
-
     def test_get_failing_get_id(self):
         repo = self.repo_impl.initialise(self.sample_url)
         result = repo.get_inserter()
