--- linux-atm-2.5.2/src/maint/atmdump.c.orig	2020-09-16 16:12:50.800622536 +0200
+++ linux-atm-2.5.2/src/maint/atmdump.c	2020-09-16 16:13:23.197409008 +0200
@@ -14,6 +14,7 @@
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/socket.h>
+#include <linux/sockios.h>
 #include <netinet/in.h> /* for htonl and ntohl */
 #include <atm.h>
 
--- linux-atm-2.5.2/src/maint/saaldump.c.orig	2020-09-16 16:15:04.823302300 +0200
+++ linux-atm-2.5.2/src/maint/saaldump.c	2020-09-16 16:15:21.321658143 +0200
@@ -15,6 +15,7 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <linux/sockios.h>
 #include <atm.h>
 
 #include "pdu.h"
