--- RTL88x2BU-Linux-Driver/Makefile.orig	2019-10-31 13:19:08.616194542 +0100
+++ RTL88x2BU-Linux-Driver/Makefile	2019-10-31 13:19:19.825060138 +0100
@@ -82,7 +82,7 @@
 CONFIG_ICMP_VOQ = n
 CONFIG_IP_R_MONITOR = n #arp VOQ and high rate
 ########################## Debug ###########################
-CONFIG_RTW_DEBUG = y
+CONFIG_RTW_DEBUG = n
 # default log level is _DRV_INFO_ = 4,
 # please refer to "How_to_set_driver_debug_log_level.doc" to set the available level.
 CONFIG_RTW_LOG_LEVEL = 4
