diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c3a42c..1b42af0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ if (BUILD_KPACKAGE_INSTALL_HANDLERS)
    find_package(KF5I18n ${KF_DEP_VERSION} REQUIRED)
 
    find_package(packagekitqt${QT_MAJOR_VERSION})
-   find_package(AppStreamQt 0.10.4)
+   find_package(AppStreamQt5 1.0)
 endif()
 
 ecm_set_disabled_deprecation_versions(
diff --git a/src/kpackage-install-handlers/CMakeLists.txt b/src/kpackage-install-handlers/CMakeLists.txt
index d0bf7f0..a7c126f 100644
--- a/src/kpackage-install-handlers/CMakeLists.txt
+++ b/src/kpackage-install-handlers/CMakeLists.txt
@@ -1,5 +1,5 @@
 add_subdirectory(kns)
 
-if(AppStreamQt_FOUND AND packagekitqt5_FOUND)
+if(AppStreamQt5_FOUND AND packagekitqt5_FOUND)
     add_subdirectory(appstream)
 endif()
diff --git a/src/kpackage-install-handlers/appstream/CMakeLists.txt b/src/kpackage-install-handlers/appstream/CMakeLists.txt
index 8db1bdb..9552c5f 100644
--- a/src/kpackage-install-handlers/appstream/CMakeLists.txt
+++ b/src/kpackage-install-handlers/appstream/CMakeLists.txt
@@ -1,3 +1,3 @@
 add_executable(appstreamhandler main.cpp)
-target_link_libraries(appstreamhandler PK::packagekitqt5 AppStreamQt)
+target_link_libraries(appstreamhandler PK::packagekitqt5 AppStreamQt5)
 install(TARGETS appstreamhandler DESTINATION ${KDE_INSTALL_LIBEXECDIR_KF}/kpackagehandlers)
diff --git a/src/kpackage-install-handlers/appstream/main.cpp b/src/kpackage-install-handlers/appstream/main.cpp
index 82328f7..8bde139 100644
--- a/src/kpackage-install-handlers/appstream/main.cpp
+++ b/src/kpackage-install-handlers/appstream/main.cpp
@@ -5,6 +5,6 @@
 */
 
-#include <AppStreamQt/pool.h>
+#include <AppStreamQt5/pool.h>
 #include <PackageKit/Daemon>
 #include <QCoreApplication>
 #include <QDebug>
