From 023ff2cd6953021b98864a718f48868ee475b113 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 27 Mar 2024 10:25:36 +0100
Subject: [PATCH 1/1] bump version to match last tag

---
 include/curl/curlver.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/curl/curlver.h b/include/curl/curlver.h
index 52009e548..b44519919 100644
--- a/include/curl/curlver.h
+++ b/include/curl/curlver.h
@@ -32,13 +32,13 @@
 
 /* This is the version number of the libcurl package from which this header
    file origins: */
-#define LIBCURL_VERSION "8.7.0-DEV"
+#define LIBCURL_VERSION "8.7.1-DEV"
 
 /* The numeric version number is also available "in parts" by using these
    defines: */
 #define LIBCURL_VERSION_MAJOR 8
 #define LIBCURL_VERSION_MINOR 7
-#define LIBCURL_VERSION_PATCH 0
+#define LIBCURL_VERSION_PATCH 1
 
 /* This is the numeric version of the libcurl version number, meant for easier
    parsing and comparisons by programs. The LIBCURL_VERSION_NUM define will
@@ -59,7 +59,7 @@
    CURL_VERSION_BITS() macro since curl's own configure script greps for it
    and needs it to contain the full number.
 */
-#define LIBCURL_VERSION_NUM 0x080700
+#define LIBCURL_VERSION_NUM 0x080701
 
 /*
  * This is the date and time when the full source package was created. The
-- 
2.44.0

