--- clang-20.1.8.src/lib/Basic/Targets/PPC.h.orig	2025-10-09 20:28:06.329765380 +0200
+++ clang-20.1.8.src/lib/Basic/Targets/PPC.h	2025-10-09 20:28:51.376870481 +0200
@@ -132,6 +132,7 @@
               .Case("7400", ArchDefineName | ArchDefinePpcgr)
               .Case("7450", ArchDefineName | ArchDefinePpcgr)
               .Case("750", ArchDefineName | ArchDefinePpcgr)
+              .Case("espresso", ArchDefineName | ArchDefinePpcgr)
               .Case("970", ArchDefineName | ArchDefinePwr4 | ArchDefinePpcgr |
                                ArchDefinePpcsq)
               .Case("a2", ArchDefineA2)
