diff --git i/runtimes/CMakeLists.txt w/runtimes/CMakeLists.txt
index 830165c799c2..fa0df7d27ff4 100644
--- i/runtimes/CMakeLists.txt
+++ w/runtimes/CMakeLists.txt
@@ -194,7 +194,7 @@ if(CMAKE_C_COMPILER_ID MATCHES "Clang")
     OUTPUT_VARIABLE output
     OUTPUT_STRIP_TRAILING_WHITESPACE)
   if(result EQUAL 0)
-    set(LLVM_DEFAULT_TARGET_TRIPLE ${output})
+    #set(LLVM_DEFAULT_TARGET_TRIPLE ${output})
   else()
     string(REPLACE ";" " " print_target_triple "${print_target_triple}")
     # TODO(#97876): Report an error.
