From 3a62b30c5078f6c77b936f1cdcf53f64a84f7282 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 26 Sep 2025 21:11:19 +0200
Subject: [PATCH] Add support for setting parent window on Wayland

---
 src/utils.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/utils.cpp b/src/utils.cpp
index 2de51c7..a71906d 100644
--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -39,4 +39,10 @@ void Utils::setParentWindow(QWidget *w, const QString &parent_window)
         w->setAttribute(Qt::WA_NativeWindow, true);
         KWindowSystem::setMainWindow(w->windowHandle(), parent_window.mid(4).toULongLong(nullptr, 16));
     }
+    if (parent_window.startsWith((QLatin1String("wayland:")))) {
+        if (!w->window()->windowHandle()) {
+            w->window()->winId(); // create QWindow
+        }
+        KWindowSystem::setMainWindow(w->window()->windowHandle(), parent_window.mid(strlen("wayland:")));
+    }
 }
