--- src/discover-5.21.3/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h.orig	2021-04-18 08:41:25.636000000 +0000
+++ src/discover-5.21.3/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h	2021-04-18 08:41:36.004000000 +0000
@@ -7,11 +7,9 @@
 #ifndef FLATPAKTRANSACTIONTHREAD_H
 #define FLATPAKTRANSACTIONTHREAD_H
 
-extern "C" {
 #include <flatpak.h>
 #include <gio/gio.h>
 #include <glib.h>
-}
 
 #include <Transaction/Transaction.h>
 #include <QThread>
