diff -aur python-feedgen-1.0.0.old/doc/conf.py python-feedgen-1.0.0/doc/conf.py
--- python-feedgen-1.0.0.old/doc/conf.py	2024-11-19 14:28:06.308582251 +0100
+++ python-feedgen-1.0.0/doc/conf.py	2024-11-19 14:28:25.955374239 +0100
@@ -249,7 +249,7 @@
 
 
 # Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {'http://docs.python.org/': None}
+intersphinx_mapping = {'python': ('https://docs.python.org/3', None)}
 
 
 # Include the GitHub readme file in index.rst
