--- rocm-llvm/amd/comgr/cmake/device-libs-id.py.orig	2026-01-23 09:30:16.757344679 +0100
+++ rocm-llvm/amd/comgr/cmake/device-libs-id.py	2026-01-23 09:59:37.001609577 +0100
@@ -1,6 +1,5 @@
 from argparse import ArgumentParser
 from hashlib import sha256
-from functools import reduce
 
 if __name__ == "__main__":
     parser = ArgumentParser(description='Generate id by computing a hash of the generated headers')
@@ -10,11 +9,12 @@
 
     args = parser.parse_args()
     args.headers.sort()
-    
+
     hash = sha256()
     for x in args.headers:
         hash.update(open(x, 'rb').read())
     digest_uchar = hash.digest()
-    digest_char = [e if e < 128 else e-256 for e in digest_uchar]
-    digest_elts = ", ".join(map(str, digest_char))
-    print(f"static const char {args.varname}[] = {{{digest_elts}, 0}};", file=open(args.output, 'w'))
+    digest_char = [e if e < 128 else e - 256 for e in digest_uchar]
+    digest_elts = ", ".join(f"static_cast<char>({e})" for e in digest_char)
+    with open(args.output, 'w') as f:
+        print(f"static const char {args.varname}[] = {{{digest_elts}, static_cast<char>(0)}};", file=f)
