--- rocm-cmake-rocm-6.2.4/test/test.cmake.bak	2024-11-23 09:36:53.749629000 +0100
+++ rocm-cmake-rocm-6.2.4/test/test.cmake	2024-11-23 09:37:42.573123886 +0100
@@ -111,7 +111,7 @@
     if(NOT EXISTS ${BUILD_DIR})
         file(MAKE_DIRECTORY ${BUILD_DIR})
     endif()
-    test_exec(COMMAND ${CMAKE_COMMAND} -Werror=dev ${ADDITIONAL_CONFIGURE_ARGS} -DCMAKE_PREFIX_PATH=${PREFIX}
+    test_exec(COMMAND ${CMAKE_COMMAND} -Wno-deprecated -Wno-error=deprecated ${ADDITIONAL_CONFIGURE_ARGS} -DCMAKE_PREFIX_PATH=${PREFIX}
                       -DCMAKE_INSTALL_PREFIX=${PREFIX} -DROCM_ERROR_TOOLCHAIN_VAR=On ${PARSE_CMAKE_ARGS} ${DIR}
               WORKING_DIRECTORY ${BUILD_DIR})
     foreach(TARGET ${PARSE_TARGETS})
