diff -ru threadpoolctl-3.6.0.orig/pyproject.toml threadpoolctl-3.6.0/pyproject.toml
--- threadpoolctl-3.6.0.orig/pyproject.toml	2026-07-05 13:12:18.272367648 +0200
+++ threadpoolctl-3.6.0/pyproject.toml	2026-07-05 13:16:46.777752258 +0200
@@ -1,20 +1,16 @@
 [build-system]
-requires = ["flit_core >=2,<4"]
+requires = ["flit_core >=2"]
 build-backend = "flit_core.buildapi"
 
-# TODO: replace the following section by the standard [project] section to be able
-# to use flit v4.
-[tool.flit.metadata]
-module = "threadpoolctl"
-author = "Thomas Moreau"
-author-email = "thomas.moreau.2010@gmail.com"
-home-page = "https://github.com/joblib/threadpoolctl"
-description-file = "README.md"
+[project]
+name = "threadpoolctl"
+authors = [{name = "Thomas Moreau", email = "thomas.moreau.2010@gmail.com"}]
+urls = {Homepage = "https://github.com/joblib/threadpoolctl"}
+readme = "README.md"
 requires-python = ">=3.9"
 license = "BSD-3-Clause"
 classifiers = [
     "Intended Audience :: Developers",
-    "License :: OSI Approved :: BSD License",
     "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
@@ -23,6 +19,7 @@
     "Programming Language :: Python :: 3.13",
     "Topic :: Software Development :: Libraries :: Python Modules",
 ]
+dynamic = ["version", "description"]
 
 [tool.black]
 line-length = 88
