--- telepathy-gabble-0.18.4/lib/ext/wocky/wocky/wocky-xmpp-reader.c.xmlcast	2016-07-02 17:35:49.000000000 +0900
+++ telepathy-gabble-0.18.4/lib/ext/wocky/wocky/wocky-xmpp-reader.c	2024-01-29 00:25:28.520953431 +0900
@@ -72,7 +72,7 @@ static void _end_element_ns (void *user_
 
 static void _characters (void *user_data, const xmlChar *ch, int len);
 
-static void _error (void *user_data, xmlErrorPtr error);
+static void _error (void *user_data, const xmlError *error);
 
 static xmlSAXHandler parser_handler = {
   /* internalSubset         */ NULL,
@@ -668,7 +668,7 @@ _end_element_ns (void *user_data, const
 }
 
 static void
-_error (void *user_data, xmlErrorPtr error)
+_error (void *user_data, const xmlError *error)
 {
   WockyXmppReader *self = WOCKY_XMPP_READER (user_data);
   WockyXmppReaderPrivate *priv = self->priv;
