--- nss/coreconf/config.gypi.vanilla	2025-08-29 10:08:14.752972341 +0100
+++ nss/coreconf/config.gypi	2025-08-29 10:10:22.455234859 +0100
@@ -112,8 +112,8 @@
     'disable_dbm%': 1,
     'disable_libpkix%': 1,
     'disable_werror%': 0,
-    'disable_altivec%': 0,
-    'disable_crypto_vsx%': 0,
+    'disable_altivec%': 1,
+    'disable_crypto_vsx%': 1,
     'disable_arm32_neon%': 0,
     'mozilla_client%': 0,
     'comm_client%': 0,
