diff --git a/tests/test_old_api.py b/tests/test_old_api.py
index 5b0b37b..2f5fad5 100644
--- a/tests/test_old_api.py
+++ b/tests/test_old_api.py
@@ -24,7 +24,7 @@ import sure
 from sure.deprecated import that
 from sure.magic import is_cpython
 from sure import VariablesBag, expect
-from nose.tools import assert_equals, assert_raises
+from nose.tools import assert_equal as assert_equals, assert_raises
 from sure.compat import compat_repr, safe_repr, text_type_name
 
 
