diff --git a/tests/test_old_api.py b/tests/test_old_api.py
index 5b0b37b..8377f25 100644
--- a/tests/test_old_api.py
+++ b/tests/test_old_api.py
@@ -562,7 +562,7 @@ def test_that_none_contains_string():
         assert that(None).contains('bungalow')
         assert False, 'should not reach here'
     except Exception as e:
-        error_msg = "argument of type 'NoneType' is not iterable" if is_cpython else "'NoneType' object is not iterable"
+        error_msg = "argument of type 'NoneType' is not a container or iterable" if is_cpython else "'NoneType' object is not iterable"
         assert_equals(
             text_type(e),
             error_msg
