From 2a67c0f26a27271f62636fa2fc7b41918ccc492b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Torsten=20Ke=C3=9Fler?= <t.kessler@posteo.de>
Date: Sat, 30 Sep 2023 19:14:47 +0200
Subject: [PATCH] Mark new symbols in ROCm 5.7.0 as global

---
 src/libhsakmt.ver | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libhsakmt.ver b/src/libhsakmt.ver
index dc17215..1ed002f 100644
--- a/src/libhsakmt.ver
+++ b/src/libhsakmt.ver
@@ -75,6 +75,10 @@ hsaKmtSetXNACKMode;
 hsaKmtGetXNACKMode;
 hsaKmtOpenSMI;
 hsaKmtExportDMABufHandle;
+hsaKmtWaitOnEvent_Ext;
+hsaKmtWaitOnMultipleEvents_Ext;
+hsaKmtReplaceAsanHeaderPage;
+hsaKmtReturnAsanHeaderPage;
 
 local: *;
 };
