--- netpbm-10.29/buildtools/makeman
+++ netpbm-10.29/buildtools/makeman
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # makeman -- compile netpbm's stereotyped HTML to troff markup
 #
@@ -176,7 +176,7 @@
         sectmap = {}
         for file in arguments:
             try: 
-                infp = open(os.path.join(dirprefix, file))
+                infp = open(os.path.join(dirprefix, file), encoding="latin-1")
             except:
                 sys.stderr.write("makeman: can't open %s\n" % file)
                 continue
@@ -207,11 +207,11 @@
         # Second pass: do formatting
         for file in arguments:
             try: 
-                infp = open(os.path.join(dirprefix, file))
+                infp = open(os.path.join(dirprefix, file), encoding="latin-1")
             except:
                 sys.stderr.write("makeman: can't open %s\n" % file)
                 continue
             indoc = infp.read()
             infp.close()
             tempfile = file + ".~%s-%d~" % (name, os.getpid())
             try:
-                outfp = open(tempfile, "w")
+                outfp = open(tempfile, "w", encoding="utf-8")
             except OSError:
                 sys.stderr.write("%s: can't open tempfile" % name)
                 return True
@@ -227,7 +227,7 @@
             else:
                 outfp.write(outdoc)
                 outfp.close()	# under Windows you can't rename an open file
                 stem = file[:file.find(".")]
-                os.rename(tempfile, stem + "." + repr(sectmap[file]))
+                os.rename(tempfile, stem + "." + str(sectmap[file]))
     except LiftException as e:
         mainerr.write("makeman: " + e.message + "\n")
         return e.retval
